{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# CheRMiT Snorkel Pipeline\n",
    "\n",
    "This notebook allows you to create a Snorkel classifier, run it on data, and analyze the results. Just run the cells in order!\n",
    "\n",
    "Precondition: You need the input dataframes to follow a particular order (sentences in the 0th column and chemical lists in the 1st column), because the labeling functions use those indices to access the columns."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Imports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 162,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import os\n",
    "import nltk\n",
    "import regex as re\n",
    "import ast\n",
    "from collections import defaultdict\n",
    "from snorkel.labeling import labeling_function"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Setup for Labeling Functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "ABSTAIN = -1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_lfs = []"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Labeling Functions + small tests written"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [],
   "source": [
    "# includes_solution_words REGEX\n",
    "# If any of the words is in a list of solution terms, we label FALSE\n",
    "# https://www.hach.com/chemGlossary   \n",
    "\n",
    "solution_terms = \"buffer|dilute|diluent|dissolv|miscib|\\\\bsolute|solvent\" + \\\n",
    "    \"|solub|saturat|saturable|aqueous|gaseous|solid\"\n",
    "@labeling_function()\n",
    "def includes_solution_words(x):\n",
    "    structure = \"(\" + solution_terms + \")\"\n",
    "    if (re.search(structure, x[0])):\n",
    "        return False\n",
    "    return ABSTAIN\n",
    "\n",
    "all_lfs.append(includes_solution_words)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [],
   "source": [
    "#includes_experiment_words\n",
    "#entropy enthalpy temperature\n",
    "#\"filter|filtr\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [],
   "source": [
    "# includes_mixture_words\n",
    "# If any of the words is in a list of mixture terms, we label FALSE\n",
    "# https://www.hach.com/chemGlossary   \n",
    "mixture_terms = [\"suspended\", \"mixture\", \"heterogenous\", \"homogeneous\"]\n",
    "@labeling_function()\n",
    "def includes_mixture_words(x):\n",
    "    for word in mixture_terms:\n",
    "        if (word in x[0]):\n",
    "            return (False)\n",
    "    return ABSTAIN\n",
    "\n",
    "all_lfs.append(includes_mixture_words)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [],
   "source": [
    "# includes_physical_words\n",
    "# If there are common physical terms, we label FALSE\n",
    "physical_terms = [\"detection\", \"composed\", \"weight\", \"characteristic\", \"metal\", \"express\",\n",
    "    \"characterization\", \"color\", \"metalic\", \"consists\", \"pure\", \"compose\", \"assay\", \"mm\", \"bound\",\n",
    "    \"permeable\", \"signal\", \"bind\", \"property\", \"stored\", \"released\", \"capacity\", \"resistance\", \"mol\"]\n",
    "@labeling_function()\n",
    "def includes_physical_words(x):\n",
    "    for word in physical_terms:\n",
    "        if (word in x[0]):\n",
    "            return (False)\n",
    "    return ABSTAIN\n",
    "\n",
    "all_lfs.append(includes_physical_words)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [],
   "source": [
    "# includes_genetic_words\n",
    "# If one of the chemicals is in a list of genetic terms such as reductase, dna, gene, we label FALSE\n",
    "genetic_terms = [\"reductase\", \"dna\", \"gene\", \"allele\", \"locus\", \"genotype\", \"phenotype\", \"dominant\", \"recessive\", \"additive\", \"phenoset\",\n",
    "    \"diallelic\", \"multiallelic\", \"polyallelic\", \"monomorphic\", \"monoallelic\", \"polymorphism\", \"complex\", \"trait\", \"multifactorial\",\n",
    "    \"polygenic\", \"monogenic\", \"mixed model\", \"transmission probability\", \"transition probability\", \"epistasis\", \"interaction\", \"pleiotropy\",\n",
    "    \"quantitative trait locus\", \"probit\", \"logit\", \"penetrance\", \"transformation\", \"scale of measurement\", \"identity by descent\", \"identity in state\",\n",
    "    \"haplotype\", \"phase\", \"multilocus\", \"genotype\", \"allelic\", \"association\", \"linkage\", \"disequilibrium\", \"gametic\", \"rna\"]\n",
    "@labeling_function()\n",
    "def includes_genetic_words(x):\n",
    "    for word in genetic_terms:\n",
    "        if (word in x[0]):\n",
    "            return False\n",
    "    return ABSTAIN\n",
    "\n",
    "all_lfs.append(includes_genetic_words)\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [],
   "source": [
    "# includes_structural_words\n",
    "# If there are common structural terms, we label FALSE\n",
    "structural_terms = [\"loop\", \"sequence\", \"encodes\", \"code\", \"codon\", \"dna\", \"rna\", \"pair\", \"group\", \"active site\",\n",
    "    \"bond\", \"chain\", \"gene\", \"structure\", \"structural\", \"encoding\", \"cdna\", \"cluster\"]\n",
    "@labeling_function()\n",
    "def includes_structural_words(x):\n",
    "    for word in structural_terms:\n",
    "        if (word in x[0]):\n",
    "            return (False)\n",
    "    return ABSTAIN\n",
    "\n",
    "all_lfs.append(includes_structural_words)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [],
   "source": [
    "# includes_general_chemical_words\n",
    "# If any of the words is in a list of common chemical terms such as amino acid, sugar, adenosine, amide, adenine, etc..., we label FALSE\n",
    "# https://www.hach.com/chemGlossary   \n",
    "chemical_terms = [\"amino acid\", \"sugar\", \"adenosine\", \"amide\", \"adenine\", \"algaecide\", \"amines\", \"base\", \"biocides\",\n",
    "    \"clarifier\", \"hydrocarbon\", \"molecule\", \"nutrients\", \"polymer\", \"peptide\", \"polypeptide\",\n",
    "    \"tag\", \"functional\", \"electron\", \"cofactor\", \"gas\"]\n",
    "@labeling_function()\n",
    "def includes_general_chemical_words(x):\n",
    "    for word in chemical_terms:\n",
    "        if (word in x[0]):\n",
    "            return (False)\n",
    "    return ABSTAIN\n",
    "\n",
    "all_lfs.append(includes_general_chemical_words)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [],
   "source": [
    "# includes_functional_group\n",
    "# If there is a close mention of a functional chemical group, we label FALSE (as it is more likely descriptive of a structure than of a reaction)\n",
    "# https://www.masterorganicchemistry.com/2010/10/06/functional-groups-organic-chemistry/\n",
    "common_functional_groups = [\"alkane\", \"alkene\", \"alkyne\", \"benzene ring\", \"phenyl\", \"amine\", \"alcohol\", \"ether\", \"alkyl halide\", \"thiol\",\n",
    "    \"aldehyde\", \"ketone\", \"ester\", \"carboxylic acid\", \"amide\", \"nitrile\", \"epoxide\", \"disulfide\", \"imine\", \"acid chloride\", \"anhydride\", \"nitro\",\n",
    "    \"sulfide\", \"thioether\", \"group\", \"functional\"]\n",
    "@labeling_function()\n",
    "def includes_functional_group(x):\n",
    "    for group in common_functional_groups:\n",
    "        if (group in x[0]):\n",
    "            return False\n",
    "    return ABSTAIN\n",
    "\n",
    "all_lfs.append(includes_functional_group)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [],
   "source": [
    "# includes_amino_acid\n",
    "# If there is an amino acid mentioned, we label FALSE\n",
    "amino_acids = ['val', 'ile', 'leu', 'glu', 'gln', \\\n",
    "    'asp', 'asn' 'his' 'trp', 'phe', 'tyr',    \\\n",
    "    'arg', 'lys', 'ser', 'thr', 'met', 'ala',    \\\n",
    "    'gly', 'pro', 'cys', \"amino\"]\n",
    "@labeling_function()\n",
    "def includes_amino_acid(x):\n",
    "    for group in amino_acids:\n",
    "        if (group in x[0]):\n",
    "            return False\n",
    "    return ABSTAIN\n",
    "\n",
    "all_lfs.append(includes_amino_acid)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [],
   "source": [
    "# includes_paper_artifacts\n",
    "# If there are common words from paper headers/footers, we label FALSE\n",
    "common_terms = [\"university\", \"univ\", \"pharma\", \"avenue\", \"street\", \"road\", \"department\", \"usa\", \"reference\", \"ref\",\n",
    "    \"keyword\", \"article\", \"http\", \"png\", \"jpg\", \"journal\", \"(20\", \"(19\"]\n",
    "@labeling_function()\n",
    "def includes_paper_artifacts(x):\n",
    "    for word in common_terms:\n",
    "        if (word in x[0]):\n",
    "            return (False)\n",
    "    return ABSTAIN\n",
    "\n",
    "all_lfs.append(includes_paper_artifacts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [],
   "source": [
    "# includes_no_terms\n",
    "# If there are words stopping or with no change, we label FALSE\n",
    "common_no_terms = [\"inactivated\", \"unaffected\", \"inactive\", \"inactivates\", \"stops\", \"prevent\", \"inhibit\",\n",
    "    \"denature\", \"block\"]\n",
    "@labeling_function()\n",
    "def includes_no_terms(x):\n",
    "    for word in common_no_terms:\n",
    "        if (word in x[0]):\n",
    "            return (False)\n",
    "    return ABSTAIN\n",
    "\n",
    "all_lfs.append(includes_no_terms)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [],
   "source": [
    "# puts chemicals separated by or for regex structures\n",
    "def helper_sep_chems_with_or(chemicals):\n",
    "    final = \"\"\n",
    "    for chem in chemicals:\n",
    "        if (final == \"\"):\n",
    "            final += re.escape(chem)\n",
    "        else:\n",
    "            final += \"|\" + re.escape(chem)\n",
    "    return final"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [],
   "source": [
    "# structure_adjacent_mentions\n",
    "# If the chemicals are adjacent, we label FALSE\n",
    "@labeling_function()\n",
    "def structure_adjacent_mentions(x):\n",
    "    chemicals = helper_sep_chems_with_or(x[1])\n",
    "    structure = \"(\" + chemicals + \") (\" + chemicals + \")\"\n",
    "    if (re.search(structure, x[0])):\n",
    "        return False\n",
    "    return ABSTAIN\n",
    "\n",
    "all_lfs.append(structure_adjacent_mentions)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [],
   "source": [
    "# structure_sep_or\n",
    "# If the chemicals are separated by or, we label FALSE\n",
    "@labeling_function()\n",
    "def structure_sep_or(x):\n",
    "    chemicals = helper_sep_chems_with_or(x[1])\n",
    "    structure = \"(\" + chemicals + \") or (\" + chemicals + \")\"\n",
    "    if (re.search(structure, x[0])):\n",
    "        return False\n",
    "    return ABSTAIN\n",
    "\n",
    "all_lfs.append(structure_sep_or)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [],
   "source": [
    "# structure_sep_and\n",
    "# If the chemicals are separated by and, we label FALSE\n",
    "@labeling_function()\n",
    "def structure_sep_and(x):\n",
    "    chemicals = helper_sep_chems_with_or(x[1])\n",
    "    structure = \"(\" + chemicals + \") and (\" + chemicals + \")\"\n",
    "    if (re.search(structure, x[0])):\n",
    "        return False\n",
    "    return ABSTAIN\n",
    "\n",
    "all_lfs.append(structure_sep_and)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [],
   "source": [
    "# structure_sep_comma\n",
    "# If the chemicals are separated by a comma, we label FALSE\n",
    "@labeling_function()\n",
    "def structure_sep_comma(x):\n",
    "    chemicals = helper_sep_chems_with_or(x[1])\n",
    "    structure = \"(\" + chemicals + \"), (\" + chemicals + \")\"\n",
    "    if (re.search(structure, x[0])):\n",
    "        return False\n",
    "    return ABSTAIN\n",
    "\n",
    "all_lfs.append(structure_sep_comma)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [],
   "source": [
    "# LF_sep_via\n",
    "# If the chemicals are separated by via, in, etc, we label FALSE\n",
    "via_terms = \"via|in\"\n",
    "@labeling_function()\n",
    "def structure_sep_via(x):\n",
    "    chemicals = helper_sep_chems_with_or(x[1])\n",
    "    structure = \"(\" + chemicals + \") (\" + via_terms + \") (\" + chemicals + \")\"\n",
    "    if (re.search(structure, x[0])):\n",
    "        return False\n",
    "    return ABSTAIN\n",
    "\n",
    "all_lfs.append(structure_sep_via)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [],
   "source": [
    "# structure_sep_sym\n",
    "# If the chemicals are separated by a single character, we label FALSE  \n",
    "@labeling_function()\n",
    "def structure_sep_sym(x):\n",
    "    chemicals = helper_sep_chems_with_or(x[1])\n",
    "    structure = \"(\" + chemicals + \").(\" + chemicals + \")\"\n",
    "    if (re.search(structure, x[0])):\n",
    "        return False\n",
    "    return ABSTAIN\n",
    "\n",
    "all_lfs.append(structure_sep_via)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [],
   "source": [
    "# structure_followed_by_ase\n",
    "# If one of the chemicals is followed by a word that ends with -ase, we label FALSE\n",
    "@labeling_function()\n",
    "def structure_followed_by_ase(x):\n",
    "    chemicals = helper_sep_chems_with_or(x[1])\n",
    "    structure = \"(\" + chemicals + r\") \\w*ase\\b\"\n",
    "    if (re.search(structure, x[0])):\n",
    "        return False\n",
    "    return ABSTAIN\n",
    "\n",
    "all_lfs.append(structure_followed_by_ase)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [],
   "source": [
    "# structure_followed_by_noun\n",
    "# If the first chemical is followed by a noun, we label FALSE\n",
    "@labeling_function()\n",
    "def structure_followed_by_noun(x):\n",
    "    sentence = x[0].replace(',', '')\n",
    "    chemicals = x[1]\n",
    "    chem_index = len(sentence)\n",
    "    chem = \"\"\n",
    "    for chemical in chemicals:\n",
    "        curr_index = sentence.find(chemical)\n",
    "        if (curr_index < chem_index and curr_index != -1):\n",
    "            chem_index = curr_index\n",
    "            chem = chemical\n",
    "    if (chem == \"\"):\n",
    "        return ABSTAIN\n",
    "    structure = \"(\" + chem + r\") \\w+\"\n",
    "    for match in re.finditer(structure, sentence):\n",
    "        if (nltk.pos_tag([match.group(0).split()[1]])[0][1] == \"NN\"):\n",
    "            return False\n",
    "    return ABSTAIN\n",
    "\n",
    "all_lfs.append(structure_followed_by_noun)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "False"
      ]
     },
     "execution_count": 102,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "structure_followed_by_noun([\"cassie cs ate\", [\"cs\", \"sdfd\"]])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [],
   "source": [
    "# structure_sep_verb\n",
    "# If the chemicals are separated by a verb, we label TRUE\n",
    "@labeling_function()\n",
    "def structure_sep_verb(x):\n",
    "    sentence = x[0].replace(',', '')\n",
    "    chemicals = helper_sep_chems_with_or(x[1])\n",
    "    if (chemicals == \"\"):\n",
    "        return ABSTAIN\n",
    "    structure = \"(\" + chemicals + r\") \\w+ (\" + chemicals + \")\"\n",
    "    for match in re.finditer(structure, sentence):\n",
    "        if (re.match(r\"(\\bVB|NNS)\", nltk.pos_tag([match.group(0).split()[1]])[0][1])):\n",
    "            return True\n",
    "    return ABSTAIN\n",
    "\n",
    "all_lfs.append(structure_sep_verb)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [],
   "source": [
    "# structure_sep_adverb\n",
    "# If the chemicals are separated by a adverb, we label FALSE\n",
    "@labeling_function()\n",
    "def structure_sep_adverb(x):\n",
    "    sentence = x[0].replace(',', '')\n",
    "    chemicals = helper_sep_chems_with_or(x[1])\n",
    "    if (chemicals == \"\"):\n",
    "        return ABSTAIN\n",
    "    structure = \"(\" + chemicals + r\") \\w+ (\" + chemicals + \")\"\n",
    "    for match in re.finditer(structure, sentence):\n",
    "        if (re.match(r\"(\\bRB)\", nltk.pos_tag([match.group(0).split()[1]])[0][1])):\n",
    "            return False\n",
    "    return ABSTAIN\n",
    "\n",
    "all_lfs.append(structure_sep_adverb)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "False"
      ]
     },
     "execution_count": 105,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "structure_sep_adverb([\"cassie crazily night away\", [\"cassie\", \"night\"]])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [],
   "source": [
    "# includes_oxidation_words\n",
    "# If the sentence contains oxidation words, we label True\n",
    "oxidation_terms = \"oxidiz|oxidis|redox|reduc|rust|corrod|oxygen\"\n",
    "@labeling_function()\n",
    "def includes_oxidation_words(x):\n",
    "    structure = \"(\" + oxidation_terms + \")\"\n",
    "    if (re.search(structure, x[0])):\n",
    "        return True\n",
    "    return ABSTAIN\n",
    "\n",
    "all_lfs.append(includes_oxidation_words)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [],
   "source": [
    "# includes_combustion_words\n",
    "# If the sentence contains combustion words, we label True\n",
    "combustion_terms = \"combust|burn|explod|gas-form\"\n",
    "@labeling_function()\n",
    "def includes_combustion_words(x):\n",
    "    structure = \"(\" + combustion_terms + \")\"\n",
    "    if (re.search(structure, x[0])):\n",
    "        return True\n",
    "    return ABSTAIN\n",
    "\n",
    "all_lfs.append(includes_combustion_words)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [],
   "source": [
    "# includes_neutralization_words\n",
    "# If the sentence contains neutralization words, we label True\n",
    "neutralization_terms = \"neutraliz|titrat|buffer|gas-form\"\n",
    "@labeling_function()\n",
    "def includes_neutralization_words(x):\n",
    "    structure = \"(\" + neutralization_terms + \")\"\n",
    "    if (re.search(structure, x[0])):\n",
    "        return True\n",
    "    return ABSTAIN\n",
    "\n",
    "all_lfs.append(includes_neutralization_words)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [],
   "source": [
    "# LF_includes_catalyze_words\n",
    "# If the sentence contains catalyze words, we label True\n",
    "# catalyze_words = [\"catalyze\", \"catalyst\", \"catalyse\", \"catalysing\", \"catalyzing\"]\n",
    "# @labeling_function()\n",
    "# def includes_catalyze_words(x):\n",
    "#     for word in catalyze_words:\n",
    "#         if(word in x[0]):\n",
    "#             return True\n",
    "#     return ABSTAIN"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [],
   "source": [
    "# includes_combination_words\n",
    "# If the sentence contains combination words, we label True\n",
    "combination_terms = \"combin\"\n",
    "@labeling_function()\n",
    "def includes_combination_words(x):\n",
    "    structure = \"(\" + combination_terms + \")\"\n",
    "    if (re.search(structure, x[0])):\n",
    "        return True\n",
    "    return ABSTAIN\n",
    "\n",
    "all_lfs.append(includes_combination_words)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {},
   "outputs": [],
   "source": [
    "# includes_decomposition_words\n",
    "# If the sentence contains decomposition words, we label True\n",
    "decomposition_terms = \"decompos\"\n",
    "@labeling_function()\n",
    "def includes_decomposition_words(x):\n",
    "    structure = \"(\" + decomposition_terms + \")\"\n",
    "    if (re.search(structure, x[0])):\n",
    "        return True\n",
    "    return ABSTAIN\n",
    "\n",
    "all_lfs.append(includes_decomposition_words)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [],
   "source": [
    "# includes_replacement_words\n",
    "# If the sentence contains decomposition words, we label True\n",
    "replacement_terms = \"replac\"\n",
    "@labeling_function()\n",
    "def includes_replacement_words(x):\n",
    "    structure = \"(\" + replacement_terms + \")\"\n",
    "    if (re.search(structure, x[0])):\n",
    "        return True\n",
    "    return ABSTAIN\n",
    "\n",
    "all_lfs.append(includes_replacement_words)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {},
   "outputs": [],
   "source": [
    "# includes_reaction_words\n",
    "# If the sentence contains reactions words, we label True\n",
    "reaction_terms_broad = \"conver|yield|produc|mak|creat|synthesiz|synthesis|transform|ferment|break|displac|exchang\" + \\\n",
    "                 \"|precipit|transfer|through|produc|activat|revers|form|ation|metaboliz|metabolis|generat|hydroly\" + \\\n",
    "                 \"|methyl|result|modif|revers|from\"\n",
    "@labeling_function()\n",
    "def includes_reaction_words(x):\n",
    "    structure = \"(\" + reaction_terms_broad + \")\"\n",
    "    if (re.search(structure, x[0])):\n",
    "        return True\n",
    "    return ABSTAIN\n",
    "\n",
    "all_lfs.append(includes_reaction_words)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {},
   "outputs": [],
   "source": [
    "# includes_creation_words\n",
    "# If the sentence contains creation words, we label True\n",
    "creation_terms = \"produc|creat|synthesis|form\"\n",
    "\n",
    "@labeling_function()\n",
    "def includes_creation_words(x):\n",
    "    sentence = x[0].replace(',', '')\n",
    "    chemicals = helper_sep_chems_with_or(x[1])\n",
    "    structure = \"(\" + chemicals + \") (\" + creation_terms + \")\"\n",
    "    if (re.search(structure, x[0])):\n",
    "        return True\n",
    "    return ABSTAIN\n",
    "\n",
    "all_lfs.append(includes_creation_words)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {},
   "outputs": [],
   "source": [
    "# included_impact_terms\n",
    "# If the sentence contains impact words, we label True\n",
    "impact_terms = \"caus|lead|result\"\n",
    "\n",
    "@labeling_function()\n",
    "def includes_impact_words(x):\n",
    "    structure = \"(\" + impact_terms + \")\"\n",
    "    if (re.search(structure, x[0])):\n",
    "        return True\n",
    "    return ABSTAIN\n",
    "\n",
    "all_lfs.append(includes_impact_words)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {},
   "outputs": [],
   "source": [
    "# includes_react\n",
    "# If the sentence contains react, we label True\n",
    "@labeling_function()\n",
    "def includes_react(x):\n",
    "    if(\"react\" in x[0]):\n",
    "        return True\n",
    "    return ABSTAIN\n",
    "\n",
    "all_lfs.append(includes_react)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [],
   "source": [
    "# includes_react_sym\n",
    "# If the sentence contains react, we label True\n",
    "@labeling_function()\n",
    "def includes_react_sym(x):\n",
    "    if(\"-->\" in x[0]):\n",
    "        return True\n",
    "    return ABSTAIN\n",
    "\n",
    "all_lfs.append(includes_react_sym)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {},
   "outputs": [],
   "source": [
    "# includes_reaction_component_words\n",
    "# If the sentence contains reaction components words, we label True\n",
    "reaction_component_terms = [\"substrate\", \"product\", \"reactant\", \"step\"]\n",
    "@labeling_function()\n",
    "def includes_reaction_component_words(x):\n",
    "    for word in reaction_component_terms:\n",
    "        if(word in x[0]):\n",
    "            return True\n",
    "    return ABSTAIN\n",
    "\n",
    "all_lfs.append(includes_reaction_component_words)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {},
   "outputs": [],
   "source": [
    "# includes_comparison_words\n",
    "# If the sentence contains comparison words, we label False\n",
    "comparison_terms = \"similar|more|greater|less|increas|decreas|compar|differ|relativ|better|time|than\"\n",
    "@labeling_function()\n",
    "def includes_comparison_words(x):\n",
    "    structure = \"(\" + comparison_terms + \")\"\n",
    "    if (re.search(structure, x[0])):\n",
    "        return False\n",
    "    return ABSTAIN\n",
    "\n",
    "all_lfs.append(includes_comparison_words)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {},
   "outputs": [],
   "source": [
    "# includes_concentration\n",
    "# If the sentence contains react, we label True\n",
    "@labeling_function()\n",
    "def includes_concentration(x):\n",
    "    if(\"concentration\" in x[0]):\n",
    "        return True\n",
    "    return ABSTAIN\n",
    "\n",
    "all_lfs.append(includes_concentration)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {},
   "outputs": [],
   "source": [
    "# includes_measure_words\n",
    "# If the sentence contains measure words, we label False\n",
    "measure_terms = [\"high\", \"low\", \"ph\", \"stability\", \"corelated\", \"more\", \"less\", \"level\", \"degree\", \"time\",\n",
    "                \"measure\"]\n",
    "@labeling_function()\n",
    "def includes_measure_words(x):\n",
    "    for word in measure_terms:\n",
    "        if(word in x[0]):\n",
    "            return False\n",
    "    return ABSTAIN\n",
    "\n",
    "all_lfs.append(includes_measure_words)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {},
   "outputs": [],
   "source": [
    "# includes_experiment_words\n",
    "# If the sentence contains experiement words, we label False\n",
    "experiment_terms = [\"mice\", \"cell\", \"mouse\", \"ovary\", \"male\", \"female\", \"animal\", \"study\", \"method\",\n",
    "    \"test\", \"treat\", \"protection\", \"brain\", \"nerve\", \"human\", \"tissue\", \"fetal\", \"vitro\", \"studies\",\n",
    "    \"membrane\", \"strain\", \"mutant\", \"regulate\", \"dependent\", \"drug\", \"therapy\", \"oral\", \"test\", \"autoantigen\"]\n",
    "@labeling_function()\n",
    "def includes_experiment_words(x):\n",
    "    for word in experiment_terms:\n",
    "        if(word in x[0]):\n",
    "            return False\n",
    "    return ABSTAIN\n",
    "\n",
    "all_lfs.append(includes_experiment_words)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {},
   "outputs": [],
   "source": [
    "# structure_next_to_conversion_words\n",
    "# If the sentence contains to, from, into, etc., we label True\n",
    "conversion_terms_next = \"to|from|into|becom|became|by\"\n",
    "@labeling_function()\n",
    "def structure_next_to_conversion_words(x):\n",
    "    sentence = x[0].replace(',', '')\n",
    "    chemicals = helper_sep_chems_with_or(x[1])\n",
    "    structure = \"((\" + chemicals + r\") (\" + conversion_terms_next + r\")\\b|((\" + conversion_terms_next + r\")\\b (\" + chemicals + \")))\"\n",
    "    if (re.search(structure, sentence)):\n",
    "            return True\n",
    "    return ABSTAIN\n",
    "\n",
    "all_lfs.append(structure_next_to_conversion_words)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {},
   "outputs": [],
   "source": [
    "# structure_sep_conversion_words\n",
    "# If the sentence contains to, from, into, etc., we label True\n",
    "conversion_terms_sep = \"to|into\"\n",
    "@labeling_function()\n",
    "def structure_sep_conversion_words(x):\n",
    "    sentence = x[0].replace(',', '')\n",
    "    chemicals = helper_sep_chems_with_or(x[1])\n",
    "    structure = \"(\" + chemicals + r\") (\" + conversion_terms_sep + r\") (\" + chemicals + \")\"\n",
    "    if (re.search(structure, sentence)):\n",
    "            return True\n",
    "    return ABSTAIN\n",
    "\n",
    "all_lfs.append(structure_sep_conversion_words)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {},
   "outputs": [],
   "source": [
    "# structure_conversion_by\n",
    "# If the sentence contains to, from, into, etc., we label True\n",
    "conversion_terms_by = \"to|from\"\n",
    "@labeling_function()\n",
    "def structure_conversion_by(x):\n",
    "    sentence = x[0].replace(',', '')\n",
    "    chemicals = helper_sep_chems_with_or(x[1])\n",
    "    structure = \"(\" + conversion_terms_by + \") (\" + chemicals + \") by\"\n",
    "    if (re.search(structure, sentence)):\n",
    "            return True\n",
    "    return ABSTAIN\n",
    "\n",
    "all_lfs.append(structure_conversion_by)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "metadata": {},
   "outputs": [],
   "source": [
    "# structure_conversion_of\n",
    "# If the sentence contains to, from, into, etc., we label True\n",
    "conversion_terms_of = \"ation of|sion of|ism of\"\n",
    "@labeling_function()\n",
    "def structure_conversion_of(x):\n",
    "    sentence = x[0].replace(',', '')\n",
    "    chemicals = helper_sep_chems_with_or(x[1])\n",
    "    structure = \"((\" + conversion_terms_of + \") (\" + chemicals + \")|of (\" + chemicals + \") (\" + conversion_terms_sep + \"))\"\n",
    "    if (re.search(structure, sentence)):\n",
    "            return True\n",
    "    return ABSTAIN\n",
    "\n",
    "all_lfs.append(structure_conversion_of)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 127,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "includes_impact_words([\"the enzyme cyclo-oxygenase catalyses the oxygenation of arachidonic acid, leading to the formation of prostaglandins\", ['prostaglandins']])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "metadata": {},
   "outputs": [],
   "source": [
    "# includes_one_chem\n",
    "# If the sentence contains one identified chemical, we label False\n",
    "@labeling_function()\n",
    "def includes_one_chem(x):\n",
    "    if(len(x[1]) == 1):\n",
    "            return False\n",
    "    return ABSTAIN\n",
    "\n",
    "all_lfs.append(includes_one_chem)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {},
   "outputs": [],
   "source": [
    "# # includes_more_than_one_chem\n",
    "# # If the sentence contains more than one identified chemical, we label True\n",
    "# @labeling_function()\n",
    "# def includes_more_than_one_chem(x):\n",
    "#     if(len(x[1]) > 1):\n",
    "#             return True\n",
    "#     return ABSTAIN\n",
    "\n",
    "# all_lfs.append(includes_more_than_one_chem)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Phil's version cleaned\n",
    "# structure_jtsui_pattern_1\n",
    "# MODIFIED STRUCTURE AND CHANGED ADJ NUM TO 0-3\n",
    "# If part of the sentence contains the specific structure\n",
    "# [trigger1] <0,3> chemical [transition] <0,3> chemical, we label True\n",
    "\n",
    "TRANS = \"from|to|into|by|are|yield\"\n",
    "TRIG1 = \"phosphoryl|condens|hydrolys|metabol|reduc|conver|produc|form|oxid|transform|bioconver|synthes|react|interconver\"\n",
    "TRANS_p = \"(\" + TRANS + \")\"\n",
    "TRIG1_p = \"(\" + TRIG1 + \")\"\n",
    "\n",
    "@labeling_function()\n",
    "def structure_jtsui_pattern_1(x):\n",
    "    chemicals = helper_sep_chems_with_or(x[1])\n",
    "    chemicals_p = \"(\" + chemicals + \")\"\n",
    "\n",
    "    structure = r\"\\b\" + r\"{}\".format(TRIG1_p) + r\"\\w*(\\s\\w*){0,3}\\s\" + r\"{}\".format(chemicals_p) + r\"\\s\" + r\"{}\".format(TRANS_p) + r\"(\\s\\w*){0,3}\\s\" + r\"{}\".format(chemicals_p)\n",
    "\n",
    "    if (re.search(structure, x[0])):\n",
    "        return True\n",
    "    return ABSTAIN"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Phil's version cleaned\n",
    "# structure_jtsui_pattern_1\n",
    "# MODIFIED STRUCTURE AND CHANGED ADJ NUM TO 0-3\n",
    "# If part of the sentence contains the specific structure\n",
    "# [trigger1] <0,3> chemical [transition] <0,3> chemical, we label True\n",
    "\n",
    "TRANS = \"from|to|into|by|are|yield\"\n",
    "TRIG1 = \"phosphoryl|condens|hydrolys|metabol|reduc|conver|produc|form|oxid|transform|bioconver|synthes|react|interconver\"\n",
    "TRANS_p = \"(\" + TRANS + \")\"\n",
    "TRIG1_p = \"(\" + TRIG1 + \")\"\n",
    "\n",
    "@labeling_function()\n",
    "def structure_jtsui_pattern_1_copy(x):\n",
    "    chemicals = helper_sep_chems_with_or(x[1])\n",
    "    chemicals_p = \"(\" + chemicals + \")\"\n",
    "\n",
    "    structure = r\"\\b\" + r\"{}\".format(TRIG1_p) + r\"\\w*(\\s\\w*){0,3}\\s\" + r\"{}\".format(chemicals_p) + r\"\\s\" + r\"{}\".format(TRANS_p) + r\"(\\s\\w*){0,3}\\s\" + r\"{}\".format(chemicals_p)\n",
    "\n",
    "    if (re.search(structure, x[0])):\n",
    "        return True\n",
    "    return ABSTAIN"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### final regex LFs ###"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "metadata": {},
   "outputs": [],
   "source": [
    "# PHIL'S VERSIONS\n",
    "# structure_jtsui_pattern_2\n",
    "# If part of the sentence contains the specific structure\n",
    "# chemical <0,1> [trigger2] <0,1> [transition] <0,1> chemical, we label True\n",
    "TRIG2 = 'conver|oxid|produc|interconver'\n",
    "TRIG2_p = \"(\" + TRIG2 + \")\"\n",
    "@labeling_function()\n",
    "def structure_jtsui_pattern_2(x):\n",
    "    chemicals = helper_sep_chems_with_or(x[1])\n",
    "    chemicals_p = \"(\" + chemicals + \")\"\n",
    "\n",
    "#                           chemical                   <0,1> space             [trigger2]         <0,1>  space               [transition]           <0,1>  space         chemical          \n",
    "    structure = r\"\\b\" + r\"{}\".format(chemicals_p) + r\"(\\s\\w*){0,1}\\s\\w*\" + r\"{}\".format(TRIG2_p) + r\"\\w*(\\s\\w*){0,1}\\s\" + r\"{}\".format(TRANS_p) + r\"(\\s\\w*){0,1}\\s\" + r\"{}\".format(chemicals_p)\n",
    "\n",
    "    if (re.search(structure, x[0])):\n",
    "        return True\n",
    "    return ABSTAIN\n",
    "\n",
    "\n",
    "# structure_jtsui_pattern_3\n",
    "# If part of the sentence contains the specific structure\n",
    "# chemical [trigger3] <0,1> chemical, we label True\n",
    "TRIG3 = 'yield|generat'\n",
    "TRIG3_p = \"(\" + TRIG3 + \")\"\n",
    "@labeling_function()\n",
    "def structure_jtsui_pattern_3(x):\n",
    "\n",
    "    chemicals = helper_sep_chems_with_or(x[1])\n",
    "    chemicals_p = \"(\" + chemicals + \")\"\n",
    "#                           chemical               space             [trigger3]         <0,1>  space             chemical          \n",
    "    structure = r\"\\b\" + r\"{}\".format(chemicals_p) + r\"\\s\\w*\" + r\"{}\".format(TRIG3_p) + r\"\\w*(\\s\\w*){0,1}\\s\" + r\"{}\".format(chemicals_p)\n",
    "\n",
    "    if (re.search(structure, x[0])):\n",
    "        return True\n",
    "    return ABSTAIN\n",
    "\n",
    "\n",
    "\n",
    "# structure_jtsui_pattern_4\n",
    "# If part of the sentence contains the specific structure\n",
    "# [trigger4] <0,1> chemical, we label True\n",
    "TRIG4 = 'conver|interconver'\n",
    "TRIG4_p = \"(\" + TRIG4 + \")\"\n",
    "@labeling_function()\n",
    "def structure_jtsui_pattern_4(x):\n",
    "\n",
    "    chemicals = helper_sep_chems_with_or(x[1])\n",
    "    chemicals_p = \"(\" + chemicals + \")\"\n",
    "\n",
    "    structure = r\"\\b\\w*\" + r\"{}\".format(TRIG4_p) + r\"\\w*(\\s\\w*){0,1}\\s\" + r\"{}\".format(chemicals_p) + r\"\\b\"\n",
    "\n",
    "    if (re.search(structure, x[0])):\n",
    "        return True\n",
    "    return ABSTAIN\n",
    "\n",
    "\n",
    "# structure_jtsui_pattern_5\n",
    "# If part of the sentence contains the specific structure\n",
    "# chemical [transition5] <0,1> [trigger5] <0,1> chemical, we label True\n",
    "TRIG5 = 'produc|metaboli'\n",
    "TRIG5_p = \"(\" + TRIG5 + \")\"\n",
    "TRANS5 = 'is|are'\n",
    "TRANS5_p = \"(\" + TRANS5 + \")\"\n",
    "@labeling_function()\n",
    "def structure_jtsui_pattern_5(x):\n",
    "    chemicals = helper_sep_chems_with_or(x[1])\n",
    "    chemicals_p = \"(\" + chemicals + \")\"\n",
    "#                           chemical               space             [transition5]         <0,1>  space             [trigger5]             <0,1> space            chemical      \n",
    "    structure = r\"\\b\" + r\"{}\".format(chemicals_p) + r\"\\s\\w*\" + r\"{}\".format(TRANS5_p) + r\"\\w*(\\s\\w*){0,1}\\s\\w*\" + r\"{}\".format(TRIG5_p) + r\"\\w*(\\s\\w*){0,1}\\s\" + r\"{}\".format(chemicals_p)\n",
    "\n",
    "    if (re.search(structure, x[0])):\n",
    "        return True\n",
    "    return ABSTAIN"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "metadata": {},
   "outputs": [],
   "source": [
    "# PHIL'S VERSIONS\n",
    "# structure_jtsui_pattern_2\n",
    "# If part of the sentence contains the specific structure\n",
    "# chemical <0,1> [trigger2] <0,1> [transition] <0,1> chemical, we label True\n",
    "TRIG2 = 'conver|oxid|produc|interconver'\n",
    "TRIG2_p = \"(\" + TRIG2 + \")\"\n",
    "@labeling_function()\n",
    "def structure_jtsui_pattern_2_copy(x):\n",
    "    chemicals = helper_sep_chems_with_or(x[1])\n",
    "    chemicals_p = \"(\" + chemicals + \")\"\n",
    "\n",
    "#                           chemical                   <0,1> space             [trigger2]         <0,1>  space               [transition]           <0,1>  space         chemical          \n",
    "    structure = r\"\\b\" + r\"{}\".format(chemicals_p) + r\"(\\s\\w*){0,1}\\s\\w*\" + r\"{}\".format(TRIG2_p) + r\"\\w*(\\s\\w*){0,1}\\s\" + r\"{}\".format(TRANS_p) + r\"(\\s\\w*){0,1}\\s\" + r\"{}\".format(chemicals_p)\n",
    "\n",
    "    if (re.search(structure, x[0])):\n",
    "        return True\n",
    "    return ABSTAIN\n",
    "\n",
    "\n",
    "# structure_jtsui_pattern_3\n",
    "# If part of the sentence contains the specific structure\n",
    "# chemical [trigger3] <0,1> chemical, we label True\n",
    "TRIG3 = 'yield|generat'\n",
    "TRIG3_p = \"(\" + TRIG3 + \")\"\n",
    "@labeling_function()\n",
    "def structure_jtsui_pattern_3_copy(x):\n",
    "\n",
    "    chemicals = helper_sep_chems_with_or(x[1])\n",
    "    chemicals_p = \"(\" + chemicals + \")\"\n",
    "#                           chemical               space             [trigger3]         <0,1>  space             chemical          \n",
    "    structure = r\"\\b\" + r\"{}\".format(chemicals_p) + r\"\\s\\w*\" + r\"{}\".format(TRIG3_p) + r\"\\w*(\\s\\w*){0,1}\\s\" + r\"{}\".format(chemicals_p)\n",
    "\n",
    "    if (re.search(structure, x[0])):\n",
    "        return True\n",
    "    return ABSTAIN\n",
    "\n",
    "\n",
    "\n",
    "# structure_jtsui_pattern_4\n",
    "# If part of the sentence contains the specific structure\n",
    "# [trigger4] <0,1> chemical, we label True\n",
    "TRIG4 = 'conver|interconver'\n",
    "TRIG4_p = \"(\" + TRIG4 + \")\"\n",
    "@labeling_function()\n",
    "def structure_jtsui_pattern_4_copy(x):\n",
    "\n",
    "    chemicals = helper_sep_chems_with_or(x[1])\n",
    "    chemicals_p = \"(\" + chemicals + \")\"\n",
    "\n",
    "    structure = r\"\\b\\w*\" + r\"{}\".format(TRIG4_p) + r\"\\w*(\\s\\w*){0,1}\\s\" + r\"{}\".format(chemicals_p) + r\"\\b\"\n",
    "\n",
    "    if (re.search(structure, x[0])):\n",
    "        return True\n",
    "    return ABSTAIN\n",
    "\n",
    "\n",
    "# structure_jtsui_pattern_5\n",
    "# If part of the sentence contains the specific structure\n",
    "# chemical [transition5] <0,1> [trigger5] <0,1> chemical, we label True\n",
    "TRIG5 = 'produc|metaboli'\n",
    "TRIG5_p = \"(\" + TRIG5 + \")\"\n",
    "TRANS5 = 'is|are'\n",
    "TRANS5_p = \"(\" + TRANS5 + \")\"\n",
    "@labeling_function()\n",
    "def structure_jtsui_pattern_5_copy(x):\n",
    "    chemicals = helper_sep_chems_with_or(x[1])\n",
    "    chemicals_p = \"(\" + chemicals + \")\"\n",
    "#                           chemical               space             [transition5]         <0,1>  space             [trigger5]             <0,1> space            chemical      \n",
    "    structure = r\"\\b\" + r\"{}\".format(chemicals_p) + r\"\\s\\w*\" + r\"{}\".format(TRANS5_p) + r\"\\w*(\\s\\w*){0,1}\\s\\w*\" + r\"{}\".format(TRIG5_p) + r\"\\w*(\\s\\w*){0,1}\\s\" + r\"{}\".format(chemicals_p)\n",
    "\n",
    "    if (re.search(structure, x[0])):\n",
    "        return True\n",
    "    return ABSTAIN"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "metadata": {},
   "outputs": [],
   "source": [
    "# includes_interconvert_sym\n",
    "# If the sentence contains the interconvert symbol, we label TRUE\n",
    "#\n",
    "# POTENTIAL CHANGE: symbol must be inbetween two chemicals?\n",
    "@labeling_function()\n",
    "def includes_interconvert_sym(x):\n",
    "    if(\"<->\" in x[0]):\n",
    "        return True\n",
    "    if(\"<-->\" in x[0]):\n",
    "        return True\n",
    "    return ABSTAIN\n",
    "\n",
    "all_lfs.append(includes_interconvert_sym)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "metadata": {},
   "outputs": [],
   "source": [
    "# LF_includes_emzyme_words\n",
    "# If the sentence contains enzyme words, we label True\n",
    "enzyme_words = [\"enzyme\", \"enzymes\"]\n",
    "@labeling_function()\n",
    "def includes_emzyze_words(x):\n",
    "    for word in enzyme_words:\n",
    "        if(word in x[0]):\n",
    "            return True\n",
    "    return ABSTAIN\n",
    "\n",
    "# LF_includes_catalyze_words\n",
    "# If the sentence contains catalyze words, we label True\n",
    "catalyze_words = [\"catalyze\", \"catalyst\", \"catalyse\", \"catalysing\", \"catalyzing\"]\n",
    "@labeling_function()\n",
    "def includes_catalyze_words(x):\n",
    "    for word in catalyze_words:\n",
    "        if(word in x[0]):\n",
    "            return True\n",
    "    return ABSTAIN\n",
    "\n",
    "# structure_followed_by_ase\n",
    "# If one of the chemicals is followed by a word that ends with -ase, we label FALSE\n",
    "@labeling_function()\n",
    "def structure_followed_by_ase(x):\n",
    "    chemicals = helper_sep_chems_with_or(x[1])\n",
    "    structure = \"(\" + chemicals + r\") \\w*ase\\b\"\n",
    "    if (re.search(structure, x[0])):\n",
    "        return False\n",
    "    return ABSTAIN\n",
    "\n",
    "all_lfs.append(structure_followed_by_ase)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {},
   "outputs": [],
   "source": [
    "# includes_reaction_words\n",
    "# If the sentence contains reactions words, we label True\n",
    "reaction_terms_specific = \"conver|yield|synthesiz|synthesis|oxid|reduc|phosphorylat\" + \\\n",
    "                 \"|form|metaboliz|metabolis|generat|hydroly\" + \\\n",
    "                 \"|methylat|brominat|aminat|dehydrat|condensat|degradat|decompos|carboxylat\"\n",
    "@labeling_function()\n",
    "def includes_reaction_words2(x):\n",
    "    structure = \"(\" + reaction_terms_specific + \")\"\n",
    "    if (re.search(structure, x[0])):\n",
    "        return True\n",
    "    return ABSTAIN\n",
    "\n",
    "all_lfs.append(includes_reaction_words2)\n",
    "\n",
    "# includes_reaction_words\n",
    "# If the sentence contains reactions words, we label True\n",
    "\n",
    "@labeling_function()\n",
    "def includes_reaction_words3(x):\n",
    "    structure = \"(\" + reaction_terms_specific + \")\"\n",
    "    if (re.search(structure, x[0])):\n",
    "        return True\n",
    "    return ABSTAIN\n",
    "\n",
    "all_lfs.append(includes_reaction_words3)\n",
    "\n",
    "# includes_reaction_words\n",
    "# If the sentence contains reactions words, we label True\n",
    "\n",
    "@labeling_function()\n",
    "def includes_reaction_words4(x):\n",
    "    structure = \"(\" + reaction_terms_specific + \")\"\n",
    "    if (re.search(structure, x[0])):\n",
    "        return True\n",
    "    return ABSTAIN\n",
    "\n",
    "all_lfs.append(includes_reaction_words4)\n",
    "\n",
    "# includes_reaction_words\n",
    "# If the sentence contains reactions words, we label True\n",
    "\n",
    "@labeling_function()\n",
    "def includes_reaction_words5(x):\n",
    "    structure = \"(\" + reaction_terms_specific + \")\"\n",
    "    if (re.search(structure, x[0])):\n",
    "        return True\n",
    "    return ABSTAIN\n",
    "\n",
    "all_lfs.append(includes_reaction_words5)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### data cleaning stuff"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [],
   "source": [
    "greek_alphabet = {\n",
    "    u'\\u0393': 'Gamma',\n",
    "    u'\\u0394': 'Delta',\n",
    "    u'\\u0398': 'Theta',\n",
    "    u'\\u039B': 'Lamda',\n",
    "    u'\\u039E': 'Xi',\n",
    "    u'\\u03A0': 'Pi',\n",
    "    u'\\u03A3': 'Sigma',\n",
    "    u'\\u03A6': 'Phi',\n",
    "    u'\\u03A7': 'Chi',\n",
    "    u'\\u03A8': 'Psi',\n",
    "    u'\\u03A9': 'Omega',\n",
    "    u'\\u03B1': 'alpha',\n",
    "    u'\\u03B2': 'beta',\n",
    "    u'\\u03B3': 'gamma',\n",
    "    u'\\u03B4': 'delta',\n",
    "    u'\\u03B5': 'epsilon',\n",
    "    u'\\u03B6': 'zeta',\n",
    "    u'\\u03B7': 'eta',\n",
    "    u'\\u03B8': 'theta',\n",
    "    u'\\u03B9': 'iota',\n",
    "    u'\\u03BA': 'kappa',\n",
    "    u'\\u03BB': 'lamda',\n",
    "    u'\\u03BC': 'mu',\n",
    "    u'\\u03BD': 'nu',\n",
    "    u'\\u03BE': 'xi',\n",
    "    u'\\u03BF': 'omicron',\n",
    "    u'\\u03C0': 'pi',\n",
    "    u'\\u03C1': 'rho',\n",
    "    u'\\u03C3': 'sigma',\n",
    "    u'\\u03C4': 'tau',\n",
    "    u'\\u03C5': 'upsilon',\n",
    "    u'\\u03C6': 'phi',\n",
    "    u'\\u03C7': 'chi',\n",
    "    u'\\u03C8': 'psi',\n",
    "    u'\\u03C9': 'omega'\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [],
   "source": [
    "def remove_greek(words):\n",
    "    for letter in greek_alphabet.keys():\n",
    "        words = words.replace(letter, greek_alphabet[letter])\n",
    "    return words"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [],
   "source": [
    "def chem_into_array(chemicals):\n",
    "    if (chemicals == \"0\"):\n",
    "        return []\n",
    "    chemicals = remove_greek(chemicals)\n",
    "    chemicals = chemicals.lower()\n",
    "    sample_chems = chemicals.replace(\"%20\", \" \")\n",
    "    sample_chems_list = sample_chems.split(\", \")\n",
    "    for index in range(len(sample_chems_list)):\n",
    "        while(sample_chems_list[index][-1] == \",\"):\n",
    "            sample_chems_list[index] = sample_chems_list[index][:-1]\n",
    "    sample_chems_list = list(set(sample_chems_list))\n",
    "    return sample_chems_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [],
   "source": [
    "def sentence_cleaned(sentence):\n",
    "    sentence = remove_greek(sentence)\n",
    "    sentence = sentence.lower()\n",
    "    sentence = sentence.replace(\".\", \"\")\n",
    "    sentence = sentence.replace(\")\", \"\")\n",
    "    sentence = sentence.replace(\"(\", \"\")\n",
    "    sentence = sentence.replace(\";\", \"\")\n",
    "    sentence = sentence.replace(\":\", \"\")\n",
    "    return sentence"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### importing data in and cleaning it (this data has truth values)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [],
   "source": [
    "# have this csv file in the same folder\n",
    "sentence_df = pd.read_csv(\"sentence_data_cleaned_csv_fixed.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sentence</th>\n",
       "      <th>chemicals</th>\n",
       "      <th>truth</th>\n",
       "      <th>substrates</th>\n",
       "      <th>products</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>The enzyme cyclo-oxygenase catalyses the oxyge...</td>\n",
       "      <td>arachidonic acid, prostaglandins</td>\n",
       "      <td>1</td>\n",
       "      <td>arachidonic, acid</td>\n",
       "      <td>prostaglandins</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Although the exact site of decarboxylation of ...</td>\n",
       "      <td>levodopa, dopamine</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>The enzyme dimethylarginine dimethylaminohydro...</td>\n",
       "      <td>dimethylarginine, arginine, citrulline, methyl...</td>\n",
       "      <td>1</td>\n",
       "      <td>arginine</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BACKGROUND AND AIMS: Glutamic acid decarboxyla...</td>\n",
       "      <td>glutamate, GABA</td>\n",
       "      <td>1</td>\n",
       "      <td>glutamate</td>\n",
       "      <td>gamma-aminobutyric, acid, GABA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Mazindane (26) was found to be a pro-drug, oxi...</td>\n",
       "      <td>Mazindane, 5-H, 5-OH, mazindol</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            sentence  \\\n",
       "0  The enzyme cyclo-oxygenase catalyses the oxyge...   \n",
       "1  Although the exact site of decarboxylation of ...   \n",
       "2  The enzyme dimethylarginine dimethylaminohydro...   \n",
       "3  BACKGROUND AND AIMS: Glutamic acid decarboxyla...   \n",
       "4  Mazindane (26) was found to be a pro-drug, oxi...   \n",
       "\n",
       "                                           chemicals  truth  \\\n",
       "0                   arachidonic acid, prostaglandins      1   \n",
       "1                                 levodopa, dopamine      1   \n",
       "2  dimethylarginine, arginine, citrulline, methyl...      1   \n",
       "3                                    glutamate, GABA      1   \n",
       "4                     Mazindane, 5-H, 5-OH, mazindol      1   \n",
       "\n",
       "          substrates                        products  \n",
       "0  arachidonic, acid                  prostaglandins  \n",
       "1                NaN                             NaN  \n",
       "2           arginine                             NaN  \n",
       "3          glutamate  gamma-aminobutyric, acid, GABA  \n",
       "4                NaN                             NaN  "
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sentence_chem_df = sentence_df[[\"sentence\", \"chemicals\", \"truth\", \"substrates\", \"products\"]]\n",
    "sentence_chem_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-62-5f200068d794>:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  sentence_chem_df[\"chemicals\"] = sentence_chem_df[\"chemicals\"].replace(np.NaN, \"0\")\n"
     ]
    }
   ],
   "source": [
    "import math\n",
    "sentence_chem_df[\"chemicals\"] = sentence_chem_df[\"chemicals\"].replace(np.NaN, \"0\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-63-3ce876c36439>:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  sentence_chem_df[\"sentence\"] = sentence_chem_df[\"sentence\"].apply(sentence_cleaned)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0    1361\n",
       "1      61\n",
       "Name: truth, dtype: int64"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sentence_chem_df[\"sentence\"] = sentence_chem_df[\"sentence\"].apply(sentence_cleaned)\n",
    "sentence_chem_df[\"truth\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-64-98d7f65adbec>:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  sentence_chem_df[\"chemicals\"] = sentence_chem_df[\"chemicals\"].apply(chem_into_array)\n"
     ]
    }
   ],
   "source": [
    "sentence_chem_df[\"chemicals\"] = sentence_chem_df[\"chemicals\"].apply(chem_into_array)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-65-de3635daabb9>:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  sentence_chem_df[\"text\"] = sentence_chem_df[[\"sentence\", \"chemicals\"]].values.tolist()\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sentence</th>\n",
       "      <th>chemicals</th>\n",
       "      <th>truth</th>\n",
       "      <th>substrates</th>\n",
       "      <th>products</th>\n",
       "      <th>text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>the enzyme cyclo-oxygenase catalyses the oxyge...</td>\n",
       "      <td>[prostaglandins, arachidonic acid]</td>\n",
       "      <td>1</td>\n",
       "      <td>arachidonic, acid</td>\n",
       "      <td>prostaglandins</td>\n",
       "      <td>[the enzyme cyclo-oxygenase catalyses the oxyg...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>although the exact site of decarboxylation of ...</td>\n",
       "      <td>[dopamine, levodopa]</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[although the exact site of decarboxylation of...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>the enzyme dimethylarginine dimethylaminohydro...</td>\n",
       "      <td>[arginine, methylamines, dimethylarginine, cit...</td>\n",
       "      <td>1</td>\n",
       "      <td>arginine</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[the enzyme dimethylarginine dimethylaminohydr...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>background and aims glutamic acid decarboxylas...</td>\n",
       "      <td>[glutamate, gaba]</td>\n",
       "      <td>1</td>\n",
       "      <td>glutamate</td>\n",
       "      <td>gamma-aminobutyric, acid, GABA</td>\n",
       "      <td>[background and aims glutamic acid decarboxyla...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>mazindane 26 was found to be a pro-drug, oxidi...</td>\n",
       "      <td>[mazindane, mazindol, 5-h, 5-oh]</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[mazindane 26 was found to be a pro-drug, oxid...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1417</th>\n",
       "      <td>only two exceptions had been reported-the ente...</td>\n",
       "      <td>[n-alpha-acetylornithine, ornithine]</td>\n",
       "      <td>1</td>\n",
       "      <td>ornithine</td>\n",
       "      <td>N-alpha-acetylornithine</td>\n",
       "      <td>[only two exceptions had been reported-the ent...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1418</th>\n",
       "      <td>the energetics is here reported for the action...</td>\n",
       "      <td>[l-beta-lysine, l-lysine]</td>\n",
       "      <td>1</td>\n",
       "      <td>l-lysine, l-beta-lysine</td>\n",
       "      <td>l-lysine, l-beta-lysine</td>\n",
       "      <td>[the energetics is here reported for the actio...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1419</th>\n",
       "      <td>estradiol 17 beta-hydroxysteroid dehydrogenase...</td>\n",
       "      <td>[estradiol, estrone]</td>\n",
       "      <td>1</td>\n",
       "      <td>estrone, estradiol</td>\n",
       "      <td>estrone, estradiol</td>\n",
       "      <td>[estradiol 17 beta-hydroxysteroid dehydrogenas...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1420</th>\n",
       "      <td>5-hydroxymethyltryptophan and 5-hydroxy-4-meth...</td>\n",
       "      <td>[5-hydroxy-4-methyltryptophan, 5-methyltryptop...</td>\n",
       "      <td>1</td>\n",
       "      <td>5-methyltryptophan</td>\n",
       "      <td>5-Hydroxymethyltryptophan, 5-hydroxy-4-methylt...</td>\n",
       "      <td>[5-hydroxymethyltryptophan and 5-hydroxy-4-met...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1421</th>\n",
       "      <td>we show that tres from mycobacterium smegmatis...</td>\n",
       "      <td>[trehalose, maltose]</td>\n",
       "      <td>1</td>\n",
       "      <td>maltose, trehalose</td>\n",
       "      <td>maltose, trehalose</td>\n",
       "      <td>[we show that tres from mycobacterium smegmati...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1422 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                               sentence  \\\n",
       "0     the enzyme cyclo-oxygenase catalyses the oxyge...   \n",
       "1     although the exact site of decarboxylation of ...   \n",
       "2     the enzyme dimethylarginine dimethylaminohydro...   \n",
       "3     background and aims glutamic acid decarboxylas...   \n",
       "4     mazindane 26 was found to be a pro-drug, oxidi...   \n",
       "...                                                 ...   \n",
       "1417  only two exceptions had been reported-the ente...   \n",
       "1418  the energetics is here reported for the action...   \n",
       "1419  estradiol 17 beta-hydroxysteroid dehydrogenase...   \n",
       "1420  5-hydroxymethyltryptophan and 5-hydroxy-4-meth...   \n",
       "1421  we show that tres from mycobacterium smegmatis...   \n",
       "\n",
       "                                              chemicals  truth  \\\n",
       "0                    [prostaglandins, arachidonic acid]      1   \n",
       "1                                  [dopamine, levodopa]      1   \n",
       "2     [arginine, methylamines, dimethylarginine, cit...      1   \n",
       "3                                     [glutamate, gaba]      1   \n",
       "4                      [mazindane, mazindol, 5-h, 5-oh]      1   \n",
       "...                                                 ...    ...   \n",
       "1417               [n-alpha-acetylornithine, ornithine]      1   \n",
       "1418                          [l-beta-lysine, l-lysine]      1   \n",
       "1419                               [estradiol, estrone]      1   \n",
       "1420  [5-hydroxy-4-methyltryptophan, 5-methyltryptop...      1   \n",
       "1421                               [trehalose, maltose]      1   \n",
       "\n",
       "                   substrates  \\\n",
       "0           arachidonic, acid   \n",
       "1                         NaN   \n",
       "2                    arginine   \n",
       "3                   glutamate   \n",
       "4                         NaN   \n",
       "...                       ...   \n",
       "1417                ornithine   \n",
       "1418  l-lysine, l-beta-lysine   \n",
       "1419       estrone, estradiol   \n",
       "1420       5-methyltryptophan   \n",
       "1421       maltose, trehalose   \n",
       "\n",
       "                                               products  \\\n",
       "0                                        prostaglandins   \n",
       "1                                                   NaN   \n",
       "2                                                   NaN   \n",
       "3                        gamma-aminobutyric, acid, GABA   \n",
       "4                                                   NaN   \n",
       "...                                                 ...   \n",
       "1417                            N-alpha-acetylornithine   \n",
       "1418                            l-lysine, l-beta-lysine   \n",
       "1419                                 estrone, estradiol   \n",
       "1420  5-Hydroxymethyltryptophan, 5-hydroxy-4-methylt...   \n",
       "1421                                 maltose, trehalose   \n",
       "\n",
       "                                                   text  \n",
       "0     [the enzyme cyclo-oxygenase catalyses the oxyg...  \n",
       "1     [although the exact site of decarboxylation of...  \n",
       "2     [the enzyme dimethylarginine dimethylaminohydr...  \n",
       "3     [background and aims glutamic acid decarboxyla...  \n",
       "4     [mazindane 26 was found to be a pro-drug, oxid...  \n",
       "...                                                 ...  \n",
       "1417  [only two exceptions had been reported-the ent...  \n",
       "1418  [the energetics is here reported for the actio...  \n",
       "1419  [estradiol 17 beta-hydroxysteroid dehydrogenas...  \n",
       "1420  [5-hydroxymethyltryptophan and 5-hydroxy-4-met...  \n",
       "1421  [we show that tres from mycobacterium smegmati...  \n",
       "\n",
       "[1422 rows x 6 columns]"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sentence_chem_df[\"text\"] = sentence_chem_df[[\"sentence\", \"chemicals\"]].values.tolist()\n",
    "sentence_chem_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sentence</th>\n",
       "      <th>chemicals</th>\n",
       "      <th>truth</th>\n",
       "      <th>substrates</th>\n",
       "      <th>products</th>\n",
       "      <th>text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>the enzyme cyclo-oxygenase catalyses the oxyge...</td>\n",
       "      <td>[prostaglandins, arachidonic acid]</td>\n",
       "      <td>1</td>\n",
       "      <td>arachidonic, acid</td>\n",
       "      <td>prostaglandins</td>\n",
       "      <td>[the enzyme cyclo-oxygenase catalyses the oxyg...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>although the exact site of decarboxylation of ...</td>\n",
       "      <td>[dopamine, levodopa]</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[although the exact site of decarboxylation of...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>the enzyme dimethylarginine dimethylaminohydro...</td>\n",
       "      <td>[arginine, methylamines, dimethylarginine, cit...</td>\n",
       "      <td>1</td>\n",
       "      <td>arginine</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[the enzyme dimethylarginine dimethylaminohydr...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>background and aims glutamic acid decarboxylas...</td>\n",
       "      <td>[glutamate, gaba]</td>\n",
       "      <td>1</td>\n",
       "      <td>glutamate</td>\n",
       "      <td>gamma-aminobutyric, acid, GABA</td>\n",
       "      <td>[background and aims glutamic acid decarboxyla...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>mazindane 26 was found to be a pro-drug, oxidi...</td>\n",
       "      <td>[mazindane, mazindol, 5-h, 5-oh]</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[mazindane 26 was found to be a pro-drug, oxid...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            sentence  \\\n",
       "0  the enzyme cyclo-oxygenase catalyses the oxyge...   \n",
       "1  although the exact site of decarboxylation of ...   \n",
       "2  the enzyme dimethylarginine dimethylaminohydro...   \n",
       "3  background and aims glutamic acid decarboxylas...   \n",
       "4  mazindane 26 was found to be a pro-drug, oxidi...   \n",
       "\n",
       "                                           chemicals  truth  \\\n",
       "0                 [prostaglandins, arachidonic acid]      1   \n",
       "1                               [dopamine, levodopa]      1   \n",
       "2  [arginine, methylamines, dimethylarginine, cit...      1   \n",
       "3                                  [glutamate, gaba]      1   \n",
       "4                   [mazindane, mazindol, 5-h, 5-oh]      1   \n",
       "\n",
       "          substrates                        products  \\\n",
       "0  arachidonic, acid                  prostaglandins   \n",
       "1                NaN                             NaN   \n",
       "2           arginine                             NaN   \n",
       "3          glutamate  gamma-aminobutyric, acid, GABA   \n",
       "4                NaN                             NaN   \n",
       "\n",
       "                                                text  \n",
       "0  [the enzyme cyclo-oxygenase catalyses the oxyg...  \n",
       "1  [although the exact site of decarboxylation of...  \n",
       "2  [the enzyme dimethylarginine dimethylaminohydr...  \n",
       "3  [background and aims glutamic acid decarboxyla...  \n",
       "4  [mazindane 26 was found to be a pro-drug, oxid...  "
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hand_anno_mult_chem_df = sentence_chem_df[sentence_chem_df['chemicals'].map(len) > 1]\n",
    "hand_anno_mult_chem_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### data with no truth values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [],
   "source": [
    "# have this data one folder up\n",
    "uncleaned_df = pd.read_csv(\"sentence_annotations_elsevier_pmid_split6.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>lit_id</th>\n",
       "      <th>indices</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>sentence</th>\n",
       "      <th>sentence_pos</th>\n",
       "      <th>enzymes</th>\n",
       "      <th>enzyme_locations</th>\n",
       "      <th>chemical_entities_full</th>\n",
       "      <th>chemical_names</th>\n",
       "      <th>chemical_smiles</th>\n",
       "      <th>name_smile_tuples</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10.1002/jps.20686</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1297</td>\n",
       "      <td>serial JL 313843 291210 291727 291789 291928 3...</td>\n",
       "      <td>[('serial', 'JJ'), ('JL', 'NN'), ('313843', 'C...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10.1002/jps.20686</td>\n",
       "      <td>1</td>\n",
       "      <td>1298</td>\n",
       "      <td>1324</td>\n",
       "      <td>Published by Elsevier Inc.</td>\n",
       "      <td>[('Published', 'VBN'), ('by', 'IN'), ('Elsevie...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10.1002/jps.20686</td>\n",
       "      <td>2</td>\n",
       "      <td>1325</td>\n",
       "      <td>1345</td>\n",
       "      <td>All rights reserved.</td>\n",
       "      <td>[('All', 'DT'), ('rights', 'NNS'), ('reserved'...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10.1002/jps.20686</td>\n",
       "      <td>3</td>\n",
       "      <td>1346</td>\n",
       "      <td>9469</td>\n",
       "      <td>KINETICANALYSESFORSPECIESDIFFERENCESINPGLYCOPR...</td>\n",
       "      <td>[('KINETICANALYSESFORSPECIESDIFFERENCESINPGLYC...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[]</td>\n",
       "      <td>[{'text': 'Diltiazem', 'start': 1718, 'end': 1...</td>\n",
       "      <td>Diltiazem,, Cyclosporin%20A,, Dexamethasone</td>\n",
       "      <td>COc1ccc(cc1)[C@@H]2Sc3ccccc3N(CCN(C)C)C(=O)[C@...</td>\n",
       "      <td>[('Diltiazem', 'COc1ccc(cc1)[C@@H]2Sc3ccccc3N(...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10.1002/jps.20686</td>\n",
       "      <td>4</td>\n",
       "      <td>9470</td>\n",
       "      <td>9599</td>\n",
       "      <td>Immunoblot analyses of P-gp expressed in MDR1 ...</td>\n",
       "      <td>[('Immunoblot', 'NN'), ('analyses', 'NNS'), ('...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[]</td>\n",
       "      <td>[{'text': 'H241', 'start': 9594, 'end': 9598, ...</td>\n",
       "      <td>H241</td>\n",
       "      <td>C[C@]12C[C@](O)(C[C@@H]1CC[C@@H]3[C@@H]2CC[C@@...</td>\n",
       "      <td>[('H241', 'C[C@]12C[C@](O)(C[C@@H]1CC[C@@H]3[C...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              lit_id  indices  start   end  \\\n",
       "0  10.1002/jps.20686        0      0  1297   \n",
       "1  10.1002/jps.20686        1   1298  1324   \n",
       "2  10.1002/jps.20686        2   1325  1345   \n",
       "3  10.1002/jps.20686        3   1346  9469   \n",
       "4  10.1002/jps.20686        4   9470  9599   \n",
       "\n",
       "                                            sentence  \\\n",
       "0  serial JL 313843 291210 291727 291789 291928 3...   \n",
       "1                         Published by Elsevier Inc.   \n",
       "2                               All rights reserved.   \n",
       "3  KINETICANALYSESFORSPECIESDIFFERENCESINPGLYCOPR...   \n",
       "4  Immunoblot analyses of P-gp expressed in MDR1 ...   \n",
       "\n",
       "                                        sentence_pos enzymes enzyme_locations  \\\n",
       "0  [('serial', 'JJ'), ('JL', 'NN'), ('313843', 'C...     NaN               []   \n",
       "1  [('Published', 'VBN'), ('by', 'IN'), ('Elsevie...     NaN               []   \n",
       "2  [('All', 'DT'), ('rights', 'NNS'), ('reserved'...     NaN               []   \n",
       "3  [('KINETICANALYSESFORSPECIESDIFFERENCESINPGLYC...     NaN               []   \n",
       "4  [('Immunoblot', 'NN'), ('analyses', 'NNS'), ('...     NaN               []   \n",
       "\n",
       "                              chemical_entities_full  \\\n",
       "0                                                 []   \n",
       "1                                                 []   \n",
       "2                                                 []   \n",
       "3  [{'text': 'Diltiazem', 'start': 1718, 'end': 1...   \n",
       "4  [{'text': 'H241', 'start': 9594, 'end': 9598, ...   \n",
       "\n",
       "                                chemical_names  \\\n",
       "0                                          NaN   \n",
       "1                                          NaN   \n",
       "2                                          NaN   \n",
       "3  Diltiazem,, Cyclosporin%20A,, Dexamethasone   \n",
       "4                                         H241   \n",
       "\n",
       "                                     chemical_smiles  \\\n",
       "0                                                NaN   \n",
       "1                                                NaN   \n",
       "2                                                NaN   \n",
       "3  COc1ccc(cc1)[C@@H]2Sc3ccccc3N(CCN(C)C)C(=O)[C@...   \n",
       "4  C[C@]12C[C@](O)(C[C@@H]1CC[C@@H]3[C@@H]2CC[C@@...   \n",
       "\n",
       "                                   name_smile_tuples  \n",
       "0                                                 []  \n",
       "1                                                 []  \n",
       "2                                                 []  \n",
       "3  [('Diltiazem', 'COc1ccc(cc1)[C@@H]2Sc3ccccc3N(...  \n",
       "4  [('H241', 'C[C@]12C[C@](O)(C[C@@H]1CC[C@@H]3[C...  "
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "uncleaned_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "serial JL 313843 291210 291727 291789 291928 31 Journal of Pharmaceutical Sciences JOURNALPHARMACEUTICALSCIENCES 2016-01-28 2016-01-28 2016-01-28 2016-01-28 2016-01-28T13:28:29 1-s2.0-S0022354916321499 S0022-3549(16)32149-9 S0022354916321499 10.1002/jps.20686 S300 S300.1 FULL-TEXT 1-s2.0-S0022354906X9500X 2016-08-24T12:59:06.644747-04:00 0 0 20061201 20061231 2006 2016-01-28T14:54:25.366606Z articleinfo articletitlenorm authfirstinitialnorm authfirstsurnamenorm cid cids contenttype copyright crossmark dateloaded dateloadedtxt datesearch datesort dateupdated dco docsubtype doctype doi eid ewtransactionid hubeid indexeddate issfirst issn issnnorm issuelist itemstage itemtransactionid itemweight openaccess openarchive pg pgfirst pglast pii piinorm pubdateend pubdatestart pubdatetxt pubyr sectiontitle sortorder srctitle srctitlenorm srctype ssids alllist content subj subheadings tomb volfirst volissue volumelist webpdf webpdfpagecount yearnav figure table body mmlmath acknowledge affil articletitle auth authfirstini authfull authkeywords authlast misctext primabst pubtype ref 0022-3549 00223549 false 95 95 12 12 Volume 95, Issue 12 7 2673 2683 2673 2683 200612 December 2006 2006-12-01 2006-12-31 2006 Research Articles article fla Copyright © 2006 Wiley-Liss, Inc., A Wiley Company.\n",
      "\n",
      "[]\n",
      "\n",
      "Published by Elsevier Inc.\n",
      "\n",
      "[]\n",
      "\n",
      "All rights reserved.\n",
      "\n",
      "[]\n",
      "\n",
      "KINETICANALYSESFORSPECIESDIFFERENCESINPGLYCOPROTEINMEDIATEDDRUGTRANSPORT KATOH M INTRODUCTION MATERIALS AND METHODS Materials Cell Culture Immunoblot Analysis of P-gp on MDR1 Transfected Cells Transcellular Transport Kinetic Study on Transcellular Transport Efflux Ratio of Transcellular Transport RESULTS Immunoblot Analysis of MDR1 Transfected Cells Kinetic Analysis of Diltiazem Transport Kinetic Analysis of Cyclosporin A Transport Kinetic Analysis of Dexamethasone Transport DISCUSSION ACKNOWLEDGMENTS REFERENCES UEDA 1992 24248 24252 K SCHINKEL 1995 1698 1705 A SCHINKEL 1994 491 502 A SCHINKEL 1997 4028 4033 A YAMAZAKI 2001 723 735 M TANGWAI 1995 32 39 D LIN 2003 59 98 J ADACHI 2001 1660 1668 Y TAKEUCHI 2006 T LAEMMLI 1970 680 685 U KATOH 2000 1189 1197 M LOWRY 1951 265 275 O HOCHMAN 2002 257 273 J CHEN 2003 272 291 C POLLI 2001 620 628 J SAEKI 1993 6077 6080 T GEORGES 1990 152 156 E ADACHI 2003 1163 1169 Y CHIOU 2000 135 140 W HE 1998 626 633 Y CHIOU 2002 868 874 W YANG 1990 10282 10288 C YAMAMOTO 2002 1055 1063 C MOLDEN 2000 107 109 E WALKER 2005 587 595 D KATOHX2006X2673 KATOHX2006X2673X2683 KATOHX2006X2673XM KATOHX2006X2673X2683XM http://creativecommons.org/licenses/by-nc-nd/4.0/ UnderEmbargo 2017-01-28T00:00:00Z item S0022-3549(16)32149-9 S0022354916321499 1-s2.0-S0022354916321499 10.1002/jps.20686 313843 2016-01-28T08:54:25.366606-05:00 2006-12-01 2006-12-31 1-s2.0-S0022354916321499-main.pdf https://s3-eu-west-1.amazonaws.com/prod-ucs-content-store-eu-west/content/pii:S0022354916321499/MAIN/application/pdf/553d479aab08135173500dd0b5b033d3/main.pdf https://s3.amazonaws.com/prod-ucs-content-store-us-east/content/pii:S0022354916321499/MAIN/application/pdf/553d479aab08135173500dd0b5b033d3/main.pdf main.pdf pdf true 211248 MAIN 11 1-s2.0-S0022354916321499-main_1.png https://s3-eu-west-1.amazonaws.com/prod-ucs-content-store-eu-west/content/pii:S0022354916321499/PREVIEW/image/png/e9226fff7b8f795aabcc079403f4d1de/main_1.png https://s3.amazonaws.com/prod-ucs-content-store-us-east/content/pii:S0022354916321499/PREVIEW/image/png/e9226fff7b8f795aabcc079403f4d1de/main_1.png main_1.png png 63704 849 656 IMAGE-WEB-PDF 1 1-s2.0-S0022354916321499-gr1.sml https://s3-eu-west-1.amazonaws.com/prod-ucs-content-store-eu-west/content/pii:S0022354916321499/gr1/THUMBNAIL/image/gif/ccb26406e522fbb3f980cf1bd41b0f92/gr1.sml https://s3.amazonaws.com/prod-ucs-content-store-us-east/content/pii:S0022354916321499/gr1/THUMBNAIL/image/gif/ccb26406e522fbb3f980cf1bd41b0f92/gr1.sml gr1 gr1.sml sml 14373 126 219 IMAGE-THUMBNAIL 1-s2.0-S0022354916321499-gr2.sml https://s3-eu-west-1.amazonaws.com/prod-ucs-content-store-eu-west/content/pii:S0022354916321499/gr2/THUMBNAIL/image/gif/dcc67e4da9b470eed1e34b96068e300f/gr2.sml https://s3.amazonaws.com/prod-ucs-content-store-us-east/content/pii:S0022354916321499/gr2/THUMBNAIL/image/gif/dcc67e4da9b470eed1e34b96068e300f/gr2.sml gr2 gr2.sml sml 6976 164 166 IMAGE-THUMBNAIL 1-s2.0-S0022354916321499-gr3.sml https://s3-eu-west-1.amazonaws.com/prod-ucs-content-store-eu-west/content/pii:S0022354916321499/gr3/THUMBNAIL/image/gif/7d2bf3180a38f29c57117ecadd49bfcb/gr3.sml https://s3.amazonaws.com/prod-ucs-content-store-us-east/content/pii:S0022354916321499/gr3/THUMBNAIL/image/gif/7d2bf3180a38f29c57117ecadd49bfcb/gr3.sml gr3 gr3.sml sml 7751 146 219 IMAGE-THUMBNAIL 1-s2.0-S0022354916321499-gr4.sml https://s3-eu-west-1.amazonaws.com/prod-ucs-content-store-eu-west/content/pii:S0022354916321499/gr4/THUMBNAIL/image/gif/fc9599dc7dc754639b42dc0348edb154/gr4.sml https://s3.amazonaws.com/prod-ucs-content-store-us-east/content/pii:S0022354916321499/gr4/THUMBNAIL/image/gif/fc9599dc7dc754639b42dc0348edb154/gr4.sml gr4 gr4.sml sml 4533 71 219 IMAGE-THUMBNAIL 1-s2.0-S0022354916321499-gr5.sml https://s3-eu-west-1.amazonaws.com/prod-ucs-content-store-eu-west/content/pii:S0022354916321499/gr5/THUMBNAIL/image/gif/1afe2415eaa901bf2ad4f37d3f23e3e6/gr5.sml https://s3.amazonaws.com/prod-ucs-content-store-us-east/content/pii:S0022354916321499/gr5/THUMBNAIL/image/gif/1afe2415eaa901bf2ad4f37d3f23e3e6/gr5.sml gr5 gr5.sml sml 4855 71 219 IMAGE-THUMBNAIL 1-s2.0-S0022354916321499-gr2.gif https://s3-eu-west-1.amazonaws.com/prod-ucs-content-store-eu-west/content/pii:S0022354916321499/gr2/DOWNSAMPLED/image/gif/1bf3ef9bfd5246de2b60b54cf7f414bc/gr2.gif https://s3.amazonaws.com/prod-ucs-content-store-us-east/content/pii:S0022354916321499/gr2/DOWNSAMPLED/image/gif/1bf3ef9bfd5246de2b60b54cf7f414bc/gr2.gif gr2 gr2.gif gif 42128 595 602 IMAGE-DOWNSAMPLED 1-s2.0-S0022354916321499-gr3.gif https://s3-eu-west-1.amazonaws.com/prod-ucs-content-store-eu-west/content/pii:S0022354916321499/gr3/DOWNSAMPLED/image/gif/9419a2056409388eff8895bda985cd1a/gr3.gif https://s3.amazonaws.com/prod-ucs-content-store-us-east/content/pii:S0022354916321499/gr3/DOWNSAMPLED/image/gif/9419a2056409388eff8895bda985cd1a/gr3.gif gr3 gr3.gif gif 31891 400 602 IMAGE-DOWNSAMPLED 1-s2.0-S0022354916321499-gr4.gif https://s3-eu-west-1.amazonaws.com/prod-ucs-content-store-eu-west/content/pii:S0022354916321499/gr4/DOWNSAMPLED/image/gif/8cb822eacd3a4f3bafe1547fbc6c4138/gr4.gif https://s3.amazonaws.com/prod-ucs-content-store-us-east/content/pii:S0022354916321499/gr4/DOWNSAMPLED/image/gif/8cb822eacd3a4f3bafe1547fbc6c4138/gr4.gif gr4 gr4.gif gif 17900 196 602 IMAGE-DOWNSAMPLED 1-s2.0-S0022354916321499-gr5.gif https://s3-eu-west-1.amazonaws.com/prod-ucs-content-store-eu-west/content/pii:S0022354916321499/gr5/DOWNSAMPLED/image/gif/9612f49e8edb2ff997cb7314b2ee1753/gr5.gif https://s3.amazonaws.com/prod-ucs-content-store-us-east/content/pii:S0022354916321499/gr5/DOWNSAMPLED/image/gif/9612f49e8edb2ff997cb7314b2ee1753/gr5.gif gr5 gr5.gif gif 19498 195 602 IMAGE-DOWNSAMPLED 1-s2.0-S0022354916321499-gr1.jpg https://s3-eu-west-1.amazonaws.com/prod-ucs-content-store-eu-west/content/pii:S0022354916321499/gr1/DOWNSAMPLED/image/jpeg/3f1aba75ea7c75741ecc5d2b9ae2f64d/gr1.jpg https://s3.amazonaws.com/prod-ucs-content-store-us-east/content/pii:S0022354916321499/gr1/DOWNSAMPLED/image/jpeg/3f1aba75ea7c75741ecc5d2b9ae2f64d/gr1.jpg gr1 gr1.jpg jpg 44907 270 470 IMAGE-DOWNSAMPLED 1-s2.0-S0022354916321499-si1.gif https://s3-eu-west-1.amazonaws.com/prod-ucs-content-store-eu-west/content/pii:S0022354916321499/STRIPIN/image/gif/493eb6067386ea527b11e5e12295762f/si1.gif https://s3.amazonaws.com/prod-ucs-content-store-us-east/content/pii:S0022354916321499/STRIPIN/image/gif/493eb6067386ea527b11e5e12295762f/si1.gif si1 si1.gif gif 285 40 113 ALTIMG 1-s2.0-S0022354916321499-si2.gif https://s3-eu-west-1.amazonaws.com/prod-ucs-content-store-eu-west/content/pii:S0022354916321499/STRIPIN/image/gif/f803a7f29e15e81219db16ce47c6d545/si2.gif https://s3.amazonaws.com/prod-ucs-content-store-us-east/content/pii:S0022354916321499/STRIPIN/image/gif/f803a7f29e15e81219db16ce47c6d545/si2.gif si2 si2.gif gif 324 18 197 ALTIMG 1-s2.0-S0022354916321499-si3.gif https://s3-eu-west-1.amazonaws.com/prod-ucs-content-store-eu-west/content/pii:S0022354916321499/STRIPIN/image/gif/4410a7dc021b46bad36691ba23b54dc7/si3.gif https://s3.amazonaws.com/prod-ucs-content-store-us-east/content/pii:S0022354916321499/STRIPIN/image/gif/4410a7dc021b46bad36691ba23b54dc7/si3.gif si3 si3.gif gif 559 18 379 ALTIMG 1-s2.0-S0022354916321499-si4.gif https://s3-eu-west-1.amazonaws.com/prod-ucs-content-store-eu-west/content/pii:S0022354916321499/STRIPIN/image/gif/b0429be46a3ea34f05cef35585bbf37a/si4.gif https://s3.amazonaws.com/prod-ucs-content-store-us-east/content/pii:S0022354916321499/STRIPIN/image/gif/b0429be46a3ea34f05cef35585bbf37a/si4.gif si4 si4.gif gif 468 43 175 ALTIMG 1-s2.0-S0022354916321499-si5.gif https://s3-eu-west-1.amazonaws.com/prod-ucs-content-store-eu-west/content/pii:S0022354916321499/STRIPIN/image/gif/934fd082640701c8695f22e003c41729/si5.gif https://s3.amazonaws.com/prod-ucs-content-store-us-east/content/pii:S0022354916321499/STRIPIN/image/gif/934fd082640701c8695f22e003c41729/si5.gif si5 si5.gif gif 2393 43 991 ALTIMG XPHS 50005140 S0022-3549(16)32149-9 10.1002/jps.20686 Wiley-Liss, Inc., A Wiley Company Copyright © 2006 Wiley-Liss, Inc., A Wiley Company Figure 1 Immunoblot analysis of P-gp in MDR1 transfected cells.\n",
      "\n",
      "[{'text': 'Diltiazem', 'start': 1718, 'end': 1727, 'smiles': 'COc1ccc(cc1)[C@@H]2Sc3ccccc3N(CCN(C)C)C(=O)[C@@H]2OC(C)=O', 'pos': 'NNP'}, {'text': 'Cyclosporin%20A', 'start': 1758, 'end': 1771, 'smiles': 'CCC1NC(=O)C(C(O)C(C)C\\\\C=C\\\\C)N(C)C(=O)C(C(C)C)N(C)C(=O)C(CC(C)C)N(C)C(=O)C(CC(C)C)N(C)C(=O)C(C)NC(=O)C(C)NC(=O)C(CC(C)C)N(C)C(=O)C(NC(=O)C(CC(C)C)N(C)C(=O)CN(C)C1=O)C(C)C'}, {'text': 'Dexamethasone', 'start': 1802, 'end': 1815, 'smiles': 'C[C@@H]1C[C@H]2[C@@H]3CCC4=CC(=O)C=C[C@]4(C)[C@@]3(F)[C@@H](O)C[C@]2(C)[C@@]1(O)C(=O)CO', 'pos': 'NNP'}]\n",
      "\n",
      "Immunoblot analyses of P-gp expressed in MDR1 transfected cells were performed using anti-P-gp antibodies, (A) C219 and (B) H241.\n",
      "\n",
      "[{'text': 'H241', 'start': 9594, 'end': 9598, 'smiles': 'C[C@]12C[C@](O)(C[C@@H]1CC[C@@H]3[C@@H]2CC[C@@]4(C)[C@H]3CC[C@@]4(O)C#C)C#C', 'pos': 'NN'}]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for i in range(5):\n",
    "    print(uncleaned_df.iloc[i]['sentence'])\n",
    "    print(\"\")\n",
    "    print(uncleaned_df.iloc[i]['chemical_entities_full'])\n",
    "    print('')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>lit_id</th>\n",
       "      <th>indices</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>sentence</th>\n",
       "      <th>sentence_pos</th>\n",
       "      <th>enzymes</th>\n",
       "      <th>enzyme_locations</th>\n",
       "      <th>chemical_entities_full</th>\n",
       "      <th>chemical_names</th>\n",
       "      <th>chemical_smiles</th>\n",
       "      <th>name_smile_tuples</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>10.1002/jps.20686</td>\n",
       "      <td>150</td>\n",
       "      <td>33180</td>\n",
       "      <td>33589</td>\n",
       "      <td>We selected diltiazem, cyclosporin A, and dexa...</td>\n",
       "      <td>[('We', 'PRP'), ('selected', 'VBD'), ('diltiaz...</td>\n",
       "      <td>ATPase</td>\n",
       "      <td>[('ATPase', 22, 22)]</td>\n",
       "      <td>[{'text': 'diltiazem', 'start': 33192, 'end': ...</td>\n",
       "      <td>diltiazem,, cyclosporin%20A,, dexamethasone,, ...</td>\n",
       "      <td>COc1ccc(cc1)[C@@H]2Sc3ccccc3N(CCN(C)C)C(=O)[C@...</td>\n",
       "      <td>[('diltiazem', 'COc1ccc(cc1)[C@@H]2Sc3ccccc3N(...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271</th>\n",
       "      <td>10.1006/abbi.1993.1114</td>\n",
       "      <td>3</td>\n",
       "      <td>2464</td>\n",
       "      <td>2912</td>\n",
       "      <td>Croteau R. Washington State Univ, Inst Biol Ch...</td>\n",
       "      <td>[('Croteau', 'NNP'), ('R.', 'NNP'), ('Washingt...</td>\n",
       "      <td>synthase, cyclase</td>\n",
       "      <td>[('synthase', 34, 34), ('cyclase', 38, 38)]</td>\n",
       "      <td>[{'text': 'Limonene', 'start': 2620, 'end': 26...</td>\n",
       "      <td>Limonene,, geranyl%20pyrophosphate,, sodium%20...</td>\n",
       "      <td>CC(=C)C1CCC(=CC1)C,, CC(C)=CCCC(/C)=C/CO[P]([O...</td>\n",
       "      <td>[('Limonene', 'CC(=C)C1CCC(=CC1)C'), ('geranyl...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>280</th>\n",
       "      <td>10.1006/abbi.1993.1129</td>\n",
       "      <td>3</td>\n",
       "      <td>2426</td>\n",
       "      <td>2514</td>\n",
       "      <td>Storage Tissue: H+/Substrate Stoichiometries f...</td>\n",
       "      <td>[('Storage', 'NNP'), ('Tissue', 'NNP'), (':', ...</td>\n",
       "      <td>ATPase, PPase</td>\n",
       "      <td>[('ATPase', 11, 11), ('PPase', 15, 15)]</td>\n",
       "      <td>[{'text': 'H+', 'start': 2442, 'end': 2444, 's...</td>\n",
       "      <td>H+,, H+,, H+</td>\n",
       "      <td>[H+],, [H+],, [H+]</td>\n",
       "      <td>[('H+', '[H+]'), ('H+', '[H+]'), ('H+', '[H+]')]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>281</th>\n",
       "      <td>10.1006/abbi.1993.1129</td>\n",
       "      <td>4</td>\n",
       "      <td>2515</td>\n",
       "      <td>2716</td>\n",
       "      <td>Briskin D.P. Univ Illinois, Dept Agron, 1201 W...</td>\n",
       "      <td>[('Briskin', 'NNP'), ('D.P.', 'NNP'), ('Univ',...</td>\n",
       "      <td>ATPase, PPase</td>\n",
       "      <td>[('ATPase', 30, 30), ('PPase', 34, 34)]</td>\n",
       "      <td>[{'text': 'H+', 'start': 2609, 'end': 2611, 's...</td>\n",
       "      <td>H+,, H+,, H+</td>\n",
       "      <td>[H+],, [H+],, [H+]</td>\n",
       "      <td>[('H+', '[H+]'), ('H+', '[H+]'), ('H+', '[H+]')]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>286</th>\n",
       "      <td>10.1006/abbi.1993.1129</td>\n",
       "      <td>9</td>\n",
       "      <td>3367</td>\n",
       "      <td>3717</td>\n",
       "      <td>From these results and the estimated level of ...</td>\n",
       "      <td>[('From', 'IN'), ('these', 'DT'), ('results', ...</td>\n",
       "      <td>ATPase, PPase</td>\n",
       "      <td>[('ATPase', 27, 27), ('PPase', 31, 31)]</td>\n",
       "      <td>[{'text': 'H+', 'start': 3509, 'end': 3511, 's...</td>\n",
       "      <td>H+,, H+,, H+</td>\n",
       "      <td>[H+],, [H+],, [H+]</td>\n",
       "      <td>[('H+', '[H+]'), ('H+', '[H+]'), ('H+', '[H+]')]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>952414</th>\n",
       "      <td>10.1263/jbb.99.623</td>\n",
       "      <td>26</td>\n",
       "      <td>11239</td>\n",
       "      <td>11493</td>\n",
       "      <td>Key words D-alanine-D-alanine ligase D-amino a...</td>\n",
       "      <td>[('Key', 'JJ'), ('words', 'NNS'), ('D-alanine-...</td>\n",
       "      <td>ligase</td>\n",
       "      <td>[('ligase', 3, 3)]</td>\n",
       "      <td>[{'text': 'D-alanine-D-alanine', 'start': 1124...</td>\n",
       "      <td>D-alanine-D-alanine,, vancomycin</td>\n",
       "      <td>C[C@@H](N)C(O)=O.C[C@@H](N)C(O)=O,, CN[C@H](CC...</td>\n",
       "      <td>[('D-alanine-D-alanine', 'C[C@@H](N)C(O)=O.C[C...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>952450</th>\n",
       "      <td>10.1263/jbb.99.623</td>\n",
       "      <td>62</td>\n",
       "      <td>17119</td>\n",
       "      <td>17420</td>\n",
       "      <td>The mobile phase consisted of a linear gradien...</td>\n",
       "      <td>[('The', 'DT'), ('mobile', 'JJ'), ('phase', 'N...</td>\n",
       "      <td>phase</td>\n",
       "      <td>[('phase', 2, 2)]</td>\n",
       "      <td>[{'text': 'methanol', 'start': 17213, 'end': 1...</td>\n",
       "      <td>methanol,, methanol,, tetrahydrofurane</td>\n",
       "      <td>CO,, CO,, C1CCOC1</td>\n",
       "      <td>[('methanol', 'CO'), ('methanol', 'CO'), ('tet...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>952517</th>\n",
       "      <td>10.1263/jbb.99.623</td>\n",
       "      <td>129</td>\n",
       "      <td>27150</td>\n",
       "      <td>27373</td>\n",
       "      <td>Weber V. Falkenhagen D. SUBPOL: a novel sucros...</td>\n",
       "      <td>[('Weber', 'NNP'), ('V.', 'NNP'), ('Falkenhage...</td>\n",
       "      <td>phase</td>\n",
       "      <td>[('phase', 16, 16)]</td>\n",
       "      <td>[{'text': 'sucrose', 'start': 27190, 'end': 27...</td>\n",
       "      <td>sucrose,, Kato</td>\n",
       "      <td>OC[C@H]1O[C@H](O[C@]2(CO)O[C@H](CO)[C@@H](O)[C...</td>\n",
       "      <td>[('sucrose', 'OC[C@H]1O[C@H](O[C@]2(CO)O[C@H](...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>952526</th>\n",
       "      <td>10.1263/jbb.99.623</td>\n",
       "      <td>138</td>\n",
       "      <td>28139</td>\n",
       "      <td>28364</td>\n",
       "      <td>Active-site mutants of the VanC2 D-alanyl-D-se...</td>\n",
       "      <td>[('Active', 'JJ'), ('-', 'HYPH'), ('site', 'NN...</td>\n",
       "      <td>ligase, ligase</td>\n",
       "      <td>[('ligase', 8, 8), ('ligase', 25, 25)]</td>\n",
       "      <td>[{'text': 'vancomycin', 'start': 28220, 'end':...</td>\n",
       "      <td>vancomycin,, D-alanyl-D-alanine</td>\n",
       "      <td>CN[C@H](CC(C)C)C(=O)NC1[C@H](O)c2ccc(Oc3cc4cc(...</td>\n",
       "      <td>[('vancomycin', 'CN[C@H](CC(C)C)C(=O)NC1[C@H](...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>952541</th>\n",
       "      <td>10.1263/jbb.99.623</td>\n",
       "      <td>153</td>\n",
       "      <td>29289</td>\n",
       "      <td>29564</td>\n",
       "      <td>In vitro ATP regeneration from polyphosphate a...</td>\n",
       "      <td>[('In', 'FW'), ('vitro', 'FW'), ('ATP', 'NN'),...</td>\n",
       "      <td>phosphotransferase, kinase</td>\n",
       "      <td>[('phosphotransferase', 12, 12), ('kinase', 15...</td>\n",
       "      <td>[{'text': 'ATP', 'start': 29298, 'end': 29301,...</td>\n",
       "      <td>ATP,, AMP,, AMP,, adenylate</td>\n",
       "      <td>Nc1ncnc2n(cnc12)C3OC(CO[P](O)(=O)O[P](O)(=O)O[...</td>\n",
       "      <td>[('ATP', 'Nc1ncnc2n(cnc12)C3OC(CO[P](O)(=O)O[P...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>50212 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                        lit_id  indices  start    end  \\\n",
       "150          10.1002/jps.20686      150  33180  33589   \n",
       "271     10.1006/abbi.1993.1114        3   2464   2912   \n",
       "280     10.1006/abbi.1993.1129        3   2426   2514   \n",
       "281     10.1006/abbi.1993.1129        4   2515   2716   \n",
       "286     10.1006/abbi.1993.1129        9   3367   3717   \n",
       "...                        ...      ...    ...    ...   \n",
       "952414      10.1263/jbb.99.623       26  11239  11493   \n",
       "952450      10.1263/jbb.99.623       62  17119  17420   \n",
       "952517      10.1263/jbb.99.623      129  27150  27373   \n",
       "952526      10.1263/jbb.99.623      138  28139  28364   \n",
       "952541      10.1263/jbb.99.623      153  29289  29564   \n",
       "\n",
       "                                                 sentence  \\\n",
       "150     We selected diltiazem, cyclosporin A, and dexa...   \n",
       "271     Croteau R. Washington State Univ, Inst Biol Ch...   \n",
       "280     Storage Tissue: H+/Substrate Stoichiometries f...   \n",
       "281     Briskin D.P. Univ Illinois, Dept Agron, 1201 W...   \n",
       "286     From these results and the estimated level of ...   \n",
       "...                                                   ...   \n",
       "952414  Key words D-alanine-D-alanine ligase D-amino a...   \n",
       "952450  The mobile phase consisted of a linear gradien...   \n",
       "952517  Weber V. Falkenhagen D. SUBPOL: a novel sucros...   \n",
       "952526  Active-site mutants of the VanC2 D-alanyl-D-se...   \n",
       "952541  In vitro ATP regeneration from polyphosphate a...   \n",
       "\n",
       "                                             sentence_pos  \\\n",
       "150     [('We', 'PRP'), ('selected', 'VBD'), ('diltiaz...   \n",
       "271     [('Croteau', 'NNP'), ('R.', 'NNP'), ('Washingt...   \n",
       "280     [('Storage', 'NNP'), ('Tissue', 'NNP'), (':', ...   \n",
       "281     [('Briskin', 'NNP'), ('D.P.', 'NNP'), ('Univ',...   \n",
       "286     [('From', 'IN'), ('these', 'DT'), ('results', ...   \n",
       "...                                                   ...   \n",
       "952414  [('Key', 'JJ'), ('words', 'NNS'), ('D-alanine-...   \n",
       "952450  [('The', 'DT'), ('mobile', 'JJ'), ('phase', 'N...   \n",
       "952517  [('Weber', 'NNP'), ('V.', 'NNP'), ('Falkenhage...   \n",
       "952526  [('Active', 'JJ'), ('-', 'HYPH'), ('site', 'NN...   \n",
       "952541  [('In', 'FW'), ('vitro', 'FW'), ('ATP', 'NN'),...   \n",
       "\n",
       "                           enzymes  \\\n",
       "150                         ATPase   \n",
       "271              synthase, cyclase   \n",
       "280                  ATPase, PPase   \n",
       "281                  ATPase, PPase   \n",
       "286                  ATPase, PPase   \n",
       "...                            ...   \n",
       "952414                      ligase   \n",
       "952450                       phase   \n",
       "952517                       phase   \n",
       "952526              ligase, ligase   \n",
       "952541  phosphotransferase, kinase   \n",
       "\n",
       "                                         enzyme_locations  \\\n",
       "150                                  [('ATPase', 22, 22)]   \n",
       "271           [('synthase', 34, 34), ('cyclase', 38, 38)]   \n",
       "280               [('ATPase', 11, 11), ('PPase', 15, 15)]   \n",
       "281               [('ATPase', 30, 30), ('PPase', 34, 34)]   \n",
       "286               [('ATPase', 27, 27), ('PPase', 31, 31)]   \n",
       "...                                                   ...   \n",
       "952414                                 [('ligase', 3, 3)]   \n",
       "952450                                  [('phase', 2, 2)]   \n",
       "952517                                [('phase', 16, 16)]   \n",
       "952526             [('ligase', 8, 8), ('ligase', 25, 25)]   \n",
       "952541  [('phosphotransferase', 12, 12), ('kinase', 15...   \n",
       "\n",
       "                                   chemical_entities_full  \\\n",
       "150     [{'text': 'diltiazem', 'start': 33192, 'end': ...   \n",
       "271     [{'text': 'Limonene', 'start': 2620, 'end': 26...   \n",
       "280     [{'text': 'H+', 'start': 2442, 'end': 2444, 's...   \n",
       "281     [{'text': 'H+', 'start': 2609, 'end': 2611, 's...   \n",
       "286     [{'text': 'H+', 'start': 3509, 'end': 3511, 's...   \n",
       "...                                                   ...   \n",
       "952414  [{'text': 'D-alanine-D-alanine', 'start': 1124...   \n",
       "952450  [{'text': 'methanol', 'start': 17213, 'end': 1...   \n",
       "952517  [{'text': 'sucrose', 'start': 27190, 'end': 27...   \n",
       "952526  [{'text': 'vancomycin', 'start': 28220, 'end':...   \n",
       "952541  [{'text': 'ATP', 'start': 29298, 'end': 29301,...   \n",
       "\n",
       "                                           chemical_names  \\\n",
       "150     diltiazem,, cyclosporin%20A,, dexamethasone,, ...   \n",
       "271     Limonene,, geranyl%20pyrophosphate,, sodium%20...   \n",
       "280                                          H+,, H+,, H+   \n",
       "281                                          H+,, H+,, H+   \n",
       "286                                          H+,, H+,, H+   \n",
       "...                                                   ...   \n",
       "952414                   D-alanine-D-alanine,, vancomycin   \n",
       "952450             methanol,, methanol,, tetrahydrofurane   \n",
       "952517                                     sucrose,, Kato   \n",
       "952526                    vancomycin,, D-alanyl-D-alanine   \n",
       "952541                        ATP,, AMP,, AMP,, adenylate   \n",
       "\n",
       "                                          chemical_smiles  \\\n",
       "150     COc1ccc(cc1)[C@@H]2Sc3ccccc3N(CCN(C)C)C(=O)[C@...   \n",
       "271     CC(=C)C1CCC(=CC1)C,, CC(C)=CCCC(/C)=C/CO[P]([O...   \n",
       "280                                    [H+],, [H+],, [H+]   \n",
       "281                                    [H+],, [H+],, [H+]   \n",
       "286                                    [H+],, [H+],, [H+]   \n",
       "...                                                   ...   \n",
       "952414  C[C@@H](N)C(O)=O.C[C@@H](N)C(O)=O,, CN[C@H](CC...   \n",
       "952450                                  CO,, CO,, C1CCOC1   \n",
       "952517  OC[C@H]1O[C@H](O[C@]2(CO)O[C@H](CO)[C@@H](O)[C...   \n",
       "952526  CN[C@H](CC(C)C)C(=O)NC1[C@H](O)c2ccc(Oc3cc4cc(...   \n",
       "952541  Nc1ncnc2n(cnc12)C3OC(CO[P](O)(=O)O[P](O)(=O)O[...   \n",
       "\n",
       "                                        name_smile_tuples  \n",
       "150     [('diltiazem', 'COc1ccc(cc1)[C@@H]2Sc3ccccc3N(...  \n",
       "271     [('Limonene', 'CC(=C)C1CCC(=CC1)C'), ('geranyl...  \n",
       "280      [('H+', '[H+]'), ('H+', '[H+]'), ('H+', '[H+]')]  \n",
       "281      [('H+', '[H+]'), ('H+', '[H+]'), ('H+', '[H+]')]  \n",
       "286      [('H+', '[H+]'), ('H+', '[H+]'), ('H+', '[H+]')]  \n",
       "...                                                   ...  \n",
       "952414  [('D-alanine-D-alanine', 'C[C@@H](N)C(O)=O.C[C...  \n",
       "952450  [('methanol', 'CO'), ('methanol', 'CO'), ('tet...  \n",
       "952517  [('sucrose', 'OC[C@H]1O[C@H](O[C@]2(CO)O[C@H](...  \n",
       "952526  [('vancomycin', 'CN[C@H](CC(C)C)C(=O)NC1[C@H](...  \n",
       "952541  [('ATP', 'Nc1ncnc2n(cnc12)C3OC(CO[P](O)(=O)O[P...  \n",
       "\n",
       "[50212 rows x 12 columns]"
      ]
     },
     "execution_count": 150,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "uncleaned_no_na_df = uncleaned_df.dropna()\n",
    "uncleaned_several_chem_df = uncleaned_no_na_df.loc[uncleaned_no_na_df[\"chemical_names\"].str.contains(\",\")]\n",
    "#uncleaned_several_chem_df = uncleaned_several_chem_df[[\"sentence\", \"chemical_names\"]]\n",
    "uncleaned_several_chem_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-151-6aebe82620ff>:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  uncleaned_several_chem_df[\"sentence\"] = uncleaned_several_chem_df[\"sentence\"].apply(sentence_cleaned)\n",
      "<ipython-input-151-6aebe82620ff>:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  uncleaned_several_chem_df[\"chemical_names\"] = uncleaned_several_chem_df[\"chemical_names\"].apply(chem_into_array)\n"
     ]
    }
   ],
   "source": [
    "uncleaned_several_chem_df[\"sentence\"] = uncleaned_several_chem_df[\"sentence\"].apply(sentence_cleaned)\n",
    "uncleaned_several_chem_df[\"chemical_names\"] = uncleaned_several_chem_df[\"chemical_names\"].apply(chem_into_array)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "metadata": {},
   "outputs": [],
   "source": [
    "# need to get rid of this line if this data should be used\n",
    "re_check_several_chem_uncleaned_df = uncleaned_several_chem_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-153-d41bf52880fc>:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  re_check_several_chem_uncleaned_df[\"text\"] = re_check_several_chem_uncleaned_df[[\"sentence\", \"chemical_names\"]].values.tolist()\n"
     ]
    }
   ],
   "source": [
    "re_check_several_chem_uncleaned_df[\"text\"] = re_check_several_chem_uncleaned_df[[\"sentence\", \"chemical_names\"]].values.tolist()\n",
    "brenda_clean_df = re_check_several_chem_uncleaned_df\n",
    "brenda_mult_chem_df = brenda_clean_df[brenda_clean_df['chemical_names'].map(len) > 1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 154,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>lit_id</th>\n",
       "      <th>indices</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>sentence</th>\n",
       "      <th>sentence_pos</th>\n",
       "      <th>enzymes</th>\n",
       "      <th>enzyme_locations</th>\n",
       "      <th>chemical_entities_full</th>\n",
       "      <th>chemical_names</th>\n",
       "      <th>chemical_smiles</th>\n",
       "      <th>name_smile_tuples</th>\n",
       "      <th>text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>10.1002/jps.20686</td>\n",
       "      <td>150</td>\n",
       "      <td>33180</td>\n",
       "      <td>33589</td>\n",
       "      <td>we selected diltiazem, cyclosporin a, and dexa...</td>\n",
       "      <td>[('We', 'PRP'), ('selected', 'VBD'), ('diltiaz...</td>\n",
       "      <td>ATPase</td>\n",
       "      <td>[('ATPase', 22, 22)]</td>\n",
       "      <td>[{'text': 'diltiazem', 'start': 33192, 'end': ...</td>\n",
       "      <td>[cyclosporin a, diltiazem, dexamethasone, calc...</td>\n",
       "      <td>COc1ccc(cc1)[C@@H]2Sc3ccccc3N(CCN(C)C)C(=O)[C@...</td>\n",
       "      <td>[('diltiazem', 'COc1ccc(cc1)[C@@H]2Sc3ccccc3N(...</td>\n",
       "      <td>[we selected diltiazem, cyclosporin a, and dex...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271</th>\n",
       "      <td>10.1006/abbi.1993.1114</td>\n",
       "      <td>3</td>\n",
       "      <td>2464</td>\n",
       "      <td>2912</td>\n",
       "      <td>croteau r washington state univ, inst biol che...</td>\n",
       "      <td>[('Croteau', 'NNP'), ('R.', 'NNP'), ('Washingt...</td>\n",
       "      <td>synthase, cyclase</td>\n",
       "      <td>[('synthase', 34, 34), ('cyclase', 38, 38)]</td>\n",
       "      <td>[{'text': 'Limonene', 'start': 2620, 'end': 26...</td>\n",
       "      <td>[limonene, sodium dodecyl sulfate, geranyl pyr...</td>\n",
       "      <td>CC(=C)C1CCC(=CC1)C,, CC(C)=CCCC(/C)=C/CO[P]([O...</td>\n",
       "      <td>[('Limonene', 'CC(=C)C1CCC(=CC1)C'), ('geranyl...</td>\n",
       "      <td>[croteau r washington state univ, inst biol ch...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>295</th>\n",
       "      <td>10.1006/abbi.1993.1275</td>\n",
       "      <td>2</td>\n",
       "      <td>1178</td>\n",
       "      <td>2551</td>\n",
       "      <td>chemicalmodificationschickenliverpyruvatecarbo...</td>\n",
       "      <td>[('CHEMICALMODIFICATIONSCHICKENLIVERPYRUVATECA...</td>\n",
       "      <td>Carboxylase</td>\n",
       "      <td>[('Carboxylase', 143, 143)]</td>\n",
       "      <td>[{'text': 'Pyruvate', 'start': 2438, 'end': 24...</td>\n",
       "      <td>[lysine, pyruvate, cysteine]</td>\n",
       "      <td>CC(=O)C([O-])=O,, N[C@@H](CS)C(O)=O,, NCCCC[C@...</td>\n",
       "      <td>[('Pyruvate', 'CC(=O)C([O-])=O'), ('Cysteine',...</td>\n",
       "      <td>[chemicalmodificationschickenliverpyruvatecarb...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>296</th>\n",
       "      <td>10.1006/abbi.1993.1275</td>\n",
       "      <td>3</td>\n",
       "      <td>2552</td>\n",
       "      <td>2895</td>\n",
       "      <td>ash de temple univ, hlth sci ctr, sch med, dep...</td>\n",
       "      <td>[('Ash', 'NNP'), ('D.E.', 'NNP'), ('Temple', '...</td>\n",
       "      <td>carboxylase</td>\n",
       "      <td>[('carboxylase', 32, 32)]</td>\n",
       "      <td>[{'text': 'pyruvate', 'start': 2696, 'end': 27...</td>\n",
       "      <td>[cysteine, n-(7-dimethylamino-4-methyl-3-couma...</td>\n",
       "      <td>CC(=O)C([O-])=O,, CN(C)c1ccc2C(=C(N3C(=O)C=CC3...</td>\n",
       "      <td>[('pyruvate', 'CC(=O)C([O-])=O'), ('N-(7-dimet...</td>\n",
       "      <td>[ash de temple univ, hlth sci ctr, sch med, de...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>298</th>\n",
       "      <td>10.1006/abbi.1993.1275</td>\n",
       "      <td>5</td>\n",
       "      <td>3008</td>\n",
       "      <td>3317</td>\n",
       "      <td>at a one- to two-fold molar excess over active...</td>\n",
       "      <td>[('At', 'IN'), ('a', 'DT'), ('one-', 'CD'), ('...</td>\n",
       "      <td>carboxylase, decarboxylase</td>\n",
       "      <td>[('carboxylase', 20, 20), ('decarboxylase', 28...</td>\n",
       "      <td>[{'text': 'pyruvate', 'start': 3120, 'end': 31...</td>\n",
       "      <td>[pyruvate, oxaloacetate, adp]</td>\n",
       "      <td>CC(=O)C([O-])=O,, Nc1ncnc2n(cnc12)C3OC(CO[P](O...</td>\n",
       "      <td>[('pyruvate', 'CC(=O)C([O-])=O'), ('ADP', 'Nc1...</td>\n",
       "      <td>[at a one- to two-fold molar excess over activ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>952414</th>\n",
       "      <td>10.1263/jbb.99.623</td>\n",
       "      <td>26</td>\n",
       "      <td>11239</td>\n",
       "      <td>11493</td>\n",
       "      <td>key words d-alanine-d-alanine ligase d-amino a...</td>\n",
       "      <td>[('Key', 'JJ'), ('words', 'NNS'), ('D-alanine-...</td>\n",
       "      <td>ligase</td>\n",
       "      <td>[('ligase', 3, 3)]</td>\n",
       "      <td>[{'text': 'D-alanine-D-alanine', 'start': 1124...</td>\n",
       "      <td>[vancomycin, d-alanine-d-alanine]</td>\n",
       "      <td>C[C@@H](N)C(O)=O.C[C@@H](N)C(O)=O,, CN[C@H](CC...</td>\n",
       "      <td>[('D-alanine-D-alanine', 'C[C@@H](N)C(O)=O.C[C...</td>\n",
       "      <td>[key words d-alanine-d-alanine ligase d-amino ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>952450</th>\n",
       "      <td>10.1263/jbb.99.623</td>\n",
       "      <td>62</td>\n",
       "      <td>17119</td>\n",
       "      <td>17420</td>\n",
       "      <td>the mobile phase consisted of a linear gradien...</td>\n",
       "      <td>[('The', 'DT'), ('mobile', 'JJ'), ('phase', 'N...</td>\n",
       "      <td>phase</td>\n",
       "      <td>[('phase', 2, 2)]</td>\n",
       "      <td>[{'text': 'methanol', 'start': 17213, 'end': 1...</td>\n",
       "      <td>[tetrahydrofurane, methanol]</td>\n",
       "      <td>CO,, CO,, C1CCOC1</td>\n",
       "      <td>[('methanol', 'CO'), ('methanol', 'CO'), ('tet...</td>\n",
       "      <td>[the mobile phase consisted of a linear gradie...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>952517</th>\n",
       "      <td>10.1263/jbb.99.623</td>\n",
       "      <td>129</td>\n",
       "      <td>27150</td>\n",
       "      <td>27373</td>\n",
       "      <td>weber v falkenhagen d subpol a novel sucrose-b...</td>\n",
       "      <td>[('Weber', 'NNP'), ('V.', 'NNP'), ('Falkenhage...</td>\n",
       "      <td>phase</td>\n",
       "      <td>[('phase', 16, 16)]</td>\n",
       "      <td>[{'text': 'sucrose', 'start': 27190, 'end': 27...</td>\n",
       "      <td>[kato, sucrose]</td>\n",
       "      <td>OC[C@H]1O[C@H](O[C@]2(CO)O[C@H](CO)[C@@H](O)[C...</td>\n",
       "      <td>[('sucrose', 'OC[C@H]1O[C@H](O[C@]2(CO)O[C@H](...</td>\n",
       "      <td>[weber v falkenhagen d subpol a novel sucrose-...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>952526</th>\n",
       "      <td>10.1263/jbb.99.623</td>\n",
       "      <td>138</td>\n",
       "      <td>28139</td>\n",
       "      <td>28364</td>\n",
       "      <td>active-site mutants of the vanc2 d-alanyl-d-se...</td>\n",
       "      <td>[('Active', 'JJ'), ('-', 'HYPH'), ('site', 'NN...</td>\n",
       "      <td>ligase, ligase</td>\n",
       "      <td>[('ligase', 8, 8), ('ligase', 25, 25)]</td>\n",
       "      <td>[{'text': 'vancomycin', 'start': 28220, 'end':...</td>\n",
       "      <td>[vancomycin, d-alanyl-d-alanine]</td>\n",
       "      <td>CN[C@H](CC(C)C)C(=O)NC1[C@H](O)c2ccc(Oc3cc4cc(...</td>\n",
       "      <td>[('vancomycin', 'CN[C@H](CC(C)C)C(=O)NC1[C@H](...</td>\n",
       "      <td>[active-site mutants of the vanc2 d-alanyl-d-s...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>952541</th>\n",
       "      <td>10.1263/jbb.99.623</td>\n",
       "      <td>153</td>\n",
       "      <td>29289</td>\n",
       "      <td>29564</td>\n",
       "      <td>in vitro atp regeneration from polyphosphate a...</td>\n",
       "      <td>[('In', 'FW'), ('vitro', 'FW'), ('ATP', 'NN'),...</td>\n",
       "      <td>phosphotransferase, kinase</td>\n",
       "      <td>[('phosphotransferase', 12, 12), ('kinase', 15...</td>\n",
       "      <td>[{'text': 'ATP', 'start': 29298, 'end': 29301,...</td>\n",
       "      <td>[adenylate, atp, amp]</td>\n",
       "      <td>Nc1ncnc2n(cnc12)C3OC(CO[P](O)(=O)O[P](O)(=O)O[...</td>\n",
       "      <td>[('ATP', 'Nc1ncnc2n(cnc12)C3OC(CO[P](O)(=O)O[P...</td>\n",
       "      <td>[in vitro atp regeneration from polyphosphate ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>44724 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                        lit_id  indices  start    end  \\\n",
       "150          10.1002/jps.20686      150  33180  33589   \n",
       "271     10.1006/abbi.1993.1114        3   2464   2912   \n",
       "295     10.1006/abbi.1993.1275        2   1178   2551   \n",
       "296     10.1006/abbi.1993.1275        3   2552   2895   \n",
       "298     10.1006/abbi.1993.1275        5   3008   3317   \n",
       "...                        ...      ...    ...    ...   \n",
       "952414      10.1263/jbb.99.623       26  11239  11493   \n",
       "952450      10.1263/jbb.99.623       62  17119  17420   \n",
       "952517      10.1263/jbb.99.623      129  27150  27373   \n",
       "952526      10.1263/jbb.99.623      138  28139  28364   \n",
       "952541      10.1263/jbb.99.623      153  29289  29564   \n",
       "\n",
       "                                                 sentence  \\\n",
       "150     we selected diltiazem, cyclosporin a, and dexa...   \n",
       "271     croteau r washington state univ, inst biol che...   \n",
       "295     chemicalmodificationschickenliverpyruvatecarbo...   \n",
       "296     ash de temple univ, hlth sci ctr, sch med, dep...   \n",
       "298     at a one- to two-fold molar excess over active...   \n",
       "...                                                   ...   \n",
       "952414  key words d-alanine-d-alanine ligase d-amino a...   \n",
       "952450  the mobile phase consisted of a linear gradien...   \n",
       "952517  weber v falkenhagen d subpol a novel sucrose-b...   \n",
       "952526  active-site mutants of the vanc2 d-alanyl-d-se...   \n",
       "952541  in vitro atp regeneration from polyphosphate a...   \n",
       "\n",
       "                                             sentence_pos  \\\n",
       "150     [('We', 'PRP'), ('selected', 'VBD'), ('diltiaz...   \n",
       "271     [('Croteau', 'NNP'), ('R.', 'NNP'), ('Washingt...   \n",
       "295     [('CHEMICALMODIFICATIONSCHICKENLIVERPYRUVATECA...   \n",
       "296     [('Ash', 'NNP'), ('D.E.', 'NNP'), ('Temple', '...   \n",
       "298     [('At', 'IN'), ('a', 'DT'), ('one-', 'CD'), ('...   \n",
       "...                                                   ...   \n",
       "952414  [('Key', 'JJ'), ('words', 'NNS'), ('D-alanine-...   \n",
       "952450  [('The', 'DT'), ('mobile', 'JJ'), ('phase', 'N...   \n",
       "952517  [('Weber', 'NNP'), ('V.', 'NNP'), ('Falkenhage...   \n",
       "952526  [('Active', 'JJ'), ('-', 'HYPH'), ('site', 'NN...   \n",
       "952541  [('In', 'FW'), ('vitro', 'FW'), ('ATP', 'NN'),...   \n",
       "\n",
       "                           enzymes  \\\n",
       "150                         ATPase   \n",
       "271              synthase, cyclase   \n",
       "295                    Carboxylase   \n",
       "296                    carboxylase   \n",
       "298     carboxylase, decarboxylase   \n",
       "...                            ...   \n",
       "952414                      ligase   \n",
       "952450                       phase   \n",
       "952517                       phase   \n",
       "952526              ligase, ligase   \n",
       "952541  phosphotransferase, kinase   \n",
       "\n",
       "                                         enzyme_locations  \\\n",
       "150                                  [('ATPase', 22, 22)]   \n",
       "271           [('synthase', 34, 34), ('cyclase', 38, 38)]   \n",
       "295                           [('Carboxylase', 143, 143)]   \n",
       "296                             [('carboxylase', 32, 32)]   \n",
       "298     [('carboxylase', 20, 20), ('decarboxylase', 28...   \n",
       "...                                                   ...   \n",
       "952414                                 [('ligase', 3, 3)]   \n",
       "952450                                  [('phase', 2, 2)]   \n",
       "952517                                [('phase', 16, 16)]   \n",
       "952526             [('ligase', 8, 8), ('ligase', 25, 25)]   \n",
       "952541  [('phosphotransferase', 12, 12), ('kinase', 15...   \n",
       "\n",
       "                                   chemical_entities_full  \\\n",
       "150     [{'text': 'diltiazem', 'start': 33192, 'end': ...   \n",
       "271     [{'text': 'Limonene', 'start': 2620, 'end': 26...   \n",
       "295     [{'text': 'Pyruvate', 'start': 2438, 'end': 24...   \n",
       "296     [{'text': 'pyruvate', 'start': 2696, 'end': 27...   \n",
       "298     [{'text': 'pyruvate', 'start': 3120, 'end': 31...   \n",
       "...                                                   ...   \n",
       "952414  [{'text': 'D-alanine-D-alanine', 'start': 1124...   \n",
       "952450  [{'text': 'methanol', 'start': 17213, 'end': 1...   \n",
       "952517  [{'text': 'sucrose', 'start': 27190, 'end': 27...   \n",
       "952526  [{'text': 'vancomycin', 'start': 28220, 'end':...   \n",
       "952541  [{'text': 'ATP', 'start': 29298, 'end': 29301,...   \n",
       "\n",
       "                                           chemical_names  \\\n",
       "150     [cyclosporin a, diltiazem, dexamethasone, calc...   \n",
       "271     [limonene, sodium dodecyl sulfate, geranyl pyr...   \n",
       "295                          [lysine, pyruvate, cysteine]   \n",
       "296     [cysteine, n-(7-dimethylamino-4-methyl-3-couma...   \n",
       "298                         [pyruvate, oxaloacetate, adp]   \n",
       "...                                                   ...   \n",
       "952414                  [vancomycin, d-alanine-d-alanine]   \n",
       "952450                       [tetrahydrofurane, methanol]   \n",
       "952517                                    [kato, sucrose]   \n",
       "952526                   [vancomycin, d-alanyl-d-alanine]   \n",
       "952541                              [adenylate, atp, amp]   \n",
       "\n",
       "                                          chemical_smiles  \\\n",
       "150     COc1ccc(cc1)[C@@H]2Sc3ccccc3N(CCN(C)C)C(=O)[C@...   \n",
       "271     CC(=C)C1CCC(=CC1)C,, CC(C)=CCCC(/C)=C/CO[P]([O...   \n",
       "295     CC(=O)C([O-])=O,, N[C@@H](CS)C(O)=O,, NCCCC[C@...   \n",
       "296     CC(=O)C([O-])=O,, CN(C)c1ccc2C(=C(N3C(=O)C=CC3...   \n",
       "298     CC(=O)C([O-])=O,, Nc1ncnc2n(cnc12)C3OC(CO[P](O...   \n",
       "...                                                   ...   \n",
       "952414  C[C@@H](N)C(O)=O.C[C@@H](N)C(O)=O,, CN[C@H](CC...   \n",
       "952450                                  CO,, CO,, C1CCOC1   \n",
       "952517  OC[C@H]1O[C@H](O[C@]2(CO)O[C@H](CO)[C@@H](O)[C...   \n",
       "952526  CN[C@H](CC(C)C)C(=O)NC1[C@H](O)c2ccc(Oc3cc4cc(...   \n",
       "952541  Nc1ncnc2n(cnc12)C3OC(CO[P](O)(=O)O[P](O)(=O)O[...   \n",
       "\n",
       "                                        name_smile_tuples  \\\n",
       "150     [('diltiazem', 'COc1ccc(cc1)[C@@H]2Sc3ccccc3N(...   \n",
       "271     [('Limonene', 'CC(=C)C1CCC(=CC1)C'), ('geranyl...   \n",
       "295     [('Pyruvate', 'CC(=O)C([O-])=O'), ('Cysteine',...   \n",
       "296     [('pyruvate', 'CC(=O)C([O-])=O'), ('N-(7-dimet...   \n",
       "298     [('pyruvate', 'CC(=O)C([O-])=O'), ('ADP', 'Nc1...   \n",
       "...                                                   ...   \n",
       "952414  [('D-alanine-D-alanine', 'C[C@@H](N)C(O)=O.C[C...   \n",
       "952450  [('methanol', 'CO'), ('methanol', 'CO'), ('tet...   \n",
       "952517  [('sucrose', 'OC[C@H]1O[C@H](O[C@]2(CO)O[C@H](...   \n",
       "952526  [('vancomycin', 'CN[C@H](CC(C)C)C(=O)NC1[C@H](...   \n",
       "952541  [('ATP', 'Nc1ncnc2n(cnc12)C3OC(CO[P](O)(=O)O[P...   \n",
       "\n",
       "                                                     text  \n",
       "150     [we selected diltiazem, cyclosporin a, and dex...  \n",
       "271     [croteau r washington state univ, inst biol ch...  \n",
       "295     [chemicalmodificationschickenliverpyruvatecarb...  \n",
       "296     [ash de temple univ, hlth sci ctr, sch med, de...  \n",
       "298     [at a one- to two-fold molar excess over activ...  \n",
       "...                                                   ...  \n",
       "952414  [key words d-alanine-d-alanine ligase d-amino ...  \n",
       "952450  [the mobile phase consisted of a linear gradie...  \n",
       "952517  [weber v falkenhagen d subpol a novel sucrose-...  \n",
       "952526  [active-site mutants of the vanc2 d-alanyl-d-s...  \n",
       "952541  [in vitro atp regeneration from polyphosphate ...  \n",
       "\n",
       "[44724 rows x 13 columns]"
      ]
     },
     "execution_count": 154,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "brenda_mult_chem_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### snorkel code for model (running on data with truth values)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [],
   "source": [
    "from snorkel.labeling import PandasLFApplier"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [],
   "source": [
    "# how to find the location\n",
    "# sentence_chem_df[sentence_chem_df[\"sentence\"] == \"even though the activities of mat and gnmt were elevated, the concentration of liver s-adenosylmethionine was decreased 24%, p<0001 and s-adenosylhomocysteine increased 113%, p<0001 in the dwarf mice\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sentence</th>\n",
       "      <th>chemicals</th>\n",
       "      <th>truth</th>\n",
       "      <th>substrates</th>\n",
       "      <th>products</th>\n",
       "      <th>text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>the enzyme cyclo-oxygenase catalyses the oxyge...</td>\n",
       "      <td>[prostaglandins, arachidonic acid]</td>\n",
       "      <td>1</td>\n",
       "      <td>arachidonic, acid</td>\n",
       "      <td>prostaglandins</td>\n",
       "      <td>[the enzyme cyclo-oxygenase catalyses the oxyg...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>although the exact site of decarboxylation of ...</td>\n",
       "      <td>[dopamine, levodopa]</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[although the exact site of decarboxylation of...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>the enzyme dimethylarginine dimethylaminohydro...</td>\n",
       "      <td>[arginine, methylamines, dimethylarginine, cit...</td>\n",
       "      <td>1</td>\n",
       "      <td>arginine</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[the enzyme dimethylarginine dimethylaminohydr...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>background and aims glutamic acid decarboxylas...</td>\n",
       "      <td>[glutamate, gaba]</td>\n",
       "      <td>1</td>\n",
       "      <td>glutamate</td>\n",
       "      <td>gamma-aminobutyric, acid, GABA</td>\n",
       "      <td>[background and aims glutamic acid decarboxyla...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>mazindane 26 was found to be a pro-drug, oxidi...</td>\n",
       "      <td>[mazindane, mazindol, 5-h, 5-oh]</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[mazindane 26 was found to be a pro-drug, oxid...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            sentence  \\\n",
       "0  the enzyme cyclo-oxygenase catalyses the oxyge...   \n",
       "1  although the exact site of decarboxylation of ...   \n",
       "2  the enzyme dimethylarginine dimethylaminohydro...   \n",
       "3  background and aims glutamic acid decarboxylas...   \n",
       "4  mazindane 26 was found to be a pro-drug, oxidi...   \n",
       "\n",
       "                                           chemicals  truth  \\\n",
       "0                 [prostaglandins, arachidonic acid]      1   \n",
       "1                               [dopamine, levodopa]      1   \n",
       "2  [arginine, methylamines, dimethylarginine, cit...      1   \n",
       "3                                  [glutamate, gaba]      1   \n",
       "4                   [mazindane, mazindol, 5-h, 5-oh]      1   \n",
       "\n",
       "          substrates                        products  \\\n",
       "0  arachidonic, acid                  prostaglandins   \n",
       "1                NaN                             NaN   \n",
       "2           arginine                             NaN   \n",
       "3          glutamate  gamma-aminobutyric, acid, GABA   \n",
       "4                NaN                             NaN   \n",
       "\n",
       "                                                text  \n",
       "0  [the enzyme cyclo-oxygenase catalyses the oxyg...  \n",
       "1  [although the exact site of decarboxylation of...  \n",
       "2  [the enzyme dimethylarginine dimethylaminohydr...  \n",
       "3  [background and aims glutamic acid decarboxyla...  \n",
       "4  [mazindane 26 was found to be a pro-drug, oxid...  "
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sentence_chem_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>lit_id</th>\n",
       "      <th>indices</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>sentence</th>\n",
       "      <th>sentence_pos</th>\n",
       "      <th>enzymes</th>\n",
       "      <th>enzyme_locations</th>\n",
       "      <th>chemical_entities_full</th>\n",
       "      <th>chemical_names</th>\n",
       "      <th>chemical_smiles</th>\n",
       "      <th>name_smile_tuples</th>\n",
       "      <th>text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10.1002/jps.20686</td>\n",
       "      <td>150</td>\n",
       "      <td>33180</td>\n",
       "      <td>33589</td>\n",
       "      <td>we selected diltiazem, cyclosporin a, and dexa...</td>\n",
       "      <td>[('We', 'PRP'), ('selected', 'VBD'), ('diltiaz...</td>\n",
       "      <td>ATPase</td>\n",
       "      <td>[('ATPase', 22, 22)]</td>\n",
       "      <td>[{'text': 'diltiazem', 'start': 33192, 'end': ...</td>\n",
       "      <td>[cyclosporin a, diltiazem, dexamethasone, calc...</td>\n",
       "      <td>COc1ccc(cc1)[C@@H]2Sc3ccccc3N(CCN(C)C)C(=O)[C@...</td>\n",
       "      <td>[('diltiazem', 'COc1ccc(cc1)[C@@H]2Sc3ccccc3N(...</td>\n",
       "      <td>[we selected diltiazem, cyclosporin a, and dex...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10.1006/abbi.1993.1114</td>\n",
       "      <td>3</td>\n",
       "      <td>2464</td>\n",
       "      <td>2912</td>\n",
       "      <td>croteau r washington state univ, inst biol che...</td>\n",
       "      <td>[('Croteau', 'NNP'), ('R.', 'NNP'), ('Washingt...</td>\n",
       "      <td>synthase, cyclase</td>\n",
       "      <td>[('synthase', 34, 34), ('cyclase', 38, 38)]</td>\n",
       "      <td>[{'text': 'Limonene', 'start': 2620, 'end': 26...</td>\n",
       "      <td>[limonene, sodium dodecyl sulfate, geranyl pyr...</td>\n",
       "      <td>CC(=C)C1CCC(=CC1)C,, CC(C)=CCCC(/C)=C/CO[P]([O...</td>\n",
       "      <td>[('Limonene', 'CC(=C)C1CCC(=CC1)C'), ('geranyl...</td>\n",
       "      <td>[croteau r washington state univ, inst biol ch...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10.1006/abbi.1993.1275</td>\n",
       "      <td>2</td>\n",
       "      <td>1178</td>\n",
       "      <td>2551</td>\n",
       "      <td>chemicalmodificationschickenliverpyruvatecarbo...</td>\n",
       "      <td>[('CHEMICALMODIFICATIONSCHICKENLIVERPYRUVATECA...</td>\n",
       "      <td>Carboxylase</td>\n",
       "      <td>[('Carboxylase', 143, 143)]</td>\n",
       "      <td>[{'text': 'Pyruvate', 'start': 2438, 'end': 24...</td>\n",
       "      <td>[lysine, pyruvate, cysteine]</td>\n",
       "      <td>CC(=O)C([O-])=O,, N[C@@H](CS)C(O)=O,, NCCCC[C@...</td>\n",
       "      <td>[('Pyruvate', 'CC(=O)C([O-])=O'), ('Cysteine',...</td>\n",
       "      <td>[chemicalmodificationschickenliverpyruvatecarb...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10.1006/abbi.1993.1275</td>\n",
       "      <td>3</td>\n",
       "      <td>2552</td>\n",
       "      <td>2895</td>\n",
       "      <td>ash de temple univ, hlth sci ctr, sch med, dep...</td>\n",
       "      <td>[('Ash', 'NNP'), ('D.E.', 'NNP'), ('Temple', '...</td>\n",
       "      <td>carboxylase</td>\n",
       "      <td>[('carboxylase', 32, 32)]</td>\n",
       "      <td>[{'text': 'pyruvate', 'start': 2696, 'end': 27...</td>\n",
       "      <td>[cysteine, n-(7-dimethylamino-4-methyl-3-couma...</td>\n",
       "      <td>CC(=O)C([O-])=O,, CN(C)c1ccc2C(=C(N3C(=O)C=CC3...</td>\n",
       "      <td>[('pyruvate', 'CC(=O)C([O-])=O'), ('N-(7-dimet...</td>\n",
       "      <td>[ash de temple univ, hlth sci ctr, sch med, de...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10.1006/abbi.1993.1275</td>\n",
       "      <td>5</td>\n",
       "      <td>3008</td>\n",
       "      <td>3317</td>\n",
       "      <td>at a one- to two-fold molar excess over active...</td>\n",
       "      <td>[('At', 'IN'), ('a', 'DT'), ('one-', 'CD'), ('...</td>\n",
       "      <td>carboxylase, decarboxylase</td>\n",
       "      <td>[('carboxylase', 20, 20), ('decarboxylase', 28...</td>\n",
       "      <td>[{'text': 'pyruvate', 'start': 3120, 'end': 31...</td>\n",
       "      <td>[pyruvate, oxaloacetate, adp]</td>\n",
       "      <td>CC(=O)C([O-])=O,, Nc1ncnc2n(cnc12)C3OC(CO[P](O...</td>\n",
       "      <td>[('pyruvate', 'CC(=O)C([O-])=O'), ('ADP', 'Nc1...</td>\n",
       "      <td>[at a one- to two-fold molar excess over activ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   lit_id  indices  start    end  \\\n",
       "0       10.1002/jps.20686      150  33180  33589   \n",
       "1  10.1006/abbi.1993.1114        3   2464   2912   \n",
       "2  10.1006/abbi.1993.1275        2   1178   2551   \n",
       "3  10.1006/abbi.1993.1275        3   2552   2895   \n",
       "4  10.1006/abbi.1993.1275        5   3008   3317   \n",
       "\n",
       "                                            sentence  \\\n",
       "0  we selected diltiazem, cyclosporin a, and dexa...   \n",
       "1  croteau r washington state univ, inst biol che...   \n",
       "2  chemicalmodificationschickenliverpyruvatecarbo...   \n",
       "3  ash de temple univ, hlth sci ctr, sch med, dep...   \n",
       "4  at a one- to two-fold molar excess over active...   \n",
       "\n",
       "                                        sentence_pos  \\\n",
       "0  [('We', 'PRP'), ('selected', 'VBD'), ('diltiaz...   \n",
       "1  [('Croteau', 'NNP'), ('R.', 'NNP'), ('Washingt...   \n",
       "2  [('CHEMICALMODIFICATIONSCHICKENLIVERPYRUVATECA...   \n",
       "3  [('Ash', 'NNP'), ('D.E.', 'NNP'), ('Temple', '...   \n",
       "4  [('At', 'IN'), ('a', 'DT'), ('one-', 'CD'), ('...   \n",
       "\n",
       "                      enzymes  \\\n",
       "0                      ATPase   \n",
       "1           synthase, cyclase   \n",
       "2                 Carboxylase   \n",
       "3                 carboxylase   \n",
       "4  carboxylase, decarboxylase   \n",
       "\n",
       "                                    enzyme_locations  \\\n",
       "0                               [('ATPase', 22, 22)]   \n",
       "1        [('synthase', 34, 34), ('cyclase', 38, 38)]   \n",
       "2                        [('Carboxylase', 143, 143)]   \n",
       "3                          [('carboxylase', 32, 32)]   \n",
       "4  [('carboxylase', 20, 20), ('decarboxylase', 28...   \n",
       "\n",
       "                              chemical_entities_full  \\\n",
       "0  [{'text': 'diltiazem', 'start': 33192, 'end': ...   \n",
       "1  [{'text': 'Limonene', 'start': 2620, 'end': 26...   \n",
       "2  [{'text': 'Pyruvate', 'start': 2438, 'end': 24...   \n",
       "3  [{'text': 'pyruvate', 'start': 2696, 'end': 27...   \n",
       "4  [{'text': 'pyruvate', 'start': 3120, 'end': 31...   \n",
       "\n",
       "                                      chemical_names  \\\n",
       "0  [cyclosporin a, diltiazem, dexamethasone, calc...   \n",
       "1  [limonene, sodium dodecyl sulfate, geranyl pyr...   \n",
       "2                       [lysine, pyruvate, cysteine]   \n",
       "3  [cysteine, n-(7-dimethylamino-4-methyl-3-couma...   \n",
       "4                      [pyruvate, oxaloacetate, adp]   \n",
       "\n",
       "                                     chemical_smiles  \\\n",
       "0  COc1ccc(cc1)[C@@H]2Sc3ccccc3N(CCN(C)C)C(=O)[C@...   \n",
       "1  CC(=C)C1CCC(=CC1)C,, CC(C)=CCCC(/C)=C/CO[P]([O...   \n",
       "2  CC(=O)C([O-])=O,, N[C@@H](CS)C(O)=O,, NCCCC[C@...   \n",
       "3  CC(=O)C([O-])=O,, CN(C)c1ccc2C(=C(N3C(=O)C=CC3...   \n",
       "4  CC(=O)C([O-])=O,, Nc1ncnc2n(cnc12)C3OC(CO[P](O...   \n",
       "\n",
       "                                   name_smile_tuples  \\\n",
       "0  [('diltiazem', 'COc1ccc(cc1)[C@@H]2Sc3ccccc3N(...   \n",
       "1  [('Limonene', 'CC(=C)C1CCC(=CC1)C'), ('geranyl...   \n",
       "2  [('Pyruvate', 'CC(=O)C([O-])=O'), ('Cysteine',...   \n",
       "3  [('pyruvate', 'CC(=O)C([O-])=O'), ('N-(7-dimet...   \n",
       "4  [('pyruvate', 'CC(=O)C([O-])=O'), ('ADP', 'Nc1...   \n",
       "\n",
       "                                                text  \n",
       "0  [we selected diltiazem, cyclosporin a, and dex...  \n",
       "1  [croteau r washington state univ, inst biol ch...  \n",
       "2  [chemicalmodificationschickenliverpyruvatecarb...  \n",
       "3  [ash de temple univ, hlth sci ctr, sch med, de...  \n",
       "4  [at a one- to two-fold molar excess over activ...  "
      ]
     },
     "execution_count": 155,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "brenda_mult_chem_df = brenda_mult_chem_df.reset_index(drop=True)\n",
    "brenda_mult_chem_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "Operator names not unique: 2 operators with name structure_sep_via",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-81-c63b36a1e46b>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0;31m#df = df of interest\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      6\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 7\u001b[0;31m \u001b[0mapplier\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mPandasLFApplier\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlfs\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mall_lfs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      8\u001b[0m \u001b[0mL_train\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mapplier\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mapply\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdf\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mdf\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      9\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/anaconda3/lib/python3.8/site-packages/snorkel/labeling/apply/core.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, lfs)\u001b[0m\n\u001b[1;32m     58\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_lfs\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlfs\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     59\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_lf_names\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0mlf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mname\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mlf\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mlfs\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 60\u001b[0;31m         \u001b[0mcheck_unique_names\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_lf_names\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     61\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     62\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_numpy_from_row_data\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlabels\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mList\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mRowData\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m->\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mndarray\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/anaconda3/lib/python3.8/site-packages/snorkel/utils/data_operators.py\u001b[0m in \u001b[0;36mcheck_unique_names\u001b[0;34m(names)\u001b[0m\n\u001b[1;32m      7\u001b[0m     \u001b[0mk\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mct\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mCounter\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mnames\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmost_common\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      8\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mct\u001b[0m \u001b[0;34m>\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 9\u001b[0;31m         \u001b[0;32mraise\u001b[0m \u001b[0mValueError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34mf\"Operator names not unique: {ct} operators with name {k}\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m: Operator names not unique: 2 operators with name structure_sep_via"
     ]
    }
   ],
   "source": [
    "from snorkel.labeling import PandasLFApplier\n",
    "from snorkel.labeling.model import MajorityLabelVoter\n",
    "\n",
    "#lfs = [add your list of labeling functions]\n",
    "#df = df of interest\n",
    "\n",
    "applier = PandasLFApplier(lfs=all_lfs)\n",
    "L_train = applier.apply(df=df)\n",
    "\n",
    "majority_model = MajorityLabelVoter()\n",
    "df[\"label_voter\"] = majority_model.predict(L=L_train)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/anaconda3/lib/python3.8/site-packages/tqdm/std.py:668: FutureWarning: The Panel class is removed from pandas. Accessing it from the top-level namespace will also be removed in the next version\n",
      "  from pandas import Panel\n",
      "100%|██████████| 44724/44724 [15:45<00:00, 47.32it/s] \n"
     ]
    }
   ],
   "source": [
    "# Define the set of labeling functions (LFs)\n",
    "# currently excluding amino_acid and followed_ase and followed_by_noun\n",
    "lfs = [includes_solution_words, includes_mixture_words, includes_physical_words, includes_genetic_words, includes_structural_words,\n",
    "      includes_general_chemical_words, includes_functional_group, includes_paper_artifacts, includes_no_terms, structure_adjacent_mentions,\n",
    "      structure_sep_or, structure_sep_comma, structure_sep_via, structure_sep_sym, structure_sep_adverb, includes_oxidation_words,\n",
    "      structure_sep_verb, structure_sep_conversion_words, includes_combustion_words, includes_neutralization_words, includes_combination_words,\n",
    "      includes_decomposition_words, includes_replacement_words, includes_reaction_words, includes_reaction_component_words, includes_comparison_words,\n",
    "      includes_one_chem, includes_react, includes_measure_words, includes_experiment_words, includes_concentration, structure_sep_and,\n",
    "      structure_next_to_conversion_words, structure_conversion_by, structure_conversion_of, includes_react_sym,\n",
    "      includes_interconvert_sym, structure_jtsui_pattern_1, structure_jtsui_pattern_2,\n",
    "      structure_jtsui_pattern_3, structure_jtsui_pattern_4, structure_jtsui_pattern_5, structure_jtsui_pattern_1_copy, structure_jtsui_pattern_2_copy,\n",
    "      structure_jtsui_pattern_3_copy, structure_jtsui_pattern_4_copy, structure_jtsui_pattern_5_copy,\n",
    "      includes_impact_words, includes_creation_words, includes_catalyze_words, includes_emzyze_words, includes_reaction_words2]\n",
    "      #  , ]\n",
    "      #includes_reaction_words5, includes_reaction_words3, includes_reaction_words4, structure_followed_by_ase,\n",
    "df_train = brenda_mult_chem_df\n",
    "applier = PandasLFApplier(lfs=lfs)\n",
    "L_train = applier.apply(df=df_train)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([-1, -1, -1,  0,  0, -1, -1,  0,  0, -1, -1, -1, -1, -1, -1, -1, -1,\n",
       "       -1, -1, -1, -1, -1, -1,  1, -1, -1, -1, -1,  0, -1, -1, -1,  1, -1,\n",
       "        1, -1, -1, -1, -1, -1,  1, -1, -1, -1, -1,  1, -1, -1, -1,  1, -1,\n",
       "        1])"
      ]
     },
     "execution_count": 138,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "L_train[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>j</th>\n",
       "      <th>Polarity</th>\n",
       "      <th>Coverage</th>\n",
       "      <th>Overlaps</th>\n",
       "      <th>Conflicts</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>includes_solution_words</th>\n",
       "      <td>0</td>\n",
       "      <td>[0]</td>\n",
       "      <td>0.078616</td>\n",
       "      <td>0.078504</td>\n",
       "      <td>0.075150</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_mixture_words</th>\n",
       "      <td>1</td>\n",
       "      <td>[0]</td>\n",
       "      <td>0.028419</td>\n",
       "      <td>0.028396</td>\n",
       "      <td>0.026541</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_physical_words</th>\n",
       "      <td>2</td>\n",
       "      <td>[0]</td>\n",
       "      <td>0.457607</td>\n",
       "      <td>0.454834</td>\n",
       "      <td>0.410250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_genetic_words</th>\n",
       "      <td>3</td>\n",
       "      <td>[0]</td>\n",
       "      <td>0.277457</td>\n",
       "      <td>0.277189</td>\n",
       "      <td>0.252392</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_structural_words</th>\n",
       "      <td>4</td>\n",
       "      <td>[0]</td>\n",
       "      <td>0.309141</td>\n",
       "      <td>0.307866</td>\n",
       "      <td>0.275065</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_general_chemical_words</th>\n",
       "      <td>5</td>\n",
       "      <td>[0]</td>\n",
       "      <td>0.238574</td>\n",
       "      <td>0.237635</td>\n",
       "      <td>0.213778</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_functional_group</th>\n",
       "      <td>6</td>\n",
       "      <td>[0]</td>\n",
       "      <td>0.262141</td>\n",
       "      <td>0.261470</td>\n",
       "      <td>0.236964</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_paper_artifacts</th>\n",
       "      <td>7</td>\n",
       "      <td>[0]</td>\n",
       "      <td>0.133396</td>\n",
       "      <td>0.133105</td>\n",
       "      <td>0.124989</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_no_terms</th>\n",
       "      <td>8</td>\n",
       "      <td>[0]</td>\n",
       "      <td>0.135475</td>\n",
       "      <td>0.134022</td>\n",
       "      <td>0.119354</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>structure_adjacent_mentions</th>\n",
       "      <td>9</td>\n",
       "      <td>[0]</td>\n",
       "      <td>0.103859</td>\n",
       "      <td>0.103859</td>\n",
       "      <td>0.093976</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>structure_sep_or</th>\n",
       "      <td>10</td>\n",
       "      <td>[0]</td>\n",
       "      <td>0.017642</td>\n",
       "      <td>0.017485</td>\n",
       "      <td>0.015584</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>structure_sep_comma</th>\n",
       "      <td>11</td>\n",
       "      <td>[0]</td>\n",
       "      <td>0.053819</td>\n",
       "      <td>0.053573</td>\n",
       "      <td>0.047447</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>structure_sep_via</th>\n",
       "      <td>12</td>\n",
       "      <td>[0]</td>\n",
       "      <td>0.004092</td>\n",
       "      <td>0.004025</td>\n",
       "      <td>0.003600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>structure_sep_sym</th>\n",
       "      <td>13</td>\n",
       "      <td>[0]</td>\n",
       "      <td>0.195354</td>\n",
       "      <td>0.194079</td>\n",
       "      <td>0.170490</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>structure_sep_adverb</th>\n",
       "      <td>14</td>\n",
       "      <td>[0]</td>\n",
       "      <td>0.000760</td>\n",
       "      <td>0.000760</td>\n",
       "      <td>0.000715</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_oxidation_words</th>\n",
       "      <td>15</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.088677</td>\n",
       "      <td>0.088655</td>\n",
       "      <td>0.086285</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>structure_sep_verb</th>\n",
       "      <td>16</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.010956</td>\n",
       "      <td>0.010934</td>\n",
       "      <td>0.010688</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>structure_sep_conversion_words</th>\n",
       "      <td>17</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.027591</td>\n",
       "      <td>0.027591</td>\n",
       "      <td>0.025512</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_combustion_words</th>\n",
       "      <td>18</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.001274</td>\n",
       "      <td>0.001274</td>\n",
       "      <td>0.001274</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_neutralization_words</th>\n",
       "      <td>19</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.049593</td>\n",
       "      <td>0.049593</td>\n",
       "      <td>0.049593</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_combination_words</th>\n",
       "      <td>20</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.024640</td>\n",
       "      <td>0.024618</td>\n",
       "      <td>0.024260</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_decomposition_words</th>\n",
       "      <td>21</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.001207</td>\n",
       "      <td>0.001207</td>\n",
       "      <td>0.001185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_replacement_words</th>\n",
       "      <td>22</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.009458</td>\n",
       "      <td>0.009391</td>\n",
       "      <td>0.009279</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_reaction_words</th>\n",
       "      <td>23</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.744835</td>\n",
       "      <td>0.740520</td>\n",
       "      <td>0.719189</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_reaction_component_words</th>\n",
       "      <td>24</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.148936</td>\n",
       "      <td>0.148690</td>\n",
       "      <td>0.143994</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_comparison_words</th>\n",
       "      <td>25</td>\n",
       "      <td>[0]</td>\n",
       "      <td>0.230860</td>\n",
       "      <td>0.229586</td>\n",
       "      <td>0.205058</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_one_chem</th>\n",
       "      <td>26</td>\n",
       "      <td>[]</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_react</th>\n",
       "      <td>27</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.101422</td>\n",
       "      <td>0.100908</td>\n",
       "      <td>0.097777</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_measure_words</th>\n",
       "      <td>28</td>\n",
       "      <td>[0]</td>\n",
       "      <td>0.583490</td>\n",
       "      <td>0.579085</td>\n",
       "      <td>0.519475</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_experiment_words</th>\n",
       "      <td>29</td>\n",
       "      <td>[0]</td>\n",
       "      <td>0.361707</td>\n",
       "      <td>0.359449</td>\n",
       "      <td>0.324457</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_concentration</th>\n",
       "      <td>30</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.050443</td>\n",
       "      <td>0.050443</td>\n",
       "      <td>0.048900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>structure_sep_and</th>\n",
       "      <td>31</td>\n",
       "      <td>[0]</td>\n",
       "      <td>0.096794</td>\n",
       "      <td>0.096369</td>\n",
       "      <td>0.087090</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>structure_next_to_conversion_words</th>\n",
       "      <td>32</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.219122</td>\n",
       "      <td>0.218272</td>\n",
       "      <td>0.208658</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>structure_conversion_by</th>\n",
       "      <td>33</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.008966</td>\n",
       "      <td>0.008966</td>\n",
       "      <td>0.007870</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>structure_conversion_of</th>\n",
       "      <td>34</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.117230</td>\n",
       "      <td>0.117230</td>\n",
       "      <td>0.112311</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_react_sym</th>\n",
       "      <td>35</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.000089</td>\n",
       "      <td>0.000089</td>\n",
       "      <td>0.000089</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_interconvert_sym</th>\n",
       "      <td>36</td>\n",
       "      <td>[]</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>structure_jtsui_pattern_1</th>\n",
       "      <td>37</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.028463</td>\n",
       "      <td>0.028463</td>\n",
       "      <td>0.026339</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>structure_jtsui_pattern_2</th>\n",
       "      <td>38</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.004919</td>\n",
       "      <td>0.004919</td>\n",
       "      <td>0.004360</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>structure_jtsui_pattern_3</th>\n",
       "      <td>39</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.000559</td>\n",
       "      <td>0.000559</td>\n",
       "      <td>0.000537</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>structure_jtsui_pattern_4</th>\n",
       "      <td>40</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.024170</td>\n",
       "      <td>0.024170</td>\n",
       "      <td>0.022538</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>structure_jtsui_pattern_5</th>\n",
       "      <td>41</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.000402</td>\n",
       "      <td>0.000402</td>\n",
       "      <td>0.000313</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>structure_jtsui_pattern_1_copy</th>\n",
       "      <td>42</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.028463</td>\n",
       "      <td>0.028463</td>\n",
       "      <td>0.026339</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>structure_jtsui_pattern_2_copy</th>\n",
       "      <td>43</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.004919</td>\n",
       "      <td>0.004919</td>\n",
       "      <td>0.004360</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>structure_jtsui_pattern_3_copy</th>\n",
       "      <td>44</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.000559</td>\n",
       "      <td>0.000559</td>\n",
       "      <td>0.000537</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>structure_jtsui_pattern_4_copy</th>\n",
       "      <td>45</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.024170</td>\n",
       "      <td>0.024170</td>\n",
       "      <td>0.022538</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>structure_jtsui_pattern_5_copy</th>\n",
       "      <td>46</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.000402</td>\n",
       "      <td>0.000402</td>\n",
       "      <td>0.000313</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_impact_words</th>\n",
       "      <td>47</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.087604</td>\n",
       "      <td>0.087559</td>\n",
       "      <td>0.085681</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_creation_words</th>\n",
       "      <td>48</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.043936</td>\n",
       "      <td>0.043936</td>\n",
       "      <td>0.042304</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_catalyze_words</th>\n",
       "      <td>49</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.066698</td>\n",
       "      <td>0.066676</td>\n",
       "      <td>0.062696</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_emzyze_words</th>\n",
       "      <td>50</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.197634</td>\n",
       "      <td>0.196606</td>\n",
       "      <td>0.191240</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_reaction_words2</th>\n",
       "      <td>51</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.411345</td>\n",
       "      <td>0.410674</td>\n",
       "      <td>0.396677</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                     j Polarity  Coverage  Overlaps  Conflicts\n",
       "includes_solution_words              0      [0]  0.078616  0.078504   0.075150\n",
       "includes_mixture_words               1      [0]  0.028419  0.028396   0.026541\n",
       "includes_physical_words              2      [0]  0.457607  0.454834   0.410250\n",
       "includes_genetic_words               3      [0]  0.277457  0.277189   0.252392\n",
       "includes_structural_words            4      [0]  0.309141  0.307866   0.275065\n",
       "includes_general_chemical_words      5      [0]  0.238574  0.237635   0.213778\n",
       "includes_functional_group            6      [0]  0.262141  0.261470   0.236964\n",
       "includes_paper_artifacts             7      [0]  0.133396  0.133105   0.124989\n",
       "includes_no_terms                    8      [0]  0.135475  0.134022   0.119354\n",
       "structure_adjacent_mentions          9      [0]  0.103859  0.103859   0.093976\n",
       "structure_sep_or                    10      [0]  0.017642  0.017485   0.015584\n",
       "structure_sep_comma                 11      [0]  0.053819  0.053573   0.047447\n",
       "structure_sep_via                   12      [0]  0.004092  0.004025   0.003600\n",
       "structure_sep_sym                   13      [0]  0.195354  0.194079   0.170490\n",
       "structure_sep_adverb                14      [0]  0.000760  0.000760   0.000715\n",
       "includes_oxidation_words            15      [1]  0.088677  0.088655   0.086285\n",
       "structure_sep_verb                  16      [1]  0.010956  0.010934   0.010688\n",
       "structure_sep_conversion_words      17      [1]  0.027591  0.027591   0.025512\n",
       "includes_combustion_words           18      [1]  0.001274  0.001274   0.001274\n",
       "includes_neutralization_words       19      [1]  0.049593  0.049593   0.049593\n",
       "includes_combination_words          20      [1]  0.024640  0.024618   0.024260\n",
       "includes_decomposition_words        21      [1]  0.001207  0.001207   0.001185\n",
       "includes_replacement_words          22      [1]  0.009458  0.009391   0.009279\n",
       "includes_reaction_words             23      [1]  0.744835  0.740520   0.719189\n",
       "includes_reaction_component_words   24      [1]  0.148936  0.148690   0.143994\n",
       "includes_comparison_words           25      [0]  0.230860  0.229586   0.205058\n",
       "includes_one_chem                   26       []  0.000000  0.000000   0.000000\n",
       "includes_react                      27      [1]  0.101422  0.100908   0.097777\n",
       "includes_measure_words              28      [0]  0.583490  0.579085   0.519475\n",
       "includes_experiment_words           29      [0]  0.361707  0.359449   0.324457\n",
       "includes_concentration              30      [1]  0.050443  0.050443   0.048900\n",
       "structure_sep_and                   31      [0]  0.096794  0.096369   0.087090\n",
       "structure_next_to_conversion_words  32      [1]  0.219122  0.218272   0.208658\n",
       "structure_conversion_by             33      [1]  0.008966  0.008966   0.007870\n",
       "structure_conversion_of             34      [1]  0.117230  0.117230   0.112311\n",
       "includes_react_sym                  35      [1]  0.000089  0.000089   0.000089\n",
       "includes_interconvert_sym           36       []  0.000000  0.000000   0.000000\n",
       "structure_jtsui_pattern_1           37      [1]  0.028463  0.028463   0.026339\n",
       "structure_jtsui_pattern_2           38      [1]  0.004919  0.004919   0.004360\n",
       "structure_jtsui_pattern_3           39      [1]  0.000559  0.000559   0.000537\n",
       "structure_jtsui_pattern_4           40      [1]  0.024170  0.024170   0.022538\n",
       "structure_jtsui_pattern_5           41      [1]  0.000402  0.000402   0.000313\n",
       "structure_jtsui_pattern_1_copy      42      [1]  0.028463  0.028463   0.026339\n",
       "structure_jtsui_pattern_2_copy      43      [1]  0.004919  0.004919   0.004360\n",
       "structure_jtsui_pattern_3_copy      44      [1]  0.000559  0.000559   0.000537\n",
       "structure_jtsui_pattern_4_copy      45      [1]  0.024170  0.024170   0.022538\n",
       "structure_jtsui_pattern_5_copy      46      [1]  0.000402  0.000402   0.000313\n",
       "includes_impact_words               47      [1]  0.087604  0.087559   0.085681\n",
       "includes_creation_words             48      [1]  0.043936  0.043936   0.042304\n",
       "includes_catalyze_words             49      [1]  0.066698  0.066676   0.062696\n",
       "includes_emzyze_words               50      [1]  0.197634  0.196606   0.191240\n",
       "includes_reaction_words2            51      [1]  0.411345  0.410674   0.396677"
      ]
     },
     "execution_count": 139,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from snorkel.labeling import LFAnalysis\n",
    "\n",
    "LFAnalysis(L=L_train, lfs=lfs).lf_summary()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.9940747696985959"
      ]
     },
     "execution_count": 140,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "LFAnalysis(L_train).label_coverage()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "-4.902110723548877"
      ]
     },
     "execution_count": 141,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# gets the average amount of labels (goal is to get this to 15)\n",
    "sum = 0\n",
    "total = 0\n",
    "for item in L_train:\n",
    "    sum += 41 - np.count_nonzero(item == -1)\n",
    "    total += 1\n",
    "sum/total\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "metadata": {},
   "outputs": [],
   "source": [
    "majority_model = MajorityLabelVoter()\n",
    "brenda_mult_chem_df[\"label_voter\"] = majority_model.predict(L=L_train)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 158,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>lit_id</th>\n",
       "      <th>indices</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>sentence</th>\n",
       "      <th>sentence_pos</th>\n",
       "      <th>enzymes</th>\n",
       "      <th>enzyme_locations</th>\n",
       "      <th>chemical_entities_full</th>\n",
       "      <th>chemical_names</th>\n",
       "      <th>chemical_smiles</th>\n",
       "      <th>name_smile_tuples</th>\n",
       "      <th>text</th>\n",
       "      <th>label_voter</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10.1002/jps.20686</td>\n",
       "      <td>150</td>\n",
       "      <td>33180</td>\n",
       "      <td>33589</td>\n",
       "      <td>we selected diltiazem, cyclosporin a, and dexa...</td>\n",
       "      <td>[('We', 'PRP'), ('selected', 'VBD'), ('diltiaz...</td>\n",
       "      <td>ATPase</td>\n",
       "      <td>[('ATPase', 22, 22)]</td>\n",
       "      <td>[{'text': 'diltiazem', 'start': 33192, 'end': ...</td>\n",
       "      <td>[cyclosporin a, diltiazem, dexamethasone, calc...</td>\n",
       "      <td>COc1ccc(cc1)[C@@H]2Sc3ccccc3N(CCN(C)C)C(=O)[C@...</td>\n",
       "      <td>[('diltiazem', 'COc1ccc(cc1)[C@@H]2Sc3ccccc3N(...</td>\n",
       "      <td>[we selected diltiazem, cyclosporin a, and dex...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10.1006/abbi.1993.1114</td>\n",
       "      <td>3</td>\n",
       "      <td>2464</td>\n",
       "      <td>2912</td>\n",
       "      <td>croteau r washington state univ, inst biol che...</td>\n",
       "      <td>[('Croteau', 'NNP'), ('R.', 'NNP'), ('Washingt...</td>\n",
       "      <td>synthase, cyclase</td>\n",
       "      <td>[('synthase', 34, 34), ('cyclase', 38, 38)]</td>\n",
       "      <td>[{'text': 'Limonene', 'start': 2620, 'end': 26...</td>\n",
       "      <td>[limonene, sodium dodecyl sulfate, geranyl pyr...</td>\n",
       "      <td>CC(=C)C1CCC(=CC1)C,, CC(C)=CCCC(/C)=C/CO[P]([O...</td>\n",
       "      <td>[('Limonene', 'CC(=C)C1CCC(=CC1)C'), ('geranyl...</td>\n",
       "      <td>[croteau r washington state univ, inst biol ch...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10.1006/abbi.1993.1275</td>\n",
       "      <td>2</td>\n",
       "      <td>1178</td>\n",
       "      <td>2551</td>\n",
       "      <td>chemicalmodificationschickenliverpyruvatecarbo...</td>\n",
       "      <td>[('CHEMICALMODIFICATIONSCHICKENLIVERPYRUVATECA...</td>\n",
       "      <td>Carboxylase</td>\n",
       "      <td>[('Carboxylase', 143, 143)]</td>\n",
       "      <td>[{'text': 'Pyruvate', 'start': 2438, 'end': 24...</td>\n",
       "      <td>[lysine, pyruvate, cysteine]</td>\n",
       "      <td>CC(=O)C([O-])=O,, N[C@@H](CS)C(O)=O,, NCCCC[C@...</td>\n",
       "      <td>[('Pyruvate', 'CC(=O)C([O-])=O'), ('Cysteine',...</td>\n",
       "      <td>[chemicalmodificationschickenliverpyruvatecarb...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10.1006/abbi.1993.1275</td>\n",
       "      <td>3</td>\n",
       "      <td>2552</td>\n",
       "      <td>2895</td>\n",
       "      <td>ash de temple univ, hlth sci ctr, sch med, dep...</td>\n",
       "      <td>[('Ash', 'NNP'), ('D.E.', 'NNP'), ('Temple', '...</td>\n",
       "      <td>carboxylase</td>\n",
       "      <td>[('carboxylase', 32, 32)]</td>\n",
       "      <td>[{'text': 'pyruvate', 'start': 2696, 'end': 27...</td>\n",
       "      <td>[cysteine, n-(7-dimethylamino-4-methyl-3-couma...</td>\n",
       "      <td>CC(=O)C([O-])=O,, CN(C)c1ccc2C(=C(N3C(=O)C=CC3...</td>\n",
       "      <td>[('pyruvate', 'CC(=O)C([O-])=O'), ('N-(7-dimet...</td>\n",
       "      <td>[ash de temple univ, hlth sci ctr, sch med, de...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10.1006/abbi.1993.1275</td>\n",
       "      <td>5</td>\n",
       "      <td>3008</td>\n",
       "      <td>3317</td>\n",
       "      <td>at a one- to two-fold molar excess over active...</td>\n",
       "      <td>[('At', 'IN'), ('a', 'DT'), ('one-', 'CD'), ('...</td>\n",
       "      <td>carboxylase, decarboxylase</td>\n",
       "      <td>[('carboxylase', 20, 20), ('decarboxylase', 28...</td>\n",
       "      <td>[{'text': 'pyruvate', 'start': 3120, 'end': 31...</td>\n",
       "      <td>[pyruvate, oxaloacetate, adp]</td>\n",
       "      <td>CC(=O)C([O-])=O,, Nc1ncnc2n(cnc12)C3OC(CO[P](O...</td>\n",
       "      <td>[('pyruvate', 'CC(=O)C([O-])=O'), ('ADP', 'Nc1...</td>\n",
       "      <td>[at a one- to two-fold molar excess over activ...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44719</th>\n",
       "      <td>10.1263/jbb.99.623</td>\n",
       "      <td>26</td>\n",
       "      <td>11239</td>\n",
       "      <td>11493</td>\n",
       "      <td>key words d-alanine-d-alanine ligase d-amino a...</td>\n",
       "      <td>[('Key', 'JJ'), ('words', 'NNS'), ('D-alanine-...</td>\n",
       "      <td>ligase</td>\n",
       "      <td>[('ligase', 3, 3)]</td>\n",
       "      <td>[{'text': 'D-alanine-D-alanine', 'start': 1124...</td>\n",
       "      <td>[vancomycin, d-alanine-d-alanine]</td>\n",
       "      <td>C[C@@H](N)C(O)=O.C[C@@H](N)C(O)=O,, CN[C@H](CC...</td>\n",
       "      <td>[('D-alanine-D-alanine', 'C[C@@H](N)C(O)=O.C[C...</td>\n",
       "      <td>[key words d-alanine-d-alanine ligase d-amino ...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44720</th>\n",
       "      <td>10.1263/jbb.99.623</td>\n",
       "      <td>62</td>\n",
       "      <td>17119</td>\n",
       "      <td>17420</td>\n",
       "      <td>the mobile phase consisted of a linear gradien...</td>\n",
       "      <td>[('The', 'DT'), ('mobile', 'JJ'), ('phase', 'N...</td>\n",
       "      <td>phase</td>\n",
       "      <td>[('phase', 2, 2)]</td>\n",
       "      <td>[{'text': 'methanol', 'start': 17213, 'end': 1...</td>\n",
       "      <td>[tetrahydrofurane, methanol]</td>\n",
       "      <td>CO,, CO,, C1CCOC1</td>\n",
       "      <td>[('methanol', 'CO'), ('methanol', 'CO'), ('tet...</td>\n",
       "      <td>[the mobile phase consisted of a linear gradie...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44721</th>\n",
       "      <td>10.1263/jbb.99.623</td>\n",
       "      <td>129</td>\n",
       "      <td>27150</td>\n",
       "      <td>27373</td>\n",
       "      <td>weber v falkenhagen d subpol a novel sucrose-b...</td>\n",
       "      <td>[('Weber', 'NNP'), ('V.', 'NNP'), ('Falkenhage...</td>\n",
       "      <td>phase</td>\n",
       "      <td>[('phase', 16, 16)]</td>\n",
       "      <td>[{'text': 'sucrose', 'start': 27190, 'end': 27...</td>\n",
       "      <td>[kato, sucrose]</td>\n",
       "      <td>OC[C@H]1O[C@H](O[C@]2(CO)O[C@H](CO)[C@@H](O)[C...</td>\n",
       "      <td>[('sucrose', 'OC[C@H]1O[C@H](O[C@]2(CO)O[C@H](...</td>\n",
       "      <td>[weber v falkenhagen d subpol a novel sucrose-...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44722</th>\n",
       "      <td>10.1263/jbb.99.623</td>\n",
       "      <td>138</td>\n",
       "      <td>28139</td>\n",
       "      <td>28364</td>\n",
       "      <td>active-site mutants of the vanc2 d-alanyl-d-se...</td>\n",
       "      <td>[('Active', 'JJ'), ('-', 'HYPH'), ('site', 'NN...</td>\n",
       "      <td>ligase, ligase</td>\n",
       "      <td>[('ligase', 8, 8), ('ligase', 25, 25)]</td>\n",
       "      <td>[{'text': 'vancomycin', 'start': 28220, 'end':...</td>\n",
       "      <td>[vancomycin, d-alanyl-d-alanine]</td>\n",
       "      <td>CN[C@H](CC(C)C)C(=O)NC1[C@H](O)c2ccc(Oc3cc4cc(...</td>\n",
       "      <td>[('vancomycin', 'CN[C@H](CC(C)C)C(=O)NC1[C@H](...</td>\n",
       "      <td>[active-site mutants of the vanc2 d-alanyl-d-s...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44723</th>\n",
       "      <td>10.1263/jbb.99.623</td>\n",
       "      <td>153</td>\n",
       "      <td>29289</td>\n",
       "      <td>29564</td>\n",
       "      <td>in vitro atp regeneration from polyphosphate a...</td>\n",
       "      <td>[('In', 'FW'), ('vitro', 'FW'), ('ATP', 'NN'),...</td>\n",
       "      <td>phosphotransferase, kinase</td>\n",
       "      <td>[('phosphotransferase', 12, 12), ('kinase', 15...</td>\n",
       "      <td>[{'text': 'ATP', 'start': 29298, 'end': 29301,...</td>\n",
       "      <td>[adenylate, atp, amp]</td>\n",
       "      <td>Nc1ncnc2n(cnc12)C3OC(CO[P](O)(=O)O[P](O)(=O)O[...</td>\n",
       "      <td>[('ATP', 'Nc1ncnc2n(cnc12)C3OC(CO[P](O)(=O)O[P...</td>\n",
       "      <td>[in vitro atp regeneration from polyphosphate ...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>44724 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                       lit_id  indices  start    end  \\\n",
       "0           10.1002/jps.20686      150  33180  33589   \n",
       "1      10.1006/abbi.1993.1114        3   2464   2912   \n",
       "2      10.1006/abbi.1993.1275        2   1178   2551   \n",
       "3      10.1006/abbi.1993.1275        3   2552   2895   \n",
       "4      10.1006/abbi.1993.1275        5   3008   3317   \n",
       "...                       ...      ...    ...    ...   \n",
       "44719      10.1263/jbb.99.623       26  11239  11493   \n",
       "44720      10.1263/jbb.99.623       62  17119  17420   \n",
       "44721      10.1263/jbb.99.623      129  27150  27373   \n",
       "44722      10.1263/jbb.99.623      138  28139  28364   \n",
       "44723      10.1263/jbb.99.623      153  29289  29564   \n",
       "\n",
       "                                                sentence  \\\n",
       "0      we selected diltiazem, cyclosporin a, and dexa...   \n",
       "1      croteau r washington state univ, inst biol che...   \n",
       "2      chemicalmodificationschickenliverpyruvatecarbo...   \n",
       "3      ash de temple univ, hlth sci ctr, sch med, dep...   \n",
       "4      at a one- to two-fold molar excess over active...   \n",
       "...                                                  ...   \n",
       "44719  key words d-alanine-d-alanine ligase d-amino a...   \n",
       "44720  the mobile phase consisted of a linear gradien...   \n",
       "44721  weber v falkenhagen d subpol a novel sucrose-b...   \n",
       "44722  active-site mutants of the vanc2 d-alanyl-d-se...   \n",
       "44723  in vitro atp regeneration from polyphosphate a...   \n",
       "\n",
       "                                            sentence_pos  \\\n",
       "0      [('We', 'PRP'), ('selected', 'VBD'), ('diltiaz...   \n",
       "1      [('Croteau', 'NNP'), ('R.', 'NNP'), ('Washingt...   \n",
       "2      [('CHEMICALMODIFICATIONSCHICKENLIVERPYRUVATECA...   \n",
       "3      [('Ash', 'NNP'), ('D.E.', 'NNP'), ('Temple', '...   \n",
       "4      [('At', 'IN'), ('a', 'DT'), ('one-', 'CD'), ('...   \n",
       "...                                                  ...   \n",
       "44719  [('Key', 'JJ'), ('words', 'NNS'), ('D-alanine-...   \n",
       "44720  [('The', 'DT'), ('mobile', 'JJ'), ('phase', 'N...   \n",
       "44721  [('Weber', 'NNP'), ('V.', 'NNP'), ('Falkenhage...   \n",
       "44722  [('Active', 'JJ'), ('-', 'HYPH'), ('site', 'NN...   \n",
       "44723  [('In', 'FW'), ('vitro', 'FW'), ('ATP', 'NN'),...   \n",
       "\n",
       "                          enzymes  \\\n",
       "0                          ATPase   \n",
       "1               synthase, cyclase   \n",
       "2                     Carboxylase   \n",
       "3                     carboxylase   \n",
       "4      carboxylase, decarboxylase   \n",
       "...                           ...   \n",
       "44719                      ligase   \n",
       "44720                       phase   \n",
       "44721                       phase   \n",
       "44722              ligase, ligase   \n",
       "44723  phosphotransferase, kinase   \n",
       "\n",
       "                                        enzyme_locations  \\\n",
       "0                                   [('ATPase', 22, 22)]   \n",
       "1            [('synthase', 34, 34), ('cyclase', 38, 38)]   \n",
       "2                            [('Carboxylase', 143, 143)]   \n",
       "3                              [('carboxylase', 32, 32)]   \n",
       "4      [('carboxylase', 20, 20), ('decarboxylase', 28...   \n",
       "...                                                  ...   \n",
       "44719                                 [('ligase', 3, 3)]   \n",
       "44720                                  [('phase', 2, 2)]   \n",
       "44721                                [('phase', 16, 16)]   \n",
       "44722             [('ligase', 8, 8), ('ligase', 25, 25)]   \n",
       "44723  [('phosphotransferase', 12, 12), ('kinase', 15...   \n",
       "\n",
       "                                  chemical_entities_full  \\\n",
       "0      [{'text': 'diltiazem', 'start': 33192, 'end': ...   \n",
       "1      [{'text': 'Limonene', 'start': 2620, 'end': 26...   \n",
       "2      [{'text': 'Pyruvate', 'start': 2438, 'end': 24...   \n",
       "3      [{'text': 'pyruvate', 'start': 2696, 'end': 27...   \n",
       "4      [{'text': 'pyruvate', 'start': 3120, 'end': 31...   \n",
       "...                                                  ...   \n",
       "44719  [{'text': 'D-alanine-D-alanine', 'start': 1124...   \n",
       "44720  [{'text': 'methanol', 'start': 17213, 'end': 1...   \n",
       "44721  [{'text': 'sucrose', 'start': 27190, 'end': 27...   \n",
       "44722  [{'text': 'vancomycin', 'start': 28220, 'end':...   \n",
       "44723  [{'text': 'ATP', 'start': 29298, 'end': 29301,...   \n",
       "\n",
       "                                          chemical_names  \\\n",
       "0      [cyclosporin a, diltiazem, dexamethasone, calc...   \n",
       "1      [limonene, sodium dodecyl sulfate, geranyl pyr...   \n",
       "2                           [lysine, pyruvate, cysteine]   \n",
       "3      [cysteine, n-(7-dimethylamino-4-methyl-3-couma...   \n",
       "4                          [pyruvate, oxaloacetate, adp]   \n",
       "...                                                  ...   \n",
       "44719                  [vancomycin, d-alanine-d-alanine]   \n",
       "44720                       [tetrahydrofurane, methanol]   \n",
       "44721                                    [kato, sucrose]   \n",
       "44722                   [vancomycin, d-alanyl-d-alanine]   \n",
       "44723                              [adenylate, atp, amp]   \n",
       "\n",
       "                                         chemical_smiles  \\\n",
       "0      COc1ccc(cc1)[C@@H]2Sc3ccccc3N(CCN(C)C)C(=O)[C@...   \n",
       "1      CC(=C)C1CCC(=CC1)C,, CC(C)=CCCC(/C)=C/CO[P]([O...   \n",
       "2      CC(=O)C([O-])=O,, N[C@@H](CS)C(O)=O,, NCCCC[C@...   \n",
       "3      CC(=O)C([O-])=O,, CN(C)c1ccc2C(=C(N3C(=O)C=CC3...   \n",
       "4      CC(=O)C([O-])=O,, Nc1ncnc2n(cnc12)C3OC(CO[P](O...   \n",
       "...                                                  ...   \n",
       "44719  C[C@@H](N)C(O)=O.C[C@@H](N)C(O)=O,, CN[C@H](CC...   \n",
       "44720                                  CO,, CO,, C1CCOC1   \n",
       "44721  OC[C@H]1O[C@H](O[C@]2(CO)O[C@H](CO)[C@@H](O)[C...   \n",
       "44722  CN[C@H](CC(C)C)C(=O)NC1[C@H](O)c2ccc(Oc3cc4cc(...   \n",
       "44723  Nc1ncnc2n(cnc12)C3OC(CO[P](O)(=O)O[P](O)(=O)O[...   \n",
       "\n",
       "                                       name_smile_tuples  \\\n",
       "0      [('diltiazem', 'COc1ccc(cc1)[C@@H]2Sc3ccccc3N(...   \n",
       "1      [('Limonene', 'CC(=C)C1CCC(=CC1)C'), ('geranyl...   \n",
       "2      [('Pyruvate', 'CC(=O)C([O-])=O'), ('Cysteine',...   \n",
       "3      [('pyruvate', 'CC(=O)C([O-])=O'), ('N-(7-dimet...   \n",
       "4      [('pyruvate', 'CC(=O)C([O-])=O'), ('ADP', 'Nc1...   \n",
       "...                                                  ...   \n",
       "44719  [('D-alanine-D-alanine', 'C[C@@H](N)C(O)=O.C[C...   \n",
       "44720  [('methanol', 'CO'), ('methanol', 'CO'), ('tet...   \n",
       "44721  [('sucrose', 'OC[C@H]1O[C@H](O[C@]2(CO)O[C@H](...   \n",
       "44722  [('vancomycin', 'CN[C@H](CC(C)C)C(=O)NC1[C@H](...   \n",
       "44723  [('ATP', 'Nc1ncnc2n(cnc12)C3OC(CO[P](O)(=O)O[P...   \n",
       "\n",
       "                                                    text  label_voter  \n",
       "0      [we selected diltiazem, cyclosporin a, and dex...            0  \n",
       "1      [croteau r washington state univ, inst biol ch...            1  \n",
       "2      [chemicalmodificationschickenliverpyruvatecarb...            0  \n",
       "3      [ash de temple univ, hlth sci ctr, sch med, de...            0  \n",
       "4      [at a one- to two-fold molar excess over activ...            1  \n",
       "...                                                  ...          ...  \n",
       "44719  [key words d-alanine-d-alanine ligase d-amino ...            0  \n",
       "44720  [the mobile phase consisted of a linear gradie...            0  \n",
       "44721  [weber v falkenhagen d subpol a novel sucrose-...            0  \n",
       "44722  [active-site mutants of the vanc2 d-alanyl-d-s...            0  \n",
       "44723  [in vitro atp regeneration from polyphosphate ...            0  \n",
       "\n",
       "[44724 rows x 14 columns]"
      ]
     },
     "execution_count": 158,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "brenda_mult_chem_df.loc[brenda_mult_chem_df[\"label_voter\"] == -1, \"label_voter\"] = 1\n",
    "\n",
    "brenda_mult_chem_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "11278"
      ]
     },
     "execution_count": 143,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#oof\n",
    "\n",
    "df_brenda_p = df_train[df_train[\"label_voter\"] == 1]\n",
    "\n",
    "len(df_brenda_p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "metadata": {},
   "outputs": [],
   "source": [
    "brenda_mult_chem_df.to_csv('voter_elsevier_split_6.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sentence</th>\n",
       "      <th>chemical_names</th>\n",
       "      <th>text</th>\n",
       "      <th>label_voter</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>croteau r washington state univ, inst biol che...</td>\n",
       "      <td>[limonene, sodium dodecyl sulfate, geranyl pyr...</td>\n",
       "      <td>[croteau r washington state univ, inst biol ch...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>biosynthesisvicinaldihydroxyfattyacidsinredalg...</td>\n",
       "      <td>[hydroperoxide, sodium]</td>\n",
       "      <td>[biosynthesisvicinaldihydroxyfattyacidsinredal...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>the transformation consisted of initial 12-lip...</td>\n",
       "      <td>[arachidonic acid, hydroperoxide]</td>\n",
       "      <td>[the transformation consisted of initial 12-li...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>the 12-lipoxygenase-catalyzed reaction consist...</td>\n",
       "      <td>[hydrogen, hydroperoxide, oxygen, carbon]</td>\n",
       "      <td>[the 12-lipoxygenase-catalyzed reaction consis...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>subsequent conversion of the product to car-3-...</td>\n",
       "      <td>[(+)-3-carene, car-3-en-5-one]</td>\n",
       "      <td>[subsequent conversion of the product to car-3...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44699</th>\n",
       "      <td>nhase exhibited an absorption peak at 710 nm, ...</td>\n",
       "      <td>[sulfur, fe]</td>\n",
       "      <td>[nhase exhibited an absorption peak at 710 nm,...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44702</th>\n",
       "      <td>concluding remarks the results indicate that f...</td>\n",
       "      <td>[fe, isonitrile]</td>\n",
       "      <td>[concluding remarks the results indicate that ...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44703</th>\n",
       "      <td>nitrogenase is reported to have an activity th...</td>\n",
       "      <td>[methylisonitrile, methylamine]</td>\n",
       "      <td>[nitrogenase is reported to have an activity t...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44704</th>\n",
       "      <td>others have reported that two enzymes are invo...</td>\n",
       "      <td>[formamide, n, isonitrile]</td>\n",
       "      <td>[others have reported that two enzymes are inv...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44717</th>\n",
       "      <td>enzyme activity was determined by monitoring t...</td>\n",
       "      <td>[beta-formylpyruvic acid, tba]</td>\n",
       "      <td>[enzyme activity was determined by monitoring ...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>11278 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                sentence  \\\n",
       "1      croteau r washington state univ, inst biol che...   \n",
       "7      biosynthesisvicinaldihydroxyfattyacidsinredalg...   \n",
       "8      the transformation consisted of initial 12-lip...   \n",
       "9      the 12-lipoxygenase-catalyzed reaction consist...   \n",
       "10     subsequent conversion of the product to car-3-...   \n",
       "...                                                  ...   \n",
       "44699  nhase exhibited an absorption peak at 710 nm, ...   \n",
       "44702  concluding remarks the results indicate that f...   \n",
       "44703  nitrogenase is reported to have an activity th...   \n",
       "44704  others have reported that two enzymes are invo...   \n",
       "44717  enzyme activity was determined by monitoring t...   \n",
       "\n",
       "                                          chemical_names  \\\n",
       "1      [limonene, sodium dodecyl sulfate, geranyl pyr...   \n",
       "7                                [hydroperoxide, sodium]   \n",
       "8                      [arachidonic acid, hydroperoxide]   \n",
       "9              [hydrogen, hydroperoxide, oxygen, carbon]   \n",
       "10                        [(+)-3-carene, car-3-en-5-one]   \n",
       "...                                                  ...   \n",
       "44699                                       [sulfur, fe]   \n",
       "44702                                   [fe, isonitrile]   \n",
       "44703                    [methylisonitrile, methylamine]   \n",
       "44704                         [formamide, n, isonitrile]   \n",
       "44717                     [beta-formylpyruvic acid, tba]   \n",
       "\n",
       "                                                    text  label_voter  \n",
       "1      [croteau r washington state univ, inst biol ch...            1  \n",
       "7      [biosynthesisvicinaldihydroxyfattyacidsinredal...            1  \n",
       "8      [the transformation consisted of initial 12-li...            1  \n",
       "9      [the 12-lipoxygenase-catalyzed reaction consis...            1  \n",
       "10     [subsequent conversion of the product to car-3...            1  \n",
       "...                                                  ...          ...  \n",
       "44699  [nhase exhibited an absorption peak at 710 nm,...            1  \n",
       "44702  [concluding remarks the results indicate that ...            1  \n",
       "44703  [nitrogenase is reported to have an activity t...            1  \n",
       "44704  [others have reported that two enzymes are inv...            1  \n",
       "44717  [enzyme activity was determined by monitoring ...            1  \n",
       "\n",
       "[11278 rows x 4 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "display(df_brenda_p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 174,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[[[[  0   1   2   3   4]\n",
      "   [  5   6   7   8   9]\n",
      "   [ 10  11  12  13  14]\n",
      "   [ 15  16  17  18  19]]\n",
      "\n",
      "  [[ 20  21  22  23  24]\n",
      "   [ 25  26  27  28  29]\n",
      "   [ 30  31  32  33  34]\n",
      "   [ 35  36  37  38  39]]\n",
      "\n",
      "  [[ 40  41  42  43  44]\n",
      "   [ 45  46  47  48  49]\n",
      "   [ 50  51  52  53  54]\n",
      "   [ 55  56  57  58  59]]]\n",
      "\n",
      "\n",
      " [[[ 60  61  62  63  64]\n",
      "   [ 65  66  67  68  69]\n",
      "   [ 70  71  72  73  74]\n",
      "   [ 75  76  77  78  79]]\n",
      "\n",
      "  [[ 80  81  82  83  84]\n",
      "   [ 85  86  87  88  89]\n",
      "   [ 90  91  92  93  94]\n",
      "   [ 95  96  97  98  99]]\n",
      "\n",
      "  [[100 101 102 103 104]\n",
      "   [105 106 107 108 109]\n",
      "   [110 111 112 113 114]\n",
      "   [115 116 117 118 119]]]]\n",
      "[[[[  0  20  40]\n",
      "   [ 60  80 100]]\n",
      "\n",
      "  [[  5  25  45]\n",
      "   [ 65  85 105]]\n",
      "\n",
      "  [[ 10  30  50]\n",
      "   [ 70  90 110]]\n",
      "\n",
      "  [[ 15  35  55]\n",
      "   [ 75  95 115]]]\n",
      "\n",
      "\n",
      " [[[  1  21  41]\n",
      "   [ 61  81 101]]\n",
      "\n",
      "  [[  6  26  46]\n",
      "   [ 66  86 106]]\n",
      "\n",
      "  [[ 11  31  51]\n",
      "   [ 71  91 111]]\n",
      "\n",
      "  [[ 16  36  56]\n",
      "   [ 76  96 116]]]\n",
      "\n",
      "\n",
      " [[[  2  22  42]\n",
      "   [ 62  82 102]]\n",
      "\n",
      "  [[  7  27  47]\n",
      "   [ 67  87 107]]\n",
      "\n",
      "  [[ 12  32  52]\n",
      "   [ 72  92 112]]\n",
      "\n",
      "  [[ 17  37  57]\n",
      "   [ 77  97 117]]]\n",
      "\n",
      "\n",
      " [[[  3  23  43]\n",
      "   [ 63  83 103]]\n",
      "\n",
      "  [[  8  28  48]\n",
      "   [ 68  88 108]]\n",
      "\n",
      "  [[ 13  33  53]\n",
      "   [ 73  93 113]]\n",
      "\n",
      "  [[ 18  38  58]\n",
      "   [ 78  98 118]]]\n",
      "\n",
      "\n",
      " [[[  4  24  44]\n",
      "   [ 64  84 104]]\n",
      "\n",
      "  [[  9  29  49]\n",
      "   [ 69  89 109]]\n",
      "\n",
      "  [[ 14  34  54]\n",
      "   [ 74  94 114]]\n",
      "\n",
      "  [[ 19  39  59]\n",
      "   [ 79  99 119]]]]\n",
      "[[  0  20  40  60  80 100   5  25  45  65  85 105  10  30  50  70  90 110\n",
      "   15  35  55  75  95 115]\n",
      " [  1  21  41  61  81 101   6  26  46  66  86 106  11  31  51  71  91 111\n",
      "   16  36  56  76  96 116]\n",
      " [  2  22  42  62  82 102   7  27  47  67  87 107  12  32  52  72  92 112\n",
      "   17  37  57  77  97 117]\n",
      " [  3  23  43  63  83 103   8  28  48  68  88 108  13  33  53  73  93 113\n",
      "   18  38  58  78  98 118]\n",
      " [  4  24  44  64  84 104   9  29  49  69  89 109  14  34  54  74  94 114\n",
      "   19  39  59  79  99 119]]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119]\n",
      "[[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "   18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "   36  37  38  39]\n",
      " [ 40  41  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57\n",
      "   58  59  60  61  62  63  64  65  66  67  68  69  70  71  72  73  74  75\n",
      "   76  77  78  79]\n",
      " [ 80  81  82  83  84  85  86  87  88  89  90  91  92  93  94  95  96  97\n",
      "   98  99 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115\n",
      "  116 117 118 119]]\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "array([[  0,   2,   4,   6,   8,  10,  12,  14,  16,  18,  20,  22,  24,\n",
       "         26,  28,  30,  32,  34,  36,  38,  40,  42,  44,  46,  48,  50,\n",
       "         52,  54,  56,  58,  60,  62,  64,  66,  68,  70,  72,  74,  76,\n",
       "         78,  80,  82,  84,  86,  88,  90,  92,  94,  96,  98, 100, 102,\n",
       "        104, 106, 108, 110, 112, 114, 116, 118],\n",
       "       [  1,   3,   5,   7,   9,  11,  13,  15,  17,  19,  21,  23,  25,\n",
       "         27,  29,  31,  33,  35,  37,  39,  41,  43,  45,  47,  49,  51,\n",
       "         53,  55,  57,  59,  61,  63,  65,  67,  69,  71,  73,  75,  77,\n",
       "         79,  81,  83,  85,  87,  89,  91,  93,  95,  97,  99, 101, 103,\n",
       "        105, 107, 109, 111, 113, 115, 117, 119]])"
      ]
     },
     "execution_count": 174,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n_examples = 2\n",
    "\n",
    "goal = np.arange(120).reshape(2,3,4,5)\n",
    "\n",
    "print(goal)\n",
    "\n",
    "print(goal.transpose(3,2,0,1))\n",
    "      \n",
    "print(goal.transpose(3,2,0,1).reshape(5, 24))\n",
    "\n",
    "forward = np.arange(120)\n",
    "\n",
    "print(forward)\n",
    "\n",
    "forward = np.arange(120).reshape(3,-1)\n",
    "\n",
    "print(forward)\n",
    "\n",
    "forward_batch = forward.reshape(-1, n_examples).T\n",
    "\n",
    "forward_batch\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 170,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "both the p furiosus and yeast enzymes required a metal ion for activity, but whereas the yeast enzyme has an absolute requirement for mg2+, the p furiosus enolase was equally active in the presence of mn2+\n",
      "\n",
      "enzymeactivityenzymaticactivityonhalogenatedcompoundswasmeasuredbychloridereleasedetectionasdescribed 13afterincubationat30â°cfor30minintabuffertris-acetate01mph85supplementedwith2mmmcaactivity of deae-bound dehalogenase in standard conditions was detected after removal of the dehalogenase-deae-sephacel complex by centrifugation, and quantitation of chloride in the supernatant\n",
      "\n",
      "the activity of the 2-haloacid dehalogenase at different phs was inspected using 01 m acetate ph 3â€“7, 01 m tris-acetate ph 7â€“10 and 01 m sodium bicarbonate ph 10â€“12 buffers to attain acidic, neutral and basic buffered environment in the assay, respectively\n",
      "\n",
      "inductionhemeoxygenase1modulatescisaconitaseactivityinlensepithelialcells rzymkiewicz d keyse 1989 99 103 s cruse 1988 3348 3353 i keyse 1987 14821 14825 s kutty 1995 1177 1181 r padgaonkar 1997 435 443 v lautier 1992 227 232 d ewing 1993 1512 1519 j stocker 1987 1043 1047 r applegate 1991 974 978 l balla 1992 18148 18153 g vile 1994 2607 2610 g basumodak 1993 4505 4510 s mascotti 1995 239 261 d reddan 1986 168 174 j andley 1990 1099 1106 u zigman 1995 899 901 s vile 1993 14678 14681 g andley 1998 31252 31261 u shigeki 1985 7865 7869 s rose 1967 1870 1879 i drapier 1993 643 649 j applegate 1995 285 291 l mccoubrey 1994 155 161 w keyse 1989 787 791 s andley 1994 575 590 u principlespracticeophthalmologyclinicalpractice gardner 1992 8757 8763 p pantopoulos 1995 2917 2924 k beinert 1997 653 659 h leenders 1974 119 147 h rzymkiewicz 1988 122 123 d nowak 1986 102 104 j graber 1997 29546 29553 m maines 1984 409 417 m rzymkiewiczx2000x324 rzymkiewiczx2000x324x328 rzymkiewiczx2000x324xd rzymkiewiczx2000x324x328xd item s0006-291x0092408-2 s0006291x00924082 1-s20-s0006291x00924082 101006/bbrc20002408 272308 2014-02-19t055041126369-0500 2000-04-02 1-s20-s0006291x00924082-mainpdf https//s3amazonawscom/prod-ucs-content-store-us-east/content/piis0006291x00924082/main/application/pdf/8cc88c752e59e7902186d96281339a24/mainpdf https//s3-eu-west-1amazonawscom/prod-ucs-content-store-eu-west/content/piis0006291x00924082/main/application/pdf/8cc88c752e59e7902186d96281339a24/mainpdf mainpdf pdf true 158175 main 5 1-s20-s0006291x00924082-main_1png https//s3-eu-west-1amazonawscom/prod-ucs-content-store-eu-west/content/piis0006291x00924082/preview/image/png/a465730701a5efaf9f5347c47be665d3/main_1png https//s3amazonawscom/prod-ucs-content-store-us-east/content/piis0006291x00924082/preview/image/png/a465730701a5efaf9f5347c47be665d3/main_1png main_1png png 96458 849 656 image-web-pdf 1 ybbrc 92408 s0006-291x0092408-2 101006/bbrc20002408 academic press ☆ abbreviations used ho-1, heme oxygenase-1 uva, ultraviolet a hsp 32, heat shock protein 32 ire, iron response element, a cis-acting sequence in 5′-untranslated regions of mrna, serving as a binding site for iron regulatory protein irp-1, iron regulatory protein 1 cacon, cytosolic cis-aconitase macon, mitochondrial cis-aconitase gapdh, glyceraldehyde 3-phosphate dehydrogenase ssc, sodium chloride, sodium phosphate buffer spee, sodium chloride, sodium phosphate, ethylenediaminetetraacetic acid buffer pbs, phosphate-buffered saline sds–page, sodium dodecyl sulfate–polyacrylamide gel electrophoresis tris, 2-amino-2-hydroxymethyl-1,3-propanediol regular article induction of heme oxygenase-1 modulates cis-aconitase activity in lens epithelial cells danuta m rzymkiewicz a 1 john r reddan b usha p andley a c 2 a department of ophthalmology and visual sciences, washington university school of medicine, st\n",
      "\n",
      "it is thus likely that interactions of the side chains of asn71 and leu73 with the uracil base are responsible for the absolute uridine specificity of rnase mc1\n",
      "\n",
      "ribostamycininhibitschaperoneactivityproteindisulfideisomerase horibe t goldberger 1963 628 635 r freedman 1984 929 932 r farquhar 1991 81 89 r gunther 1991 24536 24557 r lamantia 1991 4453 4457 m puig 1994 7764 7771 a otsu 1994 6874 6877 m otsu 1995 14958 14961 m pihlajaniemi 1987 643 649 t wetterau 1990 9800 9807 j zhou 1992 144 151 s reynolds 1982 j martindaleextrapharmacopoeia moazed 1987 389 394 d inouye 1989 465 476 s laemmli 1970 680 685 u edman 1956 761 768 p lambert 1983 225 234 n lambert 1983 235 243 n martin 1991 36 42 j shao 2000 13349 13352 f cai 1994 24550 24552 h urade 1999 610 613 r klappa 1997 37 42 p noiva 1993 19210 19217 r sun 2000 45 54 x koivunen 1999 65 74 p primm 2001 281 286 t lebeche 1994 556 561 d tsibris 1989 13967 13970 j primm 1996 33664 33669 t zapun 1992 10 15 a hayano 1995 505 511 t yao 1997 651 658 y song 1995 312 316 j horibex2001x967 horibex2001x967x972 horibex2001x967xt horibex2001x967x972xt item s0006-291x0196105-4 s0006291x01961054 1-s20-s0006291x01961054 101006/bbrc20016105 272308 2014-02-20t060951911787-0500 2001-12-21 1-s20-s0006291x01961054-mainpdf https//s3amazonawscom/prod-ucs-content-store-us-east/content/piis0006291x01961054/main/application/pdf/99fa258d9b976e81f86d26e59f3485ce/mainpdf https//s3-eu-west-1amazonawscom/prod-ucs-content-store-eu-west/content/piis0006291x01961054/main/application/pdf/99fa258d9b976e81f86d26e59f3485ce/mainpdf mainpdf pdf true 182699 main 6 1-s20-s0006291x01961054-main_1png https//s3-eu-west-1amazonawscom/prod-ucs-content-store-eu-west/content/piis0006291x01961054/preview/image/png/f41ce279815499b854c63cd010dee974/main_1png https//s3amazonawscom/prod-ucs-content-store-us-east/content/piis0006291x01961054/preview/image/png/f41ce279815499b854c63cd010dee974/main_1png main_1png png 94927 849 656 image-web-pdf 1 1-s20-s0006291x01961054-si1gif https//s3amazonawscom/prod-ucs-content-store-us-east/content/piis0006291x01961054/stripin/image/gif/5bfd3aca2804fb63c44eff05d13a934f/si1gif https//s3-eu-west-1amazonawscom/prod-ucs-content-store-eu-west/content/piis0006291x01961054/stripin/image/gif/5bfd3aca2804fb63c44eff05d13a934f/si1gif si1 si1gif gif 133 12 11 altimg ybbrc 96105 s0006-291x0196105-4 101006/bbrc20016105 elsevier science usa ☆ abbreviations used pdi, protein disulfide isomerase er, endoplasmic reticulum eia, enzyme immunoassay sds–page, sodium dodecyl sulfate-polyacrylamide gel electrophoresis dab, 3,3′-diaminobenzidine tetrahydrochloride cbb-r250, coomassie brilliant blue r250 gsh, glutathione-reduced form gssg, glutathione-oxidized form pbs, phosphate-buffered saline ru, response units hrp, horseradish peroxidase gapdh, d -glyceraldehyde-3-phosphate dehydrogenase\n",
      "\n",
      "crude extract bacteria were cultured in 10 litres luria-bertani medium and harvested three hours after heat induction to 42°c in the late exponential phase by centrifugation at 11,000 g for 30 minutes approximately 55 g wet weight, resuspended in 002 m tris buffer, ph 80 buffer a, 001 m sodium pyruvate and 01 m kcl 2 ml/g wet weight\n",
      "\n",
      "for cleavage of the his-tag and the linker region introduced into the expression vector as described above, the recombinant dhdps protein was mixed with the agarose coupled proteinase of choice in buffer iii, 2 mm beta-mercaptoethanol was added and slowly rotated overnight at room temperature\n",
      "\n",
      "table 1 refinement statistics number of non-hydrogen protein atoms 8326 number of non-hydrogen solvent atoms 827 resolution range 25–23 å 238–23 å number of reflections 43973 4432 r % 193 245 r free % 285 355 mean temperature factor b å2 all atoms 217 main-chain 208 side-chain 224 solvent 225 rms deviation from ideal values bond lengths å 0011 bond angles deg 21 improper dihedrals deg 15 ncs rms deviations for non-hydrogen atoms å restrained subunits 004 unrestrained subunits 073 regular article the 23 å x-ray crystal structure of s cerevisiae phosphoglycerate mutase 1 1 t richmond daniel j rigden 1 2 * dmitriy alexeev 2 simon ev\n",
      "\n",
      "the annealed dna was incubated with ligase buffer 50 mm tris-hcl ph 75, 10 mm mgcl2, 5 mm dtt unless stated otherwise, in the presence of enzyme and nucleotide cofactors as indicated, in a total volume of 10 mul for 15 minutes at 25 °c\n",
      "\n",
      "stearothermophilus nca1503 is approximately 55 °c baker, 1982 in ligase assay buffer 50 mm tris-hcl ph 75, 10 mm magnesium chloride, 5 mm dtt using 1 mul of ligase assay substrate in a total volume of 10 mul\n",
      "\n",
      "a suitable aliquot of aldolase was incubated at 30 °c with 2 mm potassium hexacyanoferrate iii in 50 mm tris-hcl ph 80 buffer containing 01 m potassium acetate and various concentrations of dhap or fbp were added as substrate\n",
      "\n",
      "the reaction mixture for the preparation of h2ntp consisted of 50 mm tris-hcl buffer ph 80, 1 mm edta, 100 mm kcl, 2 mm gtp and 15 mum gtp cyclohydrolase i in a total volume of 200 ml\n",
      "\n",
      "table 1 crystal and data statistics space group p 21 cell a=825 å, b=933 å, c=1473 å, beta=902° overall 30–212 å highest resolution shell 220–212 å completeness % 908 818 i/sigma i 83 39 multiplicity 26 24 r merge% 62 184 table 2 refinement statistics number of non-hydrogen protein atoms 7487 number of sulphate atoms 40 number of non-hydrogen solvent atoms 819 overall 30–212 å highest resolution shell 220–212 å number of reflections 107,503 5678 r % 197 307 r free% 229 346 mean temperature factor b å2 233 all atoms 233 main-chain 207 side-chain 255 sulphate 300 327 sulphate 301 664 solvent 288 rms deviation from ideal values bond lengths å 0016 bond angles ° 18 table 3 potential hydrogen bonds between the bound sulphate ions and the protein in subunit a protein residue atom sulphate atom distance å ser11 n 300 o3 314 og o4 290 thr20 og1 o4 268 arg87 nh2 301 o2 314 tyr89 oh o1 283 arg113 ne o1 290 nh2 o4 293 arg114 nh1 o2 292 nh2 o4 314 asn183 nd2 o1 323 table 4 kinetic data for wild-type and mutant pgams enzyme k m3pga mum k m2,3bpg mum k cats−1 wild-type 470 32 530 e86q 1100 81 47 a60s 430 72 370 s11a 710 30 20 regular article sulphate ions observed in the 212 å structure of a new crystal form of s cerevisiae phosphoglycerate mutase provide insights into understanding the catalytic mechanism 1 1 edited by k nagai daniel j rigden 1 2 2 rebecca a walter 3 simon ev\n",
      "\n",
      "keywords betaalpha8-barrel calvin cycle evolutionary conservation pentose phosphate epimerase x-ray structure analysis abbreviations mir multiple isomorphous replacement ncs non-crystallographic symmetry rmsd root-mean-square deviation rpease d-ribulose-5-phosphate 3-epimerase ru5p d-ribulose-5-phosphate sas solvent accessible surface xu5p d-xylulose-5-phosphate introduction d-ribulose-5-phosphate 3-epimerase rpease, ec 5131 catalyzes the epimerization of the phosphorylated carbohydrate d-ribulose-5-phosphate ru5p into d-xylulose-5-phosphate xu5p and vice versa as shown in figure 1\n",
      "\n",
      "crystalstructurealginatelyasea1iiisphingomonasspeciesa1178aresolution1 yoon h introduction results and discussion crystallization and structure determination quality of the refined model overall structure of alginate lyase a1-iii the alpha6/alpha5-barrel active cleft disulfide bridges comparison of alpha/alpha-barrel with other enzymes solvent structure and sulfate binding sites materials and methods crystallization and x-ray diffraction structure determination and refinement protein data bank accession number acknowledgements references aleshin 1992 19291 19298 e aleshin 1994 15631 15639 e alzari 1996 265 275 p baron 1994 61 66 a bernstein 1977 535 542 f boyd 1995 162 168 a brown 1991 91 102 b brunger 1990 585 593 a caswell 1989 127 134 r davidson 1976 707 713 i dunne 1985 562 567 w elyakova 1974 341 354 j furey 1997 590 620 w hansen 1984 704 709 j higgins 1994 307 318 d hisano 1994 109 110 t hooft 1996 272 r juy 1992 89 91 m kanehisa 1997 442 444 m kraulis 1991 946 950 p kyte 1995 210 214 j structureinproteinchemistry lange 1989 302 308 b laskowski 1993 283 291 r linhardt 1986 135 176 r linker 1984 958 964 a luzzati 1952 802 810 v matsubara 1998 29 36 y mcdonald 1994 777 793 i merrit 1994 869 873 e mikami 1994 691 692 b monday 1996 625 632 s mrsny 1994 357 366 r mrsny 1996 233 243 r muramatsu 1977 1939 1946 t murata 1993 427 437 k nicholls 1991 281 296 a pecin 1994 124 127 a preiss 1962 309 316 j ramachandran 1968 283 437 g sakon 1997 810 818 j schiller 1993 4780 4789 n sevcik 1998 854 866 j sutherland 1981 48 57 i thomassen 1987 334 351 m wang 1985 90 112 b yonemoto 1991 152 157 y yonemoto 1993 336 342 y yoonx1999x505 yoonx1999x505x514 yoonx1999x505xh yoonx1999x505x514xh item s0022-28369992883-8 s0022283699928838 1-s20-s0022283699928838 101006/jmbi19992883 272582 2010-12-06t050834278476-0500 1999-07-09 1-s20-s0022283699928838-mainpdf https//s3amazonawscom/prod-ucs-content-store-us-east/content/piis0022283699928838/main/application/pdf/6b53c6600f0511f88e721d9aeae89797/mainpdf https//s3-eu-west-1amazonawscom/prod-ucs-content-store-eu-west/content/piis0022283699928838/main/application/pdf/6b53c6600f0511f88e721d9aeae89797/mainpdf mainpdf pdf true 541074 main 10 1-s20-s0022283699928838-main_1png https//s3-eu-west-1amazonawscom/prod-ucs-content-store-eu-west/content/piis0022283699928838/preview/image/png/eb1dc7bb71f577e6fc38f0d5713bc182/main_1png https//s3amazonawscom/prod-ucs-content-store-us-east/content/piis0022283699928838/preview/image/png/eb1dc7bb71f577e6fc38f0d5713bc182/main_1png main_1png png 85452 849 656 image-web-pdf 1 1-s20-s0022283699928838-gr2jpg https//s3amazonawscom/prod-ucs-content-store-us-east/content/piis0022283699928838/gr2/downsampled/image/jpeg/d61883e26652fc3a52fd354f307d988b/gr2jpg https//s3-eu-west-1amazonawscom/prod-ucs-content-store-eu-west/content/piis0022283699928838/gr2/downsampled/image/jpeg/d61883e26652fc3a52fd354f307d988b/gr2jpg gr2 gr2jpg jpg 27818 344 358 image-downsampled 1-s20-s0022283699928838-gr2sml https//s3amazonawscom/prod-ucs-content-store-us-east/content/piis0022283699928838/gr2/thumbnail/image/gif/933f5fe521ec32b7580c7a047e9412a2/gr2sml https//s3-eu-west-1amazonawscom/prod-ucs-content-store-eu-west/content/piis0022283699928838/gr2/thumbnail/image/gif/933f5fe521ec32b7580c7a047e9412a2/gr2sml gr2 gr2sml sml 2372 93 97 image-thumbnail 1-s20-s0022283699928838-gr3jpg https//s3amazonawscom/prod-ucs-content-store-us-east/content/piis0022283699928838/gr3/downsampled/image/jpeg/73f008ef70923daef1a099ba5e1ed3b5/gr3jpg https//s3-eu-west-1amazonawscom/prod-ucs-content-store-eu-west/content/piis0022283699928838/gr3/downsampled/image/jpeg/73f008ef70923daef1a099ba5e1ed3b5/gr3jpg gr3 gr3jpg jpg 55246 255 499 image-downsampled 1-s20-s0022283699928838-gr3sml https//s3amazonawscom/prod-ucs-content-store-us-east/content/piis0022283699928838/gr3/thumbnail/image/gif/a88cb7fda7138381aa3927e2be13350f/gr3sml https//s3-eu-west-1amazonawscom/prod-ucs-content-store-eu-west/content/piis0022283699928838/gr3/thumbnail/image/gif/a88cb7fda7138381aa3927e2be13350f/gr3sml gr3 gr3sml sml 3868 64 125 image-thumbnail 1-s20-s0022283699928838-gr6jpg https//s3amazonawscom/prod-ucs-content-store-us-east/content/piis0022283699928838/gr6/downsampled/image/jpeg/f92affd909f7b69a227eaa422cd6d0e0/gr6jpg https//s3-eu-west-1amazonawscom/prod-ucs-content-store-eu-west/content/piis0022283699928838/gr6/downsampled/image/jpeg/f92affd909f7b69a227eaa422cd6d0e0/gr6jpg gr6 gr6jpg jpg 33083 254 354 image-downsampled 1-s20-s0022283699928838-gr6sml https//s3amazonawscom/prod-ucs-content-store-us-east/content/piis0022283699928838/gr6/thumbnail/image/gif/b232afb58b5b3c23bd04fa6229d037a9/gr6sml https//s3-eu-west-1amazonawscom/prod-ucs-content-store-eu-west/content/piis0022283699928838/gr6/thumbnail/image/gif/b232afb58b5b3c23bd04fa6229d037a9/gr6sml gr6 gr6sml sml 3720 90 125 image-thumbnail 1-s20-s0022283699928838-gr8jpg https//s3amazonawscom/prod-ucs-content-store-us-east/content/piis0022283699928838/gr8/downsampled/image/jpeg/c9d7d212e3ba88e00f0bf670b2bb8ae3/gr8jpg https//s3-eu-west-1amazonawscom/prod-ucs-content-store-eu-west/content/piis0022283699928838/gr8/downsampled/image/jpeg/c9d7d212e3ba88e00f0bf670b2bb8ae3/gr8jpg gr8 gr8jpg jpg 16343 90 135 image-downsampled 1-s20-s0022283699928838-gr8sml https//s3amazonawscom/prod-ucs-content-store-us-east/content/piis0022283699928838/gr8/thumbnail/image/gif/55703303bbdf86b4458dfe852b06963d/gr8sml https//s3-eu-west-1amazonawscom/prod-ucs-content-store-eu-west/content/piis0022283699928838/gr8/thumbnail/image/gif/55703303bbdf86b4458dfe852b06963d/gr8sml gr8 gr8sml sml 6859 84 125 image-thumbnail 1-s20-s0022283699928838-gr9jpg https//s3amazonawscom/prod-ucs-content-store-us-east/content/piis0022283699928838/gr9/downsampled/image/jpeg/455344187f63b6e5b998832a6f49b181/gr9jpg https//s3-eu-west-1amazonawscom/prod-ucs-content-store-eu-west/content/piis0022283699928838/gr9/downsampled/image/jpeg/455344187f63b6e5b998832a6f49b181/gr9jpg gr9 gr9jpg jpg 10111 66 100 image-downsampled 1-s20-s0022283699928838-gr9sml https//s3amazonawscom/prod-ucs-content-store-us-east/content/piis0022283699928838/gr9/thumbnail/image/gif/c0294209bdf638e6b60970fe6006cb87/gr9sml https//s3-eu-west-1amazonawscom/prod-ucs-content-store-eu-west/content/piis0022283699928838/gr9/thumbnail/image/gif/c0294209bdf638e6b60970fe6006cb87/gr9sml gr9 gr9sml sml 5402 83 125 image-thumbnail 1-s20-s0022283699928838-gr10jpg https//s3amazonawscom/prod-ucs-content-store-us-east/content/piis0022283699928838/gr10/downsampled/image/jpeg/28196c5aaefe76f5adef09330569cd6a/gr10jpg https//s3-eu-west-1amazonawscom/prod-ucs-content-store-eu-west/content/piis0022283699928838/gr10/downsampled/image/jpeg/28196c5aaefe76f5adef09330569cd6a/gr10jpg gr10 gr10jpg jpg 11630 69 141 image-downsampled 1-s20-s0022283699928838-gr10sml https//s3amazonawscom/prod-ucs-content-store-us-east/content/piis0022283699928838/gr10/thumbnail/image/gif/46d11e1b4697f5f14d2b9781ccd29581/gr10sml https//s3-eu-west-1amazonawscom/prod-ucs-content-store-eu-west/content/piis0022283699928838/gr10/thumbnail/image/gif/46d11e1b4697f5f14d2b9781ccd29581/gr10sml gr10 gr10sml sml 4482 61 125 image-thumbnail 1-s20-s0022283699928838-gr1gif https//s3amazonawscom/prod-ucs-content-store-us-east/content/piis0022283699928838/gr1/downsampled/image/gif/5e7014fb448465f4f8fdd51a8c7e9e87/gr1gif https//s3-eu-west-1amazonawscom/prod-ucs-content-store-eu-west/content/piis0022283699928838/gr1/downsampled/image/gif/5e7014fb448465f4f8fdd51a8c7e9e87/gr1gif gr1 gr1gif gif 9320 213 509 image-downsampled 1-s20-s0022283699928838-gr1sml https//s3amazonawscom/prod-ucs-content-store-us-east/content/piis0022283699928838/gr1/thumbnail/image/gif/e323124db198e936aa873b1d39d1df24/gr1sml https//s3-eu-west-1amazonawscom/prod-ucs-content-store-eu-west/content/piis0022283699928838/gr1/thumbnail/image/gif/e323124db198e936aa873b1d39d1df24/gr1sml gr1 gr1sml sml 1250 52 125 image-thumbnail 1-s20-s0022283699928838-gr4gif https//s3amazonawscom/prod-ucs-content-store-us-east/content/piis0022283699928838/gr4/downsampled/image/gif/50bff3f404c9626234fde8ff0b6a6f5c/gr4gif https//s3-eu-west-1amazonawscom/prod-ucs-content-store-eu-west/content/piis0022283699928838/gr4/downsampled/image/gif/50bff3f404c9626234fde8ff0b6a6f5c/gr4gif gr4 gr4gif gif 33574 407 743 image-downsampled 1-s20-s0022283699928838-gr4sml https//s3amazonawscom/prod-ucs-content-store-us-east/content/piis0022283699928838/gr4/thumbnail/image/gif/d077e190eacbd7756174df88d6342313/gr4sml https//s3-eu-west-1amazonawscom/prod-ucs-content-store-eu-west/content/piis0022283699928838/gr4/thumbnail/image/gif/d077e190eacbd7756174df88d6342313/gr4sml gr4 gr4sml sml 1967 69 125 image-thumbnail 1-s20-s0022283699928838-gr5gif https//s3amazonawscom/prod-ucs-content-store-us-east/content/piis0022283699928838/gr5/downsampled/image/gif/7cc0d862ea85825475c618e2dcc213ab/gr5gif https//s3-eu-west-1amazonawscom/prod-ucs-content-store-eu-west/content/piis0022283699928838/gr5/downsampled/image/gif/7cc0d862ea85825475c618e2dcc213ab/gr5gif gr5 gr5gif gif 23628 372 506 image-downsampled 1-s20-s0022283699928838-gr5sml https//s3amazonawscom/prod-ucs-content-store-us-east/content/piis0022283699928838/gr5/thumbnail/image/gif/6dd54ef7c9471103a3b3d7cf71253c04/gr5sml https//s3-eu-west-1amazonawscom/prod-ucs-content-store-eu-west/content/piis0022283699928838/gr5/thumbnail/image/gif/6dd54ef7c9471103a3b3d7cf71253c04/gr5sml gr5 gr5sml sml 2807 92 125 image-thumbnail 1-s20-s0022283699928838-gr7gif https//s3amazonawscom/prod-ucs-content-store-us-east/content/piis0022283699928838/gr7/downsampled/image/gif/cc11a631897d0fe44952269f1b0ac4fa/gr7gif https//s3-eu-west-1amazonawscom/prod-ucs-content-store-eu-west/content/piis0022283699928838/gr7/downsampled/image/gif/cc11a631897d0fe44952269f1b0ac4fa/gr7gif gr7 gr7gif gif 55950 400 523 image-downsampled 1-s20-s0022283699928838-gr7sml https//s3amazonawscom/prod-ucs-content-store-us-east/content/piis0022283699928838/gr7/thumbnail/image/gif/932a985691c7775332e0c3e344a3c3fa/gr7sml https//s3-eu-west-1amazonawscom/prod-ucs-content-store-eu-west/content/piis0022283699928838/gr7/thumbnail/image/gif/932a985691c7775332e0c3e344a3c3fa/gr7sml gr7 gr7sml sml 4000 93 122 image-thumbnail yjmbi 92883 s0022-28369992883-8 101006/jmbi19992883 academic press regular article figure 1 the cleavage site of alginate polymer beta-d-mannuronate by alginate lyase\n",
      "\n",
      "while the pectate lyases show an absolute requirement for ca2+ for activity, this is not the case with chondroitinase b michelacci & dietrich, 1974, 1975 am, wh\n",
      "\n",
      "keywords ornithine decarboxylase polyamines pyridoxal-5′-phosphate antizyme abbreviations az antizyme azbe putative locus of the antizyme-binding element asa solvent-accessible surface area hodc human ornithine decarboxylase modc full-length mouse ornithine decarboxylase 1-461 modct truncated mouse ornithine decarboxylase delta425-461 odc ornithine decarboxylase plp pyridoxal-5′-phosphate rmsd root-mean-square deviation todc trypanosoma brucei ornithine decarboxylase polyamines are ubiquitous, and are required for cell growth and differentiation in eukaryotes\n",
      "\n",
      "cells were harvested five hours later and resuspended in buffer a 50 mm sodium phosphate ph 80, 300 mm nacl, 01 mm antiprotease 4-2-aminoethyl-benzenesulfonyl fluoride hydrochloride aebsf from boehringer\n",
      "\n",
      "the amino acid residues of e coli scs that interact with the phosphate groups, the ribose and the five-membered ring of the purine base joyce et al, 2000 are absolutely conserved in the atp and gtp-specific isoforms of all five eukaryotic species for which both isoforms have been identified johnson et al, 1998a\n",
      "\n",
      "the contents of two syringes, one containing enzyme plus 02 mm tyrosine and 2 units/ml of inorganic pyrophosphatase in the standard buffer, the other containing 02 mm tyrosine and various concentrations of atp, were rapidly mixed and the change in the intrinsic fluorescence of the protein was monitored as a function of time using an excitation wavelength of 295 nm and an emission filter with a cutoff above 320 nm\n",
      "\n",
      "keywords hpar14 peptidyl prolyl cis/trans isomerase hpin1 nuclear magnetic resonance solution structure abbreviations 2d and 3d two and three-dimensional, respectively a 600 absorbance at 600 nm wavelength dmso dimethylsulfoxide gst glutathione s-transferase hsqc heteronuclear single-quantum spectroscopy iptg isopropyl-beta-d-thiogalactopyranoside nima never in mitosis a noe nuclear overhauser effect noesy noe spectroscopy pbs phosphate-buffered saline ppiase peptidyl prolyl cis/trans isomerase pna 4-nitroaniline pser phosphoserine pthr phosphothreonine rms root-mean-square suc succinate tocsy total correlation spectroscopy introduction the peptidyl prolyl cis/trans isomerases ppiases catalyze rotation around the peptide bond preceding a proline residue\n",
      "\n",
      "on the contrary, a bs-rnase-like dimer mazzarella et al, 1993 is absolutely ruled out since no disulphide bonds can be established between the two molecules residues 31 and 32 are both cysteine in bs-rnase, but arginine residues in hp-rnase\n",
      "\n",
      "refinement r-factor 0182 for 40,632 reflections r-free 0239 for 4532 reflections coordinates error luzzati plot å 018 data between 5 and 185 å rmsd on bond distances å 0010 rmsd on angle distances å 0028 average b-factor å2 protein atoms 4447 161 metal atom 1 zinc 181 solvent atoms 437 254 l-methionine atoms 9 110 regular article how methionyl-trna synthetase creates its amino acid recognition pocket upon l-methionine binding 1 1 edited by a\n",
      "\n",
      "mitochondrial pig heart citrate synthase cs was from roche diagnostics gmbh mannheim, germany and stored in te buffer 50 mm tris, 2 mm edta ph 80\n",
      "\n",
      "rcm-t1 refolding assay to measure the ppiase activity towards protein substrates refolding experiments were performed with the reduced and s-carboxymethylated s54g/p55n-variant of rnase t1 rcm-t1, which is unfolded in 100mm tris ph 80 but spontaneously refolds upon dilution into the same buffer containing 2m nacl\n",
      "\n",
      "table 1 refinement of the prorstt complexes crystal prorstt prorstt+trnaprocgg prorstt+trnaprocgg prorstt+trnaprocgg soaked with l-proline pro-ams – prolinol+atp-mn2+ resolution å 15-29 20-285 20-30 20-30 solvent content % 640 717 760 756 work reflections 67,570 52,192 43,678 46,141 test reflections 3538 5% 2797 5% 2325 48% 2452 5% r-free 0243 0242 0283 0254 r-work 0200 0210 0252 0221 no protein atoms in model occ>0 14,835 two dimers 7642 one dimer 7478 one dimer 7494 one dimer no substrate atoms 32 four proline 1439 trna 60 two pro-ams 1439 trna 1436 trna, 14 two prolinol, 62 two atp no solvent molecules 441 water 32 water four sulphate ions no water four sulphate ions 25 water two sulphate ions no metal atoms four zinc two zinc two zinc two zinc 〈b〉 protein å2 329 432 478 436 〈b〉 solvent å2 296 510 469 393 〈b〉 substrate å2 196 proline 623 trna 358 pro-ams 565 trna 538 trna, 714 atp 389 prolinol rms bonds 0005 0007 0008 0007 rms angles 1225 1383 1422 1392 ramachandran plot favourable % 917 884 857 877 additional % 77 102 136 114 generous % 06 11 07 09 disallowed % 00 02 00 00 table 2 data collection parameters of the prorstt complexes crystal prorstt prorstt+trnaprocgg prorstt+trnaprocgg prorstt+trnaprocgg prorstt+trnaprocgg prorstt+trnaprocgg soaked with l-proline pro-ams – atp-mg2+ prolinol+atp-mn2+ proline+atp-mn2+ gives pro-amp esrf beamline swissnorwegian bm1 id2 id2 id2 id2 id2 detector 300 mm mar ip 345 mm mar ip 345 mm mar ip 345 mm mar ip 345 mm mar ip 345 mm mar ip wavelength å 0873 099 099 099 099 099 exposure/image s/l deg 600 20 20 20 20 25 space-group p21212 p43212 p43212 p43212 p43212 p43212 cell dimensions å a=1326, b=1935, c=1252 a=b=1408, c= 2370 a=b=1416, c=2380 a=b=1417, c=2363 a=b=1413, c=2376 a=b=1410, c=2376 no crystals 1 1 2 1 1 1 resolution 15-29 å 20-285 å 20-30 20-335 20-30 20-33 total reflections 284,962 221,758 287,504 146,818 184,299 129,284 unique reflections 71,108 55,013 46,220 34,779 48,648 35,549 average redundancy 40 40 28 62 19 42 40 38 35 36 37 completeness 990 982 961 939 698 987 985 996 100 973 974 r-merge highest bin 0068 0223 0084 0298 0102 0309 0091 0382 0069 0269 0088 0335 table 3 data collection statistics for hisrstt structures crystal hisrstt apo-1 hisrstt apo-2 soaked with – 15h soak atp+mn2+ beamline id2 19/4/97 bm14 13/9/97 detector 180mm mar 345mm mar wavelength å 0933 0953 exposure/image s/l deg 15 30 space-group c2221 c2 cell dimensions å a=1111, b=1022, c=915 a=993, b=1034, c=572, beta=1172 no crystals 1 1 resolution 12-24 å 194-20 å highly anisotropic total reflections 39,592 120,296 unique reflections 17,744 30,177 average redundancy highest bin 22 18 40 21 completeness % highest bin 867 721 873 760 complete to 26-27 r-merge highest bin 0074 0137 0067 0197 table 4 refinement statistics for the hisrstt structures hisrstt apo-1 hisrstt apo-2 15 hour soak with atp+mn2+ resolution å 12-24 18-25 solvent content 554% 555% work reflections 16,834 16,304 test reflections 899 44% 877 49% r-free 0256 0294 r-work 0208 0255 no protein atoms in model with non-zero occupancy 3229 1 monomer 2733 1 monomer with insertion domain missing poorly ordered regions 57-62 ordering loop, 114-120 motif 2 loop, 274-279 57-62 ordering loop, 114-120 motif 2 loop, 176-224 insertion domain, 274-279 no solvent molecules 65 water, one sulphate no water, two sulphate no substrate atoms – 23 amp 〈b〉 protein å2 327 512 〈b〉 solvent å2 312 713 sulphate 〈b〉 substrate å2 – 595 amp anisotropic b-factor correction å2 b11=111, b22=197, b33=−307, b12=00, b12=00, b23=00 b11=−431, b22=2265, b33=−1834, b12=00, b12=2463, b23=00 rms bonds 0006 0008 rms angles 1280 1281 ramachandran plot favourable % 938 912 additional 53 81 generous % 09 04 disallowed % 00 04 regular article a succession of substrate induced conformational changes ensures the amino acid specificity of thermus thermophilus prolyl-trna synthetase comparison with histidyl-trna synthetase 1 1 edited by a fersht anna yaremchuk 1 2 michael tukalo 1 2 morton grøtli 3 stephen cusack 1 * cusack@embl-grenoblefr 1 european molecular biology laboratory, grenoble outstation, c/o ill, 156x, f-38042 grenoble cedex 9, france 2 institute of molecular biology and genetics, nas of ukraine, 252627 kiev-143, ukraine 3 center for solid phase organic combinatorial chemistry calsberg laboratory, gamle calsberg vej 10, dk-2300, valby, denmark * corresponding author abstract we describe the recognition by thermus thermophilusprolyl-trna synthetase prorstt of proline, atp and prolyl-adenylate and the sequential conformational changes occurring when the substrates bind and the activated intermediate is formed\n",
      "\n",
      "this is clearly reminiscent of the case of histidyl-trna synthetase and again similarly, divalent cation binding sites appear to be maintained either side of the beta- and gamma-phosphates of the atp figure 3b21 we note that gln225, thr228 and his230 are absolutely conserved in all known eukaryote/archaeon-like prors but replaced by respectively a highly conserved glutamate, histidine there is one example of a glutamine and phenylalanine in prokaryote-like prors see table 1 in the work by yaremchuk14\n",
      "\n",
      "the two oxygen atoms from the beta and gamma-phosphate groups are in the same relative position, but the oepsilon2 atom of glu143 replaces one of the ntpase solvent molecules figure 5b\n",
      "\n",
      "reactions were carried out at 37°c for three minutes lanes 1, 4, 7, 10, six minutes lanes 2, 5, 8, 11 and 12 minutes lanes 3, 6, 9, 12 in the standard buffer containing 100 mm k-hepes, 10 mm mgcl2, 02 mm edta, 2 mm atp, 5 units ml−1 yeast inorganic pyrphosphatase sigma, in the absence lanes 1–3 and presence of 05 mm cyto-metrs lanes 5–6, mito-metrs lanes 7–9, or e coli metrs lanes 10–12\n",
      "\n",
      "formation of hcy thiolactone was determined by 2d thin-layer chromatography tlc and normalized to total incorporation of 35s radioactivity into protein28 editing of homocysteine and met-s-coa synthesis in vitro kinetic measurements were carried out at 37°c in the standard buffer containing 100 mm k-hepes ph 74, 10 mm mgcl2, 02 mm edtas mm atp, 5 units ml−1 yeast inorganic pyrphosphatase sigma, and 05 mum yeast cytoplasmic or mitochondrial metrs\n",
      "\n",
      "cells were washed twice in 20 mm sodium phosphate ph 74 containing 5 mm edta, 150 mm nacl, 1 mm dtt, 1 mug ml−1 of serine protease inhibitor pefabloc pephafarm ag, basel and then resuspended in the same buffer\n",
      "\n",
      "in situ phosphorylation assay hela cells treated or not with 100 mum drb during 60 minutes were washed twice with ice-chilled pbs and lysed on ice in tpa buffer 10 mm hepes ph 79, 10 mm kcl, 15 mm mgcl2, 05% nonidet p-40, 02 mm edta, 05 mm dtt, protease inhibitor cocktail sigma supplemented with 100 mum drb\n",
      "\n",
      "for immunoprecipitation, mitotic cells or a nuclear pellet from interphase cells extracted with 420 mm nacl in tpa were dissolved in one volume of sds buffer 50 mm tris ph 75, 150 mm nacl, 02% naf, 08% sds, 1 mm edta, 1× protease inhibitor cocktail and 20 nm microcystin-lr\n",
      "\n",
      "to neutralize the sds and allow antigen-antibody interactions, one volume of triton buffer 50 mm tris ph 75, 150 mm nacl, 02% naf, 4% triton x-100, 1 mm edta, 1× protease inhibitor cocktail and 20 nm microcystin-lr was added to the lysates, which were cleared by centrifugation at 10,000 g for ten minutes\n",
      "\n",
      "real space fit no of residues with r real<025 in 3m|f o|−2d|f c| difference map main chain atoms only 1005 all protein atoms 961 table 2 analysis of inter-monomeric contacts in the crystal structure of the t cruzi pepck dimeric model i dimeric model ii no of contacting dist<5 å atoms 331 181 excluded solvent accessible area å2 1172 1134 hydrophobicity of excluded accessible surface % 723 653 contact surface shape complementarity sc a 077 061 a calculated by the method of lawrence & colman48 regular article crystal structure of the dimeric phosphoenolpyruvate carboxykinase pepck from trypanosoma cruzi at 2 å resolution 1 1 edited by r huber stefano trapani 1 4 jutta linss 2 samuel goldenberg 2 hannes fischer 3 aldo f craievich 3 glaucius oliva 4 * oliva@ifscuspbr 1 instituto de quı́mica de são carlos, universidade de são paulo, são carlos - sp,brasil brazil 2 fundação oswaldo cruz, departamento de bioquı́mica e biologia molecular, laboratório de expressão gênica, rio de janeiro - rj,brasil brazil 3 instituto de fı́sica universidade de são paulo, são paulo, sp,brasil brazil 4 instituto de fı́sica de são carlos, universidade de são paulo, são carlos - sp,brasil brazil * corresponding author abstract atp-dependent phosphoenolpyruvate carboxykinase pepck atp oxaloacetate carboxylyase transphosphorylating, ec 41149 is a key enzyme involved in the catabolism of glucose and amino acids in the parasite trypanosoma cruzi, the causative agent of chagas’ disease\n",
      "\n",
      "coli pck complexes complex i complex ii resolution limits å 100-20 100-20 no of reflections 35,230 27,047 final r-value all data 0198 0225 r free-value 0266 0285 no of non-hydrogen atoms protein + substrates 4127 4136 no of solvent molecules 275 219 mean coordinate error å46 02 02 rmsd from ideality in bond lengths å 0006 0007 rmsd from ideality in bond angles deg 140 142 rmsd from ideality in dihedral angles deg 2340 2320 rmsd from ideality in improper angles deg 109 111 mean temperature factor for pck å2 main-chain 25 16 side-chain 27 17 mean b-factor for adp å2 26 14 mean b-factor for water å2 36 25 mean b-factor for mg2+ å2 31 9 mean b-factor for alf3 å2 34 38 mean b-factor for pyruvate å2 - 44 table 2 summary of active site coordination, electrostatic interactions, hydrogen bonding and distances in pck-alf3-mg2+-adp and pck-alf3-mg2+-adp-pyruvate complexes i and ii, respectively atom 1 atom 2 distance å complex i complex ii al adp o3beta 28 29 h2o/pyruvate o 33 43 f1 arg333 nh2 30 29 ser250 ogamma 25 27 adp o3beta 27 25 f2 k213 nzeta 32 33 his232 nϵ2 27 30 asp269 odelta2 25 36 h2o 29 h2o 25 f3 mg2+ 20 21 mg2+ adp o2beta 23 22 f3 20 20 thr255 o1gamma 21 22 h2o 25 22 h2o 22 22 h2o 22 23 regular article the phosphoryl-transfer mechanism of escherichia coli phosphoenolpyruvate carboxykinase from the use of alf3 1 1 edited by r huber athena m sudom 1 lata prasad 1 hughes goldie 2 louis tj delbaere * 1 louisdelbaere@usaskca 1 department of biochemistry and department of microbiology 107 wiggins road, university of saskatchewan, saskatoon saskatchewan, s7n 5e5 canada 2 department of microbiology 107 wiggins road, university of saskatchewan, saskatoon saskatchewan, s7n 5e5 canada * corresponding author abstract the mechanism of reversible transfer of the gamma-phosphate group of atp by escherichia coli phosphoenolpyruvate carboxykinase pck on to its substrate is of great interest\n",
      "\n",
      "this observation indicates that since the active site of phosphoenolpyruvate carboxykinase is a completely different environment that is nearly shielded from bulk solvent, it may thus favor the binding of the alf3 compound, which most perfectly substitutes for the gamma-phosphate group of atp\n",
      "\n",
      "anomalous pk a values for glutamate residues have been observed for a number of enzymes including lysozyme and carboxypeptidase a where the apolar microenvironment of these residues results in an increase in their pk a values to 65 and 70, respectively28, 29 interestingly, the location of glu182 on a mobile loop of the protein would place it in a solvent exposed position in the absence of substrate, and we would expect it to have a normal pk a value\n",
      "\n",
      "a similar loop movement occurs in triose phosphate isomerase, the beta5-alpha6 loop moving 7 å on substrate binding to exclude solvent from the active site environment during the isomerisation10, 11, 30 in the e coli class ii fbp-aldolase, the beta5-alpha7 loop not only contains glu181 and glu182 described above, but also three glycine residues gly179, gly180 and gly184\n",
      "\n",
      "this provides further evidence for the importance of the sequence and dynamics of the loops in alpha/beta8 barrels for catalysis and specificity of the reaction already seen for triose phosphate isomerase10, 11, 30 and recently exploited in directed evolution of new catalytic activity on this scaffold31 it thus appears that, in the absence of substrates, the beta5-alpha7 loop of the class ii fbp-aldolase is disordered and glu182 would be solvent exposed and distant from the active site\n",
      "\n",
      "hexacyanoferrate iii oxidation of the enediolate carbanion intermediate the oxidation of the carbanionic enediolate intermediate by hexacyanoferrate iii was followed spectrophotometrically at 420 nm16, 25 a suitable aliquot of aldolase was incubated at 30°c with 2 mm potassium hexacyanoferrate iii in 50 mm tris-hcl buffer containing 01 m potassium acetate ph 80 and various concentrations of dhap and fbp\n",
      "\n",
      "oxidation-reduction indicators as oxidants of intermediary carbanions studies with aldolase, aspartate aminotransferase, pyruvate decarboxylase and 6-phosphogluconate dehydrogenase biochemistry 12 1973 35 40 26 ia rose the absolute configuration of dihydroxyacetone phosphate tritiated by aldolase reaction j am chem soc 80 1958 5835 5836 27 ar fersht structure and mechanism in protein science a guide to catalysis and protein folding 1999 w\n",
      "\n",
      "in the case of pepck, the conformation of the p-loop compatible with gtp binding in a productive mode buries the cys288 side-chain in an environment shielded from solvent\n",
      "\n",
      "the cells were harvested at the stationary phase, suspended in extraction buffer 50 mm hepes-koh ph 76 at 4°c, 1 mm edta, 1 mm dtt, 1 mm phenylmethylsulfonyl fluoride, 10 mum leupeptin, and sonicated\n",
      "\n",
      "in the assay for effect of chitin on the ribonuclease activity, reaction buffer e 40 mm tris/hcl, ph 76, 40 mm kcl, 5 mm edta contained soluble glycochitin with the concen- tration from 005 to 020%\n",
      "\n",
      "olson 15 studied the reaction of glyoxylate with semicarbazide, commented on a \"possible intermediate keto acid-dihydrogen phosphate complex,\" and attributed an inhibition of isocitrate lyase in phosphate buffer using the glyoxylate semicarbazone assay to an ionic strength effect even though potassium phosphate was more inhibitory than sodium chloride at the same ionic strength\n",
      "\n",
      "in yeast ureidoglycolatase assays 421 any case, phosphate buffers should be avoided in assay systems for glyoxylate that employ ketone reagents for the\n",
      "\n",
      "ashworth and kornberg 17 have used an imidazole buffer at ph 68 for the assay of isocitrate lyase recently\n",
      "\n",
      "phosphate buffers inhibit the reaction of glyoxylate with phenylhydrazine and appear to inhibit yeast ureidoglycolatase_ optimum activity was near ph 78 for ureidoglycolatase from c'aridida utilis and near ph 73 for ureidoglycolatase from saccharo zzyces cerevisiae\n",
      "\n",
      "in yeast ureidoglycolatase assays 421 any case, phosphate buffers should be avoided in assay systems for glyoxylate that employ ketone reagents for the\n",
      "\n",
      "ashworth and kornberg 17 have used an imidazole buffer at ph 68 for the assay of isocitrate lyase recently\n",
      "\n",
      "phosphate buffers inhibit the reaction of glyoxylate with phenylhydrazine and appear to inhibit yeast ureidoglycolatase_ optimum activity was near ph 78 for ureidoglycolatase from c'aridida utilis and near ph 73 yabio 66901771 0003-26976690177-1 101016/0003-26976690177-1 ☆ supported by grants cy2510 and am08420 from the national institutes of health, united states public health service\n",
      "\n",
      "phosphate buffers inhibit the reaction of glyoxylate with phenylhydrazine and appear to inhibit yeast ureidoglycolatase\n",
      "\n",
      "the incubation mixture to measure p-mutase activity was as follows 60 mm hepes buffer ph 70, 05 mm edta, 6 mm sodium arsenate, 25 mm magnesium acetate, 04 mm nadp, 4 rnm p-glucose-1-p, 014 u of glucose-6-p dehydrogenase and enzyme in a total volume of 05 ml, temperature 37°c\n",
      "\n",
      "the following incubation system was used 05 mm edta, 60 mm hepes buffer ph 70, 2 mm magnesium acetate, 20 mm sodium arsenate, 04 mm nadp, 014 u glucose-6-p dehydrogenase, and 20 mu phosphoglucomutase for r-glucose-l-p r-mutase extract in a total volume of 05 ml\n",
      "\n",
      "the incubation mixture contained 60 mm hepes buffer ph 70 05 mm edta 25 mm magnesium acetate 04 mm nadp, either 4 mm a- or f3-glucose-l-p sodium arsenate ph 70 at the concentrations indicated 014 u glucose-6-p dehydrogenase and extract from euglena gracilis \"protamine sup\" fraction as described in enzyme preparation\n",
      "\n",
      "the endonuclease ecori fragments of x dna were extracted once with phenol saturated with 10 mm nat-edta, ph 80, and dialyzed exhaustively against 10 mm tris-hci, ph 76\n",
      "\n",
      "the unit of activity for aldose 1-epimerase employed is that amount of enzyme which catalyzes the conversion of 1 p, mol a-d-glucose to /3-o-glucose/min employing the experimental conditions of 2 mm a-d-glucose generated in situ from sucrose hydrolysis in 001 m kh~po4/k2hpo4 buffer, ph 70, at 25°c\n",
      "\n",
      "the assay solution was 3 ml 001 m k2hpoflkh~po4, ph 70, containing 2 mm sucrose, 2 g,l of catalase 25 mg/ml, and 30/zl glucose oxidase 100/zm, and was air-saturated at 25°c\n",
      "\n",
      "the g1cnac 6-[32p]phosphate was dissolved in 32 ml 01 m tris-hc1 buffer, ph 76, to which were added the following 032 ml of disodium edta 10 mg/ml, 003 ml of mgcl, 05 m, 01 ml of glucose 1,6-diphos phate 05 mg/ml, 03 ml of utp 25 mm, 04 ml of n crassa extract, 08 ml of b licheniformis extract, 01 ml of inorganic pyrophosphatase 200 units/ml\n",
      "\n",
      "in the purification of liver argininosuccinase the enzyme emerged with the third eluate after successive fractional elution of the deae-cellulose column with phosphate buffers at ph 66, 61, and 54\n",
      "\n",
      "argininosuccinase is protected from cold inactivation by arginine and by phosphate ions but loses activity reversibly in cationic buffers at low temperatures 5,6\n",
      "\n",
      "mandelate racemase from pseudomonas putida ec 5122 catalyzes the reversible interconversion of the r- and s-enantiomers of mandelate oh oh 1 002c-c ---h c6 h5 c6 h5 r-mandelate 5-mandelate the enzyme has been isolated in a high state of purity 1-3 and has been shown to have an absolute divalent metal ion requirement eg, mg2+, mn 2+ for activity 3\n",
      "\n",
      "fractions containing rna ligase activity were pooled 31 ml, dialyzed against 1000 ml of 20 mm tris-hci buffer ph 76 containing 05 mm edta, 50 nm atp, 01 1v1 kci, 10 mm 2-mercaptoethanol, and 10% glycerol buffer b overnight and concen- 0 6 12 18 24 fracilon numb\" 5 oz 01 mei-hao, ai, and hui-fen c fig 4\n",
      "\n",
      "the conditions were the same as for step 4, except that 100 units of rna ligase were applied to a 05 x 5-cm column and the column was developed with a 50-ml linear gradient from 0 to 10 mm atp and from 0 to 02 m nacl in buffer a fig 2\n",
      "\n",
      "the conditions were the same as for step 4, except that 100 units of rna ligase were applied to a 05 x 5-cm column and the column was developed with a 50-ml linear gradient from 0 to 10 mm atp and from 0 to 02 m nacl in buffer a fig 2\n",
      "\n",
      "the detailed standard procedure for the coupled optical enzyme assay used to measure the phosphopentomutase activity is the following 17 ml of 01 lit tris-hci buffer, ph 81, was tted into a cuvette, followed by 25 al of 604 mm adenine, 75 al of the adenosine phosphorylase preparation, 50 ii of diluted adenosine deaminase, 20 al of 135 mm glucose 1,6-diphosphate, phosphopentomutase preparation, and enough water to bring the volume to 21 ml\n",
      "\n",
      "in a 3-ml cuvette containing 17 ml of 01 m tris-ci buffer, ph 81, the following additions were performed at the times indicated by the arrows a 25 jsl of 604 mm adenine b 275 jut of a mixture containing 073 mm mnc12, 98 jm glucose 1,6-diphosphate, 31 mu of phosphopentomutase, 730 mu of diluted adenosine deaminase, and 80 mu of adenosine phosphorylase and c 100 j1 of 80 mm ribose 5-phosphate\n",
      "\n",
      "the column was equilibrated against 10 mm tris-hci, ph 90, 10% ethylene glycol, 45% saturated with am table 1 summary of purification of enolases purification step total protein total protein total activity specific activity ml mg units units/mg crude extract 200 650 5360 82 tydex precipitation 215 nd° 5460 nd phenyl-sepharose chromatography 220 nd 5097 nd g-25 desalting 275 59 4914 84 chromatofocusing enolase i 12 nd 719 nd enolase ii 18 1500 enolase iii 32 1903 phenyl-sepharose chrom\n",
      "\n",
      "enolase pools were saturated with 45% ammonium sulfate and applied to a 1 x 4-cm column which was equilibrated against 10 mm potassium sulfate, ph 75, 10% ethylene glycol, 45% saturated with ammonium sulfate\n",
      "\n",
      "t-glutamylcysteine synthetase activity was determined from the rate of formation of adp assumed to be equal to the rate of oxidation of nadh as calculated from the change in absorbance at 340 nm in reaction mixtures containing the following components in a final volume of 1 ml 01 m tris-hci buffer ph 80, 150 mm kci, 5 mm na2atp, 2 mm phosphoenolpyruvate, 10 mm l-glutamate, 10 mm l-a-aminobutyrate, 20 mm mgc12, 2 mm edta, 02 mm nadh, 17 ,ug of pyruvate kinase, and 17 ag of lactate dehydrogenase 10,17\n",
      "\n",
      "kdpg aldolase activity was measured in a buffer consisting of 20 mm k-mes, ph 65, containing 50 mm nacl, 2 mm m9clz, 015 mm nadh, and 5 units ml-' rabbit muscle lactate dehydrogenase\n",
      "\n",
      "6-phosphogluconate dehydratase activity was measured in 20 mm k-mes buffer, ph 65, containing 2 mm mgc12 and 50 mm nacl, using 1 mm 6-phosphogluconate, 015 mm nadh, and 5 units ml' each of lactate i abbreviations used mes, 4-morpholineethanesulfonic acid sds, sodium dodecyl sulfate\n",
      "\n",
      "dye-screening because of the instability of the enzyme in the absence of ferrous and manganous ions, the screening of dye-ligands for adsorption of the dehydratase using phosphate buffer, as described for kdpg aldolase 1 was not successful little activity was recovered\n",
      "\n",
      "the enzyme bound to many of the adsorbents tested in this buffer including cibacron blue f3g-a which is the same as procion blue h-b we decided that the blue he-g was the most suitable, as it binds the dehydratase strongly without binding very much protein compared with most other dyes\n",
      "\n",
      "after the protein trace had been restored to the baseline, the original buffer containing 20 mm dl-a-glycerophosphate adjusted to ph 65 with hcl, was used to affinity elute 6-phosphogluconate dehydratase\n",
      "\n",
      "the elution is isocratic and the mobile phase used is 50 mm sodium acetate buffer ph 59 with 30% methanol\n",
      "\n",
      "the elution is isocratic and the mobile phase used is 50 mm sodium acetate buffer ph 59 with 30% methanol\n",
      "\n",
      "the proton exchange reactions of various straight-chain l-amino acids, with deuterium of solvent, are catalyzed by the present enzyme as well yabio 84908327 0003-26978490832-7 101016/0003-26978490832-7 purification of bacterial l-methionine gamma-lyase toru nakayama ∗ nobuyoshi esaki ∗ kunio sugie † temir t beresov ‡ 1 hidehiko tanaka ∗ kenji soda ∗ a laboratory of microbial biochemistry, institute for chemical research, kyoto university, uji, kyoto-fu 611, japan b katata research center, toyobo co, ltd, 1-1, katata 2-chome, otsu, shiga 520-02, japan c department of biochemistry, patrice lumumba people's friendship university, 117923 gsp, moscow, b-302, ordzhonikidze, 3, ussr 1 on leave\n",
      "\n",
      "the bicarbonate assay was carried out at 340 nm and 25°c in a reaction mixture of the following final composition 100 mm imidazole-hci buffer, ph 70, 2 mm mnc12, 125 mm adp, 0225 mm nadh, 02 u/ml of pure phosphoenolpyruvate carboxykinase, variable concentrations of bicarbonate, and 15 u/ml malate dehydrogenase\n",
      "\n",
      "the temperature stability of phosphoenolpyruvate carboxykinase at ph 70 in imidazole buffer was increased by mn 2+ an intermediate stability was detected in phosphate buffer fig 4a\n",
      "\n",
      "in fig 5, the initial rate of the reaction catalyzed by phosphoenolpyruvate carboxykinase in the presence of saturating adp and phosphoenolpyruvate was measured as a function of bicarbonate concentration\n",
      "\n",
      "in fig 5, the initial rate of the reaction catalyzed by phosphoenolpyruvate carboxykinase in the presence of saturating adp and phosphoenolpyruvate was measured as a function of bicarbonate concentration\n",
      "\n",
      "a ph 74 phosphate buffer at 27 to 37°c gave e a 5 0 20 15 l 50 60 70 ph fla 6 ph dependence of fucosterol epoxide lyase using various buffering agents o sodium phosphate, 76 mm a bicine [nn-bis2-hydroxyethylglycine] 100 mm o mops morpholinopropanesulfonic acid, 100 mm 0 hepes 14-2-hydroxyethyl-1-piperazineethanesulfonic acid], 100 mm\n",
      "\n",
      "to a 4-ml cuvette 1cm pathlength were added 05 ml 1 m trismaleate buffer, ph 73, 01 ml 1 m mgs04, the bacterial enzyme preparation 02 ml, containing about 14 units of d-glucarate dehydratase activity, and 20 ml of standard d-glucarate solution or glucarate-rich fraction obtained from the ion-exchange column separation of a crude urine sample see above\n",
      "\n",
      "the standard assay contains 1 mm [3-'4c]acetoacetate 10 ci//,mol, 100 mm tris-hci buffer, ph 75, 125 mm atp, 25 mm mgc12, 50 mm kci, 0050 mm coa, 10 mm dl-carnitine, and 1 unit of carnitine acetyl transferase, and the appropriately diluted cytosolic extract in a total volume of 02 ml\n",
      "\n",
      "these include separation of radiolabeled dansylarogenate' from dansylphenylalanine by thin-layer chromatography 6, selective oxidation of unutilized arogenate followed by fluorometric estimation of ' abbreviations used dansyl-, 5-dimethylaminonaphthalene-l-sulfonyl- pal, phenylalanine ammonia lyase 2-kg, 2-ketoglutarate dtt, dithiothreitol opa, o-phthalaldehyde ppa, prephenate agn, arogenate buffer a, 50 mm k phosphate buffer ph 70 containing 1 mm dtt\n",
      "\n",
      "the organisms were grown to the late exponential phase of growth, harvested by centrifugation, washed twice with 50 mm k phosphate buffer ph 70 containing 1 mm dtt buffer a, and stored at -80°c until used\n",
      "\n",
      "all the enzymatic reactions proceeded continuously in 01 m tris-hci buffer ph 75 and allowed direct assay of asal in serum by monitoring the increase in the fluorescence intensity due to resorufin\n",
      "\n",
      "all the enzymatic reactions proceeded continuously in 01 m tris-hcl buffer ph 75 and allowed direct assay of asal in serum by monitoring the increase in the fluorescence intensity due to resorufin\n",
      "\n",
      "separation of the derivatized reactants was achieved on a spherisorb c18 ods reversed-phase column 250 × 45 mm id particle size 5 t~m, perkin-elmer, bea- consfield, england eluted with a linear gradient from 100% solvent a 30% methanol, 70% sodium acetate buffer, ph 59, 50 mm to 30% solvent a and 70% metha- nol over a period of 35 min\n",
      "\n",
      "in separate experiments, three protease inhibitors were added to all sample buffers and to the gradient phenylmethylsulfonyl fluoride to 40 ag/ ml final concentration, leupeptin to 1 lag/ml, and pepstatin to 1 jig/ml\n",
      "\n",
      "3 abbreviations used nas, neutral saturated ammonium sulfate as, saturated ammonium sulfate, not neutralized nas\", e coli enzyme fraction precipitated between 50 and 70% nas and then ultracentrifuged sg, serum glucarate ug, urinary glucarate tsa, tartronate semialdehyde kdg, ketodeoxyglucarate als, amyotrophic lateral sclerosis ldh, l-lactate dehydrogenase\n",
      "\n",
      "results properties of the e coli assay enzymes the three e coli glucarate catabolic enzymes fractionated in a manner similar to that previously reported 10- serum d-glucarate assay 289 table 1 cuvette contents for spectrophotometric assay cuvette 1 2 3 4 5 6 buffer + + mga+ + nadh + + ldh + - serum - + authentic d-glucarate - - e coli nasfio-'o + - purpose control for control for serum duplicate test samples glucarate instrumental nadh oxidase nadh oxidation recovery blank 12 and were reasonably stable without the addition of reduced glutathione 3,11\n",
      "\n",
      "although glucarate dehydrase is relatively stable to the lower ph of saturated as, rather than the more neutral nas, the kdg aldolase and probably the tsa reductase is much more sensitive to the lower ph values present during fractionation with as\n",
      "\n",
      "typically, to 450 al of ligase buffer buffer c of 13 consisting of 50 mm k+mes, ph 68, 50 mm kci, 25 mm mgs04,1 mm egta,1 mm dte, and 10% glycerol, and containing 11 mg tubulin, was added 50 al of the purified ttlase, equivalent to 005-01 eu as defined by flavin et al 12\n",
      "\n",
      "after the ca-atpase was reconstituted using method a at 9001 molar ratio of lipid to protein, so that essentially all the surrounding phospholipid was of the exogenous type, high-speed centrifugation in 15% w/v sucrose/hepes buffer caused the formation of membranes with a low lipid to protein ratio, close to that for a 50 0 0 2 4 b 8 10 time mins complete annulus of surrounding phospholipid 30-40 per atpase 25\n",
      "\n",
      "during reconstitution using method a the ca-atpase is left in buffers containing a cholate concentration of up to 13 mg/ml for 1 h\n",
      "\n",
      "also after incubation the protein/lipid/detergent mixture is rapidly diluted with excess buffer in the absence of detergent so that the final cholate concentration is below the cmc, at less than 1 mg/ml, returning the atpase activity to normal levels of that observed for the purified ca-atpase when reconstituting with the endogenous phospholipids initially extracted from sr membranes\n",
      "\n",
      "the solvents used m this study for paper chromatographie separation of the lower uronides were the epiphase of an ethyl acetate-acetic acid-water mixture 212 as described by demain and phaff 5 and ethyl acetate-pyridine-water-acetic acid 5531 as re- ported by gee and mccready 6\n",
      "\n",
      "the prolyl rna synthetase activity was recovered by successive additions to the column of 005, 007, 008, 009, 012, and 015 m potassium phosphate buffers, ph 75, containing the usual concentrations of s-mercaptoethanol and magnesium chloride\n",
      "\n",
      "for enzyme preparation the bacteria were grown anaerobically in 3-liter round-bottom flasks containing 1125 ml each of mineral solution i and ii 3, 10 gm cellobiose, 2 gm trypticase, 04 mg biotin, 04 mg p-aminobenzoic acid, 90 mg pyridoxine hc1, 78 mmole acetate buffer ph 60, 03 mmole isobutyrate, 03 mmole isovalerate, 6 gm nazcoa, 038 gm na2s, and 038 gm cysteine hc1, in a total volume of 1500 ml\n",
      "\n",
      "one milliliter of reaction mixture contained triethanolamine buffer ph 78, 40 imoles ethylenediamine tetraacetic acid, 1 /imole dpnh, 01 /xmole alcohol dehydrogenase, 003 mg and 2-deoxyribose 5-phosphate, 5 ^moles\n",
      "\n",
      "the reac- tion mixture of 1 ml contained triethanolamine buffer, ph 78, 40 jumoles ethylenediamine tetra- acetic acid 1 iumole fructose diphosphate, 5 iumoles fructose diphosphate aldolase, 005 mg and acetaldehyde, 10 iumoles\n",
      "\n",
      "the specific activity of the decarboxylase could be increased slightly by gel filtration on a biogel p-200 column, which had been equilibrated with 005 m phosphate buffer ph 68 containing 10 % glycerol\n",
      "\n",
      "propionyl-coa carboxylase 26, and oxalacetate transcarboxylase 23 are table ii stoichiometry or methylmalonyl-coa decabboxylase the main compartment of each warburg flask contained in a total of 16 ml, 20 /imoles, potassium phosphate buffer, ph 70, and 52 pinoles, methylmalonyl-coa\n",
      "\n",
      "each cuvette contained methyl- malonyl-coa, 005 ^moles potassium oxalacetate, 025 fancies, nadh, 05 fancies transcarboxylase, 10 unit 10 mg lactate dehydrogenase, 1 unit and potassium phosphate buffer, ph 67, 200 fimoles all in a total volume of 30 ml\n",
      "\n",
      "erwinia carotovora, a soft-rot bacterial phytopathogen, can demethoxylate pectin by action of a pectinesterase pe,2 and split the resulting polygalacturonic acid hydrolytically by means of an exocellular polygalacturonase pg to yield galacturonic acid and saturated oligogalacturonides 11\n",
      "\n",
      "2 abbreviations used pate, polygalacturonic acid rawg-eliminase ogte, oligogalacturonide rows-eliminase tba, thiobarbituric acid rosiua , movement on chromatograms relative to saturated galacturonic acid\n",
      "\n",
      "after the first puri- fication step in which the enzyme is absorbed on deae-cellulose and eluted with potassium phos- phate buffers, the conditions that have been found optimum for measurement of /3-lysine mutase activity are as follows  100 mm 2-methyl imidazole chloride buffer, ph 81  40 mm potassium pyruvate, 6 mm atp 10 mm mgcia 20 mm kc1 8 mm 1, 4-dimercaptothreltol  40 jum fad about 3 m^m dimethylbenzimidazolyl cobamide coenzyme and 20-30 mm ^-lysine\n",
      "\n",
      "in fact, when a 70% inactive serine dehydratase preparation a liver extract incubated with homocysteine as in fig 1 was dialyzed against 01 m potassium phosphate buffer, ph 75, for 25 hr, to remove the inhibitor, more than 50% reactivation was obtained by this treatment, which does not affect the activity of a control preparation\n",
      "\n",
      "for serine dehydra- tase inhibition solid circles, a liver extract diluted in 01 m potassium phosphate buffer, ph 75, was preincubated with 1 mm dl-homocysteine\n",
      "\n",
      "table ii comparison of the effect of dl- and d-homo- cysteine on complex formation with pybidoxal 5'-phosphate and on serine dehydratase activity» inhibitornui velocity of complex formation changes in absorbance 325 nm/min serine dehydratase inhibition% dl-homocysteine, 05 0014 47 dl-homocysteine, 1 0024 70 d-homocysteine, 05 0014 33 d-homocysteine, 1 0025 50 for complex formation, the indicated amounts of homocysteine were added to 01 mm plp in 01 m potassium phosphate buffer, ph 75, to start the reaction, and changes in ab- sorbance at 325 nm were recorded at room tem- perature\n",
      "\n",
      "the spectral properties of rnase 2-methyl-2,4-pentaned iol 517 table ii free energy of transfer of amino acids and peptides fbom water to 55% mpd° solute rtina'i\", w rtin•fiu aft ni',mpt> •yimpd glycine l-tyrosine l-phenylalanine l-alanine 102 006 011 091 -017 006 085 006 011098 l-arginine l-asparagine l-glutamic acid 050 030 082 -002 050 028 082 l-threonine 096 -002 095 l-methionine 051 003 054 l-histidine 066 005 070 l-glutamine 097 001 099 l-leucine 041 001 042 l-serine 088 — 088 l-aspartic acid l-valine 102 068 — 102 068 a^-acetyl-glycine amide 002 — 002 iv-acetyl-glycyl-glycine amide 014 \" 014 ° all values in kcal/mole\n",
      "\n",
      "aliquots of the proline 62 61 enz yme 27 jag in 001 r ill w rere incubated in the glycine 124 121 ass ay cuvette at 4° w th i t ie antiserum in the alanine 156 151 am ounts indicated and 00 6b til of 10 m potassium half-cystine 17 16 pho sphate buffer, ph '' 75 , a md saline to give a valine 102 98 tôt al volume of 0056 n nl in each case\n",
      "\n",
      "the enzyme ex- - 1 - phosphate cycloaldolase henceforth tracted from neurospora crassa, as pina and called glucocycloaldolase, the enzyme which tatum 8 have reported, shows an absolute catalyzes the transformation of d-glucose- requirement for nad+ after some purifica- 32 copyright © 1972 by academic press, inc \"nad^independent\" glucocycloaldolase 33 tion steps\n",
      "\n",
      "the next step was to study the \"nad+- independent\" glucocycloaldolase activity from the 01 m tris-hcl, ph 77 extracts after enzyme precipitation with nh42s04 and dialysis against buffers containing either nh4zs04 or glycerol\n",
      "\n",
      "however, when this substance was treated with phosphomono- esterase ec 3131 from calf intestinal mucosa, or with 01 n hc1 at 100° for 15 min, it yielded two radioactive compounds with the paper chromatographic solvent 3 mobilities of d-xylose and l-arabinose respectively\n",
      "\n",
      "since udpglcua carboxy-lyase from cryptococcus laurentii has an absolute requirement for nad and is inhibited by nadh 13, the effect of these compounds on the decarboxylation of udpglcua catalyzed by extracts of a digitaia was investigated\n",
      "\n",
      "an absolute univalent cation requirement was not demonstrable for partially purified table vi summary op kinetic constants determined fob reaction 2 with desalted, partially purified tbyptophan synthase kinetic constant calculated value km for indole 50 x 10-2 mm i?,„ for l-serine 57 mm k ab 33 mm2 ki 66 x 101 mm kï 058 mm pea tryptophan synthase\n",
      "\n",
      "addition of 25 mu k4' as kc1 to partially purified enzyme extracts, initially prepared in potassium phosphate buffer and subsequently desalted on biogel p-6 using tricine buffer, ph 76, caused a maximum increase of 25 % for reac- tion 1 and a 15% stimulation of reaction 2\n",
      "\n",
      "when purified jv-an aldolase 10-20 /xg of protein was subjected to electrophoresis under the same conditions as in fig 1 at ph 32, 42 citrate phosphate buffer  52, 60,67 and 86 tris- maleate buffer, two separate and distinct protein bands were seen at each ph value tested, which indicated that the multiple bands were not an artifact produced by buffer binding to protein 16\n",
      "\n",
      "exposure of the en- zyme in 01 m phosphate buffer, ph 72, to temperatures of 70 and yabbi 72905176 0003-98617290517-6 101016/0003-98617290517-6 threonine dehydratase from goat liver mahendra nath gg\n",
      "\n",
      "the incubations were for 10 min at 25°c and con- tained in a 010-ml volume 1 mmole /v-trishy- droxymethyl methyl-2 - aminoethanesulf onate-k+ tes buffer 02 mmole 2-mercaptoethanol 2 nmoles pga ratio of 3-pga to 2-pga = 6, 10 mmoles kc1 and 123 x 10~5 unit of diphospho- glycerate mutase\n",
      "\n",
      "it can be seen from table iii that the effect of hemin and b ph fig 3 ph activity curve of a-aminolevulinic acid dehydratase of r capsulata in different buffers 3a tris chloride o—o tris maleate •—• imidazole a—a\n",
      "\n",
      "discussion the 5-aminolevulinic acid dehydratase of r capsulata appears to have properties similar both to the enzyme of r sphéroïdes and to that of eucaryotic cells in that as other photosynthetic bacterium it is not inhibited by edta, whereas the enzyme 310 nandi and shemin table iv inactivation of s-ala dehydratase of r capsulata on reduction with nabh4 in the presence of s-aminolevulinic acid and other keto acids\" additions amount inactiva- tion /timoles % none 0-6 5-aminolevulinic acid 7- 25 93 keto-s-aminovaleric acid levulinic acid 7-keto- 25 99 valeric acid ethyl levulinate 7-keto- 25 77 valerate ethyl ester a-ketoglutarate 50 18 a-ketovalerate 50 5 \",the reaction mixture 05 ml, final ph 74- 76 contained 03 m potassium phosphate buffer, ph 78, 015 ml of enzyme 0-50% ammonium sulfate fraction 20 mg of protein/ml and keto acids as indicated in the table\n",
      "\n",
      "b carboxypeptidase a-treated 3100 098 deoxyribose-5-p aldolase + acetaldehyde + nabh< ° crystalline deoxyribose-5-p aldolase 30 mg was dissolved in 20 ml of 02 m n-ethylmorphiline acetate buffer, ph 85\n",
      "\n",
      "6 the abbreviations used in this paper are as follows ab complex, glycerol dehydratase apo- enzyme abbizoh, inactive complex of glycerol dehydratase apoenzyme and hydroxocobalamin cobia, coenzyme bia, ie, 5'-deoxyadenosyl- cobalamin glycine buffer, 01 m glycine-koh buffer, ph 86 00052 m k ion phosphate buffer, 01 m potassium phosphate buffer, ph 86 slightly over 02 m with respect to k ion ethanolamine buffer, 01 m ethanolamine-hcl buffer, ph 86 tris buffer, 01 m tris hydroxymethyl amino- methane-hcl buffer, ph 86 tmac1, tetra- methylammonium chloride teac1, tetraethyl- ammonium chloride mbth, s-methyl-2-benzo- thiazolinone hydrazone\n",
      "\n",
      "'ok glyc- erol dehydratase activity in previous studies, in which phosphate buffer, ph si, was used 1-5 it has been shown that the presence of potassium ions is required together with apoenzyme, sub- strate and cobn for glycerol dehydratase activity\n",
      "\n",
      "the vi/'v,, ratio determined in the case of ethanolamine buffer increased from 143 to 175 when potassium ions ot m kc1 were present in the eluting solution\n",
      "\n",
      "since changes in optical rotation are not nec- essarily due to the action of mutarotase on i8-g-6-p, but may be due to other possible reac- tions involving i8-g-6-p, the following investiga- tions were carried out fifty microliters of a solution of g-6-p about 2 mg/ml were added to a cuvette containing 25 ml of a tris buffer 001 m, ph 80, 1 ,a g-6-p- dehydrogenase 5 mg/ml, and 04 mg of nadp\n",
      "\n",
      "table i purification of phosphoenolpyruvate carboxylase stage of purification totalactivity units protein mg specific activity units/mg rupî car-boxylase0% initial extract 465 15,624 003 — 35% saturated nhjzs04 supernatant frac- 385 10,874 004 — initial extract 35% saturated nh,,sc>4 supernatant frac- tion 35-50% saturated i^^so, fraction 407 6,940 006 deae-cellulose chromatographie fraction 181 2,534 007 hydroxylapatite chromatographie fraction 134 128 105 sephadex g-200 fraction 101 339 30 cellulose phosphate chromatographie fraction 59 138 43 deae-cellulose chromatographie fraction 204 316 65 antiserum column 149 075 199 purification 663-fold 65 ±6 77 ±7 60 ±6 60 ±6 undetect- able \"the proportion of the total protein which was ribulose diphosphate carboxylase was estimated by spectrophotometric scanning at 540 nm of sds gels stained with coomassie blue\n",
      "\n",
      "the stan- dard assay mixture contained 25 ^mol of kahpo, or 50 fimo\\ of tris buffer, ph 78, 50 nmol of na^edta, 40 ^mol of 3-mercapto-l,2-propandiol, 50 nmol of nad+, 522 nmol of udpgluua, 01 nmol of high specific activity udpfu-^clgluua 00247 nci, and udpgluua cyclase in a volume of 005 ml\n",
      "\n",
      "in checking buffer sys- tems for potential use in purification proce- dures we have found that imidazole and tris buffers are potent inhibitors while the use of multivalent anions as advocated by earlier investigators 2 slightly stimulates the formiminotransferase activity, as illus- trated in table ii\n",
      "\n",
      "the reaction was initiated by the addition of 25 ii\\ of pga isomerase purified 50-fold to a pgai solution and albumin in buffer with 5% ethanol in a 1-ml cuvette\n",
      "\n",
      "extracts for the determination of glutamate dehydrogenase 9, glucose-6-phosphate dehydrogenase 10, and chalcone-flavanone isomer- ase 11 were prepared by homogenizing 2 g fresh weight of cells and 4 ml of 01 m tris-hcl buffer, ph 76, in an ice bath with an ultra-turrax homogenizer janke and kunkel, stauten, germany three times for 30 sec at intervals of 30 sec\n",
      "\n",
      "serial jl 272380 291210 291847 291849 291857 31 archives of biochemistry and biophysics archivesbiochemistrybiophysics 2004-11-19 2004-11-19 2016-06-14t102536 1-s20-0003986175905226 0003-98617590522-6 0003986175905226 101016/0003-98617590522-6 s350 s3503 head-and-tail 1-s20-s0003986100x05292 2016-06-14t164019206265-0400 0 0 19750401 19750430 1975 2004-11-19t000000z rawtext articleinfo articletitlenorm authfirstinitialnorm authfirstsurnamenorm cid cids contenttype copyright crossmark dateloaded dateloadedtxt datesearch datesort dateupdated dco docsubtype doctype doi eid ewtransactionid hubeid indexeddate issfirst issn issnnorm issuelist itemstage itemtransactionid itemweight openaccess openarchive pg pgfirst pglast pii piinorm pubdateend pubdatestart pubdatetxt pubyr sortorder srctitle srctitlenorm srctype ssids alllist content subj tomb volfirst volissue volumelist webpdf webpdfpagecount yearnav affil articletitle auth authfirstini authfull authlast doctopic primabst ref 0003-9861 00039861 false 167 167 2 2 volume 167, issue 2 43 761 768 761 768 197504 april 1975 1975-04-01 1975-04-30 1975 converted-article fla copyright © 1975 published by elsevier inc phosphatidylserinedecarboxylaseanalysisactiontowardsunsaturatedsaturatedphosphatidylserineeffecttritonx100activity warner t borkenhagen 1961 pc28, pc30 l kanfer 1964 1720 1726 j dennis 1970 394 403 e vandenbosch 1972 13 145 h dowhan 1974 3079 3084 w patterson 1971 1062 1072 p warner 1973 595 598 t dennis 1973 152 159 e dennis 1973 165 176 e dennis 1973 485 493 e folch 1942 35 44 j chen 1956 1756 1758 p patterson 1965 854 857 m bangham 1974 1 68 a dennis 1974 764 773 e scandella 1971 4447 4456 c helenius 1972 3656 3661 a makino 1973 4926 4932 s helenius 1973 287 300 a trauble 1973 491 512 h papahadjopoulos 1973 23 42 d ribeiro 1974 26 35 a warnerx1975x761 warnerx1975x761x768 warnerx1975x761xt warnerx1975x761x768xt item 0003-98617590522-6 0003986175905226 1-s20-0003986175905226 101016/0003-98617590522-6 272380 2016-06-14t164019206265-0400 1975-04-01 1975-04-30 1-s20-0003986175905226-mainpdf https//s3-eu-west-1amazonawscom/prod-ucs-content-store-eu-west/content/pii0003986175905226/main/application/pdf/a79f0c348de4676d751b57d1edd0a024/mainpdf https//s3amazonawscom/prod-ucs-content-store-us-east/content/pii0003986175905226/main/application/pdf/a79f0c348de4676d751b57d1edd0a024/mainpdf mainpdf pdf true 662582 main 8 1-s20-0003986175905226-main_1png https//s3-eu-west-1amazonawscom/prod-ucs-content-store-eu-west/content/pii0003986175905226/preview/image/png/f8bd205a21003f8b81c83ae33ba78f75/main_1png https//s3amazonawscom/prod-ucs-content-store-us-east/content/pii0003986175905226/preview/image/png/f8bd205a21003f8b81c83ae33ba78f75/main_1png main_1png png 71684 849 656 image-web-pdf 1 1-s20-0003986175905226-si1gif https//s3-eu-west-1amazonawscom/prod-ucs-content-store-eu-west/content/pii0003986175905226/stripin/image/gif/5484cb758575db1a2a3f115ea2f851e4/si1gif https//s3amazonawscom/prod-ucs-content-store-us-east/content/pii0003986175905226/stripin/image/gif/5484cb758575db1a2a3f115ea2f851e4/si1gif si1 si1gif gif 280 18 39 altimg archives of biochemistry and biophysics 167, 761-768 1975 phosphatidylserine decarboxylase analysis of its action towards u nsaturated and saturated phosphatidylserine and the effect of triton x-100 on activity thomas g warner and edward a dennis1 department of chemistry, university of california at san diego, la jolla, california 92037 received september 26, 1974 the membrane-bound enzyme phosphatidylserine decarboxylase tetrahymena pyriformis was found to have activity both in a crude, particulate form and when it is in a soluble form in the presence of the nonionic surfactant triton x-100\n",
      "\n",
      "the activity of the enzyme towards natural phosphatidylserine is found to be greater than towards saturated phosphatidylserine, both with and without triton present this finding is considered in terms of the effect of the thermotropic phase transition of the saturated material on the physical state of the phospholipid, rather than simply in terms of the specificity of the enzyme for phosphatidylserine containing unsaturated fatty acid groups\n",
      "\n",
      "dependence of phosphatidylserine decar- boxylase activity on the molar ratio of triton x-100/ phospholipid for • 3 mm phosphatidylserine, 0 6 mm phosphatidylserine, and d 6 mm saturated phosphatidylserine all at 35°c, and for • 6 mm saturated phosphatidylserine at 22 °c\n",
      "\n",
      "we have shown that mixed micelle formation be- tween dipalmitoyl phosphatidylcholine and triton x-100 is less favorable below the thermotropic phase transition of the phospholipid 16, 23, and the data pre- sented here indicate that at a molar ratio of about 41 triton/saturated phospholipid, two phases exist a phospholipid bilayer phase and a triton-phospholipid mixed micellar phase\n",
      "\n",
      "fur yabbi 75905226 0003-98617590522-6 101016/0003-98617590522-6 lipids phosphatidylserine decarboxylase analysis of its action towards unsaturated and saturated phosphatidylserine and the effect of triton x-100 on activity thomas g warner edward a dennis 1 department of chemistry, university of california at san diego, la jolla, california 92037 usa 1 to whom correspondence should be addressed\n",
      "\n",
      "the activity of the enzyme towards natural phosphatidylserine is found to be greater than towards saturated phosphatidylserine, both with and without triton present this finding is considered in terms of the effect of the thermotropic phase transition of the saturated material on the physical state of the phospholipid, rather than simply in terms of the specificity of the enzyme for phosphatidylserine containing unsaturated fatty acid groups\n",
      "\n",
      "materials and methods male albino rats 80 to 90 g were obtained from the holtzman company, madison, wise , and main- tained until use on a dry diet containing, as a per- centage w/w vitamin-free casein, 9, corn oil, 5, mineral mixture 10, 5, vitamin mixture 10, 0 5, choline chloride, 0 2, l-methionine, 0 3, autoclaved cornstarch, 40, glucose monohydrate, 40 animals were housed in individual cages at 23°c with a 12-h light-12-h dark cycle l-histidme and l-aspartic acid were obtained from mann research laboratories, inc , new york, n y , urocamc acid, sodium pyruvate type ii, adenosine 5'-tnphosphate disodium salt, a-keto- glutanc acid, pyndoxal 5-phosphate, reduced gluta- thione, nadph type ii, dl-isocitrate tnsodium salt, and isocitrate dehydrogenase type iv were obtained from sigma chemical co , st louis, mo , carnosine and /3-alanine were obtained from cal- biochem, san diego, calif, ammopynne was ob- tained from aldnch chemical co , milwaukee, wise , ^-phenylethylhydrazine was obtained from icn-k and k laboratories, inc , cleveland, ohio, 3- pyrazolidone hydrochlonde and d-ff-hydrazmohisti- dine hereafter referred to as d-a-hydrazinoimidazo- lylpropionate, d-hipa were obtained from regis chemical co , morton grove, 111 dl-a-hydrazi- noimidazolylpropionic acid dl-hipa was prepared by a modification 8 of a published procedure 11 dl-a-hydrazinoisocaproic acid was synthesized by the procedure of niednch and gruppe 12 j8-hy- drazinopropionic acid was prepared by the procedure of van gelder 13 [carboxy-^c}ft-a\\an\\w 9 7 mci/ mmol was obtained from new england nuclear, boston, mass all other chemicals used were of re agent grade purity enzyme assays rats were killed by decapitation and the livers were immediately excised and placed in chilled 0 9% naci except where otherwise noted, livers were blotted free of blood, trimmed, and a 3-g portion was minced and homogenized in 12 ml of chilled 0 25 m sucrose with a potter-elvehjem tis- sue-grinder tube and teflon pestle homogenates were centnfuged at 40,ooqg for 50 mm resulting supernates were stored frozen below -20°c and en- zyme assays were performed within 1 week, during which time enzyme activities did not diminish sig- nificantly histidine ammonia-lyase and urocanase were as- sayed at 24°c by modifications 14 of the procedures of tabor and mehler 15 for determinations of epidermal histidine ammo- nia-lyase, 10% w/v homogenates of whole shaved rat skin were prepared in chilled 0 1 m kc1 with a ground glass tissue-grinding apparatus and centn- fuged at 300g for 10 mm, and the resulting super- nates were assayed for enzyme activity at 37°c es- sentially by the method ofladu 16, reaction mix- tures contained 0 1 ml of supernate in 1 0 ml of a solution containing 20 mm sodium pyrophosphate and 50 mm l-histidine, ph 9 0 the incubation pe- riod 60 mm was terminated by the addition of 1 0 ml of 6% perchloric acid after centnfugation of the mixture 60qg, 10 mm the absorbance of the result- ing supernates was determined at 277 nm against a blank assay in which the order of the enzyme and perchloric acid additions was reversed an extinc- tion coefficient of 18 8 miir'cm ' 13 was used for determination of urocamc acid content l-histidme decarboxylase activity was deter- mined in the pylonc region of the stomach as de- scribed by lemweber and braun 17, except that the assay pyndoxal 5-phosphate concentration was reduced from 10 to 1 um l-histidine /3-alanine ligase carnosine synthe- tase activity was estimated in gastrocnemius mus- cle essentially as described by kalyankar and meis- ter 18 using [car6o^y-l'lc]^8-alanlne and unlabeled l-histidine as substrates carnosine formed was sep- arated from ^-alanme by ascending paper chroma- tography for 15 h in tert-butyl alcohol formic acid water 70 15 15, by volume after spotting of reaction mixture aliquots including 0 5 pmol of unlabeled carnosine carrier on sheets of whatman 3mm paper, r{ values for /3-alanine and carnosine were 0 50 and 0 22, respectively ammo acid-con- taining spots located on the chromatogram by spray- ing with nmhydnn were cut out and assayed for radioactivity by the method of bosquet and chris- tian 19 histidine pyruvate ammotransferase was as- sayed at 24°c as described by schirmer and harper 14 glutamate oxalacetate ammotransferase was as- sayed by the spectrophotometnc method of gunsa- lus and stamer 20 units of activity for the above enzymes are expressed as nanomoles of substrate transformed per minute for the assay of aminopynne demethylase, liver samples were homogenized in 0 25 m sucrose con- taining 50 mm tns-hcl, ph 7 4, and centnfuged at 9000^ for 20 mm, and the resulting supernates were used for assay of enzyme activity according to the procedure of dewaide and henderson 21, substi- tuting nadph for nadp4^ in the incubation mix- inactivation of histidine ammonia-lyase 125 ture aminopynne demethylase units represent na- nomoles of formaldehyde produced per 10 mm protein assay whole homogenate protein was as- sayed by the method of lowry et al 22 preparation of liver microsomes livers were ho- mogenized in 4 vol of ice-cold 0 25 m sucrose contain- ing 50 mm tns-hcl, ph 7 4, and centnfuged for 20 mm at 9000^, and the resulting supernates were centnfuged for 50 mm at 40,000^, the pellet contain- ing the microsomal fraction was resuspended in one- tenth the original homogenate volume of the above buffered sucrose partial purification of rat liver histidine ammo- ma-lyase the enzyme preparation used for studies in vitro was purified through the first ammonium sulfate precipitation step of brand and harper 8 statistical methods the significance of the differ- ence between two arithmetic means was evaluated using student's t test results and discussion effect of hydrazines on rat liver en- zymes in vwo in order to assess the specificity of the inactivation of histidine ammonia-lyase following injection of dl-hipa, rats fasted overnight were administered intraperito- neally various hydrazines 12 mmol/kg, including dl-hipa, or an equal volume of 09% w/v saline, and were killed 6 h later for the assay of hepatic histidine ammo- nia-lyase activity\n",
      "\n",
      "methods the enzymatic synthesis ofglycerate-ls-h-'^plpz has been reported 8 the acyi phosphate and the contaminating glycerate-3-p were assayed succes- sively spectrophotometncally with glyceraldehyde- p dehydrogenase and phosphoglycerate kmase gly- cerate^s-cu-^pjpa was prepared from red blood cells 9 and assayed according to rose and liebow- itz 10 specific activities of both compounds were usually 107 to 108 cpm//xatom p phosphorylation of the enzyme was followed with the phenol extraction procedure 11, 12 enzyme assays during the purification, the bisphosphoglycerate phosphatase activity in the presence ofglycolate-2-p was followed assays were at 37°c for 10 mm and contained in 02 ml 10 mm n-tnshydroxy- methydmethyl-2-aminoethane sulfonate tes2-na buffer, ph 7 5, 5 mm /3-mercaptoethanol, 0 1 mm glycolate-2-p, 2 ^m glycerate^s-lu-^plpz, and suf- ficient enzyme to allow about 15% reaction these conditions do not give maximal rates bisphosphoglycerate synthase activity was as sayed spectrophotometncally at each step 6, 13 phosphoglycerate mutase activity was assayed cou- pled to enolase 14 in order to study the kinetics of the bisphos- phoglycerate synthase reaction, the conversion of glycerate-l^-tl-^plpz to glycerate^^-p'pip was measured directly 6 glycerate-3-p present in the glycerate-ls-n-32?]?\n",
      "\n",
      " as a result of hydrolysis of the acyi phosphate was measured and considered in the calculations of total glycerate-3-p present purification of bisphosphoglycerate synthase although it was shown by rosa et al 4 that the electrophoretic mobility at ph 7 5 of horse bisphos- phoglycerate synthase differed from that of the en zyme from human red cells, the two enzymes be- haved similarly in all purification steps all steps were carried out at 0-4°c the results are summa nzed in table i 1 hemolysate fresh horse blood 2 liters con- taining acid-citrate dextrose acd was centnfuged at 4000^ for 10 mm the serum and buffy layer were removed with suction the cells were washed three times with 3 vol of isotomc saline 0 9% containing 5 mm edta the washed cells were hemolyzed by freezing in a dry ice-acetone bath they were al- lowed to thaw overnight at 4°c three volumes of buffer a potassium phosphate, ph 7 2, 5 mm, edta, 5 mm, dithiothreitol, 2 mg/100 ml were added for a fourfold dilution of the cells the hemo lysate was centnfuged at 27,000g for 15 mm and the supernatant was poured off carefully from the stroma 2 deae-cellulose chromatography the hemoly- sate was applied equally to two columns 4 5 x 40 cm each of eastman fibrous deae-cellulose equili- brated with buffer a hemoglobin was not adsorbed the column was washed free of hemoglobin with buffer a the enzyme was eluted with 1 liter of potassium phosphate buffer, 0 1 m, ph 8 0 contain- ing 5 mm edta and dithiothreitol 2 mg/100 ml fractions with highest activity were combined 3 ammonium sulfate fractionation ammonium sulfate 0 195 g/ml was added to the enzyme from the deae-cellulose columns the mixture was al- lowed to stand 2 h before being centnfuged at 286 rose and dube table i purification of bisphosphoglycerate synthase from horse red cells fraction volumeml phosphatase activity ° units/ml protein\"mg/ml totalunits phosphatase specific activity at 37°c units/mg protein hemolysate deae-cellulose ammonium sulfate blue-dextran column 1800 270 12 85 0 185 75 1 07 2 16 2 13 1 38 0 estimated 8 9 116 333 289 195w 0 00185 0382 1 16 119 ° the maximal velocity was calculated from the equation, u = vab/k ,a + kyb + ab, wherein is 3 /zm and at , is 6 ^m,a and b are the concentrations of glycerate-2,3-p2 and glycolate-2-p, respectively 6 calculated from aasoh ss '' some activity was lost during desalting with sephadex g-25, there was 75% recovery of activity from the blue-dextran column 27,000^ for 20 mm the precipitate was discarded to the supernatant was added ammonium sulfate 0 179 g/ml after centnfugation, the precipitate was dissolved in 15 ml of buffer b 5 mm tnethanola- mme-cl\" buffer, ph 7 5 containing 5 mm edta the bulk of the phosphoglycerate mutase may be obtained by precipitation of the supernatant with additional ammonium sulfate 4 chromatography on blue-dextran column the column was prepared from sepharose-4b bio-rad denvatized with blue dextran 15 the column 2 7 x 50 cm was washed with 400 ml of buffer b and then with the same buffer containing dithiothreitol, 2 mg/100 ml, in place of edta, until there was no absorption at 280 nm the enzyme solution was equilibrated with the last-mentioned buffer by pas- sage through a g-25 column and applied to the blue- dextran column which was then washed with 400 ml of the buffer containing dithiothreitol until the a run returned to near the blank value the enzyme was eluted with a linear gradient formed from 200 ml each of starting buffer and the same buffer contain- ing 2 mm glycerate-2,3-p2 fractions 4 ml were collected flow rate, 0 5 ml/nun and assayed for bisphosphoglycerate synthase, phosphoglycerate mutase and a ago phosphoglycerate mutase activity increased after the bisphosphoglycerate synthase peak and these fractions were discarded the optical density followed the peak of synthase activity properties of purified enzyme the enzyme was stable for at least 3 months when precipitated with ammonium sulfate and stored as a paste at -80°c electrophoresis in polyacrylamide gels containing sodium dodecyi sulfate 16 of enzyme from horse or human red cells indicated one band of protein with mobility similar to rabbit muscle phosphoglycerate mutase, i e , about 30,000 daltons fig 1 at 280 nm, the absorption of a solution of enzyme contain- ing 1 mg/ml, using a 1-cm light path, was 1 65 this was calculated by relating the a ami to maximal phos- phoenzyme formation, assuming one phosphoryla- tion site per subumt the value agrees with the absorption for human synthase reported by kappel and hass 3 results kinetics of the bisphosphoglycerate synthase reaction as with synthase from human red cells 6, horse bisphosphoglycerate synthase requires glycerate-3-p for maximal activ- ity and the amount of activity observed in the absence of added glycerate-3-p is con- sistent with the amount of glycerate-3-p added with the other substrate, glycerate- 1,3-pz\n",
      "\n",
      "the purified malonyl-coa de- carboxylase was relatively stable as it could be stored in 100 mm phosphate buffer, ph 76, containing 05 mm dithio- erythritol at 4°c for 2 weeks without ap- preciable loss in activity\n",
      "\n",
      "adenylate cyclase was assayed using 3 mm atp, 15 mm mg2^ and tris buffer of varying ph plasma membrane 0 and outer acrosomal mem- branes •\n",
      "\n",
      "sperm adenylate cyclase also has a relatively high k^ for atp 27, which makes it impractical to assay this enzyme at saturating concentrations of substrate\n",
      "\n",
      "issn 0003-9861 nicotinate phosphoribosyltransferase 335 buffer containing 50 mm khapgi, 50 mm tris base, and 196 mm na+ added as naoh to adjust the ph to 80\n",
      "\n",
      "the assay mixture final vol- ume, 1 ml contained tris-hcl buffer, ph 78, 50 mm kc1, 25 mm khcoa, 10 mm mgclz, 5 mm mgatp, 2 mm na-pyruvate, 15 mm citrate syn- thase, 15 units acetyl-coa, 036 mm dtnb 100 ijlm\n",
      "\n",
      "the assay mixture final volume, 1 ml contained tris- hcl buffer, ph 78, 50 mm kc1, 25 mm khco,, 10 mm mgcia, 5 mm mgatp, 2 mm na-pyruvate, 15 mm malate dehydrogenase, 5 units nadh, 012 mm acetyl-coa, 018 mm\n",
      "\n",
      "solutions of citrate synthase and malate dehydrogenase contained bovine serum al- bumin 25 mg/ml and were dialyzed against 10 mm tris-hcl buffer, ph 78, before use to remove n1142804, which inhibits pyruvate carboxylase\n",
      "\n",
      "requirements of the a globiformis pyru- vate carboxylase reaction the highly purified enzyme sp act 270 was absolutely dependent upon pyruvate, mgatp, and mg^ for catalytic activity and showed 4 and 57% of its maximum activity in *he absence of acetyl coa and added hcos\", respectively\n",
      "\n",
      "a globiformis pyruvate carboxylase 080 mg/ml, sp act 195 was inactivated at 15°c m 10 mm tris-hcl buffer, 50 mm kc1, 1 mm edta, 01 mm dtt, ph 80, and centrifuged in this buffer at 15°c for 60 h at 4470 rpm\n",
      "\n",
      "the \"uncoupling\" of the hydrolysis of glutamine from the amination of xmp is the basis for the reported inhibitory effects of psicofuranine and ppi on the amidotransferase activity, iv tris buffer selectively inhibits the xmp-amidotransferase activity by inhibiting the glutaminase activity\n",
      "\n",
      "the xmp-aminase and amido- transferase activities were assayed by the continu- ous assay procedure as previously described 4, the glutaminase activity was initiated by adding the purified enzyme 02 unit of amidotransferase activ- ity to a 1-ml reaction mixture that contained 70 mm hepes4 buffer, ph 82, and the following concentra- tions of substrates xmp, 03 mm mgclz, 10 mm glutamine, 20 mm\n",
      "\n",
      "xmp-amidotrans- ferase was passed through a sephadex g-25 column equilibrated with 20 mm potassium phosphate buffer, ph 74, containing 01 mm edta to remove the dithiothreitol\n",
      "\n",
      "xmp-amidotransferase was passed through sephadex g-25 equilibrated with a 20 mm potassium phosphate buffer, ph 74, containing 01 mm edta to remove the dithiothreitol\n",
      "\n",
      "the enzyme 0067 unit of amidotransferase activity was incu- bated for 5 min at 30°c with the indicated concentra- tion of p-chloromercuribenzoate or bromopyruvate in a 09-ml reaction mixture that contained 90 ^mol of tris buffer, ph 83, 03 ^mol of xmp, 10 jumol of mgc12\n",
      "\n",
      "xmp-amidotrans- ferase 01 unit was added to a reaction mixture containing 01 m hepes buffer, ph 83, 01 mm edta, 01 mm dithiothreitol, 03 mm xmp, 1 mm atp, 4 mm mgcia, and 58 pm don and incubated at 30°c for the times indicated\n",
      "\n",
      "when xmp-amido- transferase is incubated for 30 min in tris buffer, xmp, atp, and mg2-*- there is a 75% loss of the amidotransferase activity fig 6 and a 60% loss of glutaminase activity\n",
      "\n",
      "since a significant amount of xmp-amidotrans- ferase activity is lost after a 30-min incu- bation in tris buffer, xmp, atp, and mg2'1', it seemed plausible that the pre- vious reports of a low glutamine-supported activity may result from this selective in- activation\n",
      "\n",
      "this hypothesis was tested as follows xmp-amidotransferase from e coli b 24-1 and e aerogenes pd1, mk 9099, and mk 9110 was partially purified by ammonium sulfate fractionation and assayed by two different methods a a non-continuous assay which involves a 30- min incubation in tris buffer b a con- tinuous assay where the specific activity is determined during the first 2 min of the reaction in tris buffer\n",
      "\n",
      "rate of inactivation of xmp-amidotrans- ferase in tris or hepes buffer\n",
      "\n",
      "however, phate synthase catalyzes the first step of these solvents were also shown to affect de novo pyrimidine biosynthesis and plays the enzymatic activity dimethyl sulfoxide a key role in the control of this pathway in or glycerol stimulated the activity at lower higher animals 1-9\n",
      "\n",
      "the column is washed with 25 mm potassium phosphate buffer, ph 70, until the effluent is free of nucleotide and the samp synthetase is eluted with a linear gradient ofo to 04 m nacl in 05 liter fig 1a\n",
      "\n",
      "molecular weight determination the molecular weight of samp synthe- tase has been determined by gel filtration on sephadex g-200 1 x 61-cm column in 01 m phosphate buffer, ph 70, containing 015 m kc1, 1 mm dte with a flow rate of a vinelandii adenylosuccinate synthetase 27 fig 1\n",
      "\n",
      "the yield and purity of [l4c]dihydroxyacetone phosphate were determined by enzymatic assay with a-glycerophosphate dehydrogenase and by chromatography on eastman no 13255 thin-layer cellulose sheets according to ban-durski and axelrod 9, using an acid solvent system composed of methanolformic acidwater 80155 and a basic system of methanolammonium hydrox-idewater 601030\n",
      "\n",
      "the assay mixture contained, in a volume of 10 ml 45 mm imidazole buffer ph 70, 10 mm dihydroxyacetone phosphate, 167 mm reduced glutathione, glyoxalase i 10 unit, and methylglyoxal synthase\n",
      "\n",
      "methylglyoxal synthase 2 mg/ml was incubated in 10 mm imidazole buffer, ph 76, containing 10 mm 2-mercaptoethanol for l h in the presence o and absence • of 1 mm potassium phosphate\n",
      "\n",
      "the precipitate containing most of the isocitrate lyase activity was collected as described earlier and resuspended in 50 ml of te m buffer ph 75, 4°c containing 005 m tris-ci, 1 him edta, and 1 mm 2-mercaptoethanol\n",
      "\n",
      "electrophoresis of the antigen isocitrate lyase ii placed in the well was carried out with sodium barbital buffer, ph 86, for 4 h at 3 ma/gel\n",
      "\n",
      "with p-hmb 61 596 815 142' 813 \" titrations of thiol content of isocitrate lyase ii with p-hmb or dtnb were carried out in 005 m potassium phosphate buffer, ph 70 and 75, respectively, at 10°c\n",
      "\n",
      "isocit-rate lyase of t aceti exhibits slightly higher activity in tris than in phosphate, but ph optima are identical in both buffers 26\n",
      "\n",
      "these findings are in contrast with those of john and syrett 29, who observed a lower ph optimum for the c pyrenoidosa isocitrate lyase in phosphate than in triethanolamine buffer\n",
      "\n",
      "in buffer a, a highly concentrated solution obtained after the ammonium sulfate pre- cipitation produced enzyme with an ap- proximate molecular weight of 210,000 de- termined by calibrated sephadex gel filtra- tion columns, as well as a lower molecular weight form of prephenate dehydratase3 the presence of phenylalanine 01 mm during sephadex g-100 chromatography fig 1b caused the enzyme to be eluted only as the low molecular weight species in high yield over 80%\n",
      "\n",
      "thus, the enzyme showed an absolute requirement for mg^ or mn2\"1', as is the case with other prenyl pyrophosphate cyclases 8, 19-21\n",
      "\n",
      "glucose 6-phosphate dehydrogenase, phosphoglucose isomerase, d-fructose 6-phosphate, and all buffers were obtained from sigma chemical co\n",
      "\n",
      "the reac- tion mixture contained 20 pmo\\ of tris-hcl buffer, ph 80, 02 fimol of dte, 2 jumol of l-malic acid, 01 junol of nad, 0025 ]nmol of nadh, 5 units of malic dehydrogenase, 085 unit of citrate synthase, 004 jumol of malonyl-coa, and malonyl-coa decarboxylase preparation in a total volume of 02 ml\n",
      "\n",
      "subcellular localization of malonyl-coa decarbox- ylase liver, excised from freshly killed young male rats ~270 g was chopped with a razor blade and gently homogenized in a chilled mortar with 01 m na-phos- phate buffer, ph 76, containing 025 m sucrose ~1 ml of buffer per 1 g of liver\n",
      "\n",
      "the fractions containing malonyl-coa decarboxylase were pooled and concen- trated by ultrafiltration and during this procedure the buffer was changed to 01 m na-phosphate, ph 76, containing 05 mm dte\n",
      "\n",
      "the gel slices were placed m test tubes with 100 id of 01 m na-phosphate buffer containing 05 mm dte, kept overnight at 4°c, and aliquots 50 jiil of the buffer were examined for malonyl-coa decarboxylase activ- ity spectrophotometrically\n",
      "\n",
      "a sephadex g-150 column 11 x 90 cm equilibrated with 01 m na- phosphate buffer, ph 70, containing 05 mm dte was calibrated with blue dextran 2000 and standard pro- teins, yeast alcohol dehy drogenase 150,000, bovine serum albumin 68,000, /?-lactoglobulin 35,000, and cytochrome c 11,700\n",
      "\n",
      "malonyl-coa decarboxylase was extracted from the isolated mitochondria by 10 mm phosphate buffer, ph 76, containing 05 mm dte as described above for the liver preparation\n",
      "\n",
      "treatment of the mitochondria with a low ionic strength buffer 10 mm phosphate for 2 h at 0°c released about 60% of the malonyl- coa decarboxylase into the soluble phase\n",
      "\n",
      "using cystine stereospecifically labeled at c-3 with tritium or with tritium and deuterium, it was found that the a,/8-elimination reaction catalyzed by s-alkylcysteine lyase involves stereo- specific replacement of the ^-substituent of the substrate by a hydrogen derived from the solvent, d2o or h2o, with retention of configuration to give pyruvate containing a chiral methyl group\n",
      "\n",
      "the reaction mixture contained 0005 m l-cystine, 02 ml 05 juci 3h 005 m tricine buffer, ph 78, 035 ml 05 mm pyridoxal phosphate, 005 ml 001 m nadh in 001 m tricine buffer, ph 78, 04 ml lactate dehydrogenase, 10 iu and s-alkyl-l-cysteine lyase, 10 iu\n",
      "\n",
      "this cysteine sample was then mixed with 001 m nacn, 01 ml 02 m tris buffer, ph 85, 015 ml and 005 ml 0-cyanoal- anine synthetase 100 lu/ml\n",
      "\n",
      "the samples ob- tained from the above incubation were diluted with 2 jumol cold ß-cyanoalanine and mixed with 05 ml of 05 m potassium phosphate buffer ph 78 containing a- ketoglutarate, 7 jumol nadh, 5 pmol asparaginase, 4 iu glutamate-oxalacetate transaminase, 4 iu and malate dehydrogenase, 60 iu\n",
      "\n",
      "the third fraction was incubated with 120 iu of malate dehydrogenase in 05 ml glycine buffer l m, ph 95 containing 2 /limol 3-acetylpyri- dine-adenine dinucleotide apad at room tempera- ture for 3 h and the reaction was followed spectro- scopically by the increase in the absorption at 365 nm 19\n",
      "\n",
      "using cystine stereospecifically labeled at c-3 with tritium or with tritium and deuterium, it was found that the alpha,beta-elimination reaction catalyzed by s-alkylcysteine lyase involves stereo-specific replacement of the beta-substituent of the substrate by a hydrogen derived from the solvent, d2o or h2o, with retention of configuration to give pyruvate containing a chiral methyl group\n",
      "\n",
      "the column was washed with buffer a 5 mm, and the valyl trna synthetase activities were eluted with a 260-ml linear gradient in buffer a from 5 to 300 mm potassium phosphate\n",
      "\n",
      "the assay mixture of 1 ml contained 0025 m tris buffer, ph 74 edta, 5 x 10--1 m sodium citrate, 3 x 10-3 m nadh, 10-'' m deoxyribose 5-p, 10-3 m alcohol dehydro- genase, 10 pg glycerol 3-p dehydrogenase-triose phosphate isomerase mixture, 10 /xg\n",
      "\n",
      "properties of mitochondrial nadh dehydrogenase 565 in routine preparations of nadh de- hy drogenase, measures were not taken for complete exclusion of oxygen, and inclusion of 5 mm dithiothreitol in the buffers was considered sufficient to prevent extensive damage to the iron-sulfur centers of nadh dehy drogenase\n",
      "\n",
      "mycobacterium tuber- culosis cells 500 g wet wt, harvested at stationary phase and kept in a freezer, were thawed and resus- pended in 1 liter of 01 m na-phosphate buffer, ph 70, containing 05 mm dte\n",
      "\n",
      "the fractions con- taining malonyl-coa decarboxylase activity were pooled, concentrated by ultrafiltration using an ami- con pm-10 membrane, dialyzed overnight against 20 mm tris-hcl buffer, ph 76, containing 05 mm dte, and any insoluble material was moved by centrifuga- tion at 20,000g for 10 min\n",
      "\n",
      "the region of bpp synthetase activity, which eluted at about 125 void volumes, was concentrated by ultrafiltration amicon pm 10, and the concentrate was applied to a 15 x 15-cm column of hydroxylapatite bio-gel ht, bio-rad laboratories that was previously equilibrated with the same buffer utilized in the above gel permeation chromatography step\n",
      "\n",
      "for a typical assay of bpp synthetase activity, the reaction mixture contained from 5 to 50 ^g of protein in a total volume of 05 ml of 5 mm sodium phosphate buffer ph 62 containing 1 mm ascorbate, 10 mm mgd2, 05 mm dithioerythritol, and 10% glycerol\n",
      "\n",
      "the aqueous phase remaining after ether extraction was treated with 025 ml of 100 mm sodium acetate buffer ph 50 and this mixture was heated in an open tube for 10 min at 100°c\n",
      "\n",
      "residual ether was driven off under n2, 025 ml of 100 mm sodium acetate buffer ph 50 containing 05 unit of wheat germ acid phosphatase was added, and the mixture was incubated in a sealed tube for 90 min at 30°c\n",
      "\n",
      "stability and molecular weight of bpp synthetase after the sephadex g-150 purification step, the synthetase was relatively stable when stored in 10 mm sodium phosphate buffer ph 62 containing 20% glycerol, 1 mm dithioerythritol, and 10 mm mgclz losing 15% activity over 10 days at 0°c\n",
      "\n",
      "effect ofph in sodium phosphate buffer 5 mm, bpp synthetase exhibited a fairly sharp ph pro- file with an optimum at approximately ph fig 4\n",
      "\n",
      "the activity of bpp synthetase in other buffers tested at the same ionic strength, at or near the ph optimum, was lower than that observed in sodium phos- phate [tris-maleate ph 62, 20% relative to phosphate barbital-sodium acetate ph 62, 65% relative to phosphate sodium acetate ph 56, 80% relative to phosphate]\n",
      "\n",
      "the re- action of drosophila and rabbit muscle aldolases with dtnb at 25°c m 20 mm n-ethylmorpholine-acetic acid buffer ph 80 was observed in a cary spectro- photometer by measuring the increase in absorbance at 412 nm resulting from the release of o-thio-2- nitrobenzoate anion, for which an extinction coef- ficient of 136 x 104 m\"' crrr' was employed 14\n",
      "\n",
      "conversion of /î-l-galactopy- ranosyl phosphate to gdp-^-l-galactose by gdp-d-mannosegdp-l-galactose epimerase characterization 501 table i epimerase reaction with different mixtures of gdp-a-d-mannose and gdp-/3-l-galactose\" gd]total get ^gaü »p-hexoseime number 0 min 180 min agdp-gal 1 00 024 +024 2 019 027 +008 3 038 037 -001 4 050 040 -010 5 064 040 -024 6 078 053 -025 7 100 058 -043 ° gdp-hexose 125 nmol was incubated with a total of 012 mg lyophilized chlorella epimerase hydoxyl- apatite fraction in 50 ki of 0016 m glycylglycine buffer, ph 81\n",
      "\n",
      "2 abbreviations used mcase, 3-methylcrotonyl- coa carboxylase sds, sodium dodecylsulfate dtt, dithiothreitol buffer k, 20 mm potassium phosphate, ph 70, containing 01 mm edta and 01 mm dtt buffer kg, 10% glycerol in buffer k source of the enzyme is bovine kidney where the enzyme is localized in the mitochondria 12\n",
      "\n",
      "to study the effect of mono- and divalent cations on mcase, a sample of purified enzyme was first dialyzed against 20 mm tris-hcl, ph 80, containing 01 mm dtt and 25% v/v glycerol, for 5 h with four changes of buffer, and then used immediately\n",
      "\n",
      "saturated short-chain coa derivatives such as propionyl- and butyryl-coa are competitive inhibitors a', = 1600-1780 /j,m, while the unsaturated long-chain coa, 2z-geranyl-coa, the substrate for the homologous acyl-coa carboxylase, geranyl- coa carboxylase 14, is a good competitive inhibitor ki = 116 /am\n",
      "\n",
      "b urocanase 08 ml, 122 /xm in the same buffer was denatured with 72 m urea and titrated as above with 194 mm p-chloromercuribenzoate\n",
      "\n",
      "when photoactivated urocanase was treated with p-chloromercuribenzoate and dialyzed over- night the enzyme had no activity and could not be restored by light or dithiothreitol table i effect of thiol reagents upon activity of urocanase\" incubation relative concen- time activity reagent tration min % hg\" 2-chloromercuri-4-nitrophenol p-chloromercuribenzoate p-chloromercuri-phenylsulfonate iodoacetate iodoacetamide 5,5'-dithiobis-2-nitrobenzoate n-ethylmaleimide 01 1 1010 l l23 5 pm fsm ^m ^l,mmm mmmm mm 15 15 1515 15 15060 90 39 2 325 100 9394 100 ° active urocanase diluted to assay concentration in 50 mm potassium phosphate buffer, ph 75, was incubated with thiol-attacking reagents and then assayed for activity\n",
      "\n",
      "sulfite- modified urocanase 30 ml, 218 jug protein/ml in 50 mm potassium phosphate buffer, ph 75 was photoactivated see materials and methods\n",
      "\n",
      "246 o'donnell, hug, and hunter table ii restoration of urocanase activity by nad4^ after inactivation by p-chloromercuribenzoate0 addition relative activity % control no pcmb dtt + nad+ nad+ dtt dtt + nadh dtt + nadph dtt + nadp+ nadh nadph nadp-^ 100 81 14 28 26 28 29 0 1 1 ° p-chloromercuribenzoate pcmb, 10 /im was incubated for 75 min at room temperature with purified photoactivated urocanase 28 ^ig/ml in 50 mm potassium phosphate buffer, ph 75\n",
      "\n",
      "sulfite-modified urocanase 110 ml, 4 mg protein/ml in 50 mm potassium phosphate buffer, ph 75 was photoactivated see materials and methods\n",
      "\n",
      "ammonia-dependent anthranilate synthase was assayed by the same procedure, but with the substitution of 50 mm triethanolamine, ph 86, for phosphate buffer, and with 30 mm ammonium sulfate as amino donor\n",
      "\n",
      "the anthranilate synthase pool from the a-05m column was concentrated as above, mixed with 025 vol of sample buffer 120 mm tris-64 mm phosphate, ph 72/25% g!ycerov0004% bromphenol blue and applied to the top of the gel under a layer of running buffer 52 mm tris/55 mm glycine, ph 88\n",
      "\n",
      "early steps in mevalonate metabolism in rat liver and kidney 135 table i effect of streptozotocin-induced diabetes on rat liver and kidney mevalonate kinase, mevalonate-p kinase, and mevalonate-pp decarboxylase\" enzyme activity organ condition mevalonate kinase mevalonate-p kinase mevalonate-pp decarboxylase liver control 5 diabetic 6 391 ± 062 237 ± 033* 109 ± 040 046 ± 01l* 0773 ± 010 0208 ± 007** kidney control 6 diabetic 6 093 ± 007 054 ± 004** 0161 ± 0020 0061 ± 0015** 0161 ± 001 0093 ± 001** \" groups of male rats were injected either with a solution of streptozotocin in citrate buffer diabetic or citrate buffer alone control as described under experimental procedures\n",
      "\n",
      "control carboxylase activity remaining in the 2x a and 4x v enzymes diluted at 10°c with bicine buffer containing sufficient mg24- and h\"coa to yield final concentrations of 10 mm each and assayed after various times at 10°c is depicted in the upper set of curves\n",
      "\n",
      "in tris-chlo- ride buffer containing 1 mm edta and 5 mm 2-mercaptoethanol ph 82, i = 0013, 25°c chorismate mutase/prephenate de- hydratase moved as a single boundary on velocity sedimentation fig 1a\n",
      "\n",
      ", and soybean phenylalanine ammonia-lyase were equilibrated with 01 m borate na'1' buffer, ph 87, by overnight dialysis\n",
      "\n",
      "the buffer extraction procedure involved adding 20 ml of cold 50 mm potassium phosphate buffer at ph 68 or tris at ph 74 if glyoxalase ii was to be assayed per gram fresh weight of tissue to the pellet in the cen- trifuge tube\n",
      "\n",
      "effect of ph on uroporphyrinogen iii formation optimal activity of pbgase was ob- served at ph 78 in 50 mm sodium phos- phate buffer or 50 mm tris-hcl buffer fig 7\n",
      "\n",
      "as indicated in table ii, mutase re- tableii effect of anions and cations on the activity of purified mutase expt° compound concentrationmm activity munit 1 ammonium phosphate 200 133 sodium phosphate 200 141 potassium phosphate 200 112 tris/phosphate 200 136 2 tria/phosphate 25 058 60 168 100 153 200 112 400 091 3 calcium chloride 0 056 17 031 52 005 4 magnesium chloride 0 078 17 047 52 012 5 tris/phosphate 200 096 tris/su if ate 200 034 tris/chloride 200 016 ° assays were carried out as indicated under experimental proce- dures, except that in experiments 1, 2, and 5, the indicated compounds, at ph 74, were substituted for the buffer normally used 100 mm tris/ phosphate, ph 74\n",
      "\n",
      "the pre- cipitated carboxylase was recovered by centrifuga- tion and dialyzed against 50 mm sodium phosphate buffer ph 70 containing 1 mm dithioerythritol\n",
      "\n",
      "electronic spectra absolute and difference of the product of the adenylosuccinate synthetase reaction with 3-nitroalanine as substrate\n",
      "\n",
      "the standard reaction mixture in an open assay system contained 2 mm naatp, 5 mm mgs04, 1 mm liacetyl-coa, 20 mm nascos, 01% v/v triton x-100,5 units lactate dehydrogenase, 35 units pyruvate kinase, 02 mm nanadh, 2 mm naphosphoenolpyruvate, and 250 mm glycylgly- cine • naoh buffer ph 80 in a final volume of 10 ml\n",
      "\n",
      "a 25-35% linear glycerol gradient prepared in 25 mm k-phosphate buffer ph 75 containing 15 mm nahcoa, 2 mm dtt, 150 mm naci, and 002% nans was overlaid with 05 ml of a 40% ammonium sulfate fraction determinedto be free of castor oil seed rubp carboxylase activity in 25 mm k-phosphate buffer ph 75 containing 10% glycerol, 15 mm nahcog, 2 mm dtt, and 002% nang to which was added 35 mg of spinach rubp carboxylase partially purified type i, free of spinach acetyl coa carboxylase activity\n",
      "\n",
      "the inclusion of triton x-100 in the purification buffer results in a 45% in- crease in the amount of enzyme recovered from the acetone powder, and a greatly increased recovery of the enzyme after col- umn chromatography\n",
      "\n",
      "the long-chain acyl-coa syn- thetase was assayed at different ph values, ranging from 35 to 95, using citrate, phosphate, or tris-hcl buffer\n",
      "\n",
      "omission of atp or coa caused more than 98% decrease in activity, suggesting their absolute requirement\n",
      "\n",
      "taken together with earlier studies which revealed that hydrogen addition to the urocanate side chain during catalysis is from solvent and not by intramolecular transfer 3-5, it appears very likely that the urocanase reaction does not in- volve an internal oxidation-reduction with nad functioning to accept and transfer a hydride ion\n",
      "\n",
      "we have previously shown that the substrate an- alog, imidazole propionate dihydrourocanate, un- dergoes an enzyme-catalyzed exchange reaction with solvent protons and also reversibly forms a covalent complex with the nad on urocanase 6\n",
      "\n",
      "taken together with earlier studies which revealed that hydrogen addition to the urocanate side chain during catalysis is from solvent and not by intramolecular transfer 3-5, it appears very likely that the urocanase reaction does not in- volve an internal oxidation-reduction with nad functioning to accept and transfer a hydride ion\n",
      "\n",
      "we have previously shown that the substrate an- alog, imidazole propionate dihydrourocanate, un- dergoes an enzyme-catalyzed exchange reaction with solvent protons and also reversibly forms a covalent complex with the nad on urocanase 6\n",
      "\n",
      "synthase has been stored either in a partially purified form table i, fractions iv and vi or as a pure enzyme in 25 mm potassium phos- phate buffer, ph 65,1 mm 2-mercaptoeth- anol at -80°c for over 3 years with no loss of activity\n",
      "\n",
      "for this procedure, 098 ml of incubation mixture containing 40 frtnol hepes buffer, ph 67, 05 ßmol mns04, 02 ^imol nadp, a nonlimiting amount of glucose 6-phosphate dehydrogenase 10 enzyme unit, variable amounts ofis-glucose 1-phosphate be- tween 001 and 10 ^mol, and 3-glucose 1,6-bisphos- phate between 01 and 300 nmol was used in a 1-ml cuvette with a 1-cm light path in a varian 365 d spectrophotometer\n",
      "\n",
      "acetyl-coa carboxylase, precipitated at 40% ammonium sulfate and resuspended in 10 mm tris-hcl buffer ph 80 containing 20 mm j8-mer- captoethanol and 1 mm edta, was subjected to gel filtration on ultrogel aca 22 under the conditions described in the texi\n",
      "\n",
      "the mixture was equilibrated at 4°c for l h and the re- duction of the schiff base was initiated by the addition of 44 mg 112 mmol of sodium borohydride in 1 ml of tricine buffer\n",
      "\n",
      "homogeneous kyn- ureninase 01 mg was added to a reaction mixture containing 65 mm tris buffer, 0065 mm pyridoxal phosphate, 25 mm kynurenine, and 0020 ml benzal- dehyde at ph 80 in a total volume of 10 ml\n",
      "\n",
      "kynureninase was added to a 10-ml re- action mixture containing 65 mm tris buffer, 0065 mm pyridoxal phosphate, 10 mm l-[side c/mw^-^cjkynurenine, and the electrophilic species, as indicated in table ii\n",
      "\n",
      "when nem was added to 8% nad^in- dependent synthase after pretreatment with glc-6-p alone or glc-6-p plus nad4, 376 gumber, loewus, and loewus table ii protection by substrate against pcmb inhibition of pine pollen synthase experiment\" nad^independent activity, % activity remaining c substrates added to preincubation mm l min 10 min l 70% none 12 06 nad+ 42 47 21 glc-6-p 10 149 107 2 8% none 68 20 glc-6-p 77 160 34 glc-6-p 77 + nad-\" 4 379'' 210e ° a reaction mixture consisting of enzyme 08 mg protein in experiment l 02 mg in experiment 2, additions as noted in the table, and 20 mm tris-acetate buffer, ph 775, was incubated for 10 min at 32° c\n",
      "\n",
      "the fractions containing isomerase and prenyltransferase activities that were obtained from the bio-gel a 15m column were pooled and dialyzed against 002 m potassium phosphate buffer, ph 70, containing 2 mm dithiothreitol, and then applied to a column of deae-celulose 3 x 20 cm, equilibrated with the same buffer\n",
      "\n",
      "the incubation system for the assay of isopentenyl pyrophosphate isomerase activity contained borate buffer, ph 82, 50 ^mol tween 80, l mg mnclg, l /miol mgclz, 75 iimol di- thiothreitol, 5 iimoï [l-\"c]isopentenyl pyrophosphate, 20 nmol and 120,000 dpm and enzyme in a final volume of 05 ml, unless otherwise stated\n",
      "\n",
      "the remainder of the petroleum ether extract was evaporated to a small volume under nitrogen and then analyzed on a reverse-phase thin-layer plate with methanolwater 8020 as the developing solvent\n",
      "\n",
      "stability of enzymes isopentenyl pyrophosphate isomerase is stable for several weeks at -20°c in 01 m potassium phosphate buffer, ph 70, con- taining 2 mm dithiothreitol\n",
      "\n",
      "the prenyltransferase recovered from the sephadex g-100 can also be stored for several weeks at —20°c in 01 m potassium phosphate buffer, ph 70, containing 2 mm dithiothreitol\n",
      "\n",
      "di- thiothreitol had a slight stimulatory effect on isopentenyl pyrophosphate isomerase activity, but it led to a slightly lower ac- tabl cofactor re' ieiiquirements isomerase ° prenyltransfer •äse nanomoles isopentenyl nanomoles isopentenyl pyrophosphate percentage pyrophosphate percentage incorporated of control incorporated of control control 1 32 100 131' 100'' + nadp 2 ftmol 32 100 10 77 + dithiothreitol 5 fimol 47 147 11 85 + atp 04 fimol 32 100 13 100 + nadp 2 »mol, dithiothreitol 5 ^mol, atp 04 minol 35 109 08 62 control 2 — — 11e 100\" -mgclz 23 72 04 36 - tween 80 31 97 08 73 - mncia 26 81 11 100 - mgcl2 and mncia 03 9 0 0 ° the control contained borate buffer, ph 82 10 /imol mnclz l jumol tween 80 l mg mgcia 75 ^mol isopentenyl pyrophosphate isomerase 49 units, obtained after sephadex g-100 chromatography and [\"c]isopentenyl pyrophosphate 20 nmol and 120,000 dpm in a final volume of 05 ml\n",
      "\n",
      "6 for prenyltransferase assayed as geranyl pyrophosphate the enzyme 0426 units was incubated with borate buffer, ph 82 10 fimol mncia 1 fimoï tween 80 l mg mgcle 75 »iniol and iodoacetamide 15 t»mol in a final volume of 017 ml\n",
      "\n",
      "gsh s-trans- f erase activity was assayed essentially as described by habig and jakoby 16 using dncb as the substrate in ph 65 buffer 50 mm each hepes and 4-morpho- lineethanesulfonic acid mes, and 1 mm glutathione and following the kinetics of the reaction on a varian model 219 recording spectrophotometer with a sample chamber thermostated to 30°c\n",
      "\n",
      "ltc synthetase reactions contained 1 mm gsh, 25 ßu lta< lithium salt, l mg/ml bsa, and variable amounts of enzyme in suspending buffer, and were incubated for 5 min\n",
      "\n",
      "purification steps include a extraction of soluble proteins by soniflcation and subsequent centrifugation b deae-cellulose ad- sorption and elution with a sodium chloride gradient 0-03 m nad in a buffer solution containing 20 mm tris-hcl, ph 75, 10% glycerol, 10 mm ß-mercapto- ethanol, 01 mm phenylmethylsulfonyl fluoride alanyl-trna synthetase was eluted between 013 and 019 m nacl\n",
      "\n",
      "each 30-ml cuvette contained 10 mm citrulline, 10 mm atp, 10 mm mgclz, 10 units each of pyruvate kinase, lactate dehydrogenase, py- rophosphatase, and adenylate kinase, 10 mm pep, 016 mm nadh, 50 mm buffer, and variable amounts of either l-aspartate or dl-3-nitro-2-nitropropionate\n",
      "\n",
      "this report describes the purification of casbene synthetase to apparent homoge- neity from castor bean seedlings that were infected with r stolonifer, the preparation of monospecific polyclonal antibodies to the enzyme, and their use in studies of 4 abbreviations used pal, phenylalanine ammonia lyase sds-page, sodium dodecyi sulfate-poly- acrylamide gel electrophoresis tbs, tris-buffered saline ggpp, geranylgeranyl pyrophosphate bsa, bovine serum albumin\n",
      "\n",
      "eighty micrograms of pure casbene synthetase in 500 ill of tris-buffered saline tbs [20 mm tris-hcl ph 75 140 mm nad] was mixed with 500 ft\\ of complete freund's adjuvant and injected intradermally into a rabbit new zea- land white at multiple sites\n",
      "\n",
      "a partially purified sample of casbene synthetase obtained from a re- active blue 2-agarose column was concentrated over a ym 10 ultraflltration membrane, and 2-ml aliquots of the concentrated sample were applied to the ultropac tsk 545 deae column equilibrated in buffer d containing 150 mm kc1\n",
      "\n",
      "these results indicated that the a hydrogen of selenocysteine was removed by a base at the active site, and was incorporated into the a position of alanine, a product, without exchange of a solvent deuterium\n",
      "\n",
      "table iii activation of inactivated selenocysteine ß-lyase with pyridoxamine-p and a-keto acid\" ff-keto acid relative activity control pyridoxal-p'' glyoxylate pyruvate a-ketobutyrate oxaloacetate o-ketoglutarate none 100 66 69 31 49 63 0 ° the inactivated selenocysteine 0-lyase with se- lenocysteine was dialyzed against 001 m potassium phosphate buffer ph 74, and then assayed after preincubation with pyridoxamine-p and o-keto acids\n",
      "\n",
      "these results indicated that the alpha hydrogen of selenocysteine was removed by a base at the active site, and was incorporated into the alpha position of alanine, a product, without exchange of a solvent deuterium\n",
      "\n",
      "imidazole was found to be the optimum buffer in the presence of phos- phate, tris, or bis-tris, rat liver aspar- tate-4-decarboxylase was 30 to 70% less active\n",
      "\n",
      "when the tissue extracts were dialyzed against hydroxyl- amine and then against standard buffer, the cysteine sulfinate desulfinase remained active as previously seen with aspartate- 4-decarboxylase activity\n",
      "\n",
      "a key piece of evidence in considering pyruvate carboxylase in this regard is that it has an absolute requirement for acetyl-coenzyme a\n",
      "\n",
      "other purifica- tions of asparagine synthetase have been characterization of beef pancreatic asparagine synthetase 343 table iii chloride ion effects on asparagine synthetase glutamine-dependent ammonia-dependent naci reaction' glutaminase reaction1' reaction\" mm nmol asn min\"1 ml\"1 nmol nhs min\"1 ml\"1 nmol asn nun\"1 ml\"1 00 224 00 591 04 390 157 501 30 368 371 358 200 468 462 209 \"[glutamine] = 15 mm, all other substrates saturating\n",
      "\n",
      "they eluted asparagine synthetase from the affinity column with a buffer containing 16 mm atp and 16 mm magnesium ions, yielding a 17-fold purification\n",
      "\n",
      "purified racemase 018 ng was assayed for activity in the standard assay as described under materials and methods except that the buffer was 10 mn hepes-koh, ph 75, and the reaction mixtures containing racemase and the various concentrations of metals were prein- cubated for l h at 4°c before adding the substrate, dl-ß^hlmethylmalonyl-coa\n",
      "\n",
      "the activity of 018 ng of purified racemase was determined with different concentrations of metals in the presence of 100 ^m go*2, using the standard assay except that the buffer was 10 mm hepes-koh, ph 75, and reaction mixtures were preincubated for l h at 4°c prior to adding the substrate, dl-p-'hjmethylmalonyl-coa\n",
      "\n",
      "the fractions in a and b were assayed for racemase activity both with • and without 100 iiîs w2 0, using the standard assay except that the buffer was 10 mm hepes-koh, ph 75\n",
      "\n",
      "'\"determined by assaying racemase over a ph range of 40 to 100 with the standard assay as described under materials and methods except that the buffer was 5 mm citric acid, 5 mm potassium phosphate, 5 mm tris-hcl, adjusted to various ph's with 10 m koh\n",
      "\n",
      "chorismate mutase-1, obtained after deae-cellulose chromatography, was ad- sorbed onto a column 14 x 20 cm of w- aminohexyl-agarose equilibrated with 50 mm glycine buffer containing 02 mm tryp- tophan and 005% /3-mercaptoethanol\n",
      "\n",
      "a 02-ml aliquot of the aqueous phase was heated at 100°c for 4 min and cooled, and 10 ^1 of 15 mm pi and 06 ml of isobutanol water saturated were added\n",
      "\n",
      "the incubations were at 25°c and contained, in 02 ml, 50 mm triethanolamine-cl^ buffer, ph 70,25 mm mgcia, 1 fim glucose-l,6-p2, 05 mm nadp, 01 mm edta, 1 mm mercaptoethanol, 02 mg bovine serum albumin, 08 unit pmi, 05 unit pgi, and 01 unit glu- cose-6-p dehydrogenase\n",
      "\n",
      "the incubations contained 100 mm imidazole-acetate buffer, ph 70,1 mm glucose- 1-p, l ßm glucose-l,6-p2,018 mm nadp, 5 mm mgclz, 05 mm edta, and 003 unit glucose-6-phosphate de- hydrogenase\n",
      "\n",
      "the molecular size of glutamine synthetase was estimated by gel filtra- tion chromatography as described by andrews 14 using a column of sephacryl s-300 1 x 54 cm equil- ibrated with hepes buffer 20 mm, ph 74, mgs04 10 mm, edta 1 mm, and, in addition when indicated, l-glutamate 50 mm, nad 100 mm, and/ or atp 4 mm\n",
      "\n",
      "molecular size estimation of the molecular weight of the purified dogfish glutamine synthetase by gel filtration chromatography on se- phacryl s-300 as described under experi- mental procedures gave a value of about 200,000 when elution was carried out in the presence of hopes buffer containing edta and mgs04 when glutamate, naci, and atp were also present, the peak of enzyme activity eluted in nearly the same position as, but just slightly later than, the peak associated with the protein stan- dard ferritin, but always significantly later than e coli glutamine synthetase, which is consistent with a molecular weight of about 400,000\n",
      "\n",
      "one milliliter of purified synthetase complex 100 nu in buffer 50 mm tris-hcl, ph 75, 5 mm mgcia, 2 mm dte, 25 mm kc1 containing l mg/ml bsa and 20% glycerol was loaded onto a column 1 ml of diami- nooctyl-sepharose, preequilibrated with the same buffer containing 20% glycerol\n",
      "\n",
      "the interpretation that the saturating amount of lysine con- stituted an irreversible step between the addition of atp and the release of pyro- phosphate fig 4a was confirmed by using a subsaturating concentration 10 jum of lysine under the same conditions, which gave a lineweaver-burk plot with a con- verging pattern intersecting at the left of the ordinate fig 4d\n",
      "\n",
      "lineweaver-burk plots of different variable substrates at changing fixed concentration of pyrophosphate and saturating concentrations of other substrates, except d with subsaturating concentration of lys 10 ßtl, using the 18 s synthetase complex\n",
      "\n",
      "lineweaver-burk plots of different variable substrates at changing fixed concentrations of amp and saturating concentrations of other substrates, except b with subsaturating concen- tration of atp 0035 mm, using the 18 s synthetase complex\n",
      "\n",
      "sion pathway of -rdp-mannose to gdp- droxymercunbenzoic acid nfcm, a'-ethylmaleimide „ mi • i dtt, dithiothreitol pmsf, phenylmethylsulfonyl fucose- thls pathway provides the major fluoride pcs, fetal calf serum con a, concanavalin source of gdp-fucose in mammalian cells a vsv, vesicular stomatitis virus pbs, phos- 4,5 and has been found in bacterial plant, phate-buffered saline glcnac, jv-acetylglucosamine and animal cells 3, 6-8 to consist of the glcnac-tl, a^-acetylglucosaminyltransferase\n",
      "\n",
      "descending paper chroma- tography was carried out on whatman no 1 paper in the following solvent systems solvent a, pyridine- ethy] acetate-water-acetic acid 5531 with pyridine- ethyl acetate-water 281 in the bottom of the tank 16 solvent b, the upper phase of pyridine-ethyl ac- etate-water 1036115 15 solvent c, 2-butanone saturated with water 13 solvent d, 2-butanone- acetic acid-saturated boric acid 911 14 solvent e, the upper phase of butanol-ethanol-02 m sodium borate buffer, ph 894 412 13\n",
      "\n",
      "further treatment of this effluent with hexokinase did not reduce this value, 536 ripka, adamany, and stanley table i chromatographic properties of sugar standards solvent sugar a b c d e mannose 078 045 034 043 035 6-deoxygalactose fucose 088 067 057 066 028 6-deoxyglucose glucomethylose nd 097 086 nd 053 6-deoxymannose rhamnose 100 100 100 100 100 6-deoxytalose talomethylose nd 141 204 139 079 note\n",
      "\n",
      "if reduction with nabh4 was performed prior to acid hydrolysis and hexokinase treatment, the impurities separated into three peaks in both solvents a and b one peak was detected between mannose and fucose, a second peak cochromatographed with rhamnose, and a third mi- grated very close to talomethylose\n",
      "\n",
      "because these impurities represented 2-3% of the ini- tial radioactivity and cochromatographed in the vi- cinity of rhamnose and talomethylose in two solvent systems, initial experiments were interpreted to in- dicate that the mutant lines actually produced the gdp-6-deoxy-keto sugar intermediates expected from the action of gdp-mannose 4,6-dehydratase 10\n",
      "\n",
      "chorismate mutase in varying amounts 0 to 10 iig in 100 ^i/well of buffer 15 mm sodium carbonate and 35 mm sodium bicarbonate, ph 98 was allowed to incubate in the microtiter plate at 4°c for 3 to 4 h\n",
      "\n",
      "biotin carboxylase activity was assayed radiometrically at 30°c as described in 10 by measuring the formation of [\"c]carboxybiotin in a reaction medium containing 8 mm kh\"c08 03-15 x 106 counts • min\"1 • ^mol~1, 50 mm biotin, 1 mm atp, 8 mm mgclz, 10% v/v ethanol unless indicated, 3 mm reduced glutathion, 06 mg/ml bovine serum al- bumin fraction v, sigma, and 01 m triethanolamine buffer, ph 80\n",
      "\n",
      "synephrinase showed an absolute requirement for a sulfhydryl compound like glutathione, dtt or js-mercaptoethanol for its activity table ii\n",
      "\n",
      "serial jl 272380 291210 291847 291849 291857 31 archives of biochemistry and biophysics archivesbiochemistrybiophysics 2004-11-02 2004-11-02 2015-09-28t183346 1-s20-0003986186904406 0003-98618690440-6 0003986186904406 101016/0003-98618690440-6 s350 s3503 head-and-tail 1-s20-s0003986100x04195 2015-10-01t022437707269-0400 0 0 19860701 19860731 1986 2004-11-02t000000z rawtext articleinfo articletitlenorm authfirstinitialnorm authfirstsurnamenorm cid cids contenttype copyright crossmark dateloaded dateloadedtxt datesearch datesort dateupdated dco docsubtype doctype doi eid ewtransactionid hubeid indexeddate issfirst issn issnnorm issuelist itemstage itemtransactionid itemweight openaccess openarchive pg pgfirst pglast pii piinorm pubdateend pubdatestart pubdatetxt pubyr sortorder srctitle srctitlenorm srctype ssids alllist content subj tomb volfirst volissue volumelist webpdf webpdfpagecount yearnav affil articletitle auth authfirstini authfull authlast primabst pubtype ref 0003-9861 00039861 false 248 248 1 1 volume 248, issue 1 47 429 434 429 434 198607 july 1986 1986-07-01 1986-07-31 1986 converted-article sco copyright © 1986 published by elsevier inc terpenecyclasecatalysisinorganicsolventminimalwatermediademonstrationoptimizationpinenecyclaseactivity wheeler c zaks 1984 1249 1251 a katchalski 1971 445 536 e advancesinenzymology diastoli 1966 44 47 f diastoli 1967 163 165 f diastoli 1967 289 291 f klibanov 1977 1351 1361 a butler 1979 253 259 l macrae 1983 291 294 a gatfield 1984 569 572 i duarte 1984 573 576 j cambou 1984 2687 2692 b koshiro 1985 47 57 s zaks 1985 3192 3196 a procnatlacadsciusa kirchner 1985 7072 7076 g martinek 1981 76 89 k kazandjian 1985 5448 5450 r martinek 1981 90 101 k croteau 1986 81 133 r herbsspicesmedicinalplantsrecentadvancesinbotanyhorticulturepharmacology wheeler 1986 c croteau 1985 5956 5962 r croteau 1986 r gambliel 1984 740 748 h gambliel 1982 2335 2342 h bradford 1976 248 253 m dixit 1981 1969 1970 v coates 1976 73 230 r progressinchemistryorganicnaturalproducts antonini 1981 291 296 e martinek 1982 889 891 k demeis 1985 7783 7789 l fersht 1977 134 155 a enzymestructuremechanism wheelerx1986x429 wheelerx1986x429x434 wheelerx1986x429xc wheelerx1986x429x434xc item 0003-98618690440-6 0003986186904406 1-s20-0003986186904406 101016/0003-98618690440-6 272380 2015-10-01t022437707269-0400 1986-07-01 1986-07-31 1-s20-0003986186904406-mainpdf https//s3amazonawscom/prod-ucs-content-store-us-east/content/pii0003986186904406/main/application/pdf/9880695999ce5cb50e61d1c1f08b5605/mainpdf https//s3-eu-west-1amazonawscom/prod-ucs-content-store-eu-west/content/pii0003986186904406/main/application/pdf/9880695999ce5cb50e61d1c1f08b5605/mainpdf mainpdf pdf true 571382 main 6 1-s20-0003986186904406-main_1png https//s3amazonawscom/prod-ucs-content-store-us-east/content/pii0003986186904406/preview/image/png/529a4d9e92c864516423267c59d891d0/main_1png https//s3-eu-west-1amazonawscom/prod-ucs-content-store-eu-west/content/pii0003986186904406/preview/image/png/529a4d9e92c864516423267c59d891d0/main_1png main_1png png 77819 849 656 image-web-pdf 1 archives of biochemistry and biophysics vol 248, no 1, july, pp 429-434,1986 communication terpene cyclase catalysis in organic solvent/minimal water media demonstration and optimization of +-a-pinene cyclase activity1 carl j wheeler and rodney croteau2 institute of biological chemistry, washington state university, pullman, washington 9916jf-631io received february 4,1986, and in revised form march 26,1986 a partially purified and lyophilized preparation of +-a-pinene cyclase from sage salvia offidnalis was shown to convert geranyl pyrophosphate to the monoterpene olefins a-pinene, camphene, limonene, and myrcene, in hexane with the addition of 01 to 10% water\n",
      "\n",
      "to ascertain whether alliin lyase dissociates into sub- units, a preparation of the purified fraction vi 2 mg was first incubated in a 6 m so- lution of urea 45 min at 65°c and its sub- unit molecular weight determined on the same column of biogel a05m, equilibrated with the same buffer containing 6 m urea\n",
      "\n",
      "effect of ferrous iron on uroporphyrinogen decarboxylase activity measured with different buffers the enzyme when assayed in the pres- ence of potassium phosphate, tris, and tes buffers showed similar activity whereas in the presence of mes and hepes buffers, the activity was lower table ii\n",
      "\n",
      "hplc and tlc analysis of pgh2 metabolites tlc analyses of inhibition experiments with pgh2 or pgh2 incubations with homogeneous thromboxane synthase were performed employing the solvent sys- tem benzenebdioxaneacetic acid, 904545 v/v, as previously described 10\n",
      "\n",
      "aliquots 102 pmol of thromboxane synthase ref 4, 95% pure according to sds-polyacrylamide gel electrophoresis were incubated in 05 ml of 50 mm potassium phosphate buffer, ph 74, with the afore- mentioned endoperoxides at concentrations of 5, 10, 15, 20, 25, 30, 40, and 60 /*m\n",
      "\n",
      "reduc- tion with triphenylphosphine, isomerisa- tion in aqueous buffer 50 mm potassium phosphate, ph 74, room temperature, 30 min, or incubation with homogeneous hu- man platelet thromboxane synthase 4, followed by tlc analysis using chloroform ethyl acetatemethanolacetic acidwater, 90451215075 v/v, or benzenedioxane acetic acid, 904545 v/v, as solvents con- firmed the identifications\n",
      "\n",
      "the main products obtained after trans- 128 hecker et al formations were pgf triphenylphosphine reduction, pge isomerisation in buffer, and thromboxane plus ci7 hydroxy fatty acid incubations with thromboxane syn- thase\n",
      "\n",
      "homogeneous thromboxane synthase 28 pmol was incubated with 30 nmol pgh2 in 500 /a of 50 mm potassium phosphate buffer, ph 74, at 20° c\n",
      "\n",
      "for measurements with microsomes we choose the hht absorption since mda is also a by-product of lipid peroxidation and may be formed from unsaturated fatty acids present in crude preparations of throm- boxane synthase\n",
      "\n",
      "homogeneous thromboxane synthase 56 pmol was incubated for 2 min at 20° c in 200 iil of 50 mm potassium phosphate buffer, ph 74, with 2 nmol of [1-14c]pgh2\n",
      "\n",
      "14c]pgh2 with 68 pmol homogeneous thromboxane synthase in 02 ml of 50 mm potassium phosphate buffer, ph 74, for 30 min at 20° c hhd and aw-hht were ob- tained after incubation of either 121 nmol [1-14c]pgh!\n",
      "\n",
      "validation of the hht assay by inhibitor studies and determination of inhibitor constants incubations were performed at 20°c in 05 ml of 50 mm potassium phosphate buffer, ph 74, containing 138 pmol ho- mogeneous thromboxane synthase\n",
      "\n",
      "the amounts of enzyme given in table iii were determined as follows purified human platelet thromboxane synthase by protein determination either with the pierce bi- cinchoninic acid- bca reagent pierce chemical co, rockford, il or according to peterson 27 solubilized human plate- let, rat liver, and human granulocyte mi- crosomes by optical cytochrome p-450 de- termination according to omura and sato 28 and solubilized sheep seminal vesicle microsomes by optical absolute spectros- copy in the presence and absence of potas- sium cyanide 29\n",
      "\n",
      "a dual buffer system was developed by substituting a mixture of 50 mm tris free base and 50 mm maleate for the tris component of buffer t\n",
      "\n",
      "pentalenene synthase activity was eluted with a further linear 400 ml gradient of 50 to 250 mm phosphate in buffer t\n",
      "\n",
      "crude pen- talenene synthase obtained after initial ammonium sulfate precipitation was resuspended in buffer h\n",
      "\n",
      "individual slices were removed with a nickel spatula and assayed for pentalenene synthase by re- suspension in a mixture of 025 ml of standard assay buffer and 015 ml of buffer t containing 5 ^m [l^hzjfarneayl pyrophosphate\n",
      "\n",
      "thus, in mammary gland of lactating rats, as in table ii assay op antizyme-ornithine decarboxylase complex in mammary gland° ornithine decarboxylase units/g tissue condition of rat + buffer + dfmo-odc 6 h fasted 5 24 h fasted 4 51 ±19 23 ± 12 84 ± 22* 34 ± ii* ° cytosol was prepared from mammary gland of lactating rats which had been fasting for 6 or 24 h dfmo-inactivated ornithine decarboxylase dfmo- odc was prepared as described under materials and methods and was added to the reaction mixture con- taining cytosol but without ornithine\n",
      "\n",
      "thus, in mammary gland of lactating rats, as in table ii assay op antizyme-ornithine decarboxylase complex in mammary gland° ornithine decarboxylase units/g tissue condition of rat + buffer + dfmo-odc 6 h fasted 5 24 h fasted 4 51 ±19 23 ± 12 84 ± 22* 34 ± ii* ° cytosol was prepared from mammary gland of lactating rats which had been fasting for 6 or 24 h dfmo-inactivated ornithine decarboxylase dfmo- odc was prepared as described under materials and methods and was added to the reaction mixture con- taining cytosol but without ornithine\n",
      "\n",
      "purified pectate lyase 500 ßg was dissolved in 05 ml phosphate-buffered saline, emulsified with an equal volume of freund's complete adjuvant difco, and injected subcutaneously into a new zealand rabbit at multiple sites\n",
      "\n",
      "galacturonic acid and the small oligomers with terminal 4,5- unsaturated 4- deoxyhexopyranuronate residues generated from polygalacturonate by the lyase were separated by thin- layer chromatography on 05 mm silica gel 60 plates developed in ethanolwaterammonium hydroxide 75 214\n",
      "\n",
      "the activity of the pectate lyase was de- pendent on the presence of calcium in the assay buffer table iii\n",
      "\n",
      "the ratio of calcium to polygalacturonic acid carboxyl, which gave maximal pectate lyase activity, was con- stant at both subsaturating and saturating substrate concentrations fig 8, inse^\n",
      "\n",
      "fractions with only cytochrome a were pooled and adjusted to ph 8 with solid tris-base and applied to a deae bio-gel a column bio-rad, 5 ml equil- ibrated with 5 mm tris, 01% tween 20, ph 8 buffer 1\n",
      "\n",
      "ac- tivity of cytochrome c oxidase was monitored spec- trophotometrically at 550 nm following the cyto- chrome c oxidation t = 28 mm\"1 cm\"1 in a buffer composed of 5 mm hepes, 02% tween 20, ph 74\n",
      "\n",
      "in order to determine the optimal incubation time for enzyme development, mung bean segments were incubated in 50 mm potassium citrate buffer, ph 46, containing 10 mm caclz, 100 mm sorbitol, 033 mm chloramphenicol, 100 ^m iaa, 10 ^m bap, and 1 mm aminooxyacetate aoa, and shaken gently in the dark at 28°c this combination was found to maxi- mize acc synthase production\n",
      "\n",
      "the acc synthase was then eluted with 400 ml of 500 mm phosphate buffer\n",
      "\n",
      "the acc synthase was then eluted with a continuous lin- ear gradient 500 ml total volume of phosphate buffer, ph 80, from 20 to 0 mm and ethylene glycol from 0 to 100%\n",
      "\n",
      "the acc synthase-containing fractions were pooled and loaded on a column of hydroxylapatite 25 cm diam and 12 cm high equilibrated with 20 mm potassium phosphate buffer\n",
      "\n",
      "four different buffers bicine-koh, tri- cine-koh, 2-amino-2-ethyl-l,3-propane- diol aep, and potassium phosphate at ph 80 were used as extraction solutions to isolate acc synthase from etiolated mung bean hypocotyl segments\n",
      "\n",
      "there was no difference among tricine, bicine, and aep buffers in the recovery of acc synthase however, phosphate buffer at the same 100 mm concentration produced over twice as much activity table i\n",
      "\n",
      "we also ob- served similar results when bicine, tri- cine, and aep were used as extraction buffers but 100 mm phosphate buffer ex- tended the half-life of acc synthase to over 10 h and there was a continual in- crease in activity when phosphate concen- trations were increased up to 15 m table i above this concentration, there was a decrease in activity, probably due to pre- cipitation of the enzyme\n",
      "\n",
      "a normal day's operation of extraction from 700 g of segments, ultracentrifuga- tion, and desalting was concluded by a stepwise elution of the activity from a short column of hydroxylapatite fig 2, after which the activity was sufficiently stable to be frozen at —20°c for an ex- tended period of time up to 30 days with table i a comparison of extraction buffers for the isolation and stabilization of acc synthase from pretreated mung bean hypocotyl segments acc synthase activity\" buffers, ph 80 initial after 3 h % activity remaining aep,6100 mm 39 ± 04 072 ± 012 18 aep, 250 mm 41 ± 04 081 ± 009 20 tricine-hcl, 100 mm 33 ± 03 067 ± 014 20 tricine-hcl, 250 mm 32 ± 05 065 ± 018 20 bicine-hcl, 100 mm 34 ± 08 068 ± 009 20 bicine-hcl, 250 mm 35 ± 04 074 ± 011 21 k phosphate, 100 mm 75 ± 11 61 ± 14 81 k phosphate, 250 mm 130 ± 20 102 ±22 78 k phosphate, 500 mm 185 ± 24 176 ± 19 95 k phosphate, 10 m 265 ± 17 257 ± 27 97 k phosphate, 15 m 321 ± 28 320 ±36 100 k phosphate, 20 m 214 ± 36 231 ±29 108 '\" acc synthase activity is expressed as nanomoles acc produced per hour per gram of segments\n",
      "\n",
      "two fractions containing the lyase were applied to a col- umn 47 x 15 cm of polybuffer ex- changer 94 pbe 94 which had been pre- equilibrated with 25 mm histidine-hcl buffer ph 62\n",
      "\n",
      "fractions containing the lyase activity were pooled, adjusted to ph 62 with l m histidine-hcl buffer ph 62, and concentrated by ultrafiltration using an amicon ultrafiltration cell containing a pm 10 membrane at a pressure of 20 psi\n",
      "\n",
      "insoluble matter was removed by centrifugation at 140,0007 for 30 min and a 15 dilution of the supernatant fluid in water was assayed for pyruvate-dependent incorporation of [\"cjkhcoa into acid-stable product in 2 ml contain- ing atp, 4 mm hepes3 buffer, 01 m, ph 78 sodium pyruvate, 10 mm mgs04,10 mm acetyl coa and ben- zoyl coa, as indicated nadh, 025 mm malate dehy- drogenase ec 11137, 20 units [\"clkhcog, 10 mm, 8 itc'v, enzyme preparation, 100 ii\\\n",
      "\n",
      "accumulation of malate was determined by measuring appearance of nadph upon incubation of neutralized aliquots of the acid- soluble fraction of the assay mixture for pyruvate carboxylase with tris buffer 01 m, mgclz 4 mm, 3 abbreviation used hepes, 4-2-hydroxyethyl-l- piperazineethanesulfonic acid\n",
      "\n",
      "aliquots of 50 {a were assayed for benzoyl coa by using a phosphate buffer 02 m, ph 40meth- anol 6535 mobile phase at a flow rate of 05 ml/ inhibition of pyruvate carboxylase by sodium benzoate 203 table i assay of pyruvate carboxylase in isolated mitochondria addition or deletion incorporation of [\"cjkhcos into acid-stable product nmol/mg protein • min none control - pyruvate + malate 5mm + phenylpyruvate 2mm + glyoxylate 3 mm + cyanohydroxy- cinnamate 5 mm 269 ± 08 07 ±02 117 ±13 76 ± 05 05 ± 01 note\n",
      "\n",
      "for studies with h^o, preparations of fenchol cy- clase, which had been adjusted to the 5 mm sodium phosphate buffer system prior to lyophilization, were employed\n",
      "\n",
      "the effect of ph on aristolochene synthase activity was analyzed as described pre- viously 14 using mes, mopso, hepes, and tris buffers\n",
      "\n",
      "the unbound ma- terial containing the corn aldolase was pooled, dia- lyzed four times against 2 liters of buffer c 20 mm mes-naoh, ph 60,1 mm edta, 1 mm dtt, 1% v/ v glycerol, and then absorbed to a 60-ml cm-sepha- rose column, ph 60\n",
      "\n",
      "the aldolase con- taining fractions were concentrated by dialysis in a degased 85% saturated ammonium sulfate solution made up in 50 mm tris-hcl, ph 74,1 mm edta, and 25 mm dtt at 4°c\n",
      "\n",
      "the patchoulol synthase was eluted from the column by raising the sodium phosphate content of the buffer to 80 mm at ph 80, with cacl and dithiothreitol as before, 10-ml fractions, 05 ml/min, and the two fractions containing the activity were combined\n",
      "\n",
      "therefore, dithiothreitol, sodium phosphate, glycerol, and divalent metal ion usually mg^ were included in all steps at which their presence in the buffer did not compromise chromatographic sepa- ration of the cyclase\n",
      "\n",
      "p-enolpyruvate carboxykinase [gtpoxaloacetate carboxy-lyase transphosphorylating ec 41132] is another of the kinase family of enzymes and catalyzes the reversible decarboxylation ofoxalacetate with a con- comitant transfer of the phosphoryl group from the nu- cleotide donor gtp or itp to yield p-enolpyruvate, cog, and gdp or idp and has an absolute requirement for the divalent cation mn24'\n",
      "\n",
      "bis-tris pro- pane buffer was substituted for tris buffer to increase the buffering capacity in the assays\n",
      "\n",
      "the ph optimum for both 3-methylcrotonyl-coa car- boxylase and propionyl-coa carboxylase activities was between 75 and 83, when tris was utilized to buffer the assays\n",
      "\n",
      "sol- ubilization of tx-synthase was achieved by homogenizing microsomal protein in 10 mm k2hpo4, 1 mm dtt, 1 mm edta, 10% v/v glyc- erol, 07% chaps solubilization buffer\n",
      "\n",
      "nadh-uqi reductase activity was measured at 340 nm in the same buffer, except for the addition of 50 jitm uqi and 10 mm kcn\n",
      "\n",
      "when several fixed concentrations of mgatp product were added in hexokinase reaction with a saturated con- centration of d-glucose 6-p and various concentrations of mgadp as substrates, double reciprocal plots of the rate of enzymatic activity vs mgadp concentrations be- come curvilinear\n",
      "\n",
      "for the enzymatic synthesis of +-camphene, 5-ml aliquots of the partially purified cyclase i preparation from sage containing 80-100 fig of pro- tein in 20 mm mes,4 5 mm potassium phosphate buffer, ph 65, con- taining 05 mm dithiothreitol, 15 mm mgclz and 10% v/v glycerol were employed\n",
      "\n",
      "the reaction mixtures were then purged with a stream of n3 to remove traces of ether, and 1 ml of 200 mm sodium acetate buffer, ph 50, containing 1 unit each of wheat germ acid phosphatase and potato apyrase, was added\n",
      "\n",
      "the optical purity of 12-oxo-pda formed from 12,13>s-eot declined when substrate concentration increased table i, indicating that the al- lene oxide cyclase activity was saturable with respect to substrate\n",
      "\n",
      "cells were lysed by sonication in lysis buffer 50 mm tris- cl, ph 78, 75 mm kc1, 1 mm dtt, 1 mm edta, 1 mm pmsf, and 001 mg/ml deoxyribonuclease i\n",
      "\n",
      "o l or • preassembled l8s8 was incubated at 25°c for 15 min in 01 m buffer hepes-naoh for ph 74 and 78 tris-hcl for ph 80, 825, 855, and 873 and ammediol-hcl for ph 90 and 94, 20 mm mgcl2,05 mm dtt, and 005 mg/ml carbonic anhydrase\n",
      "\n",
      "the column was first washed with 100 ml of the original extraction buffer, and the cyclase activity was then eluted with 200 ml of 100 mm khap04 ph 71 containing 5% sorbitol w/v and 05 mm dithiothreitol\n",
      "\n",
      "euglena gracilis absolutely requires thiamin for growth because it lacks a biosynthetic pathway for the formation of the pyrimidine moiety of thiamin 5, andposseses an active transport system for exogenous thiamin 6 it is transformed to the coenzyme by the respective thiamin pyrophosphokinase isozymes located in mitochondria, chloroplasts, and cytosol, operating for tpp-dependent enzymes such as 2-og decarboxylase, pyruvate dehy- drogenase, acetolactate synthase, and transketolase 7\n",
      "\n",
      "the molecular weight of 2-og de- carboxylase was estimated by using a column 15 x 90 cm of sepharose cl-6b equilibrated with 50 mm potassium phosphate buffer, ph 70, containing 1 mm tpp, 1 mm mgcis, 1 mm 2-mercaptoethanol, and 01 m kc1 and calibrated with blue dextran average molecular weight 2,000,000, catalase from beef liver 232,000, glucose oxidase from as- pergillus niger 186,000, lactate dehydrogenase from beef heart 140,000, malate dehydrogenase nad'1' from pig heart 70,000, and ovalbumin from egg white 43,000\n",
      "\n",
      "in ad- dition it shows that the increase in the k^ value was sim- ilar when cl in the added tris buffer was replaced by hcoß, was larger with n03 and was smaller with mes, isethionate, or acetate replacing the cf\n",
      "\n",
      "carbamoyl phosphate synthetase inhibition by anions 419 discussion although our results confirm that tris buffer inhibits carbamoyl phosphate synthetase competitively with ace- tyiglutamate 12, 13, they exonerate tris itself by dem- onstrating that the anionic component in this buffer plays the predominant role in the inhibition\n",
      "\n",
      "we also confirm 12 that k-phosphate buffers decrease considerably the ka value for acetylglutamate, but k^ appears to be in- volved in this decrease, since na or tris phosphate buffers are less effective\n",
      "\n",
      "binding of l-aspartic acid does, how- log ^ max 10-- 00-- -10 0-——0- 45 55 65 75 85 ph fig 5 ph dependence of the maximum velocity of the amination reaction of l-aspartase a at saturating mg24 10 mm with ammonia 0 or hydroxylamine d as the substrate or b in the absence of divalent metal ions with ammonia as the substrate •\n",
      "\n",
      "fractions eluting at ~180 mm kc1, containing predominantly a- and ^-pinene cyclase activity, were pooled 12 ml and concentrated to 05 ml by ultrafiltration centricon 30, amicon and the eluant buffer thereby replaced by dilution with 62 mm tris-hcl ph 68 containing 10 mm dithiothreitol, 10% v/v glycerol, and 01 mg/ml bromphenol blue\n",
      "\n",
      "glutathione transferase activity was measured spectrophotometrically at 340 nm in a reaction system containing 1 mm glutathione, 1 mm cdnb, 5% v/v ethanol solvent for cdnb, 1 mm edta, and 01 m sodium phosphate, ph 65, at 30°c\n",
      "\n",
      "immunoaffinity purified asparagine synthetase was diluted in phosphate- buffered saline pbs 10 mm phosphate, 150 mm nacl, ph 72, as indicated, and incubated overnight at 4°c in covered polyvinyl chloride 96-well microtiter plates at a concentration of 03 fag/well 125 ^i/well\n",
      "\n",
      "versiconal was prepared by incubation of 100 nmol of vha and 20 units of pork esterase in 5 ml of 20 mm potassium phosphate buffer, ph 74, for 60 min at 30°c\n",
      "\n",
      "versiconal cyclase activity was assayed by incubating 10 nmol of versiconal in 480 ill of 20 mm bistris-hcl buffer, ph 60, with 20 lu\n",
      "\n",
      "versiconal cyclase from aspergillus parasiticus 69 table i purification of versiconal cyclase purification step total volume ml total protein mg total activity nmol/min specific activity nmol/min/mg yield% purification -fold cell extract 320 1472 883 06 100 l nh42sc>4, 35-70% 95 874 699 08 79 13 de-52 65 150 374 25 42 42 hydroxyapatite 30 45 329 73 37 12 mono q hr 5/5 40 060 46 77 52 129 matrex green a 32 035 29 82 33 136 the enzyme solution eluted from the hydroxylapatite column was concentrated to a small volume on an amicon ultrafiltration cell with a pm30 membrane and dialyzed extensively against buffer c\n",
      "\n",
      "note added in proof the circular dichroism spectrum of vbc re- covered after incubation of versiconal with purified versiconal cyclase was measured in ethanol solvent\n",
      "\n",
      "pooled fractions 25 ml, 25 mg protein containing squalene oxide cyclase activity were dialyzed against 1 liter of buffer b and then applied to a second column of deae bio-gel a 15 x 115 cm equilibrated with the same buffer\n",
      "\n",
      "following anion-exchange chromatography mono q hr 10/10 pharmacia fplc, elution at ~180 mm kc1, fractions containing the pinene cyclase 12 ml were concentrated centricon 30, amicon and the buffer was changed by repeated dilution with 123 mm bistris-hcl ph 661 containing 20% v/v glycerol, 10 mm dtt, and 01 mg/1 bromophenol blue in preparation for electrophoresis\n",
      "\n",
      "cyclase assays were performed under linear conditions as described previously 5-7 using 12 ßm [l^hjgeranyl pyrophosphate in 100 jul assay buffer 50 mm hepes ph 78 containing 01 mm mncia, 5 mm dtt, and 10% glycerol with a 10-ml hexane overlay\n",
      "\n",
      "the enzyme undergoes dhfr, dihydrofolate reductase dtt, dithiothreitol iptg, isopropyi js-d-thiogalactopyranoside np-40, nonided p-40 ova, ovalbumin pbs, phosphate-buffered saline pcr, polymerase chain reaction sds- page, sodium dodecyi sulfate-polyacrylamide gel electrophoresis plp, pyridoxal 5'-phosphate\n",
      "\n",
      "further studies with the sorghum pathogen gloeocercospora sorghi 4 abbreviations used cht, cyanide hydratase hcn, hydrogen cya- nide dtt, dithiothreitol ief, isoelectric focusing igg, immunoglobulin g kda, kilodalton pbs, phosphate-buffered saline sds-page, sodium dodecyi sulfate-polyacrylamide gel electrophoresis pj, isoelectric point\n",
      "\n",
      "to date, 54 isolates from 37 species have been table iii the effect of heavy metal ions and thiol group-specific inhibitors on cyanide hydratase final inhibition compound concentration mm %° agnoa 1 15 znclz 5 52 cucl, 5 75 hg acetate l 100 iodoacetamide 5 0 p-chloromercuric benzoate l 100 ° the ions and inhibitors were preincubated in 50 mm phosphate buffer ph 75 with the deae-sephadex purified cht of gloeocercospora sorghi for 10 min before measuring cht activity\n",
      "\n",
      "however, although these nitrilases degrade a variety of complex aromatic and aliphatic nitriles, hcn is not listed as a substrate for the enzymes 11,12, while this compound is the only table iv immunoinhibition of cyanide hydratase by the 45-kda polypeptide-specific antibody and the antibody against enzymatically active cht° inhibition of i 3ht activity % amount antigen source of igg ng 05 1 2 5 8 10 45-kda polypeptide rabbit 1 0 0 8 36 98 98 45-kda polypeptide rabbit 2 0 0 8 36 86 100 deae-sephadex purified ch\" r rabbit 3 0 4 12 100 100 100 ° one microgram of deae-sephadex purified gloeocercospora sorghi cht 560 ^mol/min/mg was incubated with the antibody in 250 fil ted buffer for 30 min at 25°c, then assayed for cht activity\n",
      "\n",
      "c vinosum cells 60 g were suspended in 140 ml of memmb buffer 50 mm mops, 01 mm edta, 1 mm mgcia, 1 mm 3-mercaptoethanol, and 50 mm nahc03, ph 73 containing 1 mm p-tolylsulfonyl fluoride and 10 mg each of dnase i and rnase a\n",
      "\n",
      "the effect of ph on 4>s-limonene syn- thase activity was examined with a series of mes, pipes, mopso, hepes, and tes buffers of overlapping ph\n",
      "\n",
      "4s-limonene synthase had an absolute requirement for a divalent metal ion cofactor, either mg2'1' or mn2^ when divalent cations were removed by dialysis or com- plexation with edta, negligible enzyme activity was ob- served\n",
      "\n",
      "at lower concentra- tions 5-20 mm, sodium phosphate stimulated limonene cyclase activity two- to threefold over controls without phosphate perhaps by stabilizing the protein, and so was routinely added to assay buffers at 5 mm\n",
      "\n",
      "several amino acid-directed reagents shown to be active against other terpene cyclases 28, 51 had no effect on limonene synthase over a broad concentration range, including phenylmethylsulfonyl fluoride serine - directed and the arginine-specific reagents phenylglyoxal and 2,3-butanedione tested in tricine-borate buffer at ph 80 in the dark\n",
      "\n",
      "limonene synthase does not deviate from this pattern in being inhibited by both p- hydroxymercuribenzoate tgo = 035 ^m and n-ethyl- maleimide i so =18 mm when tested in buffer without dithiothreitol or sodium ascorbate\n",
      "\n",
      "pure isocitrate lyase can be stored without appreciable loss of activity at —70°c in 5 mm triethanolamine buffer containing 6 mm mgcia, 7 mm 2-mercaptoethanol, and 50% v/v glycerol, ph 76\n",
      "\n",
      "materials and methods enzyme assays the activity of isocitrate lyase ec 4131 was assayed spectropho- tometrically at 25°c in phosphate buffer ph 685 as described by dixon and kornberg 45\n",
      "\n",
      "most of the isocitrate lyase activity is found to adsorb onto the column, and can be eluted as a pure enzyme with buffer c containing 30 mm citrate\n",
      "\n",
      "the isocitrate lyase fraction is dialyzed against buffer d and concentrated to 4 mg/ml by dialysis against 20% w/v polyethylene glycol 10,000 in buffer d, using spectrapore membranes molecular mass cutoff 3500 da\n",
      "\n",
      "dilutions of standard glycoproteins fetuin and igg and of isocitrate lyase were prepared in 5 mm triethanolamine buffer con- taining 6 mm mgcia and 14 mm 2-mercaptoethanol\n",
      "\n",
      "they were rinsed thrice for 5 min with phosphate-buffered saline, incubated for 30 min in freshly prepared 10 mm naioi in phosphate-buffered saline, again rinsed thrice for 5 min with phosphate-buffered saline, and incubated for l h in avidin- hydrazide/biotin-peroxidase complex\n",
      "\n",
      "the latter was prepared by adding 1 vol of 15 ^g/ml avidin-hydrazide in phosphate-buffered saline con- taining 2% w/v lysozyme, to an equal volume of 75 ^g/ml biotin- peroxidase in phosphate-buffered saline containing 2% w/v lysozyme as well\n",
      "\n",
      "isocitrate lyase as a solution in 10 mm potassium phosphate buffer, ph 75, containing 14 mm 2-mer- captoethanol was precipitated with an equal volume of 20% w/v tri- chloroacetic acid, collected by centrifugation, and washed twice with 10% w/v trichloroacetic acid\n",
      "\n",
      "demetallization of isocitrate lyase maize isocitrate lyase 080 mg/ml was incubated during l h at 4°c in 80 mm triethanolamine buffer containing 14 mm 2-mercaptoethanol and 30 mm edta, ph 76\n",
      "\n",
      "maize isocitrate lyase does not bind to deae-cellulose under the conditions used 5 mm triethanolamine buffer containing 6 mm mgclg, 1 mm egta, 1 mm edta, 1 mm pmsf, and 14 mm 2-mercaptoethanol, ph 76\n",
      "\n",
      "remaining contaminating proteins, ie, es- sentially some malate dehydrogenase and a polypeptide with a molecular mass of 62 kda, can efficiently be re- moved by affinity chromatography on blue-sepharose in 5 mm triethanolamine buffer containing 6 mm mgcia, 1 mm edta, 1 mm pmsf, and 14 mm 2-mercaptoethanol, ph 76\n",
      "\n",
      "triethanolamine 80 mm containing 6 mm mgcia and 14 mm 2-mercaptoethanol apparently is the most appro- priate buffer to keep maize isocitrate lyase stable and ac- tive\n",
      "\n",
      "effect of metal ions on isocitrate lyase activity magnesium ions are absolutely essential for the activity of maize isocitrate lyase\n",
      "\n",
      "the ph dependency of isocitrate lyase activity was studied in three different buffers, ie, 50 mm phosphate, 100 mm triethanolamine, and 100 mm hepes fig 7\n",
      "\n",
      "a demetallized isocitrate lyase 08 mg/ml in 5 mm triethanolamine buffer containing 14 mm 2-mercaptoethanol, ph 76 was assayed in the presence of increasing concentrations of divalent cations mg24^ a, mn^ •, ba24^ 0, ca24^ d, and sr\" a\n",
      "\n",
      "b isocitrate lyase 08 mg/ml in the same buffer, but containing 6 mm mgcia was assayed in the solution for activity measurement, which was supplemented with various concentrations of cations added as chloride salts mn^ •, ba2-^ 0, ca\" d, and sr24- a\n",
      "\n",
      "pure isocitrate lyase can be stored without appreciable loss of activity at −70 °c in 5 mm triethanolamine buffer containing 6 mm mgcl2, 7 mm 2-mercaptoethanol, and 50%  v v  glycerol, ph 76\n",
      "\n",
      "cyclase activity was eluted with extraction buffer lacking nh,zso, and containing 01% v/v tween 20 polyoxyethylenesorbitan monolaurate and 20% v/v glycerol\n",
      "\n",
      "of enzyme extract was added to 10 ml assay buffer extraction buffer containing 100 mm kci and 1 mm mnc12 for lodgepole pine monoterpene synthases 50 mm mopso, 5 mm dithiothreitol, 20 mm mgc12, 10% v/v glycerol, ph 65, for sage pinene synthases together 50 mm mopso, 5 mat dithiothreitol, 20 mm mgc12, 10% v/v glycerol, ph 70, for --pinene synthase 50 mm mes, 5 mm dithiothreitol, 20 mm mgc12, 10% v/v glycerol, ph 60, for +-pinene synthase\n",
      "\n",
      "after incubation at ambient temperature for 30 min, the reaction was halted by the addition of 70 pl of 50 mm dithiothreitol in bistris buffer, and the reaction mixture was diluted to 2 ml with assay buffer and immediately assayed for cyclase activity as described above\n",
      "\n",
      "determination of oxaloacetate decarboxylase the standard incubation medium contained 5 mm oxaloacetate, 20 mm tris-hc1 buffer and 50 mm phosphate buffer, ph 72, 05 mm dithioerythritol, 5 mm sodium arsenite and either sonicated mitochondria, tissue extracts or the purified decarboxylase\n",
      "\n",
      "in order to check whether this mechanism applies to the mitochondrial oxalo- acetate decarboxylase, the purified enzyme was incubated with [14c]oxaloacetate, precipitated with trichloroacetic acid and the radioactivity in the precipitate was 1 77 table vi binding of [14c]oxaloacetate to oxaloacetate decarboxylase after borohydride reduction 20 nmoles of [u-14cli-aspartic acid 2 #ci were incubated with 10#moles of 2-oxoglutarate and l0/~g of crystalline glutamate-oxaloacetate transaminase in 05 ml of 01 m phosphate buffer, pig 74, for 5 min at 30 °c\n",
      "\n",
      "succinate-cyto- chrome c reductase as prepared was dialyzed against 50 mm phosphate buffer, ph 74, containing 025 m sucrose overnight with two changes of buffer and frozen at --70°c until used\n",
      "\n",
      "succinate~ytochrome c reductase, 10 mg per ml, was centrifuged at 170 000 x g for 45 rain and the collected precipitate was resuspended in 50 mm phosphate-borate buffer, ph 78, to a protein concentration of 15 mg/ml and incubated with 20 mm succinate in an argon atomosphere in the presence of i mm dithiothreitol for 30 rain before the ph of the solution was brought to 100 by anaerobic addition of n naoh\n",
      "\n",
      "the precipitate was suspended in 50 mm tris-hc1 buffer, ph 80, containing 067 m sucrose and kept at --70°c for the isolation of ubiquinone-binding protein and b~i hi complex ubiquinol-cytochrome c reductase\n",
      "\n",
      "the calcium phosphate gel precipitate was washed with an argon saturated solution containing 10 mm succinate and 1 mm dithiothreitol before the succinate dehydrogenase was eluted by 01 m phosphate buffer\n",
      "\n",
      "separation of ubiquinonebinding protein and b-c1 iii complex ubiquinol- cytochrome c reductase from the b-cl complex the frozen b-c~ complex was thawed and protein concentration adjusted to 20 mg/ml with 50 mm tris-hc1 buffer, ph 80, containing 067 m sucrose\n",
      "\n",
      "elecwophoretic profiles of t novel|us cytochrome oxidase in polyacrylamide gel, electrophoresls was performed in the polyacrylamide gel monomer, 5% which had been equilibrated with 01 m tris- acetate buffer, ph 80, containing 05% triton x-100\n",
      "\n",
      "the cholate-dialysis method was performed as follows 53 mg of the delipidated ubiquinol-cytochrome c reductase in 50 rf1m na/k phosphate buffer ph 74, containing 10% glycerol was mixed with the indicated concentraiton of the phospholipid micellar solution 20 mg/ml in 50 mm phosphate buffer, ph 74 and sodium cholate\n",
      "\n",
      "the cholesterol-containing asolectin-reductase vesicle was prepared by mixing the delipidated ubiquinone-cytochrome c reductase with a sonicated mixture of asolectin and cholesterol 1 mol cholesterol per 2 mol asolectin in the presence of 1 % cholate, followed by extensive dialysis against 50 mm phosphate buffer ph 74\n",
      "\n",
      "delipidated reductase 15 mg/ml, in 50 mm phosphate buffer ph 74, containing 10% glycerol, was treated with protease from streptomyces for 30 min at room temperature\n",
      "\n",
      "2-ml aliquots of the delipidated reductase, 20 mg/ml in 50 mm sodium potassium phosphate buffer ph 74 were incubated with the indicated concentrations of asolectin in the presence of 1% sodium cholate for 10 min and then diluted with 8 ml of phosphate buffer\n",
      "\n",
      "the reductase-azolectin com plexes were collected as precipitates and redissolved in 04 ml of 50 mm phosphate buffer containing 05`,6 sodium cholate\n",
      "\n",
      "the dsc thermograms of the delipidated ubiquinol-cytochrome c reductase reconstituted with phosphatidylcholine containing different unsaturated fatty acyl groups\n",
      "\n",
      "the delipidated reductase in 50 mm sodium potassium phosphate buffer ph 74 was reconstituted with dppc  , popc  \n",
      "\n",
      "when different kinds of phospholipids, such as dopg, dopc and dopa, which possess the same unsaturated fatty-acyl groups, were incorporated into the delipidated reductase, the resulting phospholipid-reductase vesicles varied in their thermotropic properties see table ii\n",
      "\n",
      "and keto isomers content of oxaloacetate the relative content of enol and ketone in different solvents was measured as described above in a coupled maiate dehydrogenase assay\n",
      "\n",
      "14 shown in fig 2, an additional peak oi' activity oxaloaeatate tautomerase-2 appeared, the frac- tions containing an active protein were combined, concentrated in a high pressure filtration device and dialyzed against 200 ml of a buffer containing 5 mm potassium phosphate, 02 mm edta, 1 mm ditiuothreitol ph 72 for 3 h with two changes of the buffer\n",
      "\n",
      "the initial rates of gautomeric intereonversion of 64 pm oxaloacetate were measured in the coupled mutate dehy- drogenase assay ai different ph in 2 mm tris/ci- buffer containing 02 mm edta\n",
      "\n",
      "briefly, after the extraction of ubiquinone by 3 ml hexane and following methanol wash, the extracts were analyzed by hplc equipped with c-18 reversed phase column chemcosorb, 46 x 250 mm, chemco, osaka using linear gradient from 70% of tetrahydrofuran/ acetonitrile = 1/1, v/v and 30% of methanol/water = 3/1, v/v to 100% of the former solvent with the increase of 1% per min\n",
      "\n",
      "the active fractions were combined, and then the ubiquinol oxidase was precipitated with 50% saturated nh42so4 deae- sephacel fraction\n",
      "\n",
      "the spheroplast suspension was transferred directly into 100-150 volumes of 50 mm sodium phosphate buffer ph 66 containing 50 /zg/ml pyruvate kinase and 5 mm adp to meet the requirements for the atp-re- generating system\n",
      "\n",
      "50 /xl of the super- natant, 50 /~1 of the luciferase/luciferin extract and 900 /zl 20 mm glycylglycine-buffer was used for the final atp determination\n",
      "\n",
      "nad + 56 ram, 48 mm deuterated ethanol-d6 aldrich, 99 + atom% 2h, 147 mg 25 units horse-liver alcohol dehy- drogenase and 29 mg 17 units yeast aldehyde dehydro- genase were added to 6 mm ches-koh buffer ph 90, final volume 50 ml, at 25° c\n",
      "\n",
      "nadp+ 45 mm, 50 mm atp, 60 mm deuterated o-glucose-l-d aldrich, 98 atom % 2h, 10 mm mgso4, 013 mg 35 units yeast hexoki- nase and 0014 mg 4 units yeast glucose-6-phosphate dehydrogenase were added to 6 mm tris-hc1 buffer ph 80, final volume 50 ml, at 25° c\n",
      "\n",
      "the deaminonadh-db reductase activity of ndh-1 was measured as described [12] in reaction medium containing 50 mm phosphate buffer ph 74, 25 mm kcn, 5 mm mgso4 and 10 /xm db, the final protein concentration being 15 /xg/ ml\n",
      "\n",
      "the deaminonadh-db reductase activity of ndh- 1 was mea- sured as described [12] in a reaction medium consisting of 50 mm phosphate buffer ph 74, 025 m sucrose, 2 mm kcn, 1 mm mgci2, 2 /~m antimycin a and 40 /zm db, the final protein concentration being 15 /xg/ml\n",
      "\n",
      "more than 50 % of added ouabain remained unbound in the medium at ouabain concentrations approximately one tenth of the saturating concentration for specific binding sites on the na+ + k+-activated atpase preparation\n",
      "\n",
      "na+ + k+-activated atpase preparation, containing 20 mg of protein, was suspended in io ml of a 5o-mm tris-hc1 buffer ph 75 and centrifuged at ioo ooo x g for 30 min\n",
      "\n",
      "assay of 3-octaprenyl-4-hydroxybenzoate carboxy-lyase the incubation mixture consisted of enzyme preparation 02 ml sodium phosphate buffer 005 m, ph 70, and additional co-factors as stated, in a total volume of 045 ml\n",
      "\n",
      "preliminary studies on unfractionated cell extracts experiments with unfractionated cell extracts showed that either tris - hcl or sodium phosphate buffers could be used in the assay of the carboxy-lyase and that, in phosphate buffer, the maximum enzymic activity was achieved with 005-01 m sodium phosphate buffer ph 70\n",
      "\n",
      "to decrease the share of contaminating calcium powerful calcium-egta, buffer was inchtded in the medium 045 mm \"5ctci2, 50-100 cpm per nmol and 045 mm egta\n",
      "\n",
      "ca\"- or edta-activated atpase activity was determined by measuring the amount of pi released in 15 h in 05 m kci, ph 76 20 mm histidine buffer, in the presence of 5 mm cac12 or 1 mm edta at 37° c unless otherwise noted\n",
      "\n",
      "determination of mgt+atpase was carried out in 0 or 005 m kcl/25 mm tris-maleate buffer ph 70 in the presence of 1 mm mgclz at 37°c for 15 h atp concentration was 2 mm\n",
      "\n",
      "the atpase activity was measured in 20 mm trls-msleate buffer at ph values 60-100 containing 05 m kcl in the presence of 1 told edta n or b mm cac12 • or 005 m kcl is the presence of 1 tom mgclz •\n",
      "\n",
      "the atpase activity was measured at various ph values 20 mm tris-maleate buffer in 05 m kci in the presence of edta a or caciz a and 0 m kcl in the presence of mgc12 • at 37°c for 3 h\n",
      "\n",
      "at timed intervals, samples were taken, the cells pelleted 5 min, 1500 × g, resus- pended in buffer b 90 m kci, 45 mm naci, 22 mm sucrose, 10 mm glucose, 10 mm caci2, 025 mm mgci2 and 30 mm hepes ph 74 at 5% hematocrit and incubated for either 5 min with pai-116-ampa 60 iu 125 #g per 400 #1 packed cells or 60 rain with bee venom phospholipase a 70 iu per 400 #1 packed cells\n",
      "\n",
      "the final cell pellet was homogenized in hm buffer contain- ing proteinase inhibitors 1 mm phenylmethylsulphanyl fluoride, 1 ~tg/ml pepstatin and 22 /xg/ml leupeptin referred to as hmi buffer\n",
      "\n",
      "yashphe etal assay of meso-tartrate dehydratase enzyme assays on whole cells were performed as previously described~2 with the substitution of 005 m tris buffer ph 82 for the phosphate buffer used earlier\n",
      "\n",
      "after cooling, 25ffmoles glycine-naoh buffer ph lo2 and 25 mg of crude arginase prepared from beef liver were added to a i-ml aliquot of the reaction mixture in a total voi\n",
      "\n",
      "it is known, however, that amino acids may be racemized indirectly by the combined actions of d- and l-amino-acid transaminases and alanine racemase or another amino acid racemase as follows d-arginine + pyruvate ~ d-alanine + a-keto-d-guanidinovalerate d-alanine ,~- l-alanine l-alanine + a-keto-d-guanidinovaierate ~ l-arginine + pyruvate sum n-arginine ~- l-arginine alanine and glutamate racemases were assayed by incubating 4#moles l- alanine or sodium d-glutamate, respectively, ioo #moles tris-hc1 buffer ph 80 with 13 mg protein in a total vol of 20 ml\n",
      "\n",
      "a ssay unless otherwise indicated, the reaction mixture contained 3\" io-2 #moles of unlabelled or labelled porphobilinogen 30 ooo counts/rain, 4,moles of tris-hc1 buffer ph 82, and enzyme preparation fractions 4o, 45 or 5o, see below or a mixture of porphobilinogen deaminase 05 mg of protein and uroporphyrinogen iii cosynthetase 07 mg of protein\n",
      "\n",
      "in order to prepare porphobilinogen deaminase, an aliquot of fraction 4or 45 was diluted to a protein concentration of io mg/ml after addition of sufficient 2 m tris buffer ph 82, to make the solution oi m in buffer\n",
      "\n",
      "ratio protein/ supernatant porphobili- % of total porphyrins alumina cv solution or nogen con- formed rag wash sumption as % of activity uroporphy- uroporphy- of original rin i rin iii extract i  025 supernatant 68 75 25 005 m buffer 31 25 75 005 m nac1 o 25 75 effect of cosynthetase concentration on uroporphyrinogen iii formation the amount of uroporphyrinogen iii formed when the deaminase and porpho- 7~ c~ c 25 # 14 2'1 cosynthetase mg ot 13rotein fig i\n",
      "\n",
      "incubation mixtures o---o, 0-5 mg porphobilinogen deaminase, 3\" io 2 #mole [~4c]porphobilinogen, 4ffmoles tris-hc1 buffer ph 82, final volume 2ooffl x x and o , as o--o, plus 07 mg uroporphyrinogen iii cosynthetase\n",
      "\n",
      "biophys dcta, 167 1968 414 421 oxalate decarboxylase of a niger 417 results stability of the enzyme decarboxylase dissolved in acetate 02 m buffer ph 56 did not lose its activity for several months when kept at o-3freezing of this solution caused a partial inac- tivation of the enzyme\n",
      "\n",
      "malate dehydrogenase ec 11137, glutamate dehydrogenase ec 1412, cytochrome c oxidase ec 1931, rotenone-insensitive nadh-cytochrome c reduct- ase, aspartate aminotransferase ec 2611, adenylate kinase ec 2611, acid phosphatase ec 3132 and glucose-6-phosphatase ec 3139 were assayed as de- biochim biophys acta, 178 1969 137 144 malonyl-coa decarboxylase and propionyl-coa carboxylase i~9 scribed in refs 34 see also ref 35, 36, 33, 37, 38, 34, 28 and 28, respectively, with the following modifications i 15 #m rotenone was present in the assays of malate de- hydrogenase and aspartate aminotransferase 2 the medium used for the assay of glutamate dehydrogenase contained 25 mm potassium phosphate buffer ph 75, 5 mm nh,c1, oi mm nadh, 09 #m antimycin and io mm a-oxoglutarate\n",
      "\n",
      "also in contrast to the behaviour of adenylate kinaso7,48, propionyl-coa carboxylase is not extracted from mitochondria by oi m phosphate buffer ph 72\n",
      "\n",
      "since sufficient crystalline enzyme was not available for molecular-weight determinations by methods involving schlieren optics, an estimation of the molecular weight was obtained by centrifugation in a sucrose density gradient by the lnethod of marti~ and a~i~s1 l-arabinose isomerase ooi nil was layered on one gradient and o5 mg of catalase in ooi ml buffer was layered on a second gradient\n",
      "\n",
      "the activity of &aminolevulinate dehydratase was assayed essentially by the method of gibson et al~ the assay mixture contained potassium phosphate buffer ph 7o o3 mmole 6-aminolevulinic acid hydrochloride, io #moles enzyme and any additional ingredients, plus distilled water, to a final vol of 3o ml\n",
      "\n",
      "coli a tcc 9637for assay of dihydrodipicolinate synthase during late exponential phase growth, the cells were centrifuged, washed once and resuspended prior to breakage in 002 m tris buffer ph 7-4\n",
      "\n",
      "serial jl 272805 291210 291847 291857 31 biochimica et biophysica acta bba - enzymology biochimicaetbiophysicaactabbaenzymology 2003-01-28 2003-02-05 2010-04-03t090732 1-s20-0005274469902964 0005-27446990296-4 0005274469902964 101016/0005-27446990296-4 s350 s3502 head-and-tail 1-s20-s0005274400x00551 2015-05-13t194848823392-0400 0 0 19690708 1969 2003-01-28t000000z rawtext articleinfo crossmark dco dateupdated tomb dateloaded datesearch indexeddate issuelist volumelist yearnav absattachment articletitlenorm authfirstinitialnorm authfirstsurnamenorm cid cids contenttype copyright dateloadedtxt docsubtype doctype doi eid ewtransactionid hubeid issfirst issn issnnorm itemstage itemtransactionid itemweight openaccess openarchive pg pgfirst pglast pii piinorm pubdatestart pubdatetxt pubyr sortorder srctitle srctitlenorm srctype volfirst volissue webpdf webpdfpagecount affil articletitle auth authfirstini authfull authlast primabst ref alllist content subj ssids 0005-2744 00052744 185 185 1 1 volume 185, issue 1 22 220 227 220 227 19690708 8 july 1969 1969-07-08 1969 converted-article fla copyright © 1969 published by elsevier bv anionactivationmaleatehydratase britten j taggart 1962 141 j englard 1967 2255 s malmstrom 1959 131 b gornall 1949 751 a britten 1968 330 j lineweaver 1934 658 h alberty 1956 1 r dixon 1964 116 145 m enzymes dixon 1964 429 443 m enzymes myrback 1926 1 k massey 1953 67 v alberty 1954 2494 r dixon 1964 324 m enzymes brittenx1969x220 brittenx1969x220x227 brittenx1969x220xj brittenx1969x220x227xj item 0005-27446990296-4 0005274469902964 1-s20-0005274469902964 101016/0005-27446990296-4 272805 2010-09-27t181400772927-0400 1969-07-08 1-s20-0005274469902964-mainpdf https//s3amazonawscom/prod-ucs-content-store-us-east/content/pii0005274469902964/main/application/pdf/27212027876282c28a45e3d267995fde/mainpdf https//s3-eu-west-1amazonawscom/prod-ucs-content-store-eu-west/content/pii0005274469902964/main/application/pdf/27212027876282c28a45e3d267995fde/mainpdf mainpdf pdf true 441433 main 8 1-s20-0005274469902964-main_1png https//s3-eu-west-1amazonawscom/prod-ucs-content-store-eu-west/content/pii0005274469902964/preview/image/png/2f2e5e5e0cfa93a76192786d9268e756/main_1png https//s3amazonawscom/prod-ucs-content-store-us-east/content/pii0005274469902964/preview/image/png/2f2e5e5e0cfa93a76192786d9268e756/main_1png main_1png png 47037 849 656 image-web-pdf 1 1-s20-0005274469902964-si2gif https//s3amazonawscom/prod-ucs-content-store-us-east/content/pii0005274469902964/stripin/image/gif/84477dca2b9aa9a94c61d98f3ca95c49/si2gif https//s3-eu-west-1amazonawscom/prod-ucs-content-store-eu-west/content/pii0005274469902964/stripin/image/gif/84477dca2b9aa9a94c61d98f3ca95c49/si2gif si2 true si2gif gif 982 30 224 altimg 1-s20-0005274469902964-si1gif https//s3amazonawscom/prod-ucs-content-store-us-east/content/pii0005274469902964/stripin/image/gif/99ec8781def09400355b032df9edbf51/si1gif https//s3-eu-west-1amazonawscom/prod-ucs-content-store-eu-west/content/pii0005274469902964/stripin/image/gif/99ec8781def09400355b032df9edbf51/si1gif si1 true si1gif gif 1464 15 539 altimg 220 biochimica et biopfiysica acta bba 65911 anion activation of maleate hydratase j s britten, h morell ann john v taggart department of physiology, college of physicians and surgeons, columbia university, new york, ny 10032 usa received february 3rd, 1969 summary maleate hydratase d-malate hydro-lyase has been found to have an absolute requirement for certain univalent anions\n",
      "\n",
      "results the effect of various sodium salts on the activity of maleate hydratase previ- ously dialyzed against oo2 m potassium phosphate buffer ph 74 is shown in fig i\n",
      "\n",
      "biochim biophys acta, 185 1969 220-227 bbaee 69902964 0005-27446990296-4 101016/0005-27446990296-4 anion activation of maleate hydratase js britten h morell john v taggart department of physiology, college of physicians and surgeons, columbia university, new york, ny 10032 usa abstract maleate hydratase d-malate hydro-lyase has been found to have an absolute requirement for certain univalent anions\n",
      "\n",
      "deae-cellulose column eluates l 0 1260 78 \"4 optimal ph the activity of porphyrinogen carboxy-lyase at different ph values was measured in 67 mm potassium phosphate buffer ranging from ph 65 to 77\n",
      "\n",
      "the reaction mixture contained 8o ,umoles sodium phosphate buffer ph 74  o1 ,umoles nadh 003 ml yeast alcohol dehydrogenase 110 dilution 5 ,umoles deoxyribose 5-phosphate and, 01-03 ml enzyme in a final volume of 1 ml\n",
      "\n",
      "the reaction mixture contained 8o moles sodium phosphate buffer ph 74  031 mg enzyme 003 ml nadh 2 mg/ml  003 ml yeast alcohol dehydrogenase 1io dilution  and, oo82-328 /cmoles deoxyribose 5-phosphate in a final volume of io ml\n",
      "\n",
      "evidence for the stoichiometric conversion of cystathionine to homocysteine was obtained by incubation of [1'c4]cystathionine with cystathionase, followed by paper chromatography of the amino acids, after performic oxidation, in solvent b\n",
      "\n",
      "the dehydratase was eluted with a linear gradient of ooi to oi m potassium phosphate buffered saline, ph 75, containing oor m mercaptoethanol\n",
      "\n",
      "fractionation of semipurified cystathionine ~-synthase from rat liver on biogel p-2oo colunm 23 cm 4o cm, in potassium phosphate buffer ooi m, ph 78\n",
      "\n",
      "r waygood activity was inhibited by tris buffer which had no inhibitory effect on glyoxylate carboxy-lyase from hydrogenomonas hi6 ref 13\n",
      "\n",
      "a sample of the gdp-d-mannose oxidoreductase eluted from hydroxylapatite 35 mg protein was dissolved in i ml 005 m sodium phosphate-oi m nac1 buffer, ph 70, and dialyzed overnight against i 1 of the same buffer\n",
      "\n",
      "a mixture containing 2o #1 of the purified oxidoreductase 8/~g protein in oo25 m sodium phosphate buffer, ph 7o, and ooo2/,mole ooi5#ci gdp-d-mannose labeled with a4c in the d-mannose portion in a total volume of 3o #1 was incubated at 37 for 6o rain\n",
      "\n",
      "biochim biophys acta, 276 1972 85-93 gdp-d-mannose oxidoreductase 91 table iii the effect of pyridine nucleotide on the activity of gdp-d-mannose oxidoreductase the reaction mixtures each contained 20/~1 of purified oxidoreductase 8/~g protein in 0025 m sodium phosphate buffer, ph 70, 20 nmoles 5 nci gdp-d-[14clmannose and oi/~mole of the indicated pyridine nucleotide in a total volume of 3lul\n",
      "\n",
      "table iv the effect of nucleotides on the activity of gdp-d-mannose oxldoreductase the reaction mixtures each contained 2o/*1 of purified oxidoreductase 8/2g protein in oo25 m sodium phosphate buffer, ph 70, 20 nmoles 5 nci gdp-d-~l*c]mannose and a nucleotide in the concentration indicated in a total volume of 3~1\n",
      "\n",
      "in a previous paper4, the effects of different sulfhydryl reagents on the activity and quaternary structure of chicken pyruvate carboxylase were studied in a buffer solution containing sulfate and phosphate ions\n",
      "\n",
      "the present paper shows that the rates and patterns of inactivation by n- ethylmaleimide and eystine, and the reactivity of the sulthydryl groups of pyruvate carboxylase, depend markedly on the salt composition of the buffer solution used\n",
      "\n",
      "pyruvate carboxylase was assayed speetrophotometrically in the direction of c02 fixation with the coupled malate dehydrogenase system, as described previously1 either tris-h2so4 or tris-hc1 was used as a buffer\n",
      "\n",
      "results inactivation by n-ethylmaleimide and cystine in the presence of different inorganic salts n-ethylmaleimide was shown to inactivate pyruvate carboxylase in two steps in a buffer solution containing nh42so~ and potassium phosphatea\n",
      "\n",
      "the decrease in enzymatic activity that takes place upon partial modification of pyruvate carboxylase by n-ethylmaleimide in sulfate buffer solution, could be due to either a change in the km values for the substrates or to an effect on v\n",
      "\n",
      "these apparent km values agree reasonably well with those determined previously for the native enzymet,~ the v of n-ethylmaleimide-32-pyruvate carboxylase sulfate is about 5o% of that of the native enzyme @ table v, as determined from the position of the \"break\" in the semilogarithmic plot of enzymatic activity versus time of incubation with n- ethylmaleimide, using saturating concentrations of substrates and acetyl coa in the activity assay mixture\n",
      "\n",
      "sedime~ztation patterns of the inactivated preparations the study of the sedimentation velocity patterns of pyruvate carboxylase inactivated by each of the two sh-group reagents, in all three buffer solutions chosen, was important in the construction of a unified scheme showing the effect of modifica- tion on the activity and quaternary structure of the enzyme\n",
      "\n",
      "in the sulfate buffer solu- tion, oxaloacetate increased the rate of cystine inactivation, as in the second phase of the n-ethylmaleimide inactivation  however, after the addition of cystine there was a period of time in which no increase in the rate of inactivation took place fig 4b\n",
      "\n",
      "concentration dependence of the protection by acetyl-coa and a tp against inactivation acetyl-coa protects pyruvate carboxylase completely against inactivation by n-ethylmaleimide or cystine in chloride buffer solution\n",
      "\n",
      "atp gives only partial protection against the inactivation of pyruvate carboxy- lase by n-ethylmaleimide or cystine in chloride buffer solution\n",
      "\n",
      "e neet correlation of the binding of n-ethylmaleimide and cystine with the loss of enzymatic activity fig 6a shows the binding of n-ethylmaleimide and the increased inactivation with time for a preparation of pyruvate carboxylase in sulfate buffer solution\n",
      "\n",
      "a sulfate buffer solution pyruvate carboxylase spec\n",
      "\n",
      "the extent of inactivation of pyruvate carboxylase by l-cystine, and the num- ber of sh groups modified at different times in the sulfate buffer solution are shown in fig 7a\n",
      "\n",
      "the reaction mixtures contained, in addition to sulfate buffer solution  pyruvate carboxylase o61 mg/ml spec\n",
      "\n",
      "a similar relationship between the reaction of the sh groups of pyruvate carboxylase and its inactivation by eystine was obtained in the chloride buffer solu- tion\n",
      "\n",
      "dependence of the enzymatic activity of different modified pyruvate carboxylases on acetyl- coa concentration the forms of pyruvate carboxylase obtained by partial modification with n- ethylmaleimide, in the three standard buffer solutions, present kinetics of activation by acetyl-coa different from that of the native enzyme\n",
      "\n",
      "the values for ka, n and v, ob- tained with the enzyme modified by n-ethylmaleimide in phosphate buffer solution, n-ethylmaleimide-32-pyruvate carboxylase phosphate, were similar to those of n- ethylmaleimide-32-pyruvate carboxylase sulfate table v and are, therefore, prob- ably the same molecular species\n",
      "\n",
      "the enzyme modified by n-ethylnmleimide in chloride buffer solution, n-ethylmaleimide-i6-pyruvate carboxylase chloride, pre- table v dependence of enzymatic activity on acetyl-coi~ concentration for different modi- fied pyruvate carboxylases the experiments were conducted as described in lqg\n",
      "\n",
      "the lnodified enzymes ni'm-32-pc phosphate and nem-32-pc sulfate were prepared by incubating native pyruvate carboxylase for 6-16 hours with nem, at a molar ratio of nem to enzyme equal to 3~, in \"phosphate\" and \"sulfate buffer solution\", respectively\n",
      "\n",
      "during the first phase of modification, between 4 and 32 sulfhydryls react, and this reaction takes place with either partial n-ethylmaleimide in phosphate or sulfate buffer solutions or no inactivation, and with no gross change in the sedimentation properties of the enzyme species designated as 13 in fig 9\n",
      "\n",
      "reagent buffer solution phase i phase i1 n-ethylmaleimide phosphate 332 not determined sulfate 33-' not determined chloride 16 39 l-cystine sulfate 4 6 16 chloride -t 6 16 behavior with respect to acetyl-coa table v\n",
      "\n",
      "fractionation of subcellular components of the liver mitochondria, microsomes and a soluble fraction were prepared by tile proce- dure of hogeboom33, except that the 025 m sucrose solution contained io mm sodium phosphate buffer ph 72, 02 mm pyridoxal phosphate and 5 mm dithiothreitol to prevent inactivation of ornithine decarboxylase, and that microsomes were precipi- tated by eentrifugation at lo5 ooo g for 60 rain\n",
      "\n",
      "nuclei and mitochondria were then disrupted by homogenization withglass powder in io mm sodium phosphate buffer ph 7o, containing 02 mm pyridoxal phosphate and 5 mm dithiothreitol, using a glass homogenizer and the homogenate was centrifuged at io ooo g for 20 rain, assay for ornithine decarboxylase activity tile activity of ornithine decarboxylase was determined by measuring the liberation of 1~co2 from l-ii-14cjornithine by a modification of the method of russell and snyder5\n",
      "\n",
      "of the various buffer systems tested in our assay, hepes was the best buffer system for obtaining maximal catalytic activity of the tryptophan synthase of neurospora\n",
      "\n",
      "all kinetic runs were carried out in 003 m phosphate buffer of ph 7 containing 16% of methanol using 3 times crystallized steroid 4-isomerase of specific activity 64 300 units/mg protein, which had been diluted by a factor of up to io6 with i % bovine serum albumin of ph 7\n",
      "\n",
      "the affinity of the purified epimerase for cmp-nana increased as it became more saturated with the inhibitor resulting in a sigmoidal shape curve when 100 8o z o 60 i- z 42 b 2c 01 02 03 04 01 015 06 001 10 01 0'2 o13 cmp--nana ram cmp--nana ram fig 3\n",
      "\n",
      "a 15 cm i4 cm column was packed with medium- grained deae-cellulose which was equilibrated with 002 m potassium phosphate buffer ph 74- 075 ml of the enzyme solution containing 32 mg of protein and 05 unit of total epimerase was absorbed in the column\n",
      "\n",
      "epimerase a and b, when separated on the gel and remixed in equal volumes, could again be separated on a similar deae column which was now equilibrated against 0075 m potassium phosphate buffer ph 74\n",
      "\n",
      "enzyme nad+ io4 udpg formed  m   nmoles  epimerase a epimerase b m o2 143 io 321 5o 326 -- m 02 123 io 261 50 293 requirement of nad and inhibition by ump maxwelp had shown that, unlike the yeast epimerase, the calf liver enzyme has an absolute requirement for nad added from the outside\n",
      "\n",
      "the molecular weight of malonyl-coa decarboxylase in o15 m phosphate buffer is estimated to be 25o ooo, with a possible error of lo% fig 2\n",
      "\n",
      "the effect of phosphate on the enzyme activity potassium phosphate buffer ph 7o stimulates the enzyme activity fig 3- between 40 and 200 mm phosphate the reaction velocity, measured at a high con- e e v 200 -o 100 ~o d t~ e 0 f ,-~k , , ' ' li \" 100 200 800 phosphate ram fig 3- the effect of phosphate on the activity of isolated malonyl-coa decarboxylase 42/~g/ml\n",
      "\n",
      "the column was washed with 30 ml of equilibrating buffer and 2'-epimerase activity was eluted from the column with 20 mm sodium phosphate ph 75-oi m kci-oi mm udp\n",
      "\n",
      "a tp mg~+ l, moles l~rnoles i 3 i 65 3 3 i00 3 6 75 2 i 3 6o 3 3 ioo 6 3 78 relative enzyme activity in the phosphate buffer normally used a three-fold increase in the specific activity of p-coumarate coa ligase was achieved by the mnc12 fractionation\n",
      "\n",
      "standard enzyme assay conditions adenylosuccinate synthetase is assayed by the method of lieberman4, modified as follows the reaction mixture o5 ml contains 140 mx glycine buffer ph 8, 6 mm mgc12, 4 mm imp, 03 mm gtp sodium salt, 40 mm sodium aspartate, o6 mm phosphoenolpyruvate, 15 units of pyruvate kinase and 5,u1 about 8oo,ug of protein of the partially purified enzyme\n",
      "\n",
      "measurements of epimerase rates at 290 nm ref i i were carried out in 50 mm triethanolamine-hc1 buffer, in the presence of equilibrium concentrations of ribulose-5-p generated by the addition of 22 units of isomerase to various amounts of ribose-5-p\n",
      "\n",
      "for measurements of epimerase rates at 34m, 20 ml of solution contained 50 mm glycylglycine buffer, 25 mm mgc12, i mm dithiothreitol, oi mm thiamine pyrophosphate, o13 mm nadh, 03 units of glycerol-l-phosphate dehydrogenase ec iii8, 16 units triosephosphate isomerase ec 53ii, 02 units of trans- ketolase ec 2211 and a mixture of ribose-5-p and ribulose-5-p or ribose-5-p and a ribose-5-p/ribulose-5-p equilibrium mixture to give a fixed concentration of 15 mm ribose-5-p and variable concentrations of ribulose-5-p\n",
      "\n",
      "chromatography of crude moose carbonic anhydrase ethanol chloroform fraction on a deae-cellulose column 28 cm 45 cm equilibrated with ooi m tris-hc1 buffer ph 87\n",
      "\n",
      "7-oxl-5\"-p 7-nbosyloxlpurmol 5'-phosphate, 7-rlbosyl-4,6-dlhydroxypyrazolo[3,4-d]pyrxmldme 5'-phosphate, 6-nh2-1-alo-5'-p 6-nh2-1-rlbosylallopunnol 5'-phosphate, 1-rlbosyl-4-hydroxy-6-amlnopyrazolo[3,4-d]pyrlmldme 5'-phosphate, 3-xmp 3-rlbosylxanthoslne 5'-phosphate 586 0 0 hn n n h h h xanth in e oxipu r i nol  4,6 - dihydroxypyrazoloi5,4 -d] pyrimi dine  fig 1 basic ring structures and numbering systems for xanthme and xts analogs materials and methods enzyme pur~catton and assays gmp synthetase was purified from e colt b-96 atcc no 13473 according to the method of sakamoto et al [4] the specific activity was similar to the reported value assays were performed by momtormg the formation of amp according to a modification of the coupled enzyme system of kornberg and pricer [11] in this assay the production of 1 #mole of gmp is accompanied by the production of 1 #mole of amp and the concomitant oxidation of two /~moles of nadh ae340 nm= 12 44 mm-1 cm-1 for the standard reaction, the cuvettes contained 75 mm trls-hci buffer ph 8 5, 0 125 mm xmp, 1 25 mm atp, 60 mm nh42so4, 10 mm mgso4 the following coupling reagents were also present 0025 mg/ml of myokmase, 0 025 mg/ml of pyruvate kmase, 0 025 mg/ml of lactate dehydrogenase, 0 5 mm phosphoenolpyruvate, 0 2 mm nadh reaction velocities were monitored on the 0-0 1 absorbance scale of a gllford recording spectrophotometer at 37 due to the high absorbance of 6-thloxmp and 6-thlogmp at 340 nm, reactions with these com- pounds were monitored at 360 nm ae = 8 1 mm-1 cm-1 where 6-th~oxmp and 6-thlogmp were found to be lsosbestlc e = 1 1 mm-1 cm-1 velocities measured at 340 or 360 nm were found to be hnear with respect to time following a brief lag period and concentration of gmp synthetase molar velocities determined in th~s manner agreed within 10 ~ of those determined by the direct spectral assay of moyed and magasamk [3] assay b, and a radlochemlcal assay using [8-14c]xmp performed under identical conditions, but omitting the coupling reagents reactions revolving 14c-labeled compounds were terminated wlth hc1 0 4 m final concn, and the nbo- nucleot~des were hydrolyzed to their corresponding bases by heating the resulting solutions at 100 m sealed vials for 1 h the complete hydrolysm of 3-[8j4c]xmp reqmred 2 3 m hc1 and heating for 3 h the reaction m~xtures were then spotted on whatman 3 mm paper and chromatographed m an ascending direction m water saturated butanol-nh4oh 99 1, v/v for 17 h the spots containing the uv absorbing bases rv values xanthme, 0 03, guanine, 0 13, 4,6-dlhydroxypyrazolo-[3,4-d]-pyrl- mldme oxlpurmol, 0 12, 4-hydroxy-6-ammopyrazole[3,4-d]-pynmidme, 0 23 were cut out and counted for their radloactwlty in toluene-tnton-x-omnlfluor 30 10 0 28, v/v/w scintillation fired in all assays, the reactions were initiated with gmp syn- thetase a umt of enzyme is that quantity which can produce 1 /zmole of gmp per mm under these standard condztlons, i umt was previously found to be equal to 2 nmoles of enzyme [6, 7] pt epat atton of i tbonucleottde6 6-thloxmp was synthesized enzymatlcally by incubating 6-thloxanthme and 587 5-phosphorlbosyl 1-pyrophosphate with partially purified guanlne-xanthlne phospho- nbosyltransferase from e cob b the enzyme was prepared by precipitation from a streptomycin sulfate treated extract with 80 ~ nh42so4 and redlssolved in buffer as described elsewhere [12] the product was isolated from the deprotelnlzed and diluted reaction mixture by adsorption onto an ecteola-cellulose column equilibrated with 0 l mm lici containing 0 1 mm dlthlothreltol the column was washed with a linear gradient of 0 1-200 mm l1c1 containing 0 1 mm dlthlothreltol 500 ml gra- dient/33 ml bed vol  the 6-thloxmp was eluted by a continuation of the 200 mm l1c1 the fractions containing the major portion of the uv absorbing 341 nm ma- terial were concentrated by lyophlhzation and the lici extracted with acetone- methanol 9 1, v/v the isolated rlbonucleotlde eluted as a single symmetrical peak when subjected to high-pressure liquid chromatography, had a base phosphate ratio of 0 9 1, and was dephosphorylated by 5'-nucleotldase 6-thlogmp was prepared by two distinct methods the first method was the chemical phosphorylatlon of 6-thloguanoslne and is described elsewhere [13] the other approach utilized the enzymatic synthesis from 6-thloguanlne and 5-phospho- rlbosyl 1-pyrophosphate with an extract of ehrhch ascltes acetone powder according to a previously reported procedure [14] the products from each synthetic procedure were concentrated by lyophlhzation and applied to a l0 mm fl-mercaptoethanol equilibrated blorad p-2 column and eluted with the same buffer the enzymatlcally synthesized 6-thlogmp required further purlficauon this was accomphshed by thin- layer chromatography on a preparative cellulose plate in n-propanol-water 7 3, v/v containing l0 mm fl-mercaptoethanol the product re 0 15 was eluted with water and lyophfllzed the two preparations were found to be indistinguishable and > 99 pure as analyzed by high-pressure liquid chromatography the chemically and en- zymatlcally synthesized products had base rlbose phosphate ratios of 1 00 1 01 1 01 and l 00 0 94 0 96, respectively both products were converted to 6-thloguanoslne by 5'-nucleotldase, and chromatographed as one spot on deae-cellulose thin-layer plates in 0 2 m ammonium formate ph 5 0 re 0 33 1-[6-14c]oxl-5'-p was synthesized by the same procedure used for 6-thloxmp the precursor base, [6-t4c]oxlpurlnol was enzymatlcally synthesized [15] from [6-14c]allopurlnol that had been chemically synthesized [16] after incubation with the phosphorlbosyltransferase, the rlbonucleotlde product was concentrated by lyo- phlllzatlon and was purified by the same procedure described above for the enzym- atlcally synthesized 6-thlogmp in the absence of fl-mercaptoethanol the product was found to be 99~o radtochemlcally pure as analyzed by high-pressure hquld chromatography and thin-layer chromatography on deae-cellulose in 0 2 m ammonium formate, ph 5 0 rf 0 21 6-nh2-i-alo-5'-p was enzymatically synthesized by the same procedure used for 6-thloxmp the product was then chromatographed on a ecteola-cellulose column in a manner similar to that for 6-thloxmp with the exception that dlthlo- threltol was absent and the rlbonucleotlde eluted during the l1c1 gradient this column was repeated two additional times using a 0 1-100 mm hnear gradient 400 ml gradient/82 ml bed volume the last ion-exchange chromatography step was followed by gel filtration with a water equilibrated blorad p-2 column the product was concentrated to a small volume by lyophlhzatlon and collected as a barium salt by treatment with 1 vol of 2 0 m bac104 and 2 vol of 95 ~o ethanol high-pressure liquid 588 chromatography of the ribonucleotlde resulted in a single symmetrical peak 8-azaxmp was enzymatjcally synthesized from 8-azaxanthlne and 5-phospho- ribosyl 1-pyrophosphate using partially purified xanthlne phosphorybosyltransferase from streptococcus faecahs the enzyme was purified according to a reported proce- dure [17] through the first nh42so4 precipitation the product was isolated from the deprotelnlzed and diluted reaction mixture by batch adsorption to biorad ag-1-x8 chloride form anion exchange resin equilibrated with 0 1 mm l1c1 the gel was poured into a column and eluted with a linear gradient of 0 1-1000 mm lici 200 ml gradlent/approx 3 ml bed volume the fractions containing the 8-azaxmp ultraviolet absorbing material were lyophlhzed and the l1c1 extracted with acetone- methanol 9 1, v/v the product was then chromatographed on a deae-cellulose whatman de-52 column equilibrated with 3 mm ammonium formate ph 4 4, eluting with a linear gradient from 3 to 300 mm of the same salt 600 ml gradient/ 20 ml bed volume the ribonucleotlde was concentrated and the salt concentration decreased by repeated lyophihzation and the product chromatographed on a water- equilibrated blorad p-2 column, elutlng with water the final product was found to be > 99~o pure as analyzed by high-pressure chromatography and its base rl- bose phosphate ratio was 1 00 1 01 0 99 [8-~4c]xmp was synthesized in the same manner as 8-azaxmp using [8-14c]- xanthme schwartz/mann the product was isolated with the thin-layer chromato- graphic procedure described above for the enzymatlcally synthesized 6-thlogmp the product chromatographed on thin-layer cellulose plates m 5% nazhpo4-1so- amylalcohol 2 1, v/v as one spot, which contained all the radioactivity, rv 0 84 the labeled xmp could be completely converted to [8-14c]gmp by gmp synthetase 1-oxl-5'-p, 7-oxl-5'-p and 3-[8-14c]xmp were synthesized and purified ac- cording to a previously reported procedure [18] molar lty oj rlbonucleotldes rlbonucleotlde concentrations were determined spectrally from published extinction coefficients [18-22] hlgh-pt easut e hqutd chtomatography rlbonucleotldes were analyzed with a varlan lcs-t000 equipped with a p4-38 anion exchange column, elutlng with an ammonium acetate gradient from 0 175 m ph 5 5 to 5 m ph 4 7 the details of this procedure are described elsewhere [23] other mater tals xmp, atp, gmp nadh, nad, and amp were purchased from p l blo- chemicals, cyclic xmp and 5'-nucleotjdase crotalus adamanteu~ from sigma, phosphoenolpyruvate, myokmase no 15355, pyruvate kmase no 15744, and lactate dehydrogenase no 15372 from boehrjnger mannhelm corp results substrates the kinetic parameters derived from data obtained with the coupled spectral assay are presented in table i the km value of 0 058 mm for xmp was reproduced 589 table i interactions of ribonucleotides with gmp synthetase v and km values were determined from computer analyzed lineweaver-burk graphical presentations of data obtained from the standard coupled assay described in the text varying the concentration of the analog the concentration of gmp synthetase was 6 nm for the reactions with xmp, 26 nm for 8-azaxmp 130 nm for 6-thioxmp, and 300 nm spectral and 927 nm radlolabeled for l-oxl-5'-p where no velooty was detected, the analog was at a concentration of 1-2 mm under conditions where a lower hmlt of 3 pmoles/mm could be detected and 23 000-120 000 pmoles/min of xmp would have been ammated compound v turnover/mm km kj* v/k\n",
      "\n",
      "the nature of the buffer may also influence the shape of the v versus [s] 0 plots as was distinctly demonstrated for \"biodegradative\" l-threonine dehy- dratase from the same strain of e coli [18]  in the phosphate buffer these plots had a complicated non-hyperbolic shape, without intermediate plateaux, but in phosphate-carbonate buffer with the same ionic strength and ph value these plots had one distinct intermediate plateau\n",
      "\n",
      "inactivation of aspartase by sulfhydryl reagents { when the enzyme was incubated with various sulfhydryl reagents, the 224 table i sulfhydryl groups in aspartase the reaction mixture for modification of sulfhydryl groups with dtnb contained 88 mm potassium phosphate buffer, ph 74, 080 mg without guanidine\n",
      "\n",
      "cysteine sulfinate decarboxylase is a pyridoxal 5'-p enzyme which is nor- mally saturated with coenzyme in rat liver\n",
      "\n",
      "assay of allantoinase a mixture containing 39 ymol rs- or r-allantoin, 008 pmol mns04, 5 pmol dithiothreitol, 200 pmol tris ' hcl buffer, ph 74, and an appropriate amount of the enzyme was incubated at 30c\n",
      "\n",
      "assay of allantoin racemase 02 ml enzyme solution 10-250 jug protein depending on purity were mixed with 1 ml 50 mm phosphate buffer, ph 65, containing 22 pmol r-allantoin\n",
      "\n",
      "a solution of s-allantoin was prepared by incubation of 2400 units of uricase specific activity 8 units/mg protein from b fastidiosus with a suspension of 3 g uric acid in 150 ml 50 mm phosphate buffer, ph 65\n",
      "\n",
      "the fractions indicated by the solid line 145 units/66 ml and dotted line 138 units/60 ml were separately probable oligosaccharide table i characterization of the preparations of inulobiose, inulotriose and inulo- tetraose 05 ml of 0024% fructose oligosaccharide solution was incubated with 05 ml 02 m acetate buffer ph 50 and 10 ml 01% fl-fructosidase at 30 for 3 h\n",
      "\n",
      "a 3-ml portion of 0024% sugar solution was incubated with 6 ml of 01% ~-fructosidase and 3 ml of 01 m acetate buffer ph 50 at 30 for 50 h\n",
      "\n",
      "the dnp-amino acid in the ether phase was identified by two dimen- sional paper chromatography using solvent systems of toluene/2-chloroetha- nol/pyridine 08 m/ammonia 5  3  15  3 and 15 m sodium phosphate buffer, ph 60\n",
      "\n",
      "temperature dependence of sedimentation coefficient a sample of urocanate hydratase was dialyzed extensively against 005 m potassium phosphate, ph 75, and diluted with the same buffer to give an optical density of 07 at 280 nm this corresponds to an enzyme concentration of 06 mg/ml\n",
      "\n",
      "aspaztase 160 mg was incubated with 1 pg of trypsin in 02 ml of 60 mm tris \" hc1 buffer, ph 74 at 30\n",
      "\n",
      "enzyme assays for glyoxalase-i activity in the various purification procedures were carried out using methyl- glyoxal 3 mm and glutathione 3 mm in ph 7 phosphate buffer i = 02 with 10 mm mgcl, 25c, and allowing 10 min for hemimercaptal formation to reach equilibrium\n",
      "\n",
      "the tubes were incubated for 18 h at 37 and the turbidity measured m a klett-summerson photoelectrm colonmeter with a 660 nm filter h2-folate was 50% as effective as folate m supporting the growth of l casel this value was not increased when h-folate was added aseptmally to the stenhzed medmm m the presence of 0 12% ascorbate preparatzon of ptertdmes folate, pteroate, and homofolate were reduced to the corresponding 7,8-dlhydro compounds by treatment with sodmm dlthm- rote according to the method of futterman [10] as modffmd by blakley [11] 422 h2-homopteroate was prepared by reducing homopteroate with na2 $2 04 in 2-mercaptoethanol by the method of friedkin et al [12] all reduced com- pounds were stored as acidic suspensions in the presence of 0 05--0 2 m dithio- threltol or 2-mercaptoethanol m 0 5 ml samples at --70 h4-folate was pre- pared by the reduction of h2 -folate with e colt b h2 -folate reductase and was isolated from a bio gel p-2 polyacrylamlde gel column h4-homopteroate, h4-pteroate and additional h4-folate were prepared by catalytic hydrogenation following the method of nahas and frledkin [13] for the preparation of h4- homofolate, except that the platinum oxide catalyst was removed by centnfu- gatlon and the solution was made 0 11 m with respect to 2-mercaptoethanol following hydrogenation each reduced compound was characterized by its ultraviolet spectrum [14--16] and by paper chromatography [1,17] concen- trations were determined by ultraviolet spectrum for the dlhydro compounds and by chemical formylation [18] for the tetrahydro compounds chemical reduction to the tetrahydro compounds results m the formation of -+-isomers and although the effect of having both isomers present m the enzyme assay is not known, similar results were obtained with h4-folate prepared enzymati- cally or chemically h4-homofolate was prepared by a modification of kawm and scnm- geour's [19] method of na2 $2 04 reduction a 2-ml solution of 0 4 mm homo- folate containing 0 5 m 2-mercaptoethanol was adjusted to ph 7 9 and heated to 50 in a water bath with continuous stirring na sz 04 200 mg was slowly added and the solution was allowed to sit at 50 for 30 min, unstirred it was then chilled to 5 and, while mechanically stirred, the ph was lowered to ph 3 0 with 1 0 m hc1 the solution was stored at --70 and the ph was adjusted to ph 7 0 with dilute naoh before use h4-homofolate was charac- terized by it ultrawolet spectrum [15] and obtained m 88% ymld analysts of mhtbttton data the 1/v vs 1is plots were analyzed using computer programs for the analysis of competitive and non-competitive inhibi- tions [20] the apparent km values and the v values obtained from the recip- rocal plots were analyzed by the student's t-test for statistical sigmfmance at the 0 05 level of probability the v values of the inhibited curves differed significantly from that of the uninhibited curve in the inhibition plots termed non-competitive miscellaneous a bmautogram was prepared by the method of brown et al [17] using fohc acid casel media medium containing 3% agar the seed was 1 0 ml of a 10 ml 6 h washed l case~ culture suspended in 10 ml of 0 85% nac1 paper chromatograms were developed by descending chromatography m 0 1 m sorenson's phosphate buffer, ph 7 0 protein was determined by the method of lowry et al [21] radioactive samples were placed m semtillation vials containing 10 ml of a mtx of 15 2 g 2,5-bls-25-tert-butyl-benzoxazol- ylthmpene/gal toluene and counted on a beckman ls 230 llqmd semtillatmn counter at 75--80% efficiency results development of radtochemwal assay for h2 -folate synthetase since the l casez assay was not statable for determining the effects of 423 mhlbltors and metabohtes on h2-folate synthetase, a radlochemlcal assay method was developed the reaction systems were the same as described in materials and methods for the h2-folate synthetase assay except that m place of non-radioactive l-glutamm acid, 2 mm umformally labeled l-[~4c]gluta- mm acid 1 25 c1/mol after dilution with non-radioactive l-glutamlc acid was added the use of l-[~4c] glutamm acid allows the quantltatlon of enzymatl- cally formed h2 -[ ~ 4c] folate following paper chromatography of the reaction products after incubation, the reactions were stopped by the addition of 50 mm edta and placed m ice an ahquot of each reaction 100 pl and 13 nmol of career h2-folate 10 pl were applied to whatman 3mm paper and the chromatograms were developed in 0 1 m sorenson's phosphate buffer, ph 70 l-[~4c]glutamlc acid moved to the solvent front and the product, h2- [ ~ 4c] folate, followed with an re of 0 14 career h2 -folate aided m visuahz- mg, under ultravmlet light, the position of the product on the chromatogram this area of fluorescence was cut out and the samples were counted several observations confirmed that the radioactive area was h2-[~4c]folate large amounts of enzymatmally formed h2-folate could be visualized under ultra- violet hght at an rf of 0 14 following chromatography of the complete reac- tion mixture m the absence of carner h2-folate chromatograms of ldentmal reaction mixtures were analyzed for radmactwlty and for their ability to sup- port the growth of l casez the radioactive tracer scans and the bloautograms of reaction mlxtures are seen m fig 2 the complete reactmn mixture had a peak of radmactlvlty at rf 0 12, whmh corresponds to an area of l case~ growth at rf 0 12 this peak is absent in the scan of a control reactmn mixture lacking h2-pteroate when a bloautogram of chemmally prepared h2- folate was exammed, l casez growth was detected at an re of approx 0 16 with a secondary area of growth at an rf of 0 36 corresponding to the growth m the presence of folate three identical reactmn mlxtures were assayed mi- 300 a\n",
      "\n",
      "procedure for heating dihydrodipicolinate synthase and for analysis of results in all cases the heating medium included 005 m sodium barbital buffer ph 85\n",
      "\n",
      "the reaction mixture contained a total of 8340 units of lactate racemase as a pellet obtained from a solution saturated to 75% with respect to nh42so4, 400 pmot of a-lso, 1-~4c-labeled l+-lactic acid and sufficient 01 m sodium acetate buf- fer, ph 56, to give a total volume of 310 ml\n",
      "\n",
      "the fractions from the sepharose 6b gelfiltration containing phenylalanine racemase localized by filter tests for gramicidin s biosynthesis were applied to a hydroxyapatite column 2 12 cm which was equilibrated with 001 m sodium phosphate buffer, ph 75\n",
      "\n",
      "cystathionine fl-synthase l-serine hydro-lyase adding homocysteine, ec 42122 activity present in a fresh human liver supernatant prepared in 008 m na2hpo4/k2hpo4 buffer, ph 65, was associated with two different molecular weight species 500 000 and 235 000 respectively whereas in the correspond- ing supernatant fraction prepared in 02 m tris\" hc1 buffer, ph 80, enzyme activity was associated with a species possessing a molecular weight of 130 000\n",
      "\n",
      "determination of molecular weigh ts a column 15 cm x 90 cm of sephadex g-200 was equilibrated with either 008 m na2hpojkh2po4 buffer, ph 65 or with 02 m tris hc1 buffer, ph 80 and calibrated in both systems using standard preparations of cytochrome c, myoglobin, pepsin, albumin, alcohol dehydrogenase, catalase and urease\n",
      "\n",
      "the effect of ph on the nature of cystathionine synthase in fresh tissue samples the nature of cystathionine fi-synthase activity in a crude tissue preparation was examined by preparing a 375% w/v suspension of human liver obtained as soon as possible after death, 12 h in 008 m na2hpojkh2po4 buffer, ph 65 and applying 5 ml of a clear supernatant fraction, obtained by centrifuging the suspension for 1 h at 14 000 g, to a column 2 cm 90 cm of sephadex g-200 equilibrated with the above phosphate buffer\n",
      "\n",
      "effect of salt and ageing on enzyme activity the effect of ageing and salt on the nature of cystathionine fi-synthase activ- ity was examined by maintaining portions of supernatant, prepared as previ- ously described in 008m na2hpo4/kh2po4 buffer, ph 65 at 4, in the 300 a 250 o 200 4 >u u ~ 150 ~o 09a e>,f ~2 9 u 50 e c oq 5 10 15 20 25 30 35 40 frgction no fig 1\n",
      "\n",
      "1|cystathionine beta-synthase l-serine hydro-lyase adding homocysteine, ec 42122 activity present in a fresh human liver supernatant prepared in 008 m na2hpo4/k2hpo4 buffer, ph 65, was associated with two different molecular weight species 500 000 and 235 000 respectively whereas in the corresponding supernatant fraction prepared in 02 m tris · hcl buffer, ph 80, enzyme activity was associated with a species possessing a molecular weight of 130 000\n",
      "\n",
      "fraction step total recovered specific activity number activity units/mg protein % units 1 cell-free extract 10 ml 100 3890 004--006 2 absorption and batch elution with 03 m 100 3885 007---008 phosphate buffer ph 68 from deae- cellulose 120 ml combined affinity chromatography 65--70 fraction 5 ml 3 2691 238--240 490 24 u o 8 u ~5 0 2 --10 / o / o l_ \\ volume ml 045 030 e v c 015 o affinity chromatography of methylmalonyl-coa mutase on vitamin b-12-sephaxose\n",
      "\n",
      "routinely an incubation at 30 contained 08 ml tris/ mgc12/kc1 buffer, 01 ml randomly refolded ribonuclease solution 05 mg per ml in 10 mm acetic acid, 50 pl dithiothreitol solution 01 mm in water and 01 ml membrane suspension 25 mg protein per ml\n",
      "\n",
      "an isolation and purification method for the human abbreviations fru-l,6-p2, fructose 1,6-bisphosphate fru-l-p, fructose 1-phosphate gapd, glyc- aldehyde-3-phosphate dehydrogenase dtnb, 5,5dithiobis2-nitrobenzoic acid buffer a, 25 mm tris hc1, ph 73, i mm edta, 10 mm 2-mercaptoethanol, 01 m nac1 sds, sodium dodecyl sulfate mtt tetrazolium, 3-4,5-dimethyl thiazolyl-2-2,5-diphenyl tetrazolium bromide\n",
      "\n",
      "a variety of organic solvent fractionations were 196 also attempted and while hemoglobin precipitation was easily affected, particularly with a 50  50 mixture of cold chloroform/methanol added to an equal volume of hemolysate, low aldolase yields precluded the routine use of this procedure\n",
      "\n",
      "an nadh wash in the buffer without 2-mercaptoethanol was found to be successful in eluting all the gapd prior to the aldolase elution\n",
      "\n",
      "abbreviations fru-1,6-p 2 fructose 1,6-bisphosphate fru-1-p fructose 1-phosphate gapd glycaldehyde-3-phosphate dehydrogenase dtnb 5,5′-dithiobis2-nitrobenzoic acid buffer a 25 mm tris · hcl, ph 73, 1 mm edta, 10 mm 2-mercaptoethanol, 01 m nacl sds sodium dodecyl sulfate mtt tetrazolium 3-4,5-dimethyl thiazolyl-2-2,5-diphenyl tetrazolium bromide references 1 h lebherz w rutter biochemistry 8 1969 109 121 2 e penhoet m kochman w rutter biochemistry 8 1969 4396 4402 3 b allen r gracy b harris arch\n",
      "\n",
      "glycerol dehydratase inactivation was investi- gated in the absence of substrate in thermostated cuvettes with 1 ml of 2--3 units of apoenzyme free of glycerol, 40 pmol of potassium-phosphate buffer ph 80 and 20 nmol of adocbl\n",
      "\n",
      "aldolase assay was per- formed at 25 in a gilford recording spectrophotometer as described by racker [34] in 1 ml of 40 mm triethanolamine buffer, ph 75, containing 10 mm edta, 01 mm nadh, 10 mm fructose diphosphate, and 10/~g of trio- sephosphate isomerase and of glycerol-3-phosphate dehydrogenase\n",
      "\n",
      "samples 11 and 22 pg of crystalline rabbit heart aldolase were subjected to electrophoresis on 75% polyacrylamide gel supports in tris/glycine buffer, ph 86, as described in materials and methods\n",
      "\n",
      "furylfuramide cis-trans isomerising activity of enzyme fractions toward cis-furylfuramide was routinely assayed at 21 +- i in 3 ml final volume of 01 m phosphate buffer ph 70 protein concentra- tion, about 1 mg per ml containing 25\" 10-s--810-s m cis-furylfuramide, with or without being supplemented by 30 pl of nadh or nadph 67 10-6 --16 10-4 m final concentration, by measuring the increase in the first 30 s in absorbance at 420 nm the absorbance maximum of trans-furylfuramide\n",
      "\n",
      "furylfuramide reductase activity was assayed at the same temperature and in the same buffer by measuring in the first 30 s the absorbance change at the absorption maximum of cis-furyl- furamide 394 nm or trans-furylfuramide 420 nm in the absence or presence of nadh or nadph\n",
      "\n",
      "in order to confirm that the isomerization and reduction phenomenons of furylfuramide determined by increase and decrease of absorbances at 420 or 394 nm are enzyme catalyzed, incubation of cis- and trans-furylfuramide in 01 m phosphate buffer ph 70 supplemented with nadh or nadph was 321 e c o =e c o -- / [ / trans , nadh t // io'\n",
      "\n",
      "effect of glutathione on the cis-trans isomerising activity of bacteria on furyl- furamide cis-furylfuramide in 01 m phosphate buffer final concentration, 40 10-s m was incubated with the soluble enzyme fraction final protein concentration 03 mg per ml and the effect of glutathione on the increase of absorbance at 420 nm of the compound was checked\n",
      "\n",
      "the assay mixture contained 50 mm potassium phosphate buffer ph 74, 5 mm mgc12, 2 mm formaldehyde, 1 unit of phosphoriboisomerase, 4 mm d-ribose 5-phosphate sodium salt and an aliquot of the enzyme 005--02 unit in a final volume of 05 ml\n",
      "\n",
      "ph optimum, ph optima of free and bound succinyl coa synthetase were determined by equilibrating the samples with phosphate and tris hc1 buffers over the ph range 60--90 fig 1b\n",
      "\n",
      "the reaction mixtures contained in a final volume of 2 ml, 005 m tris - hci buffer ph 80, 1 ml of enzyme 8--10 mg of soluble succinyl coa synthetase o, specific activity 12, or 2--4 mg of insoluble succinyl coa synthetase e, specific activity 04 glutatione 10 pmol and coa 025 pmol saturation levels of the other substxates were as follows mgc12 10 #mol sodium succinate 100 ~mol, atp 10/~mol\n",
      "\n",
      "pyridoxal 5-phosphate is bound firmly to dopa-decarboxylase bound pyridoxal phosphate is found after dialysis for 24 h against phosphate buffer\n",
      "\n",
      "finally, to show the persistent attachment of nad to the apoenzyme after inactivation with dithiobisnitrobenzoate the following experiment was carried out 36 mg 30 nmol of epimerase in 1 ml of 01 m glycylglycine buffer, ph 76, was completely inactivated with 02 mm dithiobisnitrobenzoate in 30 min\n",
      "\n",
      "guanylate cyclase was eluted with the buffer containing 05 m nacl and 5 mm edta fig 2\n",
      "\n",
      "guanylate cyclase was eluted with 05 m naci and 5 mm edta in the buffer containing 25 mm tris hcl ph 76, 2 mm dithiothreitol, 10 mm nan3 and 20% glycerol\n",
      "\n",
      "arrhenius plot of log, oh against reciprocal of absolute temperature t for ribulose 1,5-bisphosphate carboxylase purified from either cyanidium 0 or chlorella o, enzyme activity was measured using a pre-assay incubation period of 5 min\n",
      "\n",
      "the possibility of the presence of mg\" and hc03 influencing the stability of ribulose 1,5-bisphosphate carboxylase was examined by repeating the above experiments but incubating the enzyme, prior to assay, in buffer alone\n",
      "\n",
      "gel electrophoresis of the two most active fractions representing the two peaks of enzymic activity in the eluant, showed that the material passing directly through the column had only diol dehydratase the glycerol dehydratase was held by the column until eluted free from diol dehydratase by the buffer with 150 mm potassium phosphate\n",
      "\n",
      "in subsequent separations on a larger scale, buffer 1 containing 05 m kcl was used to elute glycerol dehydratase in a smaller volume\n",
      "\n",
      "progress of the reaction with glycerol using a purified diol dehydratase, 054 units and b purified glycerol dehydratase, 009 units, after dialysis against 100 mm triethanolamine-hcl buffer, ph 82\n",
      "\n",
      "prior to each experiment, formyltetra- hydrofolate synthetase was prepared by centrifuging a sample of the enzyme/ 50% ammonium sulfate suspension, dissolving the enzyme in 01 m tris-hcl ph 75, 01 m kc1, 005 m 2-mercaptoethanol, and dialyzing the sample against the buffer for 3 h to remove residual nh42so4\n",
      "\n",
      "the assay mixture contained in 300 ml, 50 mm glycylglycine/chloride buffer, ph 74, 25 mm magnesium chloride, 1 mm dithiothreitol, 01 mm thiamine pyrophosphate, 013 mm nadh, 034 unit a-glycerophosphate dehydrogenase, 1 unit triose phosphate isomerase, 124 mm d-ribose-5-phosphate and 008 mm d-ribulose-5-phosphate\n",
      "\n",
      "in the case of the sulphydryl reagents, cupric acetate, mercuric chloride and p-chloromercuribenzoate, the epimerase was incubated in assay buffer for 20 min at 25c with the inhibitor, then diluted and assayed by the 290 nm method at ph 74 in the case of p-mercurichlorobenzoate, and in the case of copper and mercury salts at ph 68 to avoid precipitation\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "inhibition by the specific sulphydryl group reagent 5,5'-dithiobis2-nitrobenzoic acid was tested by incubating epimerase with an excess of the reagent for 60 min at 30 in the 40 mm triethanolamine/chloride/2 mm edta buffer, ph 80\n",
      "\n",
      "a linear gradient of 50 ml of 05 m kcl in the above buffer running into 50 ml of buffer was begun at tube 26 and followed by 05 m kci in buffer , absorbance at 280 nn in a 1 cm cell a28p 0------o, ribulose-5-phosphate 3-epimerase activity - - - - - kcl concentration\n",
      "\n",
      "a linear gradient of 40 ml of 01 m kcl in buffer running into 40 ml of buffer was begun at tube 10 and followed by 05 m kcl in buffer at tube 30 , absorbance at 280 nm in a 1 cm cell az gp 0 - - _ _ _ _o, ribulose-5-phosphate 3-epixnerase activity - - - - kcl concentration\n",
      "\n",
      "except where stated otherwise, lyase activity was determined by isotopic assay using ring-labelled l-phenylalanine and a 20 h incubation 25 as described by attridge et al [9] except that 100 mm tris-hcl ph 86 replaced the borate buffer\n",
      "\n",
      "effect of ph on mustard lyase activity using three different buffer systems potassium phosphate, sodium barbital, glycine/naoh, all at 50 mm the homogeneously purified lyase gave the broad response curve to ph see eg refs 5 and 34 characteristic of this enzyme {data not presented\n",
      "\n",
      "the solvent systems used for chromatography were 1 ammonia/water/isobutyric acid 520  42  938, v/v 2 methylethylketone/ saturated boric acid/acetic acid 8  1  1, v/v 3 ethylacetate/pyridine/water 212, v/v, organic phase 4 ethylacetate/pyridine/acetic acid/water 5513, v/v\n",
      "\n",
      "the incubation mixture con- tained 200 pl 14c-labelled dtdp-4-keto-6-deoxysugars 02 pmol 2 pci, 30 pl nadph 036 pmol, 470 pl buffer b containing 15% glycerol, 100 pl buffer f, and 200 pl synthase + epimerase after the nh42so4 fractionation and desalt- ing on sephadex g-25\n",
      "\n",
      "mixture consisted of the following reagents 100 #1 05 m imidazole buffer containing 005 m mgc12 100 pl 1'5 mm phosphoenol- pyruvate 100 /a 55 mm glutamate 100 pl 55 mm cysteine 100 pl 2 mm nadh 10 pl 250 u/ml lactate dehydrogenase 10 pl 700 u/ml pyruvate kinase 100 pl 3 mm atp\n",
      "\n",
      "electrophoresis of y-glutamylcysteine synthetase was carried out for 5--6 h at 100 v in 90 mm tris-hc1, 3 mm edta, 80 mm borate buffer ph 83 on 75% pol~tacrylamide slab gels or on gradipore density gradient 25--27% polyacrylamide slab gels obtained from isolab inc, akron, oh\n",
      "\n",
      "hydroxypyruvate isomerase activity was measured at 30 in incubation mixtures which contained per ml, 130 #mol phosphate buffer ph 71/12 prnol hydroxypyruvate/01--10 gg protein dependent on the purity of the enzyme preparation\n",
      "\n",
      "the isomerase was eluted with 2 mm phosphate buffer ph 71, while contaminating proteins remained bound to the hydroxyapatite\n",
      "\n",
      "tar- tronate semialdehyde was prepared enzymatically from hydroxypyruvate by the action of hydroxypyruvate isomerase in an incubation mixture containing per ml, 130 pmol phosphate buffer ph 71/31 pjnol hydroxypyruvate/7 pg protein crude extract\n",
      "\n",
      "561 table ii enzymatic conversion of hydroxypyruvate by purified hydroxypyruvate isomerase the incubation mixture contained 390 /~mol phosphate buffer ph 71/44 /~mol hydroxypyruvate/3 ~g purified hydroxypyruvate isomerase\n",
      "\n",
      "the fractions containing adenylosuccinate synthetase activity were combined and dialyzed against 10 mm potassium phosphate buffer ph 7 to remove aspartate\n",
      "\n",
      "adenylosuccinate synthetase from the tumor cells was most stable at ph 70 in phosphate buffer\n",
      "\n",
      "equilibrium sedimentation analysis of adenylosuccinate synthetase, a, the purified enzyme in 10 mm potassium phosphate buffer ph 7, 4 mg/ml was centrifuged at a rotor speed of 5200 rev/min\n",
      "\n",
      "for assays at ph 78, a stock solution a containing 50 mm tris-hc1 buffer ph 78/20 mm nahco3/10 mm mgci/02 mm nadh/2 mm dithiothreitol/ 10 t~g malate dehydrogenase per ml was prepared at the start of each day\n",
      "\n",
      "for assays at ph 58, stock solution b containing 100 mm mes buffer ph 55/40 mm mgc12/2 mm dithiothreitol, and stock solution c containing 05 m nahco3/5 mm nadh/250 tlg malate dehydrogenase per ml were prepared at the start of each day\n",
      "\n",
      "incubation mixtures for kynureninase contained 25 p~nol potassium phosphate buffer ph 65, 6 nmol pyridoxal phosphate, 1 #mol l-kynurenine and enzyme in 10 ml total volume\n",
      "\n",
      "for hydroxy- kynureninase, assay mixtures contained 25/~mol potassium phosphate buffer ph 75, 4 #mol pyridoxal phosphate, 1 #mol l-3-hydroxykynurenine and enzyme in 10 ml total volume\n",
      "\n",
      "chromatography of a crude chloroform-ethanol extract of hagfish carbonic anhydrase on a deae- cellulose column 10 17 cm equilibrated with 5 mm tris-hc1 buffer, ph 89\n",
      "\n",
      "development in both solvent systems revealed didansyl lysine, indicating that the human milk rnaase major component contains lysine in the n-terminal position\n",
      "\n",
      "purification of d-galactonate dehydratase unless otherwise stated, all the operations were performed at 0-4 and all potassium phosphate buffers ph 68 contained 1 mm edta\n",
      "\n",
      "the dehydratase fractions were pooled, diluted with the same volume of 10 mm phosphate buffer, adsorbed onto a small deae-cellulose column 1 1 cm and concentrated by elution with 05 m nac1\n",
      "\n",
      "materials and methods determination of enzyme activity with a glucose determination reagent the glucose determination reagent diacolor, toyo spinning co, osaka contained 5 units glucose oxidase, 125 units peroxidase, 25 etg 4-aminoantipyrin, 250 ltg diethylaniline and 3 mg tris-hcl in 1 ml 01 m acetate buffer pfl 58\n",
      "\n",
      "the esr spectrum of a mixture of trans- and cis-af-2 anion free radicals observed on anaerobic incubation of 20 mm cis-af-2 with a nadph-generating system 13 units/ml glucose-6-phosphate dehydrogenase, 11 mm glucose 6-phosphate and 08 mm nadp and 4 mg/ml hepatic microsomal protein from male rats in h20-based buffer\n",
      "\n",
      "5-keto-glucose-6-p ~ nadh l-m'o-mosose- 2-1-p 2 1 l-myo-mositol- 1 -p sclleme i by the use of a substrate selectively labeled with ~s0 at the site of formation of the schlff base, the as0 being lost in the reaction with the enzyme the latter experiment can be performed m reverse, in that is0 from the incubation medium will be incorporated into the product of the enzyme reaction as a result of the hydrolysis of the schlff base the second type of classical aldolase is termed class ii these enzymes are thought to use divalent metals in the aldol transmon state and can generally be shown to be dependent on such metals utilizing d- [5-180] glucose-6-p, which as a specific and unambiguous probe of schlff base formation at c-5 of glucose-6-p[1], we find that myo-lnosltol-1- phosphate synthase from a lthum and a neurospora species do not use a class i aldolase mechanism experiments with h2180 confirm that finding and also show that schaff base formation does not occur at any other position of either the substrate or an intermediate thus, these enzymes either utilize a class ii aldolase reaction, with the participation of a divalent metal, or the cycllzatmn occurs by an atypical aldolase mechanism materials and methods preparation o] mvo-mosltol-l-phosphate synthase from l longtflorum pollen 10 g l longtllorum pollen vat ace was suspended in 60 ml 0 02 m trls- acetate, ph 8 0/0 5 mm gsh pollenkltt was removed by stirring with a glass rod, to which it sticks the suspension was homogenized with 50 strokes of a glass-glass homogemzer and the resulting mixture centrifuged for 0 5 h in a sorvall rotor at 15 000 rev / nun a 35-55% satd nh42so4 ppt of this extract was dissolved in buffer and dialyzed for 4 h with two changes of buffer this was purified on a 2 x 30 cm deae-cellulose column, elutmg with a 600 ml linear gradient of 0-0 25 m nac1 in the same buffer the fraction elutlng between 0 145 and 0 195 m nac1 was concentrated by nh42so4 60% satd precipitation and further purified on a 1 20 cm sephadex g-200 column pollen synthase elutes at 1 36 vo in the experiment in table i, synthase purified through the sephadex g-200 stage was used in the experi- ment of table ii, the enzyme was purified through the deae-cellulose step pollen enzyme assay myo-lnosltol-l-phosphate synthase was incubated at 30 for 2 5 h in 0 5 ml 0 02 m trls-acetate, ph 8 0/1 2 ~tmol d-glucose-6-p/ approx 0 5 ~c1 [1-i4c]glucose-6-p/0 5 mmol nad/ 2 5 mmol ammonium acetate/trls-edta or metal salt as specified the nlyo-mositol 1-phosphate pro- duced was determined as described previously [2,3] preparation of myo-tnosltol-l-phosphate synthase from neurospora crassa neurospora crassa, strain rl 21 was cultured as described by plfia and tatum [4] the myceha were homogenized in a warlng blendor and the supernatant fractlonated with nh42so4 and protamlne sulfate as described else- where [4,5] purification of the enzyme to homogeneity was performed according to a chromatographic procedure unpublished data consisting of the following steps gel filtration on ultrogel aca-34 lkb, anionic exchange chromatography on deae-cellulose what- man de-52, anionic exchange chromatography on ah-sepharose 4b pharmacla and adsorption chromatograpyy on calcium phosphate deposited on fibrous cellulose the enzyme obtained by this pro- cedure is homogenous according to the following criteria analytical gel filtration, sedimentation in a sucrose density gradient, analytical discontinuous acrylamlde gel electrophoresls, gel electrofocusmg and sds-polyacrylamlde gel electrophoresls the enzyme was assayed as described previously [5] preparatton of myo-mosttol-l-phosphate synthase from bovine tesns the enzyme used in these experi- ments was purified by nh42so4 precipitation on cehte-545 followed by reverse nh42so4 gradient elutton the cehte fraction containing activity was then purified on deae cellulose whatman de-52 the enzyme purification and assay procedures were as described previously [1,6] analysts of the 180 content of substrate and pro- duct lyophlhzed mcubatton mixtures were shaken overnight with a solution of butaneboronlc acid in pyrldme 44 mg/ml this produces myo-mosltol 1,2 3,5 4,6-tnsbutaneboronate [7] and a bis- butaneboronate of glucose the samples were then divided in two and the portion to be analyzed for glucose was treated with 10 #1 acetic anhydride to form o~-d-glucofuranose cychc 1,2 3,5-bisbutane- boronate 6-acetate [8] because the acetic anhydride causes some mosltol boronate to be lost, the non- acetylated preparation was used to directly analyze mos~tol and the acetylated mtxture was used to analyze glucose the lso content of the glucose and myo-lnosltol was measured as previously described [1] using electron ionization gc-ms in each case, the ions measured, using selected ion monttormg, were [m-c,ih9]+ and the 1so-labeled ion 2 daltons heavier, 1 e , m/z 297 and 299 for the glucose deriva- tive, m/z 321 and 323 for mosltol and m/z 323 and 325 for sorbltol d-[5 80] glucose-6-p was synthesized as previously described [1 ] results and discussion although the mtermedlate m the myo-mosltol- 1-phosphate synthase reaction, 5-ketoglucose-6-p of scheme i, has not been directly demonstrated there is much evidence supporting its formation the naddependence of all known myo-mosltol-l-phosphate synthases and the effect on the rate of the synthesis reactions, by deuterium and tntmm substitution at c-5 of glucose-6-p, represent some of the evidence for the oxidation of the 5-hydroxyl of glucose-6-p for a summary of these and other studies, see refs 1 and 22 supporting evidence for the intermediate is found m the facile base-catalyzed cychzatlon of 5-keto-glucose-6-p to mosose phosphates wluch can 301 table i incubations of myo-inos1tol-1-phosphate syn- thase with d-j5a 80] glucose 6-phosphate incubations of enzyme and boiled enzyme controls with labeled substrate were followed b treatment ~lth alkaline phosphatase to produce free d-glucose from the substrate and tree mvo-mosltol from the product the samples ~ere then derwatlzed and analy~ed by gc-ms enzyme from lzhum pollen was incubated tor 24 h at 37 m 0 5 ml of 40 mm tns-acetate, ph 8 0, which was 2 9 mm in d-[5aso]glucose-6-p, 5 mm in ammonium acetate, 1 mm in naiy\" and which contained 0 02% nan3 the am- tim rate ol svnthesls was 9 2 nmol myo-mosltol 1-phos- phate per h the reaction was terminated by heating at 100 for 5 mm this was followed by treatment ~lth alkaline phosphatase 2 h, 37 samples were then taken to dryness and analyzed the neurospora preparations of myo-mosltol-l-phosphate synthase were incubated tot 24 h at 37 vvlth 0 15 ml 50 mm trls-hc1, ph 7 7, which was 1 3 mm in d-[5-180]glucose-6-p, 2mm m nad 14 mm m nhac1 and which contained 0 02% nans the lmtlal rate of synthesis of myo-mosltol 1-phosphate with the deae-punfled enzyme was 1 3 ~umol/h and with the calcium phosphate-purified enzyme, 3 1 #mol/h in this case, the reaction was terminated ov incubation with alka- hne phosphatase tot 1 h enzyme percent excess 18o percent source retention d-[5aso] [2-180] of 180 m glucose m vo-inosltol myo- inosltol l longlllorum pollen 33 3 _+ 0 6 a 34 5 _+ 0 6 b 104 n crassa deae 34 +04d 345_+03c 101 calcium phosphate 34 6 _+ 0 04 c 101 a average + s d of fwe values of 53 80]glucose were not enzymatlcally treated b average + s d of three experimental samples c average _+ s d of 12 samples from inactive enzyme incu- bations lnactwated at 100 and active enzyme incuba- tions these groups were not significantly different d average _+ s d of four samples be reduced to momol phosphates [9,10] thus, if this reaction is of the class i category, the 5-carbonyl is the expected site of scbaff base formation, and labehng flus function with 180 gwes a specific probe of any covalent interaction between the 5-position of the intermediate and the enzyme table i shows that 302 table ii incubations of rnyo-inositol-1-phosphate synthase in buffered h21 so samples were incubated with d-glucose-6-p in an h2180contalnmg medmm, iollowed by treatment ~lth alkaline phosphatase products ~ere then analyzed by gc-ms as glucose or myo-mosltol lthum or following treatment ~lth sodium borohydnde neurospora when sorbltol and myo-mosltol were analyzed enzyme from ldtum pollen ~as incubated for 24 h at 37 in 30 mm trls-acetate, ph 8 0, which was enriched in lso by ddutlon with h2180 the medium was 2 4 mm in d-glucose-6-p, 1 mm in nadand 5 mm in ammonium acetate the mosltol rate ot synthesis was 3 nmol myo-mosltol l-phosphate per h the incuba- tion mixture was then dried and the residue taken up in 0 5 ml ot h20, alkaline phosphatase added and the mixture incubated a further 2 5 h, heated at 100, and the protein removed by centnfugatlon the supernatant was lyophlhzed and derwatlzed for gc-ms the ?veurospora enzyme was incubated for 24 h in an 180-enriched trls-hc1 medium, ph 7 7/d-glucose-6-p 1 mm/ 2 mm nad/nh42so4 25 mm/0 02% nan3 the initial rate ol synthesis was 4 ~mol/h alkaline phosphatase treatment was ~ollowed by reduction ~lth 160 mg nabh4 the resulting mixture was lyophxhzed and the sorbltol and mvo-mosltol ~er~ den- vatlzed and analyzed by gc-ms enzyme source 180 enrichment of incubation medium percent isotope incorporation a control substrate myo-inosltol product l longtflorum pollen 7 3% n crassa 18 8% glucose 028b 26+26c 1 6+01c sorbltol 139+15d 1344-14d 1443e a percent molecules containing one atom of 180 calculated as tollo~ s for glucose as the blsbutaneboronate acetate derivative, m/z 297 contains all the oxygen atoms ot glucose it is the derivative less a butyl radical [8] enrichment by a smgle 180 atom is thus measured at rn/z 299 the percent abundance of m/z 299 in the enriched molecule, less the percent abundance ofm/z 299 in the naturauy-occurrmg molecule is the percent isotope incorporated for lnosltol the corresponding ions are m/z 321 and 323, ior sorbltol m/z 323 and 325 b enzyme n = 2 was heated 100 to mactwate it prior to adding substrate c reaction stopped by heating, results are the average 4- s d of three samples d average values 4- s d sorbltol in controls, n = 6, from active enzyme preparations, n = 3 e average 4- s d , n = 3 ~so is retained by myo-lnosltol formed in the reaction of ttus substrate with synthase from either the ldtum or the neurospora species thus, at is tughly unlikely that the class i aldolase mechanism is used by either of these enzymes * * it is possible that the 5-carbonyl of 5-keto-glucose-6-p could react with an amme on the myo-mosltol-l-phosphate synthase to form the ammol precursor of a schfff base wtuch m~ht either, 1 dehydrate to give the schfff base, but conserve the water for later hydrolysis or, 2 stabilize the ammol so that it did not dehydrate the latter instance would not, however, produce the enamme-tmme tauto- menc system which is thought to promote the proton removal, m this case from c-6, wtuch precedes the aldol condensatmn the posslblhty that the water lost from the ammol is conserved cannot be excluded table ii shows that, when either the lily pollen or the mold enzymes are incubated with n-glucose-6-p m a buffer containing h2180, the mosltot formed does not reach the level of isotope enrichment of the solvent if a sctuff base were formed between any position of the substrate and the enzyme, it would have to be hydrolyzed with water from the buffer and thus attmn the same degree of labeling at the reacting carbon as the 180content of the buffer the observed partial uptake of label from the me&urn is due to exchange of the aldehyde carbonyl of the glucose dunng mutarotatlon, a process which xs accelerated at the ph of the incubation [ 11 ] ** ** if, in the lthum experiment, the reaction had used the schtff base route the product 2-[1801]myo-lnosltol the cause of the variability in the amount of 180 incorporated m these experlments is unknown but is similar to that wtuch was observed with mammalian myo-lnositol-l-phosphate synthase pre- parations [ 1] earher studies on the myo-lnositol-l-phosphate synthase from neurospora agree with our findmgs, m that nabh4 only lnlubited the enzyme at htgh concentrations and because the action of agents wluch inhibit schdf base formatton suggested that the enzyme was not a typical class i aldolase [12] whether the neurospora myo-mosltol-l-phosphate synthase is a class ii aldolase is uncertain the enzyme is 50% inhibited by 25 mm edta and com- pletely mtubited with 100 mm edta [12] about 2-fold activation of the neurospora enzyme occurs with 2 mm mg2but the enzyme is active without the metal, even after a 24 h chalysis against 100 mm edta [13] unless there is a metal which is very ttghfly bound, this suggests that the neurospora enzyme has neither a class i nor a class ii aldolase mechanism lily pollen synthase is not affected, positively or negatively, by tris-edta up to 100 mm mgc12 inhibited at a concentration of 10 mm and above possibly due to precipitation of the enzyme mnc12 had no affect at concentrations of 0 2 and 2 mm thus, the pollen synthase does not fit the definition of a class ii aldolase either would be enriched to the level of the medmm, 7 3% since it as evadent that the substrate takes up 180 by exchange with the medmm, presumably to form 1-[1ao1 ]glucose- 6-p, these molecules can form l-6-[1801 ]myo-mosatol to some extent by hydrolysis of the schlff base with h2160 this would raase the percentage of 1801-myo-mosltol molecules to even greater than 7 3% since the observed amount is 1 6 + 0 1% the schfff base route is clearly excluded in the case of the neurospora experament the 180 exchanged mto glucose-6-p as trapped m sorbltol if no exchange into glucose-6-p occurred prior to the syn- thase reaction, and the reactmn took the schfff base route, 18 8% of the molecules measured would be 2-[1801 ]myo-mosltol, greater than the 14 4+ 1 3% ob- served since 1 ao_uptake by glucose-6-p dad occur during the mcubatmn some of thas would be mcorporated mto myo-mosltol the rate of exchange relatwe to the syn- thase rate is not known but at as clear that some add~- tmnal 180, would be taken up the maxamum amount would be the sum of the 6-[18oa ]- and 2-[1801 ]rnyo- mosatol molecules, calculated to be 27 3% 303 '~ ~', 200 ~\\ '\\ }oo~ 0~ 005 01 05 -1 5 10 50 10c metal concentration ram fig 1 the effect of dwalent metal ions on bovine testis myo-mosltol-l-phosphate synthase purified through the de-52 stage incubations were camed out with mgc12 a, mnc12 *, cobr2 o, cac12 a, cdc12 ~, znc2h3022 *, cuc2h3022 0 for 1 h and analyzed colonmetncally concentrations where the enzyme appeared to preclpltate are indicated by a broken line we find with bovine synthase that potassium- edta is 2 7-fold stlmulatory at 5 mm but thas must be due to actwatlon by k which has been reported with tbas enzyme [6] when trls-edta is incubated with the bovme synthase there is no effect to a con- centration of 10 mm followed by a 1 5-fold stimula- tion at 50 mm rat testis myo-mosltol-l-phosphate synthase is slmdarly unaffected by edta [14] this suggests, but does not prove, that there is no divalent metal ion associated with either the bovme enzyme or that from rat testis mg2+ actwatlon of bovine testis myo-mosltol-1- phosphate synthase occurs up to concentrations of 10 mm fig 1 a slmdar stimulation occurs with mn2+ other metal ions which could be considered to be candidates for class ii aldolase activation were found to be reactive co2+ and ca2 or mtubltory cd2+, cu2and zn2+ it is thus possible that the neurospora myolnosltol-l-phosphate synthase and that from bovine testis are both being affected in a similar way, stimulated by, but not dependent on, certain dwalent cations there xs some controversy over the mechanism of the aldolase step m the myo-mosttol-l-phosphate synthase from rat testis ptttner and hoffmann- ostenhof [15] have reported that glucose-6-p can be 304 linked covalently to the enzyme following nabh4 treatment of an incubation mixture the authors report that the glucose-6-p is joined to a lyslne residue through c-1 of the glucose [16] work from another laboratory [14] suggests that this is an arti- fact, the result of an adventitous reaction between glucose and lyslne residue on a protein which co- purifies with rat testis synthase furthermore, on purely theoretical grounds, the formation of a schiff base between d-glucose-6-p and the synthase at c-1 does not produce an lntermedmte wtuch would facilitate the aldol reaction in this manner suggested by rutter [17] the myo-lnosltol-l-phosphate synthase from lemna glbba is reported to be completely inhibited by 10 mm edta, the inhibition being reversed by mn2but not by mg2[18] another myo-mositol-1- phosphate synthase, from streptomyces grlseus is inhibited by 60 mm edta, and this lnlubmon is reversed by mgz[19] these enzymes thus may be class ii aldolases of the classical variety, with clear divalent metal dependence both bovine and rat testis myo-inos~tol-l-phos- phate synthase have been found to be stimulated by kand it has been proposed that this activation is due to stablhzatlon of the carbamon generated in the step prior to c1-c6 condensation [i] since this is the formal function of the schaff base and of the divalent metal m classical aldolases, perhaps the enzymes from rat, ox, neurospora and l1hum have a previously undescnbed mechanism of action which is facilitated by certain monovalent cations such as nh4 and k+ but which is also functional m their absence in support of this it is known that both 5-ketoglu- cose [20] and 5-ketoglucose-6-p [9,10] are readily cychzed by dilute base to form inososes which can be reduced to lnositols the occurrence together, in one molecule, of an aldehyde c-l and an activated methylene c-6 may be sufficient for the c1-c6 condensation to requare no further stabilization in ttus case, the enzyme would function as an oxidore- ductase, as a basic catalyst and to provide the stereo- specificity necessary to produce only the myo-mosltol isomer perhaps the classical categones of aldolases types are too limiting it is becoming evident that aldolases cto not foiiow stnct phylogenetic lines w~th respect to mechanism using [2-180] fru-p as a substrate to detect schiff base formation in fru-p aldolases from several sources, heron and caprioh showed euglena gractlhs to simultaneously elaborate both class i and class ii enzymes [21] they also found that the fru-p2 aldolase of l casez formed a schlff base, thus showing that bactena as well as higher organisms can elaborate a class i aldolase whether the myo-lnosltol- 1-phosphate synthases we have studaed utihze another type of mechanism or have a tightly bound metal thus, being a special type of class ii enzyme has not been established however, the posslbd~ty that the class i and class ii mechamsms are not sufficient for all cases should be considered in the study of enzymes using the aldolase reaction acknowledgements m z p wishes to acknowledge the contnbutlon of edgardo escamtlla in the preparation of the neuro- spora enzyme this work was supported by n i h grants gm-22427, ns-05159 and rr-00954 and project 0266, coll agrlc res ctr sp5897, wsu, pullman references 1 sherman, w r, rasheed, a, mauck, l a and wlecko, j 1977 j blol chem 252,5672-5676 2 loewus, m w and loewus, f a 1973 plant physlol 51,263-266 3 loewus, mw and loewus, f a 1971 plant physlol 48,255-260 4 plfia, e and tatum, e l 1967 blocham blophys acta 136,265-271 5 prfia, e, saldafia, y, brunner, a and chagoya, v 1969 ann n y acad scl 165,541-558 6 mauck, l a, wong, y-h and sherman, w r 1980 biochemistry 19, 3623-3629 7 wlecko, j and sherman, w r 1979j arn chem soc 101,979-983 8 wlecko, j and sherman, w r 1976j am chem soc 98, 7631-7637 9 kaely, d e and sherman, w r 1975 j am chem soc 97, 6810-6814 10 k]ely, d e, raordan, j m, abruscato, g j, cantreu, c e and sherman, w r 1978 m cychtols and phospho- mosltldes wells, w w and elsenberg, f, jr, ods , pp 13-22 academic press ine, new york 11 rlttenberg, d and graft, c 1958j am chem soc 80 3370-3372 12 mogyoros, m, brunner, a and plfia, e 1972 blochma blophys acta 289,420-427 13 plfia, m z, castanedo, c, escamtua, e and thfia, e 1978 m cychtols and phosphomosltldes wells, w w and elsenberg, f, jr, eds pp 297-310, academic press, new york 14 maeda, t and elsenberg, f, jr, 1980 j blol chem 255, 8458-8464 15 lhttner, f and hoffmann-ostenhof, o 1976 hoppe- seyler's z physlol chem 357, 1667-1671 16 pittner, f and hoffmann-ostenhof, o 1978hoppe- seyler's z physlol chem 359, 1395-1400 17 rutter, w j 1964fed proc am soc exp blol 23, 1248-1257 305 18 ogunyeml, e o, plttner, f and hoffmann-ostenhof, o 1978 hoppe-seyler's z physlol chem 359,613-616 19 plttner, f, tovarova, i i, kormtskaya, e y, khoklov, as and hoffmann-ostenhof, o 1979 mol cell blochem 25, 43-46 20 ktely, d e and fletcher, h g, jr 1969 j org chem 34, 1386-1390 21 heron, e j and capnoh, r m , 1975 blochlm blophys acta 403,563-572 22 loewus, m w 1977 j blol chem 252, 7221-7223 bbaee 81901741 0005-27448190174-1 101016/0005-27448190174-1 studies on myo-inositol-1-phosphate synthase from lilium longiflorum pollen, neurospora crassa and bovine testis further evidence that a classical aldolase step is not utilized william r sherman a mary w loewus b marta z piña c yun-hua h wong a a department of psychiatry and biological chemistry, washington university school of medicine, st\n",
      "\n",
      "elution of adsorbed material from the column was performed by changing the eluant either to 8 m urea to elute all adsorbed pro- tein or to buffer 1 containing 50 mm carnosine in attempts to specifically elute carnosine synthetase\n",
      "\n",
      "the major difference between this prepar- ative method and earlier ones is the current inclusion of the reducing agent dithiothreitol and the protease inhibitor phenylmethanesulfonyl fluoride in the extraction buffer\n",
      "\n",
      "gmp synthetase was assayed as follows a cocktail was prepared containing 1 vol atp 40 mm 1 vol l-glutamine 40 mm 2 vol 15m tris-chloride buffer ph 76/200 mm magnesium sulfate 2vol\n",
      "\n",
      "serial jl 272805 291210 291847 291857 31 biochimica et biophysica acta bba - enzymology biochimicaetbiophysicaactabbaenzymology 2003-02-03 2003-02-05 2010-04-03t111942 1-s20-0005274481902886 0005-27448190288-6 0005274481902886 101016/0005-27448190288-6 s350 s3502 head-and-tail 1-s20-s0005274400x01362 2015-05-13t19484886767-0400 0 0 19810414 1981 2003-02-03t000000z rawtext articleinfo crossmark dco dateupdated tomb dateloaded datesearch indexeddate issuelist volumelist yearnav articletitlenorm authfirstinitialnorm authfirstsurnamenorm cid cids contenttype copyright dateloadedtxt docsubtype doctype doi eid ewtransactionid hubeid issfirst issn issnnorm itemstage itemtransactionid itemweight openaccess openarchive pg pgfirst pglast pii piinorm pubdatestart pubdatetxt pubyr sortorder srctitle srctitlenorm srctype volfirst volissue webpdf webpdfpagecount affil articletitle auth authfirstini authfull authkeywords authlast otherkwds primabst ref alllist content subj ssids 0005-2744 00052744 658 658 2 2 volume 658, issue 2 1 177 190 177 190 19810414 14 april 1981 1981-04-14 1981 converted-article fla copyright © 1981 published by elsevier bv inactivationcrystallinetobaccoribulosebisphosphatecarboxylasebymodificationarginineresidues23butanedionephenylglyoxal chollet r chollet 1977 155 159 r andrews 1978 1 9 t kung 1980 326 336 s johal 1979 75 77 s akazawa 1979 208 229 t photosynthesisii encyclopediaplantphysiology riordan 1977 884 886 j riordan 1979 71 92 j lawlis 1978 580 585 v schloss 1978 5626 5631 j chu 1974 556 559 d chollet 1976 344 351 r riordan 1973 3915 3923 j takahashi 1968 6171 6179 k takahashi 1977 395 402 k takahashi 1977 403 414 k chollet 1978 1267 1274 r chollet 1979 153 r augustus 1979 377 379 b chollet 1977 228 240 r bowman 1980 652 657 l kohlbrenner 1978 7609 7611 w chollet 1978 455 467 r kung 1977 89 94 s powers 1975 2616 2620 s procnatlacadsciusa kantrowitz 1976 2688 2695 e philips 1979 296 304 m cheung 1979 541 547 s daemen 1974 2865 2871 f cheung 1979 940 947 s frigeri 1977 3147 3152 l saluja 1978 361 363 a kameshita 1978 795 803 i kantrowitz 1977 2873 2880 e hayman 1978 4161 4168 s pal 1976 437 443 p cholletx1981x177 cholletx1981x177x190 cholletx1981x177xr cholletx1981x177x190xr item 0005-27448190288-6 0005274481902886 1-s20-0005274481902886 101016/0005-27448190288-6 272805 2010-09-27t195857537976-0400 1981-04-14 1-s20-0005274481902886-mainpdf https//s3amazonawscom/prod-ucs-content-store-us-east/content/pii0005274481902886/main/application/pdf/e91f8de35024664b4679b803cc55c806/mainpdf https//s3-eu-west-1amazonawscom/prod-ucs-content-store-eu-west/content/pii0005274481902886/main/application/pdf/e91f8de35024664b4679b803cc55c806/mainpdf mainpdf pdf true 937666 main 14 1-s20-0005274481902886-main_1png https//s3-eu-west-1amazonawscom/prod-ucs-content-store-eu-west/content/pii0005274481902886/preview/image/png/49475f5e6da03e440455118d8a886cd5/main_1png https//s3amazonawscom/prod-ucs-content-store-us-east/content/pii0005274481902886/preview/image/png/49475f5e6da03e440455118d8a886cd5/main_1png main_1png png 60258 849 656 image-web-pdf 1 177 biochimica et biophysica acta, 658 1981 177-190 elsevier/north-holland biomedical press bba 69217 inactivation of crystalline tobacco ribulosebisphosphate carboxylase by modification of arginine residues with 2,3-butanedione and phenylglyoxal raymond chollet department of agricultural biochemistry, university of nebraska, lincoln, ne 68583 usa received august 13th, 1980 key words ribulosebisphosphate carboxylase, chemical modification, arginme residue, 2,3-butanedione, phenylglyoxal summary crystalline tobacco ribulosebisphosphate carboxylase 3-phospho-d-glycerate carboxylase dimerizing, ec 41 139 is rapidly and completely inactivated by 2,3-butanedione in borate buffer or phenylglyoxal, reagents which are highly specific for the modification of arginine residues\n",
      "\n",
      "inactivation by phenylglyoxal is enhanced in bicine buffer and partially reversible, whereas inactivation by butanedione is markedly enhanced in borate buffer, irreversible in the presence of borate and partially reversed upon complete removal of borate and excess reagentwhen the modification reaction is performed in the presence of various ligands, only the substrate ribulosebisphosphate and the diphosphorylated competitive inhibitor sedoheptulosebisphosphate protect against inactivation loss of carboxylase activity is directly proportional to incorporation of [14c]phenylglyoxal until about 15% of the initial activity remains\n",
      "\n",
      "results and discusssion modification of rabulosebcsphosphate carboxylase by 2,3-butanedaone treatment of the crystalline tobacco enzyme with butanedione m borate buffer results in a time- and concentration-dependent loss of activity, although the inactivation reaction deviates from pseudo-first-order kinetics [ 16 ] \n",
      "\n",
      "tablei correlation of butanedione inactivation of tobacco ribulosebisphosphate carboxylase with loss of arginine modification of 8 1 mm enzyme by 20 mm butanedione bd was carried out for 65 mm at 30c in 50 mm borate buffer ph 8 0 in the absence or presence of 1 mm nbulose-p2 rbu-p2 the results represent the mean of four independent experiments control activity was 579 nmol h14coi fixed/nun per mg protein enzyme carboxylase arginme argumme arginme activity per modified shielded/protomer % of control protomer per protomer for 100% protection control 100 36 6 bd-modified 20 27 2 9 4 24 rbu-p2 -protected 74 285 81 * calculated from difference in number of arginme groups modified /difference um % activity x 100 183 amino acid analyses of the control and partially inactivated enzymes were performed to identify the basic amino acid residues modified by treatment with butanedione\n",
      "\n",
      "however, for our routine studies it was desirable to avoid the use of this buffer since coz is both an activator and substrate for ribulosebisphosphate carboxylase [ 5 ] comparison of several other systems with buffering capacity around ph 8 bicine, borate, tricme, tris indicated that inactivation of the tobacco enzyme was markedly enhanced in bicine buffer, whereas no inactivation was observed in tris or tricine\n",
      "\n",
      "inactivation of tobacco ribulosebisphosphate carboxylase by 2 6 or 5 1 mm phenylglyoxal in bicine buffer follows pseudo-first-order kinetics until at least 85% of the activity is lost\n",
      "\n",
      "thus, the activity of the partially inactive enzyme can be attributed to the unmodified protein still present rather than to the formation of a modified carboxylase with dif- 700 control rbu-p,protected gel filtration 300 pgo-modified i- h ioo 0 o a rbu-pz protected roo 0 m 2 e0 0 u 60 gel filtration 40 pgo-modified 20 0 0 20 40 60 0 i 2 3 4 5 6 inactivation time reactivation time minutes hours fig 1 reversibility of phenylglyoxal inactivation of tobacco nbulosebisphosphate carboxylase upon removal of excess reagent modification of 8 2 um enzyme by 5 mm phenylglyoxal pgo was earned out at wc in 50 mm bicine buffer ph 8 0 in the absence 0 - or presence a ' of 1 mm nbulose-p2 rbu-p2 after a 15 or 70 mm inactivation period an aliquot was passed through a sephadex g-25 column equilibrated with 50 mm bicine/25 mm nacl ph 7 7 and the protein eluant assayed for carboxylase activity immediately after and up to 5 5 h after gel filtration\n",
      "\n",
      "the control ~ ~ was subjected to identical conditions except that neither pgo nor rbu-p2 was added 185 table ii effect of various ligands on phenylglyoxal inactivation of t intermedius and tobacco ribulosebisphosphate carboxylases modification of the t in termedius enzyme 85 gg/ml by 5 2 mm phenylglyoxal was earned out at 30 in 50 mm bicme buffer ph 8 0 in the absence or presence of 2 mm nbulose-p2 rbu-p2, sedoheptulosepz sed 1,7-p2 or 6 p-gluconate after a 70 mm inactivation period an aliquot was passed through a sephadex g-25 column equilibrated with 50 mm bicme/1 mm na2edta/1 mm dxtluothreitol ph 8 0 and the protein eluant was immediately assayed for carboxylase activity control activity was 1 70 umol hi 4c03 fixed/mm per mg protein modification of the tobacco enzyme 4 4 mg/ml by 5 1 mm phenylglyoxal was earned out for 65 mm at 30 as described above control activity was 590 nmol hi 4c03 faxed/mm per mg protein ligand carboxylase activity % of control premcubated t intermedius tobacco none 8 9 rbu-p2 69 79 sed 1,7-p2 64 69 6p-gluconate 8 22 ferent properties\n",
      "\n",
      "direct titration of free 186 table iii correlation of phenylglyoxal inactivation of tobacco ribulosebisphosphate carboxylase with loss of arginine modification of 8 6 gm enzyme by 4 5 mm phenylglyoxal pgo was earned out for 65 min at 30`c m 50 mm bicine buffer ph 8 0 m the absence or presence of 1 mm nbulose-p2 rbu-p2 the results represent the mean of two independent experiments control activity was 625 nmol h14co g fixed/min per mg protein enzyme carboxylase argmme argmme arginine activity per modified shielded/protomer % of control protomer per protomer for 100% protection control 100 37 6 pgo-modified 15 32 8 4 8 1 9 rbu-p2-protected 87 34 2 3 4 * calculated from difference m number of argmme groups modified /difference m % activity x 100 sulfhydryl groups with nbs2 in the presence of sds [19] demonstrated that there is no irreversible loss of -sh residues in the inactivated or substrate-protected tobacco enzymes\n",
      "\n",
      "amino acid analyses of the control and modified proteins indicated that an 85% loss of carboxylase activity is associated with the table iv correlation of inactivation of tobacco ribulosebisphosphate carboxylase with incorporation of [2-14c]phenylglyoxal in expt i, modification of 7 8iim enzyme by 5 3 mm [2-14c]phenylglyoxal pgo 0 79 ci/mol was carried out at 30 m 50 mm bicme buffer ph 8 0 m the absence or presence of 2 mm nbulose p2 rbu p2 or 6-p-gluconate control carboxylase activity was 659 nmol h14cog fixed/min per mg protein in expt ii, modification of 6 4 tm enzyme by 5 2 mm [2-14c]phenylglyoxal 0 48 ci/mol was earned out as described above control carboxylase activity was 462 nmol h14con fixed/mm per mg protein in expt iii, modification of 8 4 um enzyme by 5 3 mm [2-14c]phenylglyoxal 1 25 ci/mol was earned out as described above m the absence or presence of 2 mm rbu-p2 or sedoheptulose-p2 sed 1,7-pz control carboxylase activity was 940 nmol h14c03 fixed/mm per mg protein expt enzyme time carboxylase [14c]pgo argmme no at 30 activity incorporation shielded/ mm % of control mol/mol protomer for protomer 100% protection i [14c]pgo-modified 5 77 127 12 67 2 32 28 34 3 69 rbu-p2 -protected 5 102 0 73 1 1 12 104 1 43 1 2 28 97 2 45 1 0 6-p-gluconate-protected 28 58 3 87 ii [14c]pgo-modified 15 52 2 87 35 25 4 53 65 6 6 31 rbu-p2-protected 15 97 183 12 35 93 3 04 1 1 65 76 4 14 1 6 iii [14c]pgo-modified 45 15 503 rbu-p2 -protected 45 79 331 13 sed 1,7-pz-protected 45 78 2 87 1 7 * calculated from difference m [14c]incorporated/protomer/difference m % activity x 100, assuming a 21 stoichiometry between phenylglyoxal incorporation and argmme modification [9,13,14] 187 modification of about 5 of the 37 arginine residues present per protomer table iii\n",
      "\n",
      "since both arginine modification and inactivation of the table v effect of phenylglyoxal modification of tobacco ribulosebisphosphate carboxylase on activation by anionic effectors modihcation of 8 4 1tm enzyme by 4 7 mid phenylglyoxal pgo was carved out at 30 in 50 mm bi buffer ph 8 0 m the absence or presence of 1 mm nbulose pz rbu p2 after 50 and 145 mm, aliquots were passed through a sephadex g-25 column and the protem eluant was immediately assayed for carboxylase activity m the absence or presence of 0 1 mm 6p-gluconate or 0 4 mm nadph control activity was 24 amol hi4cog fixed/min per mg protem enzyme carboxylase activity % of control +/- 6-p-gluconate +/- nadph control 100 3 48 2 80 pgo-modified 23 2 46 2 09 <1 - rbu p2-protected 103 2 82 2 20 70 2 85 2 03 190 higher plant-type carboxylase isolated from tobacco leaves and the procaryote t intermedaus are decreased by the substrate rlbulose-p2 or the diphosphorylated competitive inhibitor sedoheptulose-p2, these residues likely serve as binding sites for negatively charged phosphate groups of the non-gaseous substrate\n",
      "\n",
      "inactivation of crystalline tobacco ribulosebisphosphate carboxylase by modification of arginine residues with 2,3-butanedione and phenylglyoxal raymond chollet department of agricultural biochemistry, university of nebraska, lincoln, ne 68583 usa abstract crystalline tobacco ribulosebisphosphate carboxylase 3-phospho-d-glycerate carboxylase dimerizing, ec 41139 is rapidly and completely inactivated by 2,3-butanedione in borate buffer or phenylglyoxal, reagents which are highly specific for the modification of arginine residues\n",
      "\n",
      "the v for acetate activation is strongly influenced by the buffer used, while for propio- nate and butyrate activations this is not the case table i\n",
      "\n",
      "the mutual ratio of the v values for ca , and 64 activation in both buffers is in good agreement with the results of schte et al [5] who measured the acyl-coa synthetase in the same fraction by means of the coenzyme a disappearance under similar condi- tions\n",
      "\n",
      "this phenomenon together with the different behaviour of the v of acetate and propionate activation by change of the assay buffer see table i, made schte et al [5] suppose that propionate is ac- tivated by a distinct propionyl-coa synthetase\n",
      "\n",
      "the major contribution to propionate activation is given by acetyl-coa synthe- tase k^ for acetate 078 mm and km for propionate 640 mm, while a minor contribution is given by a second acyl-coa synthetase km for butyrate 038 mm, depending on the assay buffer, and -km for propionate 640 mm\n",
      "\n",
      "acetyl-coa synthetase showed a sharp ph optimum at 76, similar to that reported for rhodospirillum rubrum [5] and the activity in tris hc1 was almost twice that in potassium phosphate buffer\n",
      "\n",
      "unlabeled ethanolamine 1-phosphate 1 ^mol, water 2 ml, and a 2  1 v/v mixture of chloroform and methanoi 20 ml were added and, after thorough mixing and 436 centrifugation, the upper aqueous phase containing 49 10s cpm was concen- trated at 40c to 5 ml and the unreacted sphinganine 1-phosphate was extract- ed three times with 5-ml portions of tert-amyi alcohol saturated with water\n",
      "\n",
      "ethanolamine 1-phosphate to ethanolamine [i-3] ethanolamine 1-phosphate 43 nmol 107 104 cpm from experi- ment 2 was incubated for 4 h at 37c with alkaline phosphatase from calf mucosa l mg sigma chemical co, lot p81b-091-1 in tris hc1 buffer 005 m ph 74 in a final volume of 02 ml\n",
      "\n",
      "solubilization and parital purification of throm- boxane synthetase 10 ml ofmicrosomes fraction i were mixed with an equal volume of 50 mm potassium phosphate buffer, ph 74, containing 02 mm dithiothreitol, 2 mm edta, 20% glycerin and 1% lubrol px w/v\n",
      "\n",
      "throm- boxane synthetase activity was eluted from the aff gel blue by washing two times with 15 ml of 50 mm potassium phosphate buffer, ph 74, containing 01% lubrol px, 01 mm dithiothreitol, 1 mm edta, 10% glycerin and 2 m kcl\n",
      "\n",
      "specificity of microsomal fatty-acyl-coa ligase the kinetic constants v^^ and the apparent michaelis constant k^ of the fatty-acyl-coa ligase with saturated fatty acids of various chain lengths as well as with elaidic acid 18 1?9 and oleic acid 18lc9 were found from the mea- tablei kinetic constants of microsomal acyl-coa ligase assays were carried out as described in materials and methods\n",
      "\n",
      "fractions 2 ml with 16a-dehydroxylase activity were pooled, the pro- tein was precipitated with nh4,s04 70% saturation, and dialyzed against 25 mm imidazole hc1 buffer ph 74 +1 mm dithiothreitol\n",
      "\n",
      "this acyl-coa synthetase was active with a wide range of substrates ranging from saturated fatty acid palmitic acid to highly un- saturated fatty acids arachidonic acid and eicosapentaenoic acid fig 3\n",
      "\n",
      "acetate incorporation into nonsaponifiable lipds washed cells, harvested in the late logarithmic phase, were resuspended in 25 mm phosphate buffer ph 65 containing 1% glucose\n",
      "\n",
      "2 to 4-106 cells, preincubated in sabouraud medium and collected at the late logarithmic phase, were transferred to 5 ml of 25 mm potassium phosphate buffer ph 65 containing 1% glucose, inhibitors and 2 p ci [^cjacetate 50 ci/mol\n",
      "\n",
      "incubation of ltc4 with y-glutarnyl trans- peptidase g-4756, bovine kidney, sigma was performed in the ph 70 hepes buffer to produce lt\n",
      "\n",
      "gsh s-transferase activity gsh s-transferase activity was assayed as de- scribed by habig and jakoby [24] using dncb as the substrate in a buffer containing 50 mm of both hepes and mes at ph 65\n",
      "\n",
      "the residue was suspended in ph 75 hepes buffer containing 1% bovine serum albumin, and a portion of this lt solution was injected on the c-18 column of the reverse-phase hplc system and eluted with acetonitrile in borate at ph 10\n",
      "\n",
      "heat inactivation of acyl-coa synthetase activities membranes were heated for various times at 45c in isolation buffer supplemented with 667 m m atp, 20 mm mgcl and 20 mm 2- mercaptoethanol [20]\n",
      "\n",
      "activity was measured using 015 mm ^oleate 400000 dpm as the substrate, as de- scribed in materials and methods, and using buffers sodium acetate, sodium maleate, hepes, tris and sodium borate within the ph interval of their p , 1 had no inhibitory effect on oleyl-coa synthetase activity in p knowlesi-infected simian erythro- cytes, as shown in table ii\n",
      "\n",
      "palmitoyl- and stearoyl-coa synthetases dis- played the same inhibition pattern by other fatty acids they were highly ie, with a low k, and competitively inhibited only by the other saturated fatty acid and poorly inhibited by the unsaturated species except for the inhibition of palmitoyl-coa synthetase by oleate, although there was a non- competitive component\n",
      "\n",
      "in the latter case, linoleate, which behaved like arachidonate, would be activated at the same site, specific for polyunsaturated species, whereas palmitate and strate would be activated at the other site\n",
      "\n",
      "unless otherwise stated, the assay mixture contained 50 mm tris-hcl buffer ph 75, 10 mm mgci, 1 mm dtt, 075 mg/ml bovine serum albumin fatty-acid-poor, 20 mm khc03, 10 mm potassium citrate, 375 mm atp, 0125 mm acetyl-coa, 05 mm phosphoeno/pyruvate 0125 mm nadh, 80 u/ml pyruvate kinase, 44 u/ml lactic dehydrogenase and either medica 16, or medica 16-coa as stated in a final volume of 08 ml\n",
      "\n",
      "chemicals phosphate-buffered saline pbs, sucrose, tris base, dithiothreitol, 1-chloro-2,4-dinitrobenzene cdnb, ethy- lenediaminetetraacetic acid edta disodium salt, triton x-100, reduced glutathione gsh, magnesium chloride, n-ethylmaleimide, eia reagent grade gelatin, bovine serum albumin essentially globulin-free and rat serum albumin essentially fatty acid-free were purchased from sigma st\n",
      "\n",
      "the resulting supernatant, which contained ltc4 synthase ac- tivity, was applied to a mono q hr 5/5 anion-exchange column, which had been equilibrated with running buffer containing 20 mm tris-hcl ph 74, 1 mm edta, 4 mm gsh, 1 mm dithiothreitol, 01% w/v taurocholic acid and 05% w/v n-octylglucoside, at a flow rate of 1 ml/min\n",
      "\n",
      "this gene belongs to the class of early inducible genes, which are generally rapidly turned abbreviations txa2, thromboxane a fcs, foetal calf serum pgh2, prostaglandin h csf-1, colony stimulating factor-1 pghs, prostaglandin h synthase pdgf, platelet derived growth factor il-1, interleukin-1 egf, epidermal growth factor pge2, prostaglandin e pgi2, prostacyclin pbs, phosphate-buffered saline tbs, tris-buffered saline aa, arachidonic acid phla2, phospholipase a\n",
      "\n",
      "m orlandi et al /biochimica et biophysica acta 1215 1994 285-290 289 table 1 inactivation of txa2 synthase by pgh2 in human monocytes and platelets a pgh2 6 fim l 100 nm 1st incubation 2nd incubation 3rd incubation 27 100% 36  > 100% 21 78% 06 100% 07  > 100% 04 76% 01 100% 009 75% 009 75% b pgh2 6u,m l^m 100 nm 1st incubation 2nd incubation 3rd incubation 29 100% 98 34% 36 12% 8 100% 44 55% 22 27% 09 100% 06 56% 03 34% monocytes a and washed platelets b, suspended in krebs buffer containing 1 mm edta, were incubated for 10 min at 37 in the presence of different concentrations of pgh2\n",
      "\n",
      "the cyclase was solubilized by smooth shaking of the membranes m buffer b 10 mm potassium phosphate ph 70, 2 mm dtt, 20% v/v glyceroi, 15% v/w chaps for l h at 30 followed by centrifugation at 100 000 x g for 90 min\n",
      "\n",
      "results and discussion purification of the squalene-hopene cyclase the addition of sucrose and ascorbic acid to the membrane extraction buffer enhanced the stability of the enzyme, while inhibitors for proteases did not show any influence\n",
      "\n",
      "the reaction mixture contained in a total volume of 10 ml tris-acetate buffer ph 65, mo µmoles glucose, 40 µmoles glucose i-phosphate, i µmole nadp+, 03 µmole glucose-6-phosphate dehydrogenase, i unit and enzyme\n",
      "\n",
      "ribonuclease activity was assayed in reaction mixtures containing, in a total volume of i ml, 3 mg yeast rna and 4*moles of phosphate buffers of various ph, in the presence +-+ or absence o-o of 2/~m atp\n",
      "\n",
      "the possibility that the two synthetases might differentially acylate these isoacceptor trnas was investigated by the acylation of a small quan- tity of trna with one of the two synthetases the other synthetase then was added after the trna appeared to be saturated with tyrosine ie plateau or 'complete' biochim biophys dcta, 281 1972 33 44 tyrosyl-trna synthetases of pea roots 41 acylation to determine if a new saturation level could be established\n",
      "\n",
      "the partially purified asparaginyl-trna synthetase 120 units of transfer activity 816 mg protein specific activity, 0147 was placed on a column at 4 of deae-cellulose 268 x 2 cm, in a buffer of tris\n",
      "\n",
      "a ribonuclease rnaase cl ec 31423, ribonucleate 3'-oligonucleo- tide hydrolase was extracted by edta/acetate buffer, ph 56 from acetone- dried cells of candida lipolytica and purified 1350-fold by acetone and nh42so4 fractionation, deae-cellulose and deae-sephadex chromato- graphy\n",
      "\n",
      "the nh42 so4 fraction was chromatographed on a deae-cellulose column 2 cm 30 cm, flow rate = 20 ml/h, equilibrated with 002 m sodium phosphate buffer, ph 64 and the rnaase activity was divided into two frac- tions by elution with a linear 00--015 m nac1 gradient in the equilibration buffer\n",
      "\n",
      "the 1st deae-sephadex fraction was rechromatographed on a deae- sephadex column equilibrated with 01 m tris hc1 buffer, ph 72 fig 1 and the rnaase active fraction was collected, dialyzed and called the purified rnaase preparation 210 ml\n",
      "\n",
      "rnaase assay was carried out according to a previous report [8] using acetate buffer at ph 56 during the purification, however, the acetate buffer was replaced by 02 m citrate buffer, ph 60, to examine the properties of purified enzyme\n",
      "\n",
      "endo- or exo-nucleolytic cleavage by rnase cl was determined as fol- lows in a reaction vessel were added 5 #1 of poly[3h] cytidylatesolution {143 nmol phosphate/pl, 50 pl of rnaase cl 20 units/ml, and 50 ~1 of 02 m sodium citrate buffer, ph 60\n",
      "\n",
      "protein rag activity unit % unit/rng protein solvent-dehydration buffer extract 4660 56 828 100 12, 2 acetone fraction 1176 61740 108 526 nh42so4 fraction 202 49 836 88 2465 deae-cellulose fraction 118 33 289 59 3505 1st deae-sephadex fraction 193 25202 44 13 298 2nd deae-sephadex fraction 082 13 440 24 16 452 purified rnaase preparation 495 e ~oc z 5o rnaqse 0j c o < v -0c6c o q_ o  0 t t adsorption o08m naci 01m nqci 0 139 410 6oo 8oo 1000 ml trough column  fig i reehromatography of rnaase on deae-sephadex a-25\n",
      "\n",
      "a comparison of synthetic homopolymers as substrate showed that polyadenylic acid is cleaved at half the rate of polycytidylic acid, while polyguanylic acid was table ii effect of metal ions on rnaase activity the relative activity was assayed using sodium acetate buffer, ph 56, at a final concentration of 01 m\n",
      "\n",
      "the reaction mixture contain- ing 02 ml of 02 m sodium citrate buffer, ph 60, 01 ml of 15 mm as mononucleotide of polynucleotide, 01 ml of water and 01 ml of x12 dilution of purified rnaase in a total volume of 05 ml was incubated at 87 for the indicated time period, 497 08 100\" degradation value for ~_p_ya  l poly,~ poly u __ _~___o o~  ~z  e \" ipoly u 1o !r  o4- / / ,a-/ /o // [3 polya poly u o ' ~ ~ ~ 2b incubation period hrs\n",
      "\n",
      "a ribonuclease rnaase cl ec 31423, ribonucleate 3′-oligonucleotide hydrolase was extracted by edta/acetate buffer, ph 56 from acetone-dried cells of candida lipolytica and purified 1350-fold by acetone and nh42so4 fractionation, deae-cellulose and deae-sephadex chromatography\n",
      "\n",
      "the nuclease, 12 000 activity units in 45 ml of 001 m tris hc1 buffer, ph 75, was applied to a col- umn 09 12 cm of sepharose 4b to which 18 pmol of putrescine had been bound\n",
      "\n",
      "extraction of the synthetases of muscle with a buffer containing 4 mm adenosine triphosphate atp resulted in a 50--100% increase in the activities of glutaminyl-, glutamyl-, isoleucyl-, leucyl-, lysyl-, methionyl- and prolyl- trna synthetases in the posrtibosomal fraction, over those activities extracted in the absence of atp\n",
      "\n",
      "half of the solution 25 1 of the diluted polyethyleneglycol phase, or 1 1 of the 100 000 x g supernatant was then adsorbed on a 7 cm x 30 cm deae-cellulose type de-52 column, which was then washed with 1 1 of 10 mm potassium phosphate buffer, ph 72\n",
      "\n",
      "fraction deae-cellulose obtained from the chromatography of the 5-1 fraction of the diluted poly- ethyleneglycol phase or the 2 1 one of the 100000 x g supernatant was adsorbed onto a 35 cm x 35 cm column of phosphocellulose whatman pll equilibrated with a 10 mm potassium phosphate buffer, ph 70\n",
      "\n",
      "each fraction from the hydroxyapatite column was diluted 130 times a or 1000 times b in the aminoacylation mixture and the amounts of aminoacyl-trna determined in 50-pl aliquots after 10 rain incubation at 30   a280nm,  - molarity of phosphate buffer, m = synthe- tase activity\n",
      "\n",
      "step 5 for alanyl-trna synthetase the hydroxyapatite column 4 x 12 cm was eleuted with a linear gradient of potassium phosphate buffer, ph 75, from 002 to 07 m total volume of the gradient 1 1 for leucyl-trna synthetase the cm-sephadex column 42 x 42 cm was eluted with a linear gradient from 01 to 03 m kc1 in a 002 m potassium phosphate buffer, ph 65, total volume of the gradient 200 ml\n",
      "\n",
      "the reaction, which was carried out at 30 in a final volume of 075 ml containing 67 mm kc1, 77 mm mgso4, 32 mm edta, 18 mm adp,048 mm nadh, 38 mm triethanolamine buffer ph 75, 15 units/ml pyruvate kinase and 36 units/ml lactate dehydrogenase, was started by addition of 1 mm 2-phosphoglycerate\n",
      "\n",
      "the enolase activity on the strips was detected by laying them on similar sized pieces of whatman no 1 filter paper wetted with a modified assay mixture 67 mm kc1, 77 mm mgso4, 32 mm edta, 18 mm adp, 10 mm nadh, 38 mm triethanol- 287 amine buffer ph 75, 15 units/ml pyruvate kinase, 36 units/ml lactate dehydrogen- ase and 1 mm 2-phosphoglycerate\n",
      "\n",
      "purification of rat muscle enolase approx 100 g of fresh skeletal muscle from male adult rats was collected and homogenized in a waring blender with 4 vol of 15 mm sodium phosphate buffer ph 70, containing 5 mm mgso4\n",
      "\n",
      "a 20 lag muscle enolase 120 units/nag on 45 % acrylamide gels in tris-hcl-glycine buffer ph 89, stained with amido black\n",
      "\n",
      "tryptic digestion and peptide analysis apo-tryptophanase was dialyzed first for 2 h against 5 m guanidine - hcl in 01 m potassium phosphate buffer ph 68, then for 12 h against distilled water\n",
      "\n",
      "105 mg reduced and aminoethylated eland ribonuclease were cleaved by trypsin in 02 m n-ethylmorpholine acetate buffer, ph 80, for 4 h at 37 c enzyme substrate = 1 500, by weight and 183 mg oxidized eland ribonuclease were cleaved by thermolysin in 02 m ammonium acetate buffer, ph 80, for 2 h at 37 c enzyme substrate --- 1 50, by weight\n",
      "\n",
      "experimental extraction and purification of alfalfa ribulose-l,5-bisphosphate carboxylase was carried out at 4 in 002 m potassium phosphate buffer *, ph 75, con- taining 10-~m edta and 002% nan3\n",
      "\n",
      "for this purpose 950 mg of aldolase in 38 ml of 10 mm triethanolamine buffer, ph 74, containing 25 mm edta was treated with 15 ml of 05 m dihydroxyacetone phosphate solution at 0\n",
      "\n",
      "for carboxypeptidase digestion 01-02 pmol of peptide or protein was dried with a stream of nz at the bottom of a conical tube, dissolved with 01-02 ml of 02 m n-ethylmorpholine acetate buffer, ph 80, and treated with 2 pl of carboxypeptidase a 4 pg/pl solution in 2 m licl at room temperature\n",
      "\n",
      "absorption spectrum of lysine decarboxylase 177 mg per ml in 001 m potassium phosphate buffer, ph 62\n",
      "\n",
      "the reaction mixture 0-15 ml contained 5 pmoles of 2-keto-3-deoxy-l-arabonate, 1 pmole of mgc12, 015 pmole of 2-thioethanol, 007 pmole of nadh, 13 pgrams of lactic acid dehydrogenase, 15 pmoles of hepes buffer 12, ph 78, and enzyme\n",
      "\n",
      "the reaction mixture 040 ml for the cleavage reaction contained 197 pmoles of 2-keto-3-deoxy-l-arabonate, 2 pmoles of mncl2, 03 mole of 2-thioethanol, 6 pmoles of hepes buffer 12, ph 74, and 03 mg of 17-fold purified 2-keto-3deoxy-l-arabonate aldolase\n",
      "\n",
      "the reaction mixture 1o ml for the condensation reaction contained 100 pmoles of sodium pyruvate, 100 pmoles of glycolaldehyde, 5 funoles of mnc12, 12 pmoles of 2-thioethanol, 15 pmoles of hepes buffer, ph 74, and 06 mg of 17-fold purified 2-keto-3-deoxy-l-arabonate aldolase\n",
      "\n",
      "thin-layer chromatography of derivatives of the 2-keto-3-deoxy-l-arabonate cleavage products 2,4-dinitrophenylhydrazone color after rf va ues of koh spray 1st dimension 2nd dimension pyruvate standard purple-brown 038 o4o cleavage product a purple-brown 037 o41 glycolaldehyde standard yellow-brown 059 052 cleavage product b yellow-brown o6o 052 a reaction mixture 10 ml containing 10 jimoles of 2-keto-3-deoxy-larabonate, 5 pmoles of mgc12, 735 pmoles of 2-thioethanol, 15 pmoles of hepes buffer, ph 74, and 05 mg of 17-fold purified 2-keto-3-deoxy-larabonate aldolase was incubated at 27°c for 1 hour\n",
      "\n",
      "z, biochem prep, 4, 56 1955• _ eto-3-deoxy-l-arabonate cleavage products 2,4-dinitrophenylhydrazone color after rf va ues of koh spray 1st dimension 2nd dimension pyruvate standard purple-brown 038 o4o cleavage product a purple-brown 037 o41 glycolaldehyde standard yellow-brown 059 052 cleavage product b yellow-brown o6o 052 a reaction mixture 10 ml containing 10 jimoles of 2-keto-3-deoxy-larabonate, 5 pmoles of mgc12, 735 pmoles of 2-thioethanol, 15 pmoles of hepes buffer, ph 74, and 05 mg of 17-fold purified 2-keto-3-deoxy-larabonate aldolase was incubated at 27°c for 1 hour\n",
      "\n",
      "crude extract 800 460 1105 0138 1 5000 1st deae 136 60 560 411 29 3350 2nd deae 058 590 415 720 52 2450 3rd deaf 180 154 1135 655 474 1850 amberlite øc-50 0056 155 915 162 1170 1410 sephadex g-75 0110 40 320 290 2100 1305 ribonuclease activity was assayed in a reaction mixture containing per ml 3 mg yeast rna and 40 p moles of phosphate buffer ph 57 at 30°c\n",
      "\n",
      "phosphodiesterase activity was measured by determining the absorbancy of liberated p-nitrophenol at 420 mp employing bis-g-nitrophenyl phosphate 05 mm in 01 m acetate buffer ph 57 as the substrate\n",
      "\n",
      "purification of threonine aldolase from c humicola threonine aldolase was assayed in a reaction mixture containing enzyme, 50 pmoles of l-threonine, 02 pmoles of pyridoxal phosphate, 400 pmoles of kc1 and 400 pmoles of tris-chloride buffer, ph 86, in a total volume of 35 ml\n",
      "\n",
      "the crystalline threonine aldolase preparation gave a single band on acrylamide gel electrophoresis carried out in tris-glycine buffer, ph 83, davis, 1964 containing 1 x 10-2 m kc1, 5 x 10-3 m mercaptoethanol and x 10-5 m pyridoxal phosphate\n",
      "\n",
      "absorption spectrum of crystalline threonine aldolase 276 mg per ml in 003 m potassium phosphate buffer, ph 64, containing 5 x 10 3 m mercaptoethanol and 1 x 10 3 m edta\n",
      "\n",
      "absorption spectrum of crystalline threonine aldolase 276 mg per ml in 003 m potassium phosphate buffer, ph 64, containing 5 x 10 3 m mercaptoethanol and 1 x 10 3 m edta\n",
      "\n",
      "a saturated solution of ammonium sulfate containing 50 mm trizma base was added dropwise to crude supernatant at 50c precipitated material was collected at 20%, 40%, 60%, and 75% saturation by centrifugation at 35,000 x g for 15 min\n",
      "\n",
      "further, at ph 70, ligase i was more active in 2-n-morpholinoethanesulfonic acid - koh buffer than in tris-hc1 buffer, while the reverse was observed for vol 53, no 3, 1973 biochemical and biophysical research communications 180, 120 e a u 60 10 20 fraction no\n",
      "\n",
      "the holoenzyme of diol dehydrase is not resolved into apoenzyme and coenzyme b12 by gel filtration on sephadex c-25 using potassium phosphate buffer containing both kc1 and substrate 2, 3\n",
      "\n",
      "o-carbamyl-l-serine deaminase was assayed in mixtures containing 01 mm pyridoxal-5'-phosphate and 50 mm o-carbamyl-l-serine in 100 mm sodium borate buffer ph 85 in a final volume of 01 ml\n",
      "\n",
      "thin-layer chromatography of the 2,4-dinitrophenylhydrazones of the aldolase cleavage products 2,4-dinitrophenylhydrazone of i k0li spray ilst dimension 2nd dimension pyruvate standard cleavage product 1 glycolaldehyde standard cleavage product 2 040 042 050 050 a reaction mixture 10 ml containing 20 4moles 3-deoxy-d-pentulosonic acid, 10 pnoles mnc12, 10 pmoles bicine buffer, ph 80, and 2 mg of 32-fold purified aldolase was incubated at 300c for 2 hours\n",
      "\n",
      "stoichiometry of the aldolase-mediated reaction initial concentrations i equilibrium concentrations 3-deoxy-d- pyruvate glycolaldehyde 3-deoxy-d- pyruvate glycolaldehyde pentulosonic pentulosonic acid acid 500 0 0 365 131 128 0 100 100 821 163 169 millimolarities the reaction mixture 10 ml for the cleavage reaction contained 5 pmoles of 3-deoxy-d-pentulosonic acid, 5 amoles mnc12, 10 pmoles bicine buffer, ph 80, and 1 mg of 32-fold purified 3-deoxy-d-pentulosonic acid aldolase\n",
      "\n",
      "the reaction mixture 10 ml for the condensation reaction contained 100 j„unoles sodium pyruvate, 10 4moles glycolaldehyde, 5 ~jmoles mnc12, 10 ~jmoles bicine buffer, ph 80, and 1 mg of 32-fold purified 3-deoxy-d-pentulosonic acid aldolase\n",
      "\n",
      "d-xylose dehydrogenase was assayed spectrophotometrically at 340 nm the reaction mixture 015 ml contained 25 4moles d-xylose, 033 15 llmoles bicine buffer, ph 80, and enzyme\n",
      "\n",
      "005 ~lmole nadh, 10 pgrams lactic acid dehydrogenase, 5 idmoles bicine buffer, ph 80 and enzyme\n",
      "\n",
      "the solution contained 74 nmoles of pyridoxal phosphate, 15 umoles of glutathione, 120 umoles of alanine, 25 umoles of potassium phosphate buffer pd 81 in 045 ml of d20 and alanine racemase 2 units in 10 uliters of water\n",
      "\n",
      "the reaction mixture contained 74 nmoles of pyridoxal phosphate, 2 umoles of sodium pyruvate, 120 umoles of l-alanine, 25 umoles of potassium phosphate buffer pd 81 in 045 ml of d20 and transaminase 93 units of gpt or 18 units of got in 10 miters of water\n",
      "\n",
      "gmp synthetase was eluted with a gradient of potassium chloride from 005 to 04 m in the above buffer\n",
      "\n",
      "gmp synthetase was eluted with a gradient of potassium chloride from 005 to 04 m in the above buffer\n",
      "\n",
      "gel filtration in 6 m urea citrate lyase, dissociated in 6 m urea containing 10 mm dtt and 10 mm sodium phosphate, ph 70 11 mg protein in 08 ml was subjected to gel filtration on a bio-gel a-5 m 200-400 mesh column 15 x 90 cm equilibrated with the same solvent\n",
      "\n",
      "2-benzimidazolecarboxylate, 2,3-pyridinedicarboxylate, and 4,5-imidazoledicarboxylate, when present at concentrations equal to that of saturating substrate, all fail to inhibit proline racemase\n",
      "\n",
      "whole tissue homogenates used in the isolation of prenyl transferase enzyme activity were prepared as described by green and west 13, except the homogenizing buffer was 50 mm potassium phosphate in 10 mm potassium pyrophos phate, ph 63\n",
      "\n",
      "the assay for triose phosphate isomerase contained 023 mm r,s glyceraldehyde-3-phosphate, 01 mm nadh, 7 units of a-glycerophosphate dehydrogenase, and 50 mm triethanolamine buffer, ph 79 in a total volume of 10 ml\n",
      "\n",
      "residual [14c]acid-labile products in the reaction mixtures, and geranyl and farnesyl transferase activities, were all assayed by previously described methods 13 routine assays of farnesyl transferase employed fpp in place of gpp, 257 glycerol was included in the assay and the buffer was adjusted to ph 63\n",
      "\n",
      "acetyl-coa for iso table i reverse phase tlc analysis of isoprenoid products derived from [4-\"c]ipp exp 1 exp 2 exp 3 activity proplastid proplastid mitochondrion fraction fraction fraction sdh a od/min/200 ul 006 0 085 tpi a od/min/10 pl 037 128 046 farnesol synthesis 183 cpm 115 cpm 169 cpm geranylgeraniol synthesis 522 cpm 584 cpm 688 cpm 02 ml enzyme aliquots of each fraction were mixed with [14c]ipp 21,000 cpm and phosphate buffer, ph 63, made up in 01% triton x-100 in a total volume of 06 ml and allowed to stand overnight at room temperature\n",
      "\n",
      "acetyl-coa for iso table i reverse phase tlc analysis of isoprenoid products derived from [4-\"c]ipp exp 1 exp 2 exp 3 activity proplastid proplastid mitochondrion fraction fraction fraction sdh a od/min/200 ul 006 0 085 tpi a od/min/10 pl 037 128 046 farnesol synthesis 183 cpm 115 cpm 169 cpm geranylgeraniol synthesis 522 cpm 584 cpm 688 cpm 02 ml enzyme aliquots of each fraction were mixed with [14c]ipp 21,000 cpm and phosphate buffer, ph 63, made up in 01% triton x-100 in a total volume of 06 ml and allowed to stand overnight at room temperature\n",
      "\n",
      "a uridine nucleotide, isolated from the reaction mixture after incubation of udp-g1cnac with component a as described under materials and methods, was identified as udp-mannac by the following criteria hydrolysis with 001 n hc1 for 20 min at 1000, as well as treatment with snake venom phosphodiesterase followed by alkaline phosphomonoesterase, released a radioactive compound coincident with n-acetylmannosamine electrophoretically 50 mm sodium borate buffer, ph 95 and table 2\n",
      "\n",
      "cystathionine synthase activity was eluted by washing with portions 650ml of the starting buffer containing potassium chloride concentrations of o1m, 02m and 03m\n",
      "\n",
      "the reaction mixture 3 ml contained 15 ml phosphate buffer 01 m, ph 74, 05 ml enzyme preparation 005 ml in case of cellfree extract and 03 ml isocitric acid 001 m as substrate\n",
      "\n",
      "coli b cell pack was suspended in 20 ml of 02 m potassium phosphate buffer ph 70 containing 001 m 2-mercaptoethanol, 10% glycerol, 10-4 m phenylmethylsulfonylfluoride and 6000 kunitz units of deoxy ribonuclease\n",
      "\n",
      "other properties of the methionvl-trna synthetase the enzyme activity was found to be absolutely dependent on mg t+ 10 mm mg2+ was the optimal concentration\n",
      "\n",
      "isocitrate dehydrogenase nadp activity, - the enzyme activity was measured in tris buffer, ph 74 by the method of ruffo et al 9\n",
      "\n",
      "716 vol 71, no 3, 1976 biochemical and biophysical research communications the various pieces of evidence which clearly demonstrate that the reactions described in equations i and ii are catalysed by proteins distinct from isocitrate dehydrogenase are a the isocitrate synthase system has an absolute requirement for atp, b the increase in the ratio of isocitrate synthase to isocitrate dehydrogenase activity measured in the forward direction at successive steps in the purification procedure, c the greater sensitivity of the synthase activity to mild heat inactivation, d its response to dietary manipulation, and e the inhibition by 80 um palmitoyl coa of the synthase activity by 42% whilst having no effect on the activity of isocitrate dehydrogenase at this concentration\n",
      "\n",
      "at saturating levels of aryl-coa substrates the geranyl-coa carboxylase preparation carboxylated geranyl-coa and 3-methylcrotonyl-coa at relative rates of 10 and 025 respectively fig 3a this ratio of activities is 748 vol 71, no 3,11076 biochemical and biophysical research communications 15 10 is 10 is 0s 4 figure 1\n",
      "\n",
      "the carboxylase peaks obtained from deae-cellulose fig 1 were concentrated by ammonium sulfate precipitation, dissolved in 3-ml of buffer a and chromatographed on a column of sepharose 4b 25 x 80 cm in buffer a 3-ml fractions were collected\n",
      "\n",
      "results and discussion separation and characterization of two types of adenylosuccinate synthetase from rat liver fresh rat liver 40 g was homogenized with 3 vol of 005 m tris-hc1 buffer ph 74 containing 01 mm dithiothreitol dtt\n",
      "\n",
      "reaction mixtures, buffered with 50 mm imidazole ph 70, contained methylglyoxal synthase 005 mg/ml, and various concentrations of bromo- and iodoacetol phosphate at 37°c\n",
      "\n",
      "the mixture was then neutralized with 005 ml of 1 m tris base and diluted with the radioimmunoassay buffer for further measurement of txb2 immunoreactivity\n",
      "\n",
      "contained, in a final volume of 01 ml 25 mm tris-hcl buffer, ph 72, 3 mm cacl2 or mgcl2, 30 mm or 300 mm kc1, 1 mm y-[32p]atp, and 0003 unit/ml of atpase\n",
      "\n",
      "the standard assay mixture contained, in a final volume of 03 ml 25 mm tris-hc1 buffer, ph 72, 3 mm mgc12, 4 mm or 30 mm kc1, 1 mm atp, 1 mm phosphoenolpyruvate, 02 mm nadh, 2 units/ml pyruvate kinase, 125 units/ml lactate dehydrogenase,,and about 0001-0003 unit of atpase\n",
      "\n",
      "relative activity % dialysis additions before dialysis after irradiated dark none 100 81 90 nahs03 42 26 89 thioglycolate 21 89 89 thioglycolate + nahs03 22 80 87 inactive purified urocanase 395 ug protein/ml in 33 mm potassium phosphate buffer, ph 74, total volume 40 ml, was irradiated by the standard method with near uv for 15 min\n",
      "\n",
      "absorption spectra in 1 ml cells with 10 mm path length -a-, photoactivated urocanase, 236 mg protein/ml in 01 m potassium phosphate, ph 74, vs buffer only -b-, the same solutions brought to 1 mm sodium thioglycolate\n",
      "\n",
      "difference spectrum taken in tandem cells each with two, 1 ml, 4 mm path length compartments reference cell photoactivated urocanase,224 mg protein/ml, 33 mm potassium phosphate, ph 74 / 1 mm sodium thioglycolate, same buffer sample cell photoactivated urocanase, 224 mg protein/ml, 1 mm sodium thioglycolate, same buffer / same buffer\n",
      "\n",
      "absorption spectra in 1 ml cells with 10 mm path length -a-, photoactivated urocanase, 236 mg protein/ml in 01 m potassium phosphate, ph 74, vs buffer only -b-, the same solutions brought to 1 mm sodium thioglycolate\n",
      "\n",
      "difference spectrum taken in tandem cells each with two, 1 ml, 4 mm path length compartments reference cell photoactivated urocanase,224 mg protein/ml, 33 mm potassium phosphate, ph 74 / 1 mm sodium thioglycolate, same buffer sample cell photoactivated urocanase, 224 mg protein/ml, 1 mm sodium thioglycolate, same buffer / same buffer\n",
      "\n",
      "it is of interest that the hydroperoxide except those of unsaturated fatty acids is a potent inhibitor of pgi2 synthetase\n",
      "\n",
      "percent arginyl-tfna synthetase x elution without mgc12 elution with 5 mm mgcl2 at 016 1rm trna at 16 pm trna at 5 11m trna trnaa3g 85 90 65 trnaasp 30 60 15 trnaphe 27 70 1 trnaser - 30 - trnaval 12 34 i 0 x tfna eluate + kc1 eluate input activity 15 pg of arginyl-trna synthetase fraction in 50 mm tris-hcl buffer ph 75 containing 1 mm dtt were loaded onto a blue-dextran-sepharose column 05 x 27cm\n",
      "\n",
      "for surveying the tissue distribution of leucine 2,3-aminomutase, the organs were suspended in a volume of 002 m potassium phosphate buffer, ph 70, containing 08 percent triton x-100 equal to twice the tissue weight\n",
      "\n",
      "the importance of cryoprotectants present in studies on carbamoyl phosphate introduction carbamoyl phosphate synthetase i of liver mitochondria has been reported to have an absolute requirement for acetylglutamate1 or an analogue for the synthesis of carbamoyl phosphate, it has been suggested that acetylglutamate may be directly involved in the catalytic process although attempts to demonstrate a covalent intermediate have been unsuccessful 1,2\n",
      "\n",
      "3-chloro-d-alanine hydrogen chloride-lyase, in the presence of an ammonium chloride buffer, has an optimal reactivity of around ph 85 and at 40°c for the synthesis of d-cysteine fig 2a and 2b\n",
      "\n",
      "synthesis and isolation of s-substituted d-cysteine derivatives large scale synthesis of d-cysteine-related amino acid was carried 810 vol 111, no 3, 1983 biochemical and biophysical research communications out at 30°c in 100 ml of reaction mixture containing 10 mmol of 3chloro-d-alanine, 50 mmol of mercaptan, 10 mmol of potassium phosphate buffer ph 75 and 100 units of 3-chloro-d-alanine chloridelyase\n",
      "\n",
      "to verify the configuration of synthesized amino acids, each of the synthesized cysteine derivatives 1 pmol was incubated in 20 ml of the reaction mixture containing 75 units of d-amino acid oxidase, 03 pmol of fad, 400 units of catalase and 05 mmol of potassium phosphate buffer ph 80 at 30°c for 1 h with shaking\n",
      "\n",
      "substrate km vmax m umol/mg/min 3-chloro-d-alanine 61 x 10-5 311 d-cysteine 18 x 10-4 1057 s-methyl-d-cysteine 23 x 10-4 75 s-ethyl-d-cysteine 52 x 10-4 44 s-_n-propyl-d-cysteine 46 x 10-4 28 s-n-butyl-d-cysteine 70 x 10-4 44 s-phenyl-d-cysteine 32 x 10-4 1104 s-benzyl-d-cysteine 39 x 10-4 16 s-2-hydroxyethyl-d-cysteine 14 x 10-2 417 s-2,3-dihydroxypropyl-d-cysteine 32 x 10-2 177 various amounts of each d-cysteine derivative were incubated at 30°c in the reaction mixture 2 ml containing an appropriate amount of the enzyme, 100 umol of potassium phosphate buffer ph 80, 01 umol of pyridoxal 5'-phosphate, 026 umol of nadh 3 units of lactate dehydrogenase\n",
      "\n",
      "substrate km vmax m umol/mg/min 3-chloro-d-alanine 61 x 10-5 311 d-cysteine 18 x 10-4 1057 s-methyl-d-cysteine 23 x 10-4 75 s-ethyl-d-cysteine 52 x 10-4 44 s-_n-propyl-d-cysteine 46 x 10-4 28 s-n-butyl-d-cysteine 70 x 10-4 44 s-phenyl-d-cysteine 32 x 10-4 1104 s-benzyl-d-cysteine 39 x 10-4 16 s-2-hydroxyethyl-d-cysteine 14 x 10-2 417 s-2,3-dihydroxypropyl-d-cysteine 32 x 10-2 177 various amounts of each d-cysteine derivative were incubated at 30°c in the reaction mixture 2 ml containing an appropriate amount of the enzyme, 100 umol of potassium phosphate buffer ph 80, 01 umol of pyridoxal 5'-phosphate, 026 umol of nadh 3 units of lactate dehydrogenase\n",
      "\n",
      "aliquots of purified ubiquinol-cytochrome _c reductase in chloroform were dried and dispersed in 10 mm tris-acetate buffer, ph o 78 containingol mn edta by sonication and incubated at 0 -o-o-, 23 -x-x- and 37 at indicated times, samples were withdrawn and activity assayed at room temperature\n",
      "\n",
      "carbamoyl phosphate synthetase i from mammalian liver mitochondria has a nearly absolute requirement for n-acetyl-l-glutamate, an allosteric ac tivator 1\n",
      "\n",
      "and developed using the solvent system ethyl acetateacetic acidhexanewater 54122560, organic phase\n",
      "\n",
      "since vitamin k-dependent carboxylase rapidly loses activity in the absence of dtt, buffers used for enzyme purification contained 1 mm dtt\n",
      "\n",
      "assay of the enzyme dhba decarboxylase was assayed as follows the typical assay mixture 1 ml contained 50mm sodium acetate buffer ph 52, 2 mm dhba and the enzyme\n",
      "\n",
      "spectral properties of dhba decarboxylase in 50mm so ti- um acetate buffer ph 52 28°c\n",
      "\n",
      "cytidylate cyclase was solubilized with 06 %w/v doc, followed by dialysis against the homogenizing buffer to remove doc\n",
      "\n",
      "modification of the enzyme by diethylryrocarbonate depc n-bromosuccinimide nbs and n-ethximaleiimidenem1-dhba decarboxylase 225 gm in 50 mm sodium phosphate buffer, ph 70, was mixed with different concentrations 1,2,3 and 4 mm of depc freshly prepared in ethanol and incubated at room temperature 28 oc\n",
      "\n",
      "reversal of depc modification by hydroxyla in - dhba decarboxylase 5 gmwas inactivated with 4 mm depc in 50 mm sodium phophate buffer at ph 70 for 10 min\n",
      "\n",
      "the oxidase preparation was dissolved in 50 mm tris-hcl buffer, ph 80, containing 01 octylthioglucoside and 03 m nacl, at the haem _a concentration of 072 wm\n",
      "\n",
      "the standard fapase assay includes preincubation of the enzyme in a hepes-buffered solution containing potassium and magnesium, and in the absence or presence of ouabain\n",
      "\n",
      "we decided, therefore, to preincubate the enzyme with ouabain in a buffered solution containing only magnesium, and to add potassium necessary for the fapase reaction along with fap see materials and methods\n",
      "\n",
      "the determination of udp-n-acetylglucosamine 4-epimerase activity two-ml incubation mixture contained 2 fcmol udp-g1cnhac 04 pmol nad 50 gmol tris-hci buffer, ph 80 4 amol mgc12 and postmicrosomal supernatant\n",
      "\n",
      "printed in great britain short communications inhibition of nicotinate phosphoribosyltransferase in human platelet lysate by nicotinic acid analogs received 21 october 1970 accepted 26 february 1971 the human platelet accumulates radioactivity by a saturable process when incubated with 7-^l*c] nicotinic acid\n",
      "\n",
      "assay of the medium chain acyl-coa synthetase the activities of the purified medium chain acyl-coa synthe- tase with various carboxylic acids were determined by a cou- pled enzyme assay, which monitors the formation of ampt the reaction mixture consisted of each carboxylic acid 1 umol, nadh 036 umol, coa 06 umol, atp 5 umol, mgcl^ 20 u,mol, phosphoenolpyruvate 1 umol, kc1 012 umol, myokinase l u, pyruvate kinase l u, lactate de- hydrogenase l u in 15 ml of 02 m tris-hcl buffer ph 85\n",
      "\n",
      "o-ala dehydratase was assayed as described by labbe [18] but 025 m hepes buffer, ph 70, replaced the citrate- phosphate buffer used in the original method\n",
      "\n",
      "the activity of 8-ala dehydratase in 025 m hepes at ph 68 to 72 was com- parable to that in the citrate-phosphate buffer\n",
      "\n",
      "the supernatant, which contained most of the dhs reductase, was discarded, and the dehydroquinase was eluted from the gel by a similar treatment with m/io phosphate buffer at ph 7o\n",
      "\n",
      "fl-hydroxybutyryl coa racemase activity was assayed by following dpn+ reduction in this system components in #moles diol buffer ph 95, ioo dpn+, o25 i---boh-s-coa, 003 crystalline fl-hydroxy- butyryl dehydrogenase, ool5 mg and enzyme\n",
      "\n",
      "after the cells were harvested they were resuspended in phosphate-calcium buffer and the cyanase activity was measured manometrically\n",
      "\n",
      "fraction protein i ~h/- ldh citrate na+, k +- 2\",3\"-cyclic molarity of carnosine synthase atpase nucleotide sucrose 3\"-phosphohydrolase cytoplasm <03 19 57 30 62 33 3 02 0 0 0 0 synaptic vesicles ~03, <08 14 2216 2619 00 i 01 12t09 myelin ~06, <08 13 5 04 3 02 0 0 33 25 53 41 membranes >08, <10 14 6t04 302 504 4230 1914 synaptosomes ~l0, ~12 30 1003 401 3411 2408 1405 mitochondria ~12 10 1 01 202 58t58 00 202 105 table iv elution of [3h]¢arnosine from a sephadex g-50 column [3h]carnosine or blue dextran were eluted from a 15 × 25 cm sephadex g-50 column with 025 m sucrose buffer\n",
      "\n",
      "the standard assay conditions used for dihydrofolate synthetase from n gonorrhoeae are as follows the total volume of 200 il of reaction mixture was composed of 100 mm tris-hcl buffer ph 93 25 mm 2-mercaptoethanol 50 mm kci 5 mm mgc12 5 mm atp 100 pm dihydropteroate 1 mm l-['4c]glutamic acid 05 pci of uniform labeled and 5-10 units* of enzyme\n",
      "\n",
      "however the subsequent pyridoxal phosphate dependent lyase aldolase absolutely requires the 2s, 3r configuration for the cleavage into the respective long-chain aldehyde and phosphorylethanolamine\n",
      "\n",
      "kc1 3 af was added to increase the conductivity to that of 300 mafkcl and the solution was diluted with a buffer 20 maf hepes, ph 75, 02 maf edta, 2 maf dtt, 02 maf pmsf, and 1 /ml leupeptin containing 300 maf kc1 to lower the protein concentration to 5 mg/ml\n",
      "\n",
      "to stain for serine sulphydrase activity, the fo- cused gel was submerged in 150 ml of reaction mixture containing l-cysteine 33 mm, lead acetate 033 mm, 2-mercaptoethanol 284 mm, and tris-hci buffer, ph 85 100 mm, and incubated at 37 c for approximately 1-15 hr\n",
      "\n",
      "the staining for homocys- teine desulphurase was carried out similarly by sub- merging the focused gel in 150 ml of reaction mixture containing dl-homocysteine 33 mm, lead acetate 033 mm, 2-mercaptoethanol 284 mm, and tris- hci buffer, ph 75 100 mm, and incubated at 37 c for approximately 1-15 hr\n",
      "\n",
      "these results suggest that odc from p falciparum bears some structural differences as compared to the mammalian enzyme, o 1988 academic press, inc index descriptors and abbreviations plasmodium falciparum malaria poly- amines putrescine ornithine decarboxylase odc dl-a-difluoromethylornithine dfmo parasitized erythrocytes pe phosphate-buffered saline pbs\n",
      "\n",
      "ornithine decarboxylase from crude cytosolic prep- arations was purified on an affinity chromatography column 17 x 12 cm of pyridoxamine 5'-phosphate coupled to affi-gel 10 beads boucek and lembach 1977 which was washed with 20 vol of cold buffer 10 mm tris, ph 75 5 mm dithiothreitol and 01 mm edta\n",
      "\n",
      "ornithine decar- boxylase was eluted 05 ml/min with buffer containing 10 jam pyridoxal 5'-phosphate\n",
      "\n",
      "the ph optimum of nematode acti- vated l-serine sulphydrase was determined by mea- suring enzyme activity at intervals of 05 ph units, using the following buffers citric acid/sodium phos- phate, 100 mm ph 40-65 hepes, 100 mm ph 60- 80 tris-hcl, 100 mm ph 75-90 glycine/naoh, 100 mm ph 90-95 sodium carbonate/sodium hy- drogen carbonate, 100 mm ph 95-105\n",
      "\n",
      "inhibitors were dissolved in either assay buffer or distilled water or, in the case of hydrophobic compounds, dimethylsulph- oxide dmso, and preincubated for l min at 37 in the standard assay mixture containing enzyme solu- tion prior to initiation of the enzyme reaction by the addition of l-cysteine\n",
      "\n",
      "the lower es- timate of absolute vmax derived from the secondary plot of 1/vm^, against i/[l- cysteine] may have been influenced by the restricted activity of the activated l-serine sulphydrase enzyme at the lower of the fixed l-cysteine concentrations\n",
      "\n",
      "by contrast, the absolute k^ of the nematode enzyme for l-cysteine is very similar to that of t vaginalis activated l-serine sulphydrase ^l-cysteine = 0-42 mm thong and coombs 1985\n",
      "\n",
      "crude homogenates of n bra- sitiensis and rat liver were prepared in 20 mm potas- sium phosphate buffer, ph 60 containing 01 mm plp, 1 mm dtt, and 0025% v/v triton x-100, and immediately assayed for activated l-serine sulfhy- drase activity, forward l-serine sulfhydrase activity, and protein content\n",
      "\n",
      "the native m,s of nem- atode-activated l-serine sulfhydrase and cystathionine synthase and rat liver cystathionine synthase were determined by sephadex g-200 gel filtration chroma- tography column dimensions 16 x 90 cm at a flow rate of 12 ml hr\"1, using 50 mm tris-hcl buffer, ph 85 containing 100 mm kc1, 01 mm plp, and 1 mm dtt as the running solution\n",
      "\n",
      "following separation of the activated l-serine sulfhy- drase enzyme and cystathionine synthase from fresh cytosolic extracts of selected nematode species via gel filtration chromatography, the activity peaks of each enzyme were pooled, concentrated, and exchanged into the appropriate assay buffers\n",
      "\n",
      "to reveal isoenzymes of both the activated l-serine sulf- hydrase enzyme and cystathionine synthase, the fo- cused gel was submersed in a reaction mixture con- taining 100 mm tris-hcl buffer ph 85, 33 mm l-cysteine, 30 mm 2-mercaptoethanol, 033 mm lead acetate, and 01 mm plp, and incubated at 37 for 45-60 min\n",
      "\n",
      "the michaelis-menten constants x of isoenzymes of serine threonine dehydratase kmm - 1 isoenzyme serine dehydratase threonine dehydratase 909 133 128 125 isoenzyme i is eluted with lower buffer concentration from a deae cellulose column during purification as described in the text\n",
      "\n",
      "sephadex g'-200 gel filtration chromatography the purified uroporphyrinogen decarboxylase 10 mg was applied to a sephadex g-200 column 16x90 cm equilibrated with 001 m potassium phosphate buffer, ph 68 containing 01 mm dithiothreitol\n",
      "\n",
      "the apparent molecu- lar weights of the two purified and separated uroporphyrin- ogen decarboxylase isoenzymes were determined by chromatography on a calibrated sephadex g-100 column 09 x 55 cm equilibrated and eluted with 05 m kc1 m 10 mm potassium phosphate buffer ph 68 containing 01 mm dithiothreitol\n",
      "\n",
      "axenic t augusta strain 103 atcc-30081 was grown in modified c 13 medium 40 mg l-ascorbic acid, 200 mg glucose, l g liver no 2 us biochem- icals, 267 mg l20 nutritional biochemicals, 32 mg thiomalic acid, 34 g trypticase, 120 mg yeast extract, 16 mg tem-4t heinz and 08 mg cholesterol in 100 ml of distilled water buffered to ph 7 with koh\n",
      "\n",
      "samples were defined as 5 ~,1 of purified glutamine synthetase diluted with an equal volume of sds loading buffer 1% sds h tumani et al /journal of immunological methods 188 1995155-163 157 and 2% 2-mercaptoethanol were heated at 90-100 for 5 min and electrophoresed on a pre-cast 12%o sds polyacrylamide gel novex, san diego, ca according to the method of laemmli 1970\n",
      "\n",
      "antigen capture enzyme immunoassay immulon ii microtiter plates dynatech laboratories, chantilly, va were coated overnight at 4 with 100 lcl of 1 ecg/ml mouse monoclonal antibody to native sheep brain glutamine synthetase chemicon in carbonate buffer 50 mm na2c03, ph 95\n",
      "\n",
      "keywords astrocyte glutamine synthetase cerebrospinal fluid enzyme immunoassay enzyme activity electrophoresis abbreviations csf cerebrospinal fluid fplc fast protein liquid chromatography hap hydroxyapatite hplc high pressure liquid chromatography sds sodium dodecyl sulfate page polyacrylamide gel electrophoresis ief isoelectric focusing bsa bovine serum albumin pbs-t phosphate-buffered saline containing 005% tween 20 igg immunoglobulin g eia enzyme immunoassay deae diethylaminoethyl references banik and hogan, 1983 nl\n",
      "\n",
      "triton-binding see helenius et al, 1979 was measured by subjecting 8 mg of nadh  q reductase protein in 0-4 ml of 0-05 m- tris-acetate ph 7-0 and 0-05% v/v triton to size- exclusion hplc on a 7-5 mm x 600 mm ultro-pac tsk g-4000 sw lkb column in a buffer containing 0-05 m- tris-acetate ph 7-0 and 0-05% r-'h]triton spec\n",
      "\n",
      "the triton-binding of nadh  q reductase was measured by subjecting the preparation to size- exclusion hplc in a buffer containing [^hjtriton\n",
      "\n",
      "crystals of the complex lys25-rnase ti/2',5'-gpg were grown by vapor diffusion from a 1 % w/v protein solution m 10 mm-sodium acetate buffer ph 5 containing 2 mm-cacl, 0-2% w/v nan3 and 48% v/v 2-methyl-2,4-pentanediol as precipitating agent\n",
      "\n",
      "ida-hydroxyprogesterone dehydratase and 16-dehydroprogesterone reductase activities in cell fractions of eubacterium sp no 144 105,000 g preparation 16a-hydroxyprogesteronc dehydratase* nmol 16-dehydroprogesterone/min/mg protein 16-dehydroprogesterone reductaset nmol i-progesterone/min/mg protein supernatant fluid membrane enriched fraction 36 35 <1 <1 * the assay mixture contained 50 mm potassium phosphate buffer ph 68 75 /ig 16a-hydroxyprogesterone c^oh, 25% v/v and 05 mg protein supernatant or 05 mg protein membrane in a final volume of 025 ml\n",
      "\n",
      "the 3/?-hydroxysteroid dehydrogenase activity was assayed spectrophotometrically at 25 by measuring the time-dependent change in absorbance at 340 nm in 100 mm potassium phosphate buffer ph 75, 20% glycerol, 01 mm edta, 1 mm nad+ and 01 mm dha\n",
      "\n",
      "the steroid isomerase activity was assayed at 25 by measuring the time-dependent increase in absorb- ance at 248 nm due to 4-ene-3-oxosteroid pro- gesterone formation in loomm potassium phos- phate buffer ph 75, 20% glycerol, 01 mm edta, 01 mm nad4^ and 20//m 5-progesterone\n",
      "\n",
      "effect of fsba on the 3js-hydroxysteroid dehydrogenase and steroid isomerase activities of the purified enzyme 3/hydroxysteroid dehydrogenase activity % steroid isomerase activity % control fsba-treated 100 133 100 168 the purified enzyme was incubated with 27 mm fsba in 560^1 of 100 mm potassium phosphate buffer ph 75 containing 20% glycerol, 01 mm edta and 10% dmso at 25 for 30min\n",
      "\n",
      "the isomerase activities reported in table 1 were assayed spectrophotometrically 241 nm at 37 using 5-pregnene-3,20-dione 200 ^m or 5-androstene-3,17-dione 500  as substrate with nad\"1' 005 mm and purified enzyme 001 mg in 002 m potassium phosphate buffer, ph75, 40% methanoi 10ml total volume\n",
      "\n",
      "the kinetic constants for the isomerase substrates were determined in duplicate by the spec- trophotometric assay 241 nm described above except that five concentrations of 5-pregnene-3,20- dione 50-150^m or 5-androstene-3,17-dione 150-350 , 50 of purified enzyme, and 3^-hydroxysteroid dehydrogenase/isomerase 787 002 m potassium phosphate buffer, ph 74, were used\n",
      "\n",
      "a linear phosphate concentration gradient 006-025 m of the wash buffer without sodium cholate eluted the 3-hsd/isomerase activity fig 2\n",
      "\n",
      "all other chemicals were of best grade available carnosine-synthetase~rep#fation tissues were homogenized in 25 volumes of a 50 mm tris hci buffer, ph 74, containing 32 mm sodium bicarbonate, 20 mm glucose, and 5 mm dithioery- thritol\n",
      "\n",
      "adenylate cyclase activity was determined in a similar manner by employing the reaction mixture consisted of 50 mm tris-hc1 buffer ph 74, 50 mm atp, 4 mm mncl2, 10 mm theophylline and an enzyme solution\n",
      "\n",
      "in brief, the antibody vol 52, no 1, 1993 cytidylate cyclase and g-protein 15 specific to ccmp was adjusted to the protein concentration at 3 jig/ml in 005 m phosphate buffer ph 68-05 m naci-5 mm mgclz buffer a\n",
      "\n",
      "effect of gtp and gdp analogues on cyclase activity brain membrane preparation 1-2 mg protein was preincubated with various concentrations of gtprs at 30°c for 30 min in a total volume of 400 ml of 30 mm tris-hc1 buffer ph 75-l mm dtt\n",
      "\n",
      "effect of bacterial toxin on cyclase activity brain membrane preparation 1-2 mg protein was preincubated with various concentrations of activated iap at 30°c for 30 min in a total volume of 500 ul of 30 mm tris-hc1 buffer ph 752mm dtt-60 mm nad-06 mm atp-60 um gtp\n",
      "\n",
      "an extraction procedure has been developed which yields high sesquiterpene cyclase activity by in- cluding in the extraction buffer polyethylene glycol peg to bind the polyphenols [i1], as well as a non- ionic detergent tween 80 and a substrate analogue tetrasodium pyrophosphate to stabilize the enzyme\n",
      "\n",
      "typically, sesquiterpene cyclase activity was found in phenyl-agarose fractions containing 50 or 0 mm nh42so4 in 50 mm tris buffer, indicating that the enzyme is, like other terpene cyclases, rather hydro- phobic\n",
      "\n",
      "following electrophoresis, 1-mm-thick gel slices were incubated in hepes buffer containing 01% tween 80 to displace the sds and allow the cyclase to renature\n",
      "\n",
      "to renature the cyclase, each slice was transferred to a microcentrifuge tube, homogenized with a plastic pestle in 100 #1 30 mm hepes buffer ph 70, containing 01% tween 80, 10% glycerol and 5mm dtt, and incubated for 2-3 hr\n",
      "\n",
      "steroids, 1991, vol 56, january 13 papers table 1 corticosteroid side-chain isomerase in blood plasma additions none co2 ni2 mouse 5066 201 5368 305 7126 + 5123 rat 7382 191 8376 050\" 7924 + 375 guinea pig 2309 160 2946 424 3233 139\" rabbit 10734 087 2198 382e 1946 +- 055e dog 9912 383 2692 186e 2689 075e pi9 274 112 160 - 023 horse 1649 220 620 165'1 1069 -t- 088-' sheep 2921 191 2414 119-' 2009 086' cow 3225 264 2869 154 3285 153 chicken 2015 138 2615 310 3002 114'' a mixture containing 1,200 pmol of ^i^hidoc in 5 ni of metha- noi, 50 iof pooled blood plasma, 30 y-il of 2 mm of cobalt or nickel chloride, and 25 mm tris-hci buffer, ph 80, in a total volume of 05 ml, was incubated at 37 c for 2 hours and then lyophilized\n",
      "\n",
      "the data in table iii demonstrate that the purified preparation is capable of converting vitamin khz to vitamin k epoxide table i purification of the vitamin k-dependent carboxylase on an antiprothrombin-sephadex column fraction volume protein carboxylase sp act ml mg units ula prot a solubilized microsomes 100 340 439 129 b1 prothrombin antibody gel 10 - 199 - b2 antibody gel supernatant 100 320 379 118 c1 buffer eluate of gel 150 15 45 30 cz propeptide eluate of gel 36 036 224 608 d post elution antibody gel 10 - 86 - carboxylase units are expressed as 10dpm of 1'c02 fixed in the standard assay\n",
      "\n",
      "a land snail was recently shown to contain a peculiar carbamyl-p synthetase 43 which utilizes only glutamine as the nitrogen donor and has an absolute requirement for acetylglutamate\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for i in range(10000):\n",
    "    if L_train[df_train.index[i]][0] == 0:\n",
    "        print(df_train.iloc[i]['sentence'])\n",
    "        print('')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 165,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "476\n",
      "\n",
      "SENTENCE:\n",
      "the catalytic mechanism of ligases proceeds via a number of steps beginning with the hydrolysis of atp or nad+ to yield amp covalently attached to the active site lysine residue i, with the release of pyrophosphate or nicotinamide mononucleotide\n",
      "\n",
      "CHEMICALS:\n",
      "['nad+', 'nicotinamide mononucleotide', 'lysine', 'amp', 'atp', 'pyrophosphate']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 5 POS: 6\n",
      "\n",
      "['includes_structural_words', 'includes_general_chemical_words', 'includes_functional_group', 'structure_sep_or', 'includes_measure_words']\n",
      "\n",
      "['includes_reaction_words', 'includes_reaction_component_words', 'structure_next_to_conversion_words', 'structure_jtsui_pattern_1', 'structure_jtsui_pattern_1_copy', 'includes_reaction_words2']\n",
      "\n",
      "481\n",
      "\n",
      "SENTENCE:\n",
      "this gmp-adduct is equivalent to the amp-adduct formed upon reaction of ligase with atp\n",
      "\n",
      "CHEMICALS:\n",
      "['amp', 'atp', 'gmp']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1  1  1]\n",
      "\n",
      "NEGS: 1 POS: 3\n",
      "\n",
      "['includes_general_chemical_words']\n",
      "\n",
      "['includes_reaction_words', 'includes_react', 'includes_reaction_words2']\n",
      "\n",
      "496\n",
      "\n",
      "SENTENCE:\n",
      "this would explain why the c-terminal domain of t7 ligase is able to stimulate the reduced self-adenylation rate of the n-terminal domain of this enzyme doherty & wigley, 1998\n",
      "\n",
      "CHEMICALS:\n",
      "['c', 'n']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 1 POS: 4\n",
      "\n",
      "['includes_general_chemical_words']\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'includes_emzyze_words', 'includes_reaction_words2']\n",
      "\n",
      "504\n",
      "\n",
      "SENTENCE:\n",
      "like hpr, the histidine residue of ndp kinase is phosphorylated on ndelta1\n",
      "\n",
      "CHEMICALS:\n",
      "['ndp', 'histidine']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1  0 -1 -1  0 -1 -1 -1 -1  0 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1  1  1]\n",
      "\n",
      "NEGS: 1 POS: 2\n",
      "\n",
      "['includes_measure_words']\n",
      "\n",
      "['includes_reaction_words', 'includes_reaction_words2']\n",
      "\n",
      "517\n",
      "\n",
      "SENTENCE:\n",
      "dd-ligase binds adp or atp, and one residue that would reduce the enzyme’s affinity for the guanine nucleotide is trp182\n",
      "\n",
      "CHEMICALS:\n",
      "['atp', 'trp', 'adp']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1  1  1]\n",
      "\n",
      "NEGS: 3 POS: 4\n",
      "\n",
      "['includes_physical_words', 'includes_general_chemical_words', 'structure_sep_or']\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'includes_emzyze_words', 'includes_reaction_words2']\n",
      "\n",
      "521\n",
      "\n",
      "SENTENCE:\n",
      "collier js nishimura affinity labeling of succinyl-coa synthetase from porcine heart and escherichia coli with oxidized coenzyme a disulfide j biol chem 253 1977 4938 4943 connolly 1983 ml\n",
      "\n",
      "CHEMICALS:\n",
      "['oxidized coenzyme a', 'succinyl-coa']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1  0 -1  0  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      "  1  0 -1  1  0 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1  1  1]\n",
      "\n",
      "NEGS: 1 POS: 4\n",
      "\n",
      "['includes_functional_group']\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'includes_emzyze_words', 'includes_reaction_words2']\n",
      "\n",
      "526\n",
      "\n",
      "SENTENCE:\n",
      "effects of phosphoenzyme formation and of substrate binding on the reactivity and stability of the enzyme j biol chem 247 1972 8139 8144 morera et al 1995 s morera m chiadmi g lebras i lascu j janin mechanism of phosphate transfer by nucleoside diphosphate kinase x-ray structures of the phosphohistidine intermediate of the enzymes from drosophila and dictyostelium biochemistry 34 1995 11062 11070 murzin 1996 ag murzin structural classification of proteins new superfamilies curr opin struct biol 6 1996 386 394 murzin et al 1995 ag murzin se brenner t hubbard c chothia scop a structural classification of proteins database for the investigation of sequences and structures j mol\n",
      "\n",
      "CHEMICALS:\n",
      "['phosphate', 'phosphohistidine']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 5 POS: 6\n",
      "\n",
      "['includes_physical_words', 'includes_structural_words', 'includes_general_chemical_words', 'includes_comparison_words', 'includes_measure_words']\n",
      "\n",
      "['includes_reaction_words', 'includes_reaction_component_words', 'includes_react', 'structure_conversion_of', 'includes_emzyze_words', 'includes_reaction_words2']\n",
      "\n",
      "532\n",
      "\n",
      "SENTENCE:\n",
      "this disruptive approach has been successful in other systems, including glutathione reductase scrutton et al, 1992 and malate dehydrogenase breiter et al, 1994\n",
      "\n",
      "CHEMICALS:\n",
      "['malate', 'glutathione']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1  1  1]\n",
      "\n",
      "NEGS: 1 POS: 3\n",
      "\n",
      "['includes_genetic_words']\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'includes_reaction_words2']\n",
      "\n",
      "541\n",
      "\n",
      "SENTENCE:\n",
      "table 3 the effect of mutation of the residue responsible for c4 hydroxy proton abstraction in both classes of fbp-aldolase enzyme cleavage of fbp hexacyanoferrate iii oxidation of intermediate dhap carbanion k mfbp mm k cat min−1 reduction in k catfold k′catdhap min−1 reduction in k′catdhap fold k′catfbp min−1 reduction in k′catfbp fold class i wild-type 002±001 21±17 - 72±11 - 79±14 - d33a class i 008±001 0004±18×10−6 5250 14±026 51 006±001 132 class ii wild-type 019±003 490±26 - 170±5 - 140±5 - d109a class ii 018±003 017±001 2900 20±06 81 nm - kinetic parameters for the wild-type and d33a mutant of the class i fbp-aldolase are taken from morris & tolan 1993 and those for the wild-type class ii enzyme from qamar et al 1996\n",
      "\n",
      "CHEMICALS:\n",
      "['tolan', 'dhap']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1  1  1  1]\n",
      "\n",
      "NEGS: 2 POS: 4\n",
      "\n",
      "['includes_physical_words', 'includes_experiment_words']\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'includes_emzyze_words', 'includes_reaction_words2']\n",
      "\n",
      "544\n",
      "\n",
      "SENTENCE:\n",
      "the wild-type enzyme is able to catalyse the formation of an intermediate carbanion species from either fbp or dhap qamar et al, 1996 in a manner similar to the class i fbp-aldolase from rabbit muscle healy & christen, 1973 morris & tolan, 1993\n",
      "\n",
      "CHEMICALS:\n",
      "['tolan', 'dhap']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      "  1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 1 POS: 4\n",
      "\n",
      "['includes_comparison_words']\n",
      "\n",
      "['includes_reaction_words', 'includes_catalyze_words', 'includes_emzyze_words', 'includes_reaction_words2']\n",
      "\n",
      "545\n",
      "\n",
      "SENTENCE:\n",
      "in that case, the mutation of the residue responsible for c4 hydroxy proton abstraction asp33 morris & tolan, 1993 results in a reduction of k′cat of 51-fold when dhap is the substrate, but of 130-fold when fbp is the substrate compared with an 81-fold reduction with dhap, and a substantial reduction with fbp in the class ii fbp-aldolase reported here table 3\n",
      "\n",
      "CHEMICALS:\n",
      "['tolan', 'dhap']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      "  1  0 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1  1  1]\n",
      "\n",
      "NEGS: 1 POS: 5\n",
      "\n",
      "['includes_comparison_words']\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'includes_reaction_component_words', 'includes_impact_words', 'includes_reaction_words2']\n",
      "\n",
      "547\n",
      "\n",
      "SENTENCE:\n",
      "the finding that the folds adopted by the fuculose-1-phosphate aldolase fuca dreyer & schulz, 1993, 1996 and fructose-1,6-bisphosphate aldolases blom et al, 1996 cooper et al, 1996 were different suggested that these enzymes were not as closely related as had been earlier thought and had probably arisen through a convergent evolution to common aldolase function albeit on slightly different substrates\n",
      "\n",
      "CHEMICALS:\n",
      "['fuculose-1-phosphate', 'fructose-1,6-bisphosphate']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1  0 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1  0 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 3 POS: 4\n",
      "\n",
      "['includes_physical_words', 'includes_comparison_words', 'includes_measure_words']\n",
      "\n",
      "['includes_reaction_words', 'includes_reaction_component_words', 'includes_emzyze_words', 'includes_reaction_words2']\n",
      "\n",
      "559\n",
      "\n",
      "SENTENCE:\n",
      "the protein had glutamine synthetase activity determined according to shapiro & stadtman 1970by measuring the phosphate pi produced during the biosynthesis of glutamine from atp, nh4 + and glutamate results not shown\n",
      "\n",
      "CHEMICALS:\n",
      "['glutamine', 'glutamate', 'phosphate', 'atp', 'nh4']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      "  1 -1 -1 -1  0 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1  1  1  1]\n",
      "\n",
      "NEGS: 3 POS: 4\n",
      "\n",
      "['includes_functional_group', 'structure_sep_comma', 'includes_measure_words']\n",
      "\n",
      "['includes_reaction_words', 'structure_next_to_conversion_words', 'includes_impact_words', 'includes_reaction_words2']\n",
      "\n",
      "562\n",
      "\n",
      "SENTENCE:\n",
      "abstract the enzyme 6-pyruvoyl tetrahydropterin synthase ptps catalyses the second step in the de novo biosynthesis of tetrahydrobiopterin, the conversion of dihydroneopterin triphosphate to 6-pyruvoyl tetrahydropterin\n",
      "\n",
      "CHEMICALS:\n",
      "['6-pyruvoyl tetrahydropterin', 'tetrahydrobiopterin']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1  0 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      "  1 -1 -1 -1  0 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1  1  1  1]\n",
      "\n",
      "NEGS: 1 POS: 6\n",
      "\n",
      "['includes_measure_words']\n",
      "\n",
      "['includes_reaction_words', 'includes_reaction_component_words', 'structure_next_to_conversion_words', 'includes_catalyze_words', 'includes_emzyze_words', 'includes_reaction_words2']\n",
      "\n",
      "564\n",
      "\n",
      "SENTENCE:\n",
      "the subunits of the other trimer are denoted by b, b′, b″ introduction the enzyme 6-pyruvoyl tetrahydropterin synthase ptps ec 46110 catalyses the conversion of dihydroneopterin triphosphate h2ntp to 6-pyruvoyl tetrahydropterin ptp figure 1, the second of three enzymatic steps in the biosynthesis of tetrahydrobiopterin bh4 from guanosine triphosphate gtp\n",
      "\n",
      "CHEMICALS:\n",
      "['gtp', 'bh4', '6-pyruvoyl tetrahydropterin', 'ptp', 'guanosine triphosphate', 'tetrahydrobiopterin']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1  0  0 -1  0 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      "  1 -1 -1 -1  0  0 -1 -1  1 -1  1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1  1 -1 -1\n",
      " -1  1 -1  1]\n",
      "\n",
      "NEGS: 3 POS: 6\n",
      "\n",
      "['structure_adjacent_mentions', 'structure_sep_sym', 'includes_measure_words']\n",
      "\n",
      "['includes_reaction_words', 'includes_reaction_component_words', 'structure_next_to_conversion_words', 'includes_catalyze_words', 'includes_emzyze_words', 'includes_reaction_words2']\n",
      "\n",
      "565\n",
      "\n",
      "SENTENCE:\n",
      "the first step, the conversion of gtp to h2ntp, is catalysed by gtp cyclohydrolase i, the third step, the nadph dependent reduction of the two side-chain keto groups of ptp, by sepiapterin reductase\n",
      "\n",
      "CHEMICALS:\n",
      "['nadph', 'gtp', 'ptp', 'sepiapterin']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1  0  0 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1  1  1]\n",
      "\n",
      "NEGS: 5 POS: 9\n",
      "\n",
      "['includes_genetic_words', 'includes_structural_words', 'includes_functional_group', 'includes_measure_words', 'includes_experiment_words']\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'includes_reaction_component_words', 'structure_next_to_conversion_words', 'structure_conversion_of', 'structure_jtsui_pattern_4', 'structure_jtsui_pattern_4_copy', 'includes_catalyze_words', 'includes_reaction_words2']\n",
      "\n",
      "566\n",
      "\n",
      "SENTENCE:\n",
      "bh4 is the cofactor for several enzymes such as phenylalanine hydroxylase, tyrosine hydroxylase and tryptophan hydroxylase nichol et al, 1985 duch & smith, 1991, nitric oxide synthase known et al, 1989 muhl & pfeilschifter, 1994 and glycerol ether monooxygenase kaufman et al, 1990\n",
      "\n",
      "CHEMICALS:\n",
      "['tryptophan', 'phenylalanine', 'bh4', 'tyrosine']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 3 POS: 4\n",
      "\n",
      "['includes_general_chemical_words', 'includes_functional_group', 'includes_measure_words']\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'includes_emzyze_words', 'includes_reaction_words2']\n",
      "\n",
      "568\n",
      "\n",
      "SENTENCE:\n",
      "et al, unpublished results, in sepiapterin reductase auerbach et al, 1997 and in 7,8-dihydroneopterin aldolase hennig et al, 1998\n",
      "\n",
      "CHEMICALS:\n",
      "['sepiapterin', '7,8-dihydroneopterin']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      "  1 -1 -1  1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 1 POS: 4\n",
      "\n",
      "['includes_genetic_words']\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'includes_impact_words', 'includes_reaction_words2']\n",
      "\n",
      "570\n",
      "\n",
      "SENTENCE:\n",
      "we have not yet determined the exact product of this reaction, but it can be reduced by sepiapterin reductase and afterwards oxidised by iodine to biopterin\n",
      "\n",
      "CHEMICALS:\n",
      "['sepiapterin', 'iodine']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      "  1 -1 -1 -1  0 -1 -1 -1  1  1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1  1 -1 -1\n",
      " -1  1 -1  1]\n",
      "\n",
      "NEGS: 1 POS: 6\n",
      "\n",
      "['includes_genetic_words']\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'includes_reaction_component_words', 'includes_react', 'structure_next_to_conversion_words', 'includes_reaction_words2']\n",
      "\n",
      "572\n",
      "\n",
      "SENTENCE:\n",
      "in the second step the intermediate is converted to ptp by a second base-catalysed proton abstraction from c2′, followed by a triphosphate elimination and tautomerization step\n",
      "\n",
      "CHEMICALS:\n",
      "['ptp', 'triphosphate']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      "  1 -1 -1  1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 2 POS: 8\n",
      "\n",
      "['includes_general_chemical_words', 'includes_measure_words']\n",
      "\n",
      "['includes_reaction_words', 'includes_reaction_component_words', 'structure_next_to_conversion_words', 'structure_conversion_by', 'structure_jtsui_pattern_4', 'structure_jtsui_pattern_4_copy', 'includes_catalyze_words', 'includes_reaction_words2']\n",
      "\n",
      "573\n",
      "\n",
      "SENTENCE:\n",
      "in addition, kinetic studies with nmp have shown that it reacts with ptps yielding a product which reacts with sepiapterin reductase and can be converted after subsequent jodine oxydation to biopterin\n",
      "\n",
      "CHEMICALS:\n",
      "['nmp', 'sepiapterin']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      "  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1]\n",
      "\n",
      "NEGS: 2 POS: 5\n",
      "\n",
      "['includes_genetic_words', 'includes_experiment_words']\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'includes_reaction_component_words', 'includes_react', 'includes_reaction_words2']\n",
      "\n",
      "575\n",
      "\n",
      "SENTENCE:\n",
      "in the case of the nmp substrate this may be the rate-limiting step, because of the loss of the aromatic system in the pyrazine ring\n",
      "\n",
      "CHEMICALS:\n",
      "['nmp', 'pyrazine']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      "  1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1 -1]\n",
      "\n",
      "NEGS: 0 POS: 2\n",
      "\n",
      "[]\n",
      "\n",
      "['includes_reaction_component_words', 'includes_impact_words']\n",
      "\n",
      "598\n",
      "\n",
      "SENTENCE:\n",
      "in the second pathway, cp is coupled to ornithine via ornithine carbamoyltransferase otcase, constituting the sixth step starting from glutamate in the arginine biosynthetic pathway\n",
      "\n",
      "CHEMICALS:\n",
      "['glutamate', 'arginine', 'ornithine']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1  0  0 -1 -1 -1  0 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      "  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1  1 -1  1]\n",
      "\n",
      "NEGS: 1 POS: 2\n",
      "\n",
      "['structure_sep_via']\n",
      "\n",
      "['includes_reaction_component_words', 'structure_next_to_conversion_words']\n",
      "\n",
      "607\n",
      "\n",
      "SENTENCE:\n",
      "in the presence of arginine, the synthesis of ornithine is strongly reduced due to repression of the biosynthetic genes and feedback inhibition of the n-acetyl-l-glutamate synthase, catalyzing the first step of arginine biosynthesis\n",
      "\n",
      "CHEMICALS:\n",
      "['arginine', 'ornithine', 'n-acetyl-l-glutamate']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1  0 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 3 POS: 5\n",
      "\n",
      "['includes_genetic_words', 'includes_structural_words', 'includes_no_terms']\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'includes_reaction_component_words', 'includes_catalyze_words', 'includes_reaction_words2']\n",
      "\n",
      "610\n",
      "\n",
      "SENTENCE:\n",
      "on that basis, the n-terminal half is described as the carboxyphosphate-forming domain by analogy with biotin carboxylase climent & rubio, 1986\n",
      "\n",
      "CHEMICALS:\n",
      "['carboxyphosphate', 'n', 'biotin']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1  0 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1  0  0 -1 -1  1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 1 POS: 3\n",
      "\n",
      "['includes_measure_words']\n",
      "\n",
      "['includes_reaction_words', 'structure_next_to_conversion_words', 'includes_reaction_words2']\n",
      "\n",
      "611\n",
      "\n",
      "SENTENCE:\n",
      "the three mutated cpsases p165s, p170l, and a182v present a reduced affinity for bicarbonate, which could be correlated with the observed growth stimulation of the original strains bearing these mutations, by the addition of co2 to the gas phase mergeay et al, 1974\n",
      "\n",
      "CHEMICALS:\n",
      "['bicarbonate', 'co2']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      "  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1  1  1]\n",
      "\n",
      "NEGS: 4 POS: 5\n",
      "\n",
      "['includes_genetic_words', 'includes_general_chemical_words', 'includes_measure_words', 'includes_experiment_words']\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'structure_next_to_conversion_words', 'structure_conversion_of', 'includes_reaction_words2']\n",
      "\n",
      "614\n",
      "\n",
      "SENTENCE:\n",
      "arginine in the medium also directly affects the rate of enzyme production through repression of the p2 promoter and influences the enzyme activity, as arginine inhibits the synthesis of ornithine, the activator of cpsase\n",
      "\n",
      "CHEMICALS:\n",
      "['arginine', 'ornithine']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      "  1 -1 -1  1]\n",
      "\n",
      "NEGS: 1 POS: 4\n",
      "\n",
      "['includes_no_terms']\n",
      "\n",
      "['includes_reaction_words', 'includes_reaction_component_words', 'includes_emzyze_words', 'includes_reaction_words2']\n",
      "\n",
      "616\n",
      "\n",
      "SENTENCE:\n",
      "the rate of cp synthesis was assayed by measuring the rate of citrulline formation in a coupled assay containing otcase and ornithine using a protocol derived from the method by archibald 1944\n",
      "\n",
      "CHEMICALS:\n",
      "['citrulline', 'ornithine']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1  1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1 -1]\n",
      "\n",
      "NEGS: 2 POS: 3\n",
      "\n",
      "['includes_physical_words', 'includes_experiment_words']\n",
      "\n",
      "['includes_reaction_words', 'includes_creation_words', 'includes_reaction_words2']\n",
      "\n",
      "622\n",
      "\n",
      "SENTENCE:\n",
      "biol 226 1992 367 386 climent and rubio 1986 i climent v rubio atpase activity of biotin carboxylase provides evidence for initial activation of hco3 −by atp in the carboxylation of biotin arch\n",
      "\n",
      "CHEMICALS:\n",
      "['biotin', 'atp', 'hco3']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1  0 -1 -1 -1  0 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1  1  1  1]\n",
      "\n",
      "NEGS: 0 POS: 2\n",
      "\n",
      "[]\n",
      "\n",
      "['structure_next_to_conversion_words', 'structure_conversion_of']\n",
      "\n",
      "633\n",
      "\n",
      "SENTENCE:\n",
      "a closely related enzyme, bisphosphoglycerate mutase bpgam, ec 5424/ec 31313 catalyses the synthesis of 2,3-bisphosphoglycerate 2,3-bpg and hence plays a critical role in controlling haemoglobin oxygen affinity\n",
      "\n",
      "CHEMICALS:\n",
      "['2,3-bpg', 'oxygen', 'bisphosphoglycerate', '2,3-bisphosphoglycerate']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1\n",
      " -1  0 -1 -1  0  0 -1 -1  1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1]\n",
      "\n",
      "NEGS: 3 POS: 5\n",
      "\n",
      "['structure_adjacent_mentions', 'structure_sep_sym', 'includes_measure_words']\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'includes_catalyze_words', 'includes_emzyze_words', 'includes_reaction_words2']\n",
      "\n",
      "644\n",
      "\n",
      "SENTENCE:\n",
      "mutation of glu86 or equivalent also leads to less than 5% of wild-type activity in s cerevisiae pgam table 4 and f2,6bpase lin et al, 1992 with the exception of the conservative glu to asp mutant of f2,6bpase which retains 20% activity\n",
      "\n",
      "CHEMICALS:\n",
      "['glu', 'asp']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 3 POS: 4\n",
      "\n",
      "['includes_comparison_words', 'includes_measure_words', 'includes_experiment_words']\n",
      "\n",
      "['structure_sep_conversion_words', 'structure_next_to_conversion_words', 'structure_conversion_of', 'includes_impact_words']\n",
      "\n",
      "663\n",
      "\n",
      "SENTENCE:\n",
      "the class ii fbp-aldolasepgh complex crystallises in a distinct trigonal form\n",
      "\n",
      "CHEMICALS:\n",
      "['pgh', 'fbp']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 1 POS: 2\n",
      "\n",
      "['includes_genetic_words']\n",
      "\n",
      "['includes_reaction_words', 'includes_reaction_words2']\n",
      "\n",
      "669\n",
      "\n",
      "SENTENCE:\n",
      "mechanism of the class ii fbp-aldolase the aldol condensation carried out by fbp-aldolases involves three major covalency changes\n",
      "\n",
      "CHEMICALS:\n",
      "['aldol', 'fbp']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1  1\n",
      "  1 -1 -1 -1 -1  0 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 1 POS: 3\n",
      "\n",
      "['structure_sep_sym']\n",
      "\n",
      "['includes_reaction_words', 'structure_next_to_conversion_words', 'includes_reaction_words2']\n",
      "\n",
      "675\n",
      "\n",
      "SENTENCE:\n",
      "sect d 50 1994 869 873 mildvan et al 1971 as mildvan rd kobes wj rutter magnetic resonance studies of the divalent cation in the mechanism of yeast aldolase biochemistry 10 1971 1191 1204 morris et al 1996 aj morris rc davenport dr tolan a lysine to arginine substitution at position 146 of rabbit aldolase a changes the rate-determining step to schiff base formation protein eng\n",
      "\n",
      "CHEMICALS:\n",
      "['lysine', 'arginine', 'tolan']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1]\n",
      "\n",
      "NEGS: 2 POS: 5\n",
      "\n",
      "['includes_general_chemical_words', 'includes_experiment_words']\n",
      "\n",
      "['structure_sep_conversion_words', 'includes_reaction_words', 'includes_reaction_component_words', 'structure_next_to_conversion_words', 'includes_reaction_words2']\n",
      "\n",
      "687\n",
      "\n",
      "SENTENCE:\n",
      "we report that rsub is a missense mutation resulting in a glutamate to lysine substitution within the second atp binding site ie cpsb2 domain of cpsase\n",
      "\n",
      "CHEMICALS:\n",
      "['lysine', 'atp', 'glutamate']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1  0 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1  1 -1 -1]\n",
      "\n",
      "NEGS: 1 POS: 3\n",
      "\n",
      "['includes_physical_words']\n",
      "\n",
      "['structure_sep_conversion_words', 'structure_next_to_conversion_words', 'includes_impact_words']\n",
      "\n",
      "689\n",
      "\n",
      "SENTENCE:\n",
      "these reactions are catalyzed by carbamyl phosphate synthetase ii cpsase ec 6355, aspartate transcarbamylase atcase ec 2132, and dihydroorotase dhoase ec 3523, respectively shoaf & jones, 1973\n",
      "\n",
      "CHEMICALS:\n",
      "['carbamyl phosphate', 'aspartate']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1  1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1  1 -1  1]\n",
      "\n",
      "NEGS: 1 POS: 3\n",
      "\n",
      "['includes_measure_words']\n",
      "\n",
      "['includes_react', 'structure_next_to_conversion_words', 'includes_catalyze_words']\n",
      "\n",
      "690\n",
      "\n",
      "SENTENCE:\n",
      "the cpsase domain of cad catalyzes the synthesis of carbamyl phosphate according to the overall reaction scheme 2atp + bicarbonate + glutamine → 2adp + pi + glutamate + carbamyl phosphate\n",
      "\n",
      "CHEMICALS:\n",
      "['pi', 'carbamyl phosphate', 'bicarbonate', 'glutamate']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1  1 -1]\n",
      "\n",
      "NEGS: 2 POS: 4\n",
      "\n",
      "['includes_functional_group', 'includes_measure_words']\n",
      "\n",
      "['includes_reaction_words', 'includes_react', 'includes_catalyze_words', 'includes_reaction_words2']\n",
      "\n",
      "691\n",
      "\n",
      "SENTENCE:\n",
      "the cpsase reaction is rate-determining for the pathway and enzyme activity responds allosterically to both utp and prpp jones, 1980\n",
      "\n",
      "CHEMICALS:\n",
      "['utp', 'prpp']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 1 POS: 2\n",
      "\n",
      "['structure_sep_and']\n",
      "\n",
      "['includes_react', 'includes_emzyze_words']\n",
      "\n",
      "698\n",
      "\n",
      "SENTENCE:\n",
      "substituting ser1406 with glutamate in the cpsb3 domain of hamster cad cpsase was shown to reduce utp inhibition, whereas an alanine substitution of the same residue had no effect banerjei & davidson, 1997\n",
      "\n",
      "CHEMICALS:\n",
      "['alanine', 'utp', 'ser', 'glutamate']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1  0  0 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 1 POS: 3\n",
      "\n",
      "['includes_no_terms']\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'includes_reaction_words2']\n",
      "\n",
      "699\n",
      "\n",
      "SENTENCE:\n",
      "these data suggest that the drosophila cpsase is not regulated by phosphorylation of the serine residue that corresponds to ser1406 in hamster cad\n",
      "\n",
      "CHEMICALS:\n",
      "['serine', 'ser']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1  0  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 2 POS: 3\n",
      "\n",
      "['includes_measure_words', 'includes_experiment_words']\n",
      "\n",
      "['includes_reaction_words', 'structure_next_to_conversion_words', 'includes_reaction_words2']\n",
      "\n",
      "705\n",
      "\n",
      "SENTENCE:\n",
      "for example, the cpsase from e coli is involved in both pyrimidine and arginine biosynthesis, and it responds positively to ornithine and imp but negatively to ump meister, 1989\n",
      "\n",
      "CHEMICALS:\n",
      "['arginine', 'pyrimidine', 'ornithine']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1]\n",
      "\n",
      "NEGS: 1 POS: 3\n",
      "\n",
      "['structure_sep_and']\n",
      "\n",
      "['includes_reaction_words', 'structure_next_to_conversion_words', 'includes_reaction_words2']\n",
      "\n",
      "706\n",
      "\n",
      "SENTENCE:\n",
      "this mutation results in a charge reversal glutamate to lysine substitution at a residue that is universally conserved across species and corresponds to glu780 in e coli cpsase figure 2\n",
      "\n",
      "CHEMICALS:\n",
      "['lysine', 'glu', 'glutamate']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1  0 -1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1  1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1]\n",
      "\n",
      "NEGS: 1 POS: 3\n",
      "\n",
      "['includes_paper_artifacts']\n",
      "\n",
      "['structure_sep_conversion_words', 'structure_next_to_conversion_words', 'includes_impact_words']\n",
      "\n",
      "707\n",
      "\n",
      "SENTENCE:\n",
      "when inhibition resulting from the chelation of mg2+ by utp in our assays is taken into consideration figure 7, it can be concluded that utp feedback inhibition is essentially abolished in the cpsase of sub flies\n",
      "\n",
      "CHEMICALS:\n",
      "['mg2+', 'utp']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1  0  0 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 2 POS: 3\n",
      "\n",
      "['includes_physical_words', 'includes_no_terms']\n",
      "\n",
      "['structure_next_to_conversion_words', 'structure_conversion_of', 'includes_impact_words']\n",
      "\n",
      "708\n",
      "\n",
      "SENTENCE:\n",
      "first, in vitro phosphorylation of ser1406 with protein kinase a shaw & carrey, 1992 or its substitution with glutamate banerjei & davidson, 1997 results in a partial loss of utp inhibition of hamster cpsase\n",
      "\n",
      "CHEMICALS:\n",
      "['utp', 'ser', 'glutamate']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1  0 -1 -1  0 -1 -1  1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1  1 -1 -1\n",
      " -1  1 -1  1]\n",
      "\n",
      "NEGS: 3 POS: 4\n",
      "\n",
      "['includes_no_terms', 'includes_measure_words', 'includes_experiment_words']\n",
      "\n",
      "['includes_reaction_words', 'structure_conversion_of', 'includes_impact_words', 'includes_reaction_words2']\n",
      "\n",
      "716\n",
      "\n",
      "SENTENCE:\n",
      "3, d-06466 gatersleben, germany * corresponding author abstract ribulose-5-phosphate 3-epimerase ec 5131 catalyzes the interconversion of ribulose-5-phosphate and xylulose-5-phosphate in the calvin cycle and in the oxidative pentose phosphate pathway\n",
      "\n",
      "CHEMICALS:\n",
      "['ribulose-5-phosphate', 'xylulose-5-phosphate']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1  0 -1  0  0 -1 -1 -1 -1 -1 -1 -1  0 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1  0  1 -1  1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1  1 -1 -1\n",
      " -1  1 -1  1]\n",
      "\n",
      "NEGS: 2 POS: 6\n",
      "\n",
      "['includes_measure_words', 'structure_sep_and']\n",
      "\n",
      "['includes_reaction_words', 'structure_conversion_of', 'structure_jtsui_pattern_4', 'structure_jtsui_pattern_4_copy', 'includes_catalyze_words', 'includes_reaction_words2']\n",
      "\n",
      "718\n",
      "\n",
      "SENTENCE:\n",
      "the only reported three-dimensional epimerase structure is that from udp-galactose 4-epimerase bauer et al, 1992, which catalyzes the interconversion of udp-galactose and udp-glucose by hydride transfer through the transient reduction of a tightly bound nad+ molecule thoden et al, 1997\n",
      "\n",
      "CHEMICALS:\n",
      "['udp', 'nad+', 'hydride', 'glucose', 'galactose']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1  0  0  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1  1\n",
      " -1 -1 -1  1  0  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1  1  1  1]\n",
      "\n",
      "NEGS: 5 POS: 8\n",
      "\n",
      "['includes_physical_words', 'includes_structural_words', 'includes_general_chemical_words', 'structure_sep_sym', 'structure_sep_and']\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'structure_next_to_conversion_words', 'structure_conversion_of', 'structure_jtsui_pattern_4', 'structure_jtsui_pattern_4_copy', 'includes_catalyze_words', 'includes_reaction_words2']\n",
      "\n",
      "724\n",
      "\n",
      "SENTENCE:\n",
      "sect d 55 1999 225 229 williamson and wood 1966 wt williamson wa wood d-ribulose-5-phosphate 3-epimerase methods enzymol 9 1966 605 608 wilmanns et al 1991 m wilmanns cc hyde dr davies k kirschner jn jansonius structural conservation in parallel beta/alpha-barrel enzymes that catalyze three sequential reactions in the pathway of tryptophan biosynthesis biochemistry 30 1991 9161 9169 wood 1979 t wood purification and properties of d-ribulose-5-phosphate 3-epimerase from calf liver biochim biophys acta 570 1979 352 362 zhang et al 1994 z zhang s sugio ea komives kd liu jr knowles ga petsko d ringe crystal structure of recombinant chicken triosephosphate isomerase-phosphoglycolohydroxamate complex at 18 å resolution biochemistry 33 1994 2830 2837\n",
      "\n",
      "CHEMICALS:\n",
      "['tryptophan', 'd-ribulose-5-phosphate']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1  0  0  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      "  1 -1 -1  1  0  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      "  1 -1  1  1]\n",
      "\n",
      "NEGS: 5 POS: 6\n",
      "\n",
      "['includes_physical_words', 'includes_genetic_words', 'includes_structural_words', 'includes_measure_words', 'includes_experiment_words']\n",
      "\n",
      "['includes_combination_words', 'includes_reaction_words', 'includes_react', 'includes_catalyze_words', 'includes_emzyze_words', 'includes_reaction_words2']\n",
      "\n",
      "738\n",
      "\n",
      "SENTENCE:\n",
      "enzymatic properties of human mtphers atpppi exchange the aminoacylation reaction is a two-step process which results in the covalent attachment of the amino acid to the appropriate trna 1 e+aa+atp ⇋ [eaa-amp]+ppi 2 [eaa-amp]+trna→aa-trna+amp+e in the first step, the enzyme e condenses the cognate amino acid aa and atp forming an aminoacyl-adenylate aa-amp intermediate with the release of pyrophosphate\n",
      "\n",
      "CHEMICALS:\n",
      "['amp', 'pyrophosphate', 'atp']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1  1  1]\n",
      "\n",
      "NEGS: 5 POS: 7\n",
      "\n",
      "['includes_genetic_words', 'includes_structural_words', 'includes_general_chemical_words', 'includes_measure_words', 'includes_experiment_words']\n",
      "\n",
      "['includes_reaction_words', 'includes_reaction_component_words', 'includes_react', 'includes_impact_words', 'includes_creation_words', 'includes_emzyze_words', 'includes_reaction_words2']\n",
      "\n",
      "744\n",
      "\n",
      "SENTENCE:\n",
      "two hyperthermostable enzymes of the tryptophan biosynthesis have been characterized so far phosphoribosyl anthranilate isomerase from the bacterium t maritima tprai and indoleglycerol phosphate synthase from the archaeon sulfolobus solfataricus sigpshennig et al, 1995 sterner et al, 1996 andreotti et al, 1997\n",
      "\n",
      "CHEMICALS:\n",
      "['indoleglycerol phosphate', 'tryptophan', 'phosphoribosyl anthranilate']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1  0 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 1 POS: 3\n",
      "\n",
      "['includes_measure_words']\n",
      "\n",
      "['includes_reaction_words', 'includes_emzyze_words', 'includes_reaction_words2']\n",
      "\n",
      "757\n",
      "\n",
      "SENTENCE:\n",
      "rose s dube rates of phosphorylation and dephosphorylation of phosphoglycerate mutase and bisphosphoglycerate synthase j biol chem 251 1976 4817 4822 23 b rost c\n",
      "\n",
      "CHEMICALS:\n",
      "['phosphoglycerate', 'bisphosphoglycerate']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1  0  0  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      "  1 -1 -1  1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1]\n",
      "\n",
      "NEGS: 1 POS: 3\n",
      "\n",
      "['includes_measure_words']\n",
      "\n",
      "['includes_reaction_words', 'structure_conversion_of', 'includes_reaction_words2']\n",
      "\n",
      "766\n",
      "\n",
      "SENTENCE:\n",
      "several steps of polymerase chain reaction perkin elmer gene amp 2400 of this groe operon region using primers +,−strands containing the mutation atc ile to tgg trp and the m13 sequencing primers p7, p8 of puc118, resulted in a dna fragment containing the desired mutation\n",
      "\n",
      "CHEMICALS:\n",
      "['ile', 'trp']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1  0 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1  1  0 -1 -1 -1  1 -1 -1 -1 -1  1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 3 POS: 4\n",
      "\n",
      "['includes_genetic_words', 'includes_structural_words', 'includes_general_chemical_words']\n",
      "\n",
      "['includes_reaction_component_words', 'includes_react', 'structure_next_to_conversion_words', 'includes_impact_words']\n",
      "\n",
      "769\n",
      "\n",
      "SENTENCE:\n",
      "a carboxylase reaction produces two molecules of 3-phosphoglycerate from one molecule of ribulose 1,5-bisphosphate, carbon dioxide and water\n",
      "\n",
      "CHEMICALS:\n",
      "['carbon dioxide', '3-phosphoglycerate', 'ribulose']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1  0  0  0  0 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1  1  0 -1 -1  0  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1  1 -1  1]\n",
      "\n",
      "NEGS: 3 POS: 6\n",
      "\n",
      "['includes_physical_words', 'includes_general_chemical_words', 'includes_measure_words']\n",
      "\n",
      "['includes_reaction_words', 'includes_react', 'structure_next_to_conversion_words', 'structure_jtsui_pattern_1', 'structure_jtsui_pattern_1_copy', 'includes_reaction_words2']\n",
      "\n",
      "770\n",
      "\n",
      "SENTENCE:\n",
      "rubisco also catalyzes an oxygenase reaction, which generates one molecule of 3-phosphoglycerate and 2-phosphoglycolate from ribulose 1,5-bisphosphate and oxygen, leading to photorespiration\n",
      "\n",
      "CHEMICALS:\n",
      "['oxygen', '2-phosphoglycolate', '3-phosphoglycerate', 'ribulose']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1  0 -1 -1  0  0 -1 -1  0 -1 -1 -1  0 -1 -1 -1  1 -1 -1 -1 -1 -1  1\n",
      "  1 -1 -1 -1 -1 -1 -1 -1  1 -1  1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1  1 -1 -1\n",
      " -1  1 -1  1]\n",
      "\n",
      "NEGS: 6 POS: 7\n",
      "\n",
      "['includes_physical_words', 'includes_genetic_words', 'includes_structural_words', 'includes_general_chemical_words', 'includes_measure_words', 'structure_sep_and']\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'includes_react', 'structure_next_to_conversion_words', 'includes_impact_words', 'includes_catalyze_words', 'includes_reaction_words2']\n",
      "\n",
      "792\n",
      "\n",
      "SENTENCE:\n",
      "the committed step in this pathway, the conversion of s-adenosylmethionine sam to 1-aminocyclopropane-1-carboxylic acid acc, is catalyzed by acc synthase s-adenosyl-l-methionine methylthioadenosine-lyase, ec 44114\n",
      "\n",
      "CHEMICALS:\n",
      "['methylthioadenosine', 's-adenosyl-l-methionine', 'acc', '1-aminocyclopropane-1-carboxylic acid', 'sam', 's-adenosylmethionine']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      "  1 -1 -1 -1 -1 -1 -1 -1  1  1 -1 -1 -1  1  1 -1  1 -1  1  1 -1  1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 5 POS: 9\n",
      "\n",
      "['includes_physical_words', 'includes_general_chemical_words', 'includes_functional_group', 'structure_adjacent_mentions', 'structure_sep_sym']\n",
      "\n",
      "['structure_sep_conversion_words', 'includes_reaction_words', 'includes_reaction_component_words', 'structure_next_to_conversion_words', 'structure_conversion_of', 'structure_jtsui_pattern_4', 'structure_jtsui_pattern_4_copy', 'includes_catalyze_words', 'includes_reaction_words2']\n",
      "\n",
      "793\n",
      "\n",
      "SENTENCE:\n",
      "the product acc is converted to ethylene by acc oxidase scheme 1\n",
      "\n",
      "CHEMICALS:\n",
      "['acc', 'ethylene']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1  0 -1 -1 -1  0 -1 -1  1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1  1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 0 POS: 11\n",
      "\n",
      "[]\n",
      "\n",
      "['includes_reaction_words', 'includes_reaction_component_words', 'structure_next_to_conversion_words', 'structure_conversion_by', 'structure_jtsui_pattern_1', 'structure_jtsui_pattern_2', 'structure_jtsui_pattern_4', 'structure_jtsui_pattern_1_copy', 'structure_jtsui_pattern_2_copy', 'structure_jtsui_pattern_4_copy', 'includes_reaction_words2']\n",
      "\n",
      "798\n",
      "\n",
      "SENTENCE:\n",
      "li et al 1996 have reported an interesting fusion construct with the n-terminal segment of acc synthase fused to an n-terminal deletion of acc oxidase that is active in converting sam directly to ethylene in escherichia coli\n",
      "\n",
      "CHEMICALS:\n",
      "['li', 'acc', 'n', 'sam', 'ethylene']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1  1 -1]\n",
      "\n",
      "NEGS: 2 POS: 6\n",
      "\n",
      "['structure_adjacent_mentions', 'structure_sep_sym']\n",
      "\n",
      "['structure_sep_verb', 'includes_reaction_words', 'structure_next_to_conversion_words', 'structure_jtsui_pattern_4', 'structure_jtsui_pattern_4_copy', 'includes_reaction_words2']\n",
      "\n",
      "804\n",
      "\n",
      "SENTENCE:\n",
      "the y85a mutation leads to an inactive acc synthase li et al, 1997, whereas y85f and y85w produce a partially active enzyme tarun et al, 1998\n",
      "\n",
      "CHEMICALS:\n",
      "['acc', 'li']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1  0 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 1 POS: 2\n",
      "\n",
      "['includes_no_terms']\n",
      "\n",
      "['includes_impact_words', 'includes_emzyze_words']\n",
      "\n",
      "811\n",
      "\n",
      "SENTENCE:\n",
      "the pyridine ring rests against the side-chain of both residues, but the smaller ala224 of aatase allows plp to assume an orientation consistent with the formation of a strong hydrogen bond between the 3′o− of plp and the oh of tyr225\n",
      "\n",
      "CHEMICALS:\n",
      "['plp', 'ala', 'tyr', 'pyridine', 'hydrogen', 'oh']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1  0 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1  1  1]\n",
      "\n",
      "NEGS: 2 POS: 3\n",
      "\n",
      "['includes_structural_words', 'includes_measure_words']\n",
      "\n",
      "['includes_reaction_words', 'structure_next_to_conversion_words', 'includes_reaction_words2']\n",
      "\n",
      "816\n",
      "\n",
      "SENTENCE:\n",
      "arg407 acc synthase, equivalent to arg386 in aatase is conserved in nearly all plp-dependent enzymes of amino acid metabolism alexander et al, 1994 mehta et al, 1993\n",
      "\n",
      "CHEMICALS:\n",
      "['acc', 'arg', 'plp']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      "  1 -1 -1  1  0 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1 -1]\n",
      "\n",
      "NEGS: 2 POS: 4\n",
      "\n",
      "['includes_general_chemical_words', 'includes_experiment_words']\n",
      "\n",
      "['includes_reaction_words', 'structure_next_to_conversion_words', 'includes_emzyze_words', 'includes_reaction_words2']\n",
      "\n",
      "820\n",
      "\n",
      "SENTENCE:\n",
      "modeling of a reaction intermediate recognition of the dicarboxylate substrates of aatase is effected by arg386 alpha-carboxylate and arg292∗ second subunit omega-carboxylate\n",
      "\n",
      "CHEMICALS:\n",
      "['arg', 'dicarboxylate']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1  0 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 1 POS: 3\n",
      "\n",
      "['includes_measure_words']\n",
      "\n",
      "['includes_reaction_component_words', 'includes_react', 'structure_next_to_conversion_words']\n",
      "\n",
      "833\n",
      "\n",
      "SENTENCE:\n",
      "kirsch the tyrosine-225 to phenylalanine mutation of escherichia coli aspartate aminotransferase results in an alkaline transition in the spectrophotometric and kinetic pk a values and reduced values of both k cat and k m biochemistry 30 1991 305 312 graf-hausner et al 1983 u graf-hausner kj\n",
      "\n",
      "CHEMICALS:\n",
      "['phenylalanine', 'tyrosine', 'aspartate']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 2 POS: 5\n",
      "\n",
      "['includes_functional_group', 'includes_measure_words']\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'structure_next_to_conversion_words', 'includes_impact_words', 'includes_reaction_words2']\n",
      "\n",
      "851\n",
      "\n",
      "SENTENCE:\n",
      "a region of positive potential is located on the reducing side of the disaccharide, consisting of his116, arg184 and arg219, which could represent a second sulfate binding site in the case of a bound tetrasaccharide\n",
      "\n",
      "CHEMICALS:\n",
      "['sulfate', 'his', 'disaccharide', 'arg']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1 -1]\n",
      "\n",
      "NEGS: 1 POS: 3\n",
      "\n",
      "['includes_physical_words']\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'includes_reaction_words2']\n",
      "\n",
      "856\n",
      "\n",
      "SENTENCE:\n",
      "arginine 236 of aspergillus niger pectin lyases a and b, analogous to arg218 in e chrysanthemi pectate lyase c, arg230 of e chrysanthemi pectate lyase e and arg279 in bacillus subtilis pectate lyase, is expected to be a key catalytic residue\n",
      "\n",
      "CHEMICALS:\n",
      "['arg', 'arginine', 'pectate']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1  1\n",
      "  1 -1 -1 -1 -1 -1 -1 -1  1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 0 POS: 1\n",
      "\n",
      "[]\n",
      "\n",
      "['structure_next_to_conversion_words']\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "868\n",
      "\n",
      "SENTENCE:\n",
      "ornithine decarboxylase odc, ec 41117 performs the first step in polyamine biosynthesis, the decarboxylation of ornithine to putrescine\n",
      "\n",
      "CHEMICALS:\n",
      "['putrescine', 'ornithine']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1  1\n",
      "  1 -1 -1 -1 -1 -1 -1 -1  1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1  1 -1  1]\n",
      "\n",
      "NEGS: 1 POS: 6\n",
      "\n",
      "['includes_functional_group']\n",
      "\n",
      "['structure_sep_conversion_words', 'includes_reaction_words', 'includes_reaction_component_words', 'structure_next_to_conversion_words', 'structure_conversion_of', 'includes_reaction_words2']\n",
      "\n",
      "870\n",
      "\n",
      "SENTENCE:\n",
      "ornithine decarboxylase odc catalyzes the first and rate-limiting step in polyamine biosynthesis, the decarboxylation of ornithine to putrescine\n",
      "\n",
      "CHEMICALS:\n",
      "['putrescine', 'ornithine']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1  0 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 1 POS: 7\n",
      "\n",
      "['includes_functional_group']\n",
      "\n",
      "['structure_sep_conversion_words', 'includes_reaction_words', 'includes_reaction_component_words', 'structure_next_to_conversion_words', 'structure_conversion_of', 'includes_catalyze_words', 'includes_reaction_words2']\n",
      "\n",
      "875\n",
      "\n",
      "SENTENCE:\n",
      "a 47 1991 110 119 kanamoto et al 1986 r kanamoto k utsunomiya t kameji s hayashi effects of putrescine on synthesis and degradation of ornithine decarboxylase in primary cultured hepatocytes eur j biochem 154 1986 539 544 katz and kahana 1987 a katz c kahana transcriptional activation of mammalian ornithine decarboxylase during stimulated growth mol cell\n",
      "\n",
      "CHEMICALS:\n",
      "['putrescine', 'ornithine']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1  0 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      "  1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1  1  1  1]\n",
      "\n",
      "NEGS: 2 POS: 3\n",
      "\n",
      "['includes_physical_words', 'includes_experiment_words']\n",
      "\n",
      "['includes_reaction_words', 'structure_conversion_of', 'includes_reaction_words2']\n",
      "\n",
      "886\n",
      "\n",
      "SENTENCE:\n",
      "keywords x-ray crystallography protein structure purine nucleotide biosynthesis gtp-hydrolyzing enzymes introduction adenylosuccinate synthetase ec 6344adss catalyzes the first step in the de novo biosynthesis of amp\n",
      "\n",
      "CHEMICALS:\n",
      "['amp', 'adenylosuccinate']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1  0  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 3 POS: 5\n",
      "\n",
      "['includes_structural_words', 'includes_paper_artifacts', 'includes_measure_words']\n",
      "\n",
      "['includes_reaction_words', 'includes_reaction_component_words', 'includes_catalyze_words', 'includes_emzyze_words', 'includes_reaction_words2']\n",
      "\n",
      "910\n",
      "\n",
      "SENTENCE:\n",
      "the extended conformations of active-site lysine residues are strictly conserved between rnase le lys96 and recombinant rnase rh lys108 figure 5\n",
      "\n",
      "CHEMICALS:\n",
      "['lys', 'lysine']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1  0  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1  1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 2 POS: 3\n",
      "\n",
      "['includes_genetic_words', 'includes_structural_words']\n",
      "\n",
      "['includes_combination_words', 'includes_reaction_words', 'includes_reaction_words2']\n",
      "\n",
      "912\n",
      "\n",
      "SENTENCE:\n",
      "thus, it is concluded that the extended side-chain conformations of active-site lysine residues lys96 of rnase le and lys108 of recombinant rnase rh presented here figure 5 show a truly active conformation of the active-site lysine residues\n",
      "\n",
      "CHEMICALS:\n",
      "['lys', 'lysine']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1  0  0  0  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      "  1 -1 -1  1  0  0 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      "  1 -1  1  1]\n",
      "\n",
      "NEGS: 2 POS: 4\n",
      "\n",
      "['includes_genetic_words', 'includes_structural_words']\n",
      "\n",
      "['structure_sep_verb', 'includes_combination_words', 'includes_reaction_words', 'includes_reaction_words2']\n",
      "\n",
      "958\n",
      "\n",
      "SENTENCE:\n",
      "table 2 influence of the mutations k993a and v994a on the ammonia-dependent carbamoyl phosphate synthetase activity and on the partial activities of the enzyme mutation full reaction from ammonia partial reactions cp synthesis adp production atpase atp synthesis mumol/minute per mg protein wild-type 092 156 028 035 k993a 028 069 027 017 v994a→a994v a 0005092 a 029 021025 a 013034 a a data in parentheses are those for the revertant to wild-type, generated by site-directed mutagenesis of the corresponding mutant\n",
      "\n",
      "CHEMICALS:\n",
      "['atp', 'ammonia', 'adp', 'carbamoyl phosphate']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1  0 -1  1  0 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      "  1 -1  1  1]\n",
      "\n",
      "NEGS: 6 POS: 7\n",
      "\n",
      "['includes_physical_words', 'includes_genetic_words', 'includes_structural_words', 'includes_general_chemical_words', 'includes_measure_words', 'includes_experiment_words']\n",
      "\n",
      "['includes_reaction_words', 'includes_reaction_component_words', 'includes_react', 'structure_next_to_conversion_words', 'includes_creation_words', 'includes_emzyze_words', 'includes_reaction_words2']\n",
      "\n",
      "963\n",
      "\n",
      "SENTENCE:\n",
      "ump reduces to one-third of the v max of the partial reaction of atp synthesis from carbamoyl phosphate, decreases the apparent affinity for atp and favors the formation of enzyme dimers, whereas the activators increase the apparent affinity for atp and favor tetramer formation powers et al, 1980 meister, 1989 braxton et al, 1996\n",
      "\n",
      "CHEMICALS:\n",
      "['atp', 'carbamoyl phosphate']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      "  1 -1 -1  1  0  0 -1  0  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      "  1 -1 -1  1]\n",
      "\n",
      "NEGS: 2 POS: 7\n",
      "\n",
      "['includes_comparison_words', 'includes_measure_words']\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'includes_react', 'structure_next_to_conversion_words', 'includes_creation_words', 'includes_emzyze_words', 'includes_reaction_words2']\n",
      "\n",
      "965\n",
      "\n",
      "SENTENCE:\n",
      "to determine the influence of v994a and k993a on the individual steps of the cps reaction see introduction the effects of these mutations on the partial reactions of bicarbonate-dependent atp hydrolysis atpase and of atp synthesis from adp and carbamoyl phosphate was investigated\n",
      "\n",
      "CHEMICALS:\n",
      "['atp', 'bicarbonate', 'adp', 'carbamoyl phosphate']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1  0 -1  1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      "  1 -1 -1  1]\n",
      "\n",
      "NEGS: 3 POS: 6\n",
      "\n",
      "['includes_measure_words', 'includes_experiment_words', 'structure_sep_and']\n",
      "\n",
      "['includes_reaction_words', 'includes_reaction_component_words', 'includes_react', 'structure_next_to_conversion_words', 'includes_creation_words', 'includes_reaction_words2']\n",
      "\n",
      "966\n",
      "\n",
      "SENTENCE:\n",
      "both v994a and k993a decreased the rate of the partial reaction of atp synthesis without altering much the bicarbonate-dependent atpase partial reaction table 2\n",
      "\n",
      "CHEMICALS:\n",
      "['atp', 'bicarbonate']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1  0 -1  1  0 -1 -1  0  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      "  1 -1 -1  1]\n",
      "\n",
      "NEGS: 2 POS: 4\n",
      "\n",
      "['includes_comparison_words', 'includes_experiment_words']\n",
      "\n",
      "['includes_reaction_words', 'includes_react', 'includes_creation_words', 'includes_reaction_words2']\n",
      "\n",
      "976\n",
      "\n",
      "SENTENCE:\n",
      "the partial reaction of atp synthesis from adp and carbamoyl phosphate was assayed continuously at 37°c by following the increase in absorbance at 340 nm in a system containing 01 m tris-hcl ph 80, 01 m kcl, 15 mm mgso4, 15 mm glucose, 5 mm adp, 5 mm carbamoyl phosphate, 1 mm nadp, 01 mg/ml hexokinase and 0025 mg/ml glucose-6-phosphate dehydrogenase\n",
      "\n",
      "CHEMICALS:\n",
      "['glucose', 'kcl', 'glucose-6-phosphate', 'mgso4', 'atp', 'nadp', 'adp', 'carbamoyl phosphate']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1  0 -1 -1  0 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1  0 -1 -1 -1  1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1  1  1]\n",
      "\n",
      "NEGS: 4 POS: 5\n",
      "\n",
      "['includes_physical_words', 'includes_comparison_words', 'includes_measure_words', 'structure_sep_and']\n",
      "\n",
      "['includes_reaction_words', 'includes_react', 'structure_next_to_conversion_words', 'includes_creation_words', 'includes_reaction_words2']\n",
      "\n",
      "978\n",
      "\n",
      "SENTENCE:\n",
      "references alonso and rubio 1983 e alonso v rubio binding of n-acetyl-l-glutamate to rat liver carbamoyl phosphate synthetase ammonia eur j biochem 135 1983 331 337 alonso and rubio 1995 e alonso v rubio affinity cleavage of carbamoyl-phosphate synthetase i localizes regions of the enzyme interacting with the molecule of atp that phosphorylates carbamate eur j biochem 229 1995 377 384 alonso et al 1992 e alonso j cervera a garcı́a-españa e bendala v rubio oxidative inactivation of carbamoyl phosphate synthetase ammonia\n",
      "\n",
      "CHEMICALS:\n",
      "['carbamoyl-phosphate', 'ammonia', 'n-acetyl-l-glutamate', 'carbamate', 'atp', 'carbamoyl phosphate']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1  0 -1 -1 -1  0 -1 -1 -1 -1  0 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1  0 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1  1  1]\n",
      "\n",
      "NEGS: 4 POS: 5\n",
      "\n",
      "['includes_physical_words', 'includes_general_chemical_words', 'includes_paper_artifacts', 'includes_measure_words']\n",
      "\n",
      "['includes_reaction_words', 'structure_next_to_conversion_words', 'structure_conversion_of', 'includes_emzyze_words', 'includes_reaction_words2']\n",
      "\n",
      "979\n",
      "\n",
      "SENTENCE:\n",
      "mechanism and sites of oxidation, degradation of the oxidized enzyme, and inactivation by glycerol, edta, and thiol protecting agents j biol chem 267 1992 4524 4532 anderson 1977 pm anderson binding of allosteric affectors to carbamyl-phosphate synthetase from escherichia coli biochemistry 16 1977 587 593 ausubel et al 1997 fm\n",
      "\n",
      "CHEMICALS:\n",
      "['carbamyl-phosphate', 'thiol', 'edta', 'glycerol']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1  0 -1 -1 -1  1  1 -1 -1 -1  1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 4 POS: 5\n",
      "\n",
      "['includes_physical_words', 'includes_functional_group', 'structure_sep_comma', 'includes_measure_words']\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'structure_next_to_conversion_words', 'includes_emzyze_words', 'includes_reaction_words2']\n",
      "\n",
      "989\n",
      "\n",
      "SENTENCE:\n",
      "lusty in vivo synthesis of carbamyl phosphate from nh3 by the large subunit of escherichia coli carbamyl phosphate synthetase j biol chem 262 1987 4382 4386 rubio et al 1991 v rubio j cervera cj\n",
      "\n",
      "CHEMICALS:\n",
      "['nh3', 'carbamyl phosphate']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1  0 -1 -1  0 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 1 POS: 6\n",
      "\n",
      "['includes_measure_words']\n",
      "\n",
      "['includes_reaction_words', 'structure_next_to_conversion_words', 'structure_conversion_by', 'structure_jtsui_pattern_1', 'structure_jtsui_pattern_1_copy', 'includes_reaction_words2']\n",
      "\n",
      "1015\n",
      "\n",
      "SENTENCE:\n",
      "analogous experiments were conducted at four different protein concentrations 2, 4, 6, 8 mum, three ph values 56, 75, 88 and two temperatures 10°c to avoid atp hydrolysis and, in the case of adp, at a physiologically relevant temperature of 55°c\n",
      "\n",
      "CHEMICALS:\n",
      "['atp', 'adp']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1  0  0 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 2 POS: 3\n",
      "\n",
      "['includes_comparison_words', 'includes_measure_words']\n",
      "\n",
      "['includes_reaction_words', 'includes_concentration', 'includes_reaction_words2']\n",
      "\n",
      "1018\n",
      "\n",
      "SENTENCE:\n",
      "interestingly, atp hydrolysis by the sulfolobus solfataricus thermosome was shown to be completely dependent on k+ guagliardi et al, 1994, whereas pyrodictium occultum thermosome is a very effective atpase even in its absence phipps et al, 1991 and the hydrolytic activity of tric/cct is enhanced only tenfold by k+\n",
      "\n",
      "CHEMICALS:\n",
      "['k+', 'atp']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1  1 -1]\n",
      "\n",
      "NEGS: 2 POS: 3\n",
      "\n",
      "['includes_measure_words', 'includes_experiment_words']\n",
      "\n",
      "['includes_reaction_words', 'structure_next_to_conversion_words', 'includes_reaction_words2']\n",
      "\n",
      "1029\n",
      "\n",
      "SENTENCE:\n",
      "all three proposed catalytic residues his257, tyr246, lys285 in the af endonuclease are superimposable with those in the mj enzyme his125, tyr115, lys156 and these residues are absent from the n repeat of the af endonuclease figure 4a and b\n",
      "\n",
      "CHEMICALS:\n",
      "['lys', 'his', 'tyr']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      "  1  0 -1  1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1  1  1 -1]\n",
      "\n",
      "NEGS: 0 POS: 1\n",
      "\n",
      "[]\n",
      "\n",
      "['includes_emzyze_words']\n",
      "\n",
      "1039\n",
      "\n",
      "SENTENCE:\n",
      "both enzymes have a modular, multi-domain structure and catalyse a similar atp-dependent reaction involving the addition of a glutamate residue to a carboxylate-containing substrate, tetrahydrofolate in the case of fpgs, and udp-n-acetylmuramoyl-l-alanine in the case of murd\n",
      "\n",
      "CHEMICALS:\n",
      "['udp', 'tetrahydrofolate', 'carboxylate', 'glutamate', 'atp']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      "  1  0 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 3 POS: 4\n",
      "\n",
      "['includes_structural_words', 'includes_comparison_words', 'includes_experiment_words']\n",
      "\n",
      "['includes_reaction_component_words', 'includes_react', 'includes_catalyze_words', 'includes_emzyze_words']\n",
      "\n",
      "1073\n",
      "\n",
      "SENTENCE:\n",
      "if a negative charge develops on the carbonyl oxygen atom during formation of the transition state for the second step of the reaction, one would predict that this hydrogen bond will increase in strength\n",
      "\n",
      "CHEMICALS:\n",
      "['hydrogen', 'carbonyl oxygen']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      "  1  0 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 2 POS: 5\n",
      "\n",
      "['includes_structural_words', 'includes_comparison_words']\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'includes_reaction_component_words', 'includes_react', 'includes_reaction_words2']\n",
      "\n",
      "1074\n",
      "\n",
      "SENTENCE:\n",
      "the second hypothesis proposes that the strengths of the interactions between asp78, tyr169, gln173, and/or asp194 and the tyrosyl-adenylate moiety increase during formation of the transition state for the second step of the reaction\n",
      "\n",
      "CHEMICALS:\n",
      "['gln', 'asp', 'tyr']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1  0  0  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1  1\n",
      "  1 -1 -1  1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 2 POS: 4\n",
      "\n",
      "['includes_genetic_words', 'includes_comparison_words']\n",
      "\n",
      "['includes_reaction_words', 'includes_reaction_component_words', 'includes_react', 'includes_reaction_words2']\n",
      "\n",
      "1076\n",
      "\n",
      "SENTENCE:\n",
      "if this model is correct, it implies that the bond between the 2′-hydroxyl group of adenosine 76 and the carbonyl carbon atom of tyrosyl-adenylate is partially formed in the transition state for the second step of the reaction if this were not the case, replacement of thr40, lys82, and arg86 by alanine would not destabilize this transition state\n",
      "\n",
      "CHEMICALS:\n",
      "['lys', 'alanine', 'thr', 'carbonyl carbon', 'adenosine']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1  0  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1 -1 -1 -1  1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1  1 -1]\n",
      "\n",
      "NEGS: 3 POS: 6\n",
      "\n",
      "['includes_structural_words', 'includes_general_chemical_words', 'includes_functional_group']\n",
      "\n",
      "['includes_replacement_words', 'includes_reaction_words', 'includes_reaction_component_words', 'includes_react', 'structure_next_to_conversion_words', 'includes_reaction_words2']\n",
      "\n",
      "1081\n",
      "\n",
      "SENTENCE:\n",
      "at this concentration of atp, over 90 % of the tyrosyl-trna synthetase is present as the free unliganded enzyme, and the dissociation constant k tyr calculated from the tyrosine dependence of the initial rate is equal to the dissociation constant for the e·tyr complex wells et al, 1991\n",
      "\n",
      "CHEMICALS:\n",
      "['atp', 'tyrosine', 'tyr']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 2 POS: 3\n",
      "\n",
      "['includes_genetic_words', 'includes_structural_words']\n",
      "\n",
      "['includes_concentration', 'structure_conversion_of', 'includes_emzyze_words']\n",
      "\n",
      "1083\n",
      "\n",
      "SENTENCE:\n",
      "known examples include nal, dihydrodipicolinate synthetase dhdps, d-5-keto-4-deoxyglucarate dehydratase, 2-keto-3-deoxygluconate aldolase, trans-o-hydroxybenzylidenepyruvate hydrolase-aldolase and trans-2′-carboxybenzalpyruvate hydratase-aldolase\n",
      "\n",
      "CHEMICALS:\n",
      "['2-keto-3-deoxygluconate', 'dihydrodipicolinate', 'trans-o-hydroxybenzylidenepyruvate', 'trans-2-carboxybenzalpyruvate']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 0 POS: 2\n",
      "\n",
      "[]\n",
      "\n",
      "['includes_reaction_words', 'includes_reaction_words2']\n",
      "\n",
      "1085\n",
      "\n",
      "SENTENCE:\n",
      "sub-family members identified thus far include the archetype nal, dihydrodipicolinate synthetase dhdps, d-5-keto-4-deoxyglucarate dehydratase kdgdh and trans-o-hydroxybenzylidenepyruvate hydrolase-aldolase hbpha, with recent additions being trans-2′-carboxybenzalpyruvate hydratase-aldolase cbpha iwabuchi & harayama, 1998 and 2-keto-3-deoxygluconate aldolase kdga buchanan et al, 1999\n",
      "\n",
      "CHEMICALS:\n",
      "['trans-o-hydroxybenzylidenepyruvate', 'dihydrodipicolinate', 'trans-2-carboxybenzalpyruvate', '2-keto-3-deoxygluconate']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      "  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 1 POS: 2\n",
      "\n",
      "['includes_measure_words']\n",
      "\n",
      "['includes_reaction_words', 'includes_reaction_words2']\n",
      "\n",
      "1102\n",
      "\n",
      "SENTENCE:\n",
      "the catalytic cycle of class i aldolases involves the formation of a schiff base intermediate between the carbonyl carbon atom of the substrate and an essential lysine residue horecker et al, 1972\n",
      "\n",
      "CHEMICALS:\n",
      "['lysine', 'carbonyl carbon']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1  0 -1 -1  0 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      "  1 -1 -1 -1  0 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1  1 -1  1]\n",
      "\n",
      "NEGS: 1 POS: 3\n",
      "\n",
      "['includes_general_chemical_words']\n",
      "\n",
      "['includes_reaction_words', 'includes_reaction_component_words', 'includes_reaction_words2']\n",
      "\n",
      "1104\n",
      "\n",
      "SENTENCE:\n",
      "the class ii aldolase fuca catalyzes the cleavage of l-fuculose 1-phosphate fuc1p to dihydroxyacetone phosphate dhap and l-lactaldehyde figure 1, which is a crucial step in the bacterial l-fucose metabolism ghalambor & heath, 1962\n",
      "\n",
      "CHEMICALS:\n",
      "['l-fucose', 'l-fuculose 1-phosphate', 'dihydroxyacetone phosphate', 'dhap']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 4 POS: 5\n",
      "\n",
      "['includes_functional_group', 'structure_adjacent_mentions', 'structure_sep_sym', 'includes_measure_words']\n",
      "\n",
      "['includes_reaction_words', 'includes_reaction_component_words', 'structure_next_to_conversion_words', 'includes_catalyze_words', 'includes_reaction_words2']\n",
      "\n",
      "1108\n",
      "\n",
      "SENTENCE:\n",
      "the produced dhap was reduced using nadh and rabbit muscle glycerol-3-phosphate dehydrogenase boehringer-mannheim or fluka and monitored at 366 nm eppendorf\n",
      "\n",
      "CHEMICALS:\n",
      "['nadh', 'glycerol-3-phosphate', 'dhap']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1  0  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 1 POS: 3\n",
      "\n",
      "['includes_measure_words']\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'includes_reaction_words2']\n",
      "\n",
      "1109\n",
      "\n",
      "SENTENCE:\n",
      "subsequently, nadh was added and the remaining dha or bhp was reduced after adding glycerol dehydrogenase from enterobacter aerogenes boehringer-mannheim or rabbit muscle lactate dehydrogenase fluka, respectively\n",
      "\n",
      "CHEMICALS:\n",
      "['bhp', 'nadh', 'lactate', 'glycerol']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1  0 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      "  1 -1 -1 -1  0 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1  1  1  1]\n",
      "\n",
      "NEGS: 2 POS: 3\n",
      "\n",
      "['includes_genetic_words', 'includes_structural_words']\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'includes_reaction_words2']\n",
      "\n",
      "1115\n",
      "\n",
      "SENTENCE:\n",
      "sect d 53 1997 240 255 ozaki et al 1990 a ozaki ej toone ch von der osten aj sinskey gm whitesides overproduction and substrate specificity of a bacterial fuculose-1-phosphate aldolase a new enzymatic catalyst for stereocontrolled aldol condensation j am chem soc 112 1990 4970 4971 silverman and lindskog 1988 dn silverman s lindskog the catalytic mechanism of carbonic anhydrase implications of a rate-limiting protolysis of water accts chem res 21 1988 30 36 vahrenkamp 1999 h vahrenkamp transitions, transition states, transition state analogues zinc pyrazolylborate chemistry related to zinc enzymes accts chem res 32 1999 589 596 wong et al 1995 c-h wong rl halcomb y ichikawa t kajimoto enzymes in organic synthesis application to the problems of carbohydrate recognition angew chem int ed engl 34 1995 412 432\n",
      "\n",
      "CHEMICALS:\n",
      "['fuculose-1-phosphate', 'osten', 'zinc pyrazolylborate', 'aldol', 'zinc']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1 -1]\n",
      "\n",
      "NEGS: 3 POS: 6\n",
      "\n",
      "['structure_adjacent_mentions', 'structure_sep_sym', 'includes_measure_words']\n",
      "\n",
      "['includes_reaction_words', 'includes_reaction_component_words', 'structure_next_to_conversion_words', 'includes_catalyze_words', 'includes_emzyze_words', 'includes_reaction_words2']\n",
      "\n",
      "1117\n",
      "\n",
      "SENTENCE:\n",
      "2 the hpin1 protein has the n-terminal ww domain residues 1–39 in addition to the c-terminal ppiase domain residues 45–163\n",
      "\n",
      "CHEMICALS:\n",
      "['c', 'n']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1  1  1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 0 POS: 1\n",
      "\n",
      "[]\n",
      "\n",
      "['structure_next_to_conversion_words']\n",
      "\n",
      "1122\n",
      "\n",
      "SENTENCE:\n",
      "3,10 in fact, the ppiase activity of hpin1 is greatly reduced when arg68 and arg69 are replaced by alanine residues\n",
      "\n",
      "CHEMICALS:\n",
      "['arg', 'alanine']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1  0 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1  0 -1 -1  0 -1 -1  1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1  1 -1 -1\n",
      " -1  1 -1  1]\n",
      "\n",
      "NEGS: 0 POS: 5\n",
      "\n",
      "[]\n",
      "\n",
      "['includes_oxidation_words', 'includes_replacement_words', 'includes_reaction_words', 'structure_next_to_conversion_words', 'includes_reaction_words2']\n",
      "\n",
      "1124\n",
      "\n",
      "SENTENCE:\n",
      "keywords phosphoglycerate mutase nmr structure dynamics protein abbreviations pgam phosphoglycerate mutase bpg bis-phosphosphoglycerate noesy-hsqc 15n separated nuclear overhauser effect spectroscopy hsqc heteronuclear single quantum coherence csi chemical shift index rmsd root-mean-square deviation introduction phosphoglycerate mutase pgam, ec 5421 catalyses the interconversion of 2-phosphoglycerate and 3-phosphoglycerate\n",
      "\n",
      "CHEMICALS:\n",
      "['2-phosphoglycerate', 'phosphoglycerate', '3-phosphoglycerate', 'bpg']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1  1  1]\n",
      "\n",
      "NEGS: 4 POS: 6\n",
      "\n",
      "['includes_structural_words', 'includes_paper_artifacts', 'includes_measure_words', 'structure_sep_and']\n",
      "\n",
      "['includes_reaction_words', 'structure_conversion_of', 'structure_jtsui_pattern_4', 'structure_jtsui_pattern_4_copy', 'includes_catalyze_words', 'includes_reaction_words2']\n",
      "\n",
      "1129\n",
      "\n",
      "SENTENCE:\n",
      "dtdp-d-glucose 4,6-dehydratase rmlb ec 42146 is the second enzyme in the dtdp-l-rhamnose biosynthetic pathway\n",
      "\n",
      "CHEMICALS:\n",
      "['dtdp-d-glucose', 'dtdp-l-rhamnose']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1  0  0 -1 -1 -1 -1 -1 -1 -1 -1  0 -1  1 -1  1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1  0  0 -1 -1  1 -1  1 -1 -1  1 -1 -1  1 -1  1 -1 -1  1 -1 -1\n",
      " -1  1  1  1]\n",
      "\n",
      "NEGS: 0 POS: 3\n",
      "\n",
      "[]\n",
      "\n",
      "['includes_reaction_words', 'includes_emzyze_words', 'includes_reaction_words2']\n",
      "\n",
      "1133\n",
      "\n",
      "SENTENCE:\n",
      "the sequence of rmlb from s enterica reveals it to be a tyrosine-dependent oxidoreductase, also known as the short-chain dehydrogenase/reductase family sdr, containing the highly conserved tyrxxxlys catalytic couple28–30 closely related enzymes for which structures are known include udp-galactose 4-epimerase from e coli egale and homo sapiens hgale which catalyses the interconversion of udp-galactose to udp-glucose, and gdp-mannose-4,6-dehydratase from e coli gmd, which catalyses the conversion of gdp-d-mannose to gdp-4-keto 6-deoxy-d-glucose in the gdp-fucose pathway\n",
      "\n",
      "CHEMICALS:\n",
      "['udp', 'glucose', 'fucose', 'galactose', 'tyrosine', 'gdp', 'gdp-d-mannose']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1  0 -1 -1  0  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1  1  1]\n",
      "\n",
      "NEGS: 5 POS: 12\n",
      "\n",
      "['includes_genetic_words', 'includes_structural_words', 'structure_sep_sym', 'includes_measure_words', 'includes_experiment_words']\n",
      "\n",
      "['includes_oxidation_words', 'structure_sep_conversion_words', 'includes_reaction_words', 'structure_next_to_conversion_words', 'structure_conversion_of', 'structure_jtsui_pattern_1', 'structure_jtsui_pattern_4', 'structure_jtsui_pattern_1_copy', 'structure_jtsui_pattern_4_copy', 'includes_catalyze_words', 'includes_emzyze_words', 'includes_reaction_words2']\n",
      "\n",
      "1134\n",
      "\n",
      "SENTENCE:\n",
      "nad+ binding region the enzyme is a class b dehydrogenase,43 in that the nicotinamide ring of the cofactor adopts the syn conformation, presenting the proper si face for pro-s hydride transfer from c4 of the dtdp-d-glucose to c4 of the dinucleotide cofactor\n",
      "\n",
      "CHEMICALS:\n",
      "['nicotinamide', 'hydride', 'nad+', 'dtdp-d-glucose']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1  0  0  0 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1  0 -1 -1 -1  1 -1 -1 -1 -1  1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 3 POS: 4\n",
      "\n",
      "['includes_physical_words', 'includes_general_chemical_words', 'includes_functional_group']\n",
      "\n",
      "['includes_reaction_words', 'structure_next_to_conversion_words', 'includes_emzyze_words', 'includes_reaction_words2']\n",
      "\n",
      "1137\n",
      "\n",
      "SENTENCE:\n",
      "lys171 stabilises the phenolic form of tyr by hydrogen bonding to the 2 and 3 hydroxyl groups of the nicotinamide ribose moiety of nad+ figure 4 creating a positive electrostatic field around itself and the nicotinamide ring, lowering the pk a of tyr16731,45 our model, as does egale, shows that the phenolic oxygen atom of tyr167 is over 40 å from the 4′-hydroxyl group of the glucose moiety, precluding tyr acting directly as the active site base\n",
      "\n",
      "CHEMICALS:\n",
      "['lys', 'nad+', 'glucose', 'nicotinamide ribose', 'tyr', 'nicotinamide', 'hydrogen', 'hydroxyl']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1  0 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 4 POS: 6\n",
      "\n",
      "['includes_structural_words', 'includes_general_chemical_words', 'includes_functional_group', 'includes_measure_words']\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'structure_next_to_conversion_words', 'structure_jtsui_pattern_1', 'structure_jtsui_pattern_1_copy', 'includes_reaction_words2']\n",
      "\n",
      "1142\n",
      "\n",
      "SENTENCE:\n",
      "isolation and crystallisation of thymidine diphosphate-d-glucose oxidoreductase from escherichia coli j biol chem 244 1969 3430 3437 22 m giraud jh naismith the rhamnose pathway curr opin struct biol 10 2000 687 696 23 m tonetti l sturla a bisso d zanardi u benatti a\n",
      "\n",
      "CHEMICALS:\n",
      "['rhamnose', 'thymidine diphosphate-d-glucose']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1  0 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      "  1 -1 -1 -1  0  0 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 2 POS: 4\n",
      "\n",
      "['includes_genetic_words', 'includes_measure_words']\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'structure_conversion_of', 'includes_reaction_words2']\n",
      "\n"
     ]
    }
   ],
   "source": [
    "oof = 101\n",
    "\n",
    "for i in range(100):\n",
    "    negs = []\n",
    "    neg_count = 0\n",
    "    pos = []\n",
    "    pos_count = 0\n",
    "    \n",
    "    print(df_brenda_p.index[oof])\n",
    "    print('')\n",
    "    \n",
    "    for j in range(len(L_train[i])):\n",
    "        if L_train[df_brenda_p.index[oof]][j] == 0:\n",
    "            negs.append(str(lfs[j].name))\n",
    "            neg_count = neg_count + 1\n",
    "        if L_train[df_brenda_p.index[oof]][j] == 1:\n",
    "            pos.append(str(lfs[j].name))\n",
    "            pos_count = pos_count + 1\n",
    "    print(\"SENTENCE:\")\n",
    "    print(df_brenda_p.iloc[oof]['sentence'])\n",
    "    print('')\n",
    "    print(\"CHEMICALS:\")\n",
    "    print(df_brenda_p.iloc[oof]['chemical_names'])\n",
    "    print(\"\")\n",
    "    oof = oof + 1\n",
    "    print(\"LABELS:\")\n",
    "    print(L_train[df_brenda_p.index[oof]])\n",
    "    print('')\n",
    "    print(\"NEGS: \" + str(neg_count) + \" POS: \" + str(pos_count))\n",
    "    print('')\n",
    "    print(negs)\n",
    "    print('')\n",
    "    print(pos)\n",
    "    print('')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'df_train' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-1-f45937856810>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mdf_train\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m: name 'df_train' is not defined"
     ]
    }
   ],
   "source": [
    "df_train"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# DEBUGGING MAJORITY LABEL VOTER\n",
    "\n",
    "## The following code should print out the fps and fns and other useful info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [],
   "source": [
    "from snorkel.labeling.model import MajorityLabelVoter\n",
    "from snorkel.labeling.model import LabelModel"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sentence</th>\n",
       "      <th>chemicals</th>\n",
       "      <th>truth</th>\n",
       "      <th>substrates</th>\n",
       "      <th>products</th>\n",
       "      <th>text</th>\n",
       "      <th>label_voter</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>the enzyme cyclo-oxygenase catalyses the oxyge...</td>\n",
       "      <td>[prostaglandins, arachidonic acid]</td>\n",
       "      <td>1</td>\n",
       "      <td>arachidonic, acid</td>\n",
       "      <td>prostaglandins</td>\n",
       "      <td>[the enzyme cyclo-oxygenase catalyses the oxyg...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>although the exact site of decarboxylation of ...</td>\n",
       "      <td>[levodopa, dopamine]</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[although the exact site of decarboxylation of...</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>the enzyme dimethylarginine dimethylaminohydro...</td>\n",
       "      <td>[citrulline, arginine, methylamines, dimethyla...</td>\n",
       "      <td>1</td>\n",
       "      <td>arginine</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[the enzyme dimethylarginine dimethylaminohydr...</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>background and aims glutamic acid decarboxylas...</td>\n",
       "      <td>[gaba, glutamate]</td>\n",
       "      <td>1</td>\n",
       "      <td>glutamate</td>\n",
       "      <td>gamma-aminobutyric, acid, GABA</td>\n",
       "      <td>[background and aims glutamic acid decarboxyla...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>mazindane 26 was found to be a pro-drug, oxidi...</td>\n",
       "      <td>[5-h, mazindol, mazindane, 5-oh]</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[mazindane 26 was found to be a pro-drug, oxid...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>418</th>\n",
       "      <td>only two exceptions had been reported-the ente...</td>\n",
       "      <td>[n-alpha-acetylornithine, ornithine]</td>\n",
       "      <td>1</td>\n",
       "      <td>ornithine</td>\n",
       "      <td>N-alpha-acetylornithine</td>\n",
       "      <td>[only two exceptions had been reported-the ent...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>419</th>\n",
       "      <td>the energetics is here reported for the action...</td>\n",
       "      <td>[l-beta-lysine, l-lysine]</td>\n",
       "      <td>1</td>\n",
       "      <td>l-lysine, l-beta-lysine</td>\n",
       "      <td>l-lysine, l-beta-lysine</td>\n",
       "      <td>[the energetics is here reported for the actio...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420</th>\n",
       "      <td>estradiol 17 beta-hydroxysteroid dehydrogenase...</td>\n",
       "      <td>[estrone, estradiol]</td>\n",
       "      <td>1</td>\n",
       "      <td>estrone, estradiol</td>\n",
       "      <td>estrone, estradiol</td>\n",
       "      <td>[estradiol 17 beta-hydroxysteroid dehydrogenas...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>421</th>\n",
       "      <td>5-hydroxymethyltryptophan and 5-hydroxy-4-meth...</td>\n",
       "      <td>[5-hydroxy-4-methyltryptophan, 5-methyltryptop...</td>\n",
       "      <td>1</td>\n",
       "      <td>5-methyltryptophan</td>\n",
       "      <td>5-Hydroxymethyltryptophan, 5-hydroxy-4-methylt...</td>\n",
       "      <td>[5-hydroxymethyltryptophan and 5-hydroxy-4-met...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>422</th>\n",
       "      <td>we show that tres from mycobacterium smegmatis...</td>\n",
       "      <td>[trehalose, maltose]</td>\n",
       "      <td>1</td>\n",
       "      <td>maltose, trehalose</td>\n",
       "      <td>maltose, trehalose</td>\n",
       "      <td>[we show that tres from mycobacterium smegmati...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>423 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                              sentence  \\\n",
       "0    the enzyme cyclo-oxygenase catalyses the oxyge...   \n",
       "1    although the exact site of decarboxylation of ...   \n",
       "2    the enzyme dimethylarginine dimethylaminohydro...   \n",
       "3    background and aims glutamic acid decarboxylas...   \n",
       "4    mazindane 26 was found to be a pro-drug, oxidi...   \n",
       "..                                                 ...   \n",
       "418  only two exceptions had been reported-the ente...   \n",
       "419  the energetics is here reported for the action...   \n",
       "420  estradiol 17 beta-hydroxysteroid dehydrogenase...   \n",
       "421  5-hydroxymethyltryptophan and 5-hydroxy-4-meth...   \n",
       "422  we show that tres from mycobacterium smegmatis...   \n",
       "\n",
       "                                             chemicals  truth  \\\n",
       "0                   [prostaglandins, arachidonic acid]      1   \n",
       "1                                 [levodopa, dopamine]      1   \n",
       "2    [citrulline, arginine, methylamines, dimethyla...      1   \n",
       "3                                    [gaba, glutamate]      1   \n",
       "4                     [5-h, mazindol, mazindane, 5-oh]      1   \n",
       "..                                                 ...    ...   \n",
       "418               [n-alpha-acetylornithine, ornithine]      1   \n",
       "419                          [l-beta-lysine, l-lysine]      1   \n",
       "420                               [estrone, estradiol]      1   \n",
       "421  [5-hydroxy-4-methyltryptophan, 5-methyltryptop...      1   \n",
       "422                               [trehalose, maltose]      1   \n",
       "\n",
       "                  substrates  \\\n",
       "0          arachidonic, acid   \n",
       "1                        NaN   \n",
       "2                   arginine   \n",
       "3                  glutamate   \n",
       "4                        NaN   \n",
       "..                       ...   \n",
       "418                ornithine   \n",
       "419  l-lysine, l-beta-lysine   \n",
       "420       estrone, estradiol   \n",
       "421       5-methyltryptophan   \n",
       "422       maltose, trehalose   \n",
       "\n",
       "                                              products  \\\n",
       "0                                       prostaglandins   \n",
       "1                                                  NaN   \n",
       "2                                                  NaN   \n",
       "3                       gamma-aminobutyric, acid, GABA   \n",
       "4                                                  NaN   \n",
       "..                                                 ...   \n",
       "418                            N-alpha-acetylornithine   \n",
       "419                            l-lysine, l-beta-lysine   \n",
       "420                                 estrone, estradiol   \n",
       "421  5-Hydroxymethyltryptophan, 5-hydroxy-4-methylt...   \n",
       "422                                 maltose, trehalose   \n",
       "\n",
       "                                                  text  label_voter  \n",
       "0    [the enzyme cyclo-oxygenase catalyses the oxyg...            1  \n",
       "1    [although the exact site of decarboxylation of...           -1  \n",
       "2    [the enzyme dimethylarginine dimethylaminohydr...           -1  \n",
       "3    [background and aims glutamic acid decarboxyla...            1  \n",
       "4    [mazindane 26 was found to be a pro-drug, oxid...            1  \n",
       "..                                                 ...          ...  \n",
       "418  [only two exceptions had been reported-the ent...            1  \n",
       "419  [the energetics is here reported for the actio...            1  \n",
       "420  [estradiol 17 beta-hydroxysteroid dehydrogenas...            1  \n",
       "421  [5-hydroxymethyltryptophan and 5-hydroxy-4-met...            1  \n",
       "422  [we show that tres from mycobacterium smegmati...            1  \n",
       "\n",
       "[423 rows x 7 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Reset indices so that majority label voter's predict fn aligns in the df\n",
    "\n",
    "df_test = hand_anno_mult_chem_df\n",
    "df_test = df_test.reset_index(drop=True)\n",
    "display(df_test)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/anaconda3/lib/python3.8/site-packages/tqdm/std.py:668: FutureWarning: The Panel class is removed from pandas. Accessing it from the top-level namespace will also be removed in the next version\n",
      "  from pandas import Panel\n",
      "100%|██████████| 423/423 [00:08<00:00, 49.07it/s]\n"
     ]
    }
   ],
   "source": [
    "# Define the set of labeling functions (LFs)\n",
    "# currently excluding amino_acid and followed_ase and followed_by_noun\n",
    "\n",
    "# removing physical_words increases recall but causes large drop in precision\n",
    "# sep_conversion_word and sep_verb removal increase precision to 0.71 with recall at 0.38\n",
    "# Apply the LFs to the unlabeled training data\n",
    "lfs = [includes_solution_words, includes_mixture_words, includes_physical_words, includes_genetic_words, includes_structural_words,\n",
    "      includes_general_chemical_words, includes_functional_group, includes_paper_artifacts, includes_no_terms, structure_adjacent_mentions,\n",
    "      structure_sep_or, structure_sep_comma, structure_sep_via, structure_sep_sym, structure_sep_adverb, includes_oxidation_words,\n",
    "      structure_sep_verb, structure_sep_conversion_words, includes_combustion_words, includes_neutralization_words, includes_combination_words,\n",
    "      includes_decomposition_words, includes_replacement_words, includes_reaction_words, includes_reaction_component_words, includes_comparison_words,\n",
    "      includes_one_chem, includes_react, includes_measure_words, includes_experiment_words, includes_concentration, structure_sep_and,\n",
    "      structure_next_to_conversion_words, structure_conversion_by, structure_conversion_of, includes_react_sym,\n",
    "      includes_interconvert_sym, structure_jtsui_pattern_1, structure_jtsui_pattern_2,\n",
    "      structure_jtsui_pattern_3, structure_jtsui_pattern_4, structure_jtsui_pattern_5, structure_jtsui_pattern_1_copy, structure_jtsui_pattern_2_copy,\n",
    "      structure_jtsui_pattern_3_copy, structure_jtsui_pattern_4_copy, structure_jtsui_pattern_5_copy,\n",
    "      includes_impact_words, includes_creation_words, includes_catalyze_words, includes_emzyze_words, includes_reaction_words2]\n",
    "      #  , ]\n",
    "      #includes_reaction_words5, includes_reaction_words3, includes_reaction_words4, structure_followed_by_ase,\n",
    "\n",
    "applier_test = PandasLFApplier(lfs=lfs)\n",
    "L_test = applier_test.apply(df=df_test)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>j</th>\n",
       "      <th>Polarity</th>\n",
       "      <th>Coverage</th>\n",
       "      <th>Overlaps</th>\n",
       "      <th>Conflicts</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>includes_solution_words</th>\n",
       "      <td>0</td>\n",
       "      <td>[0]</td>\n",
       "      <td>0.018913</td>\n",
       "      <td>0.018913</td>\n",
       "      <td>0.018913</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_mixture_words</th>\n",
       "      <td>1</td>\n",
       "      <td>[]</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_physical_words</th>\n",
       "      <td>2</td>\n",
       "      <td>[0]</td>\n",
       "      <td>0.283688</td>\n",
       "      <td>0.283688</td>\n",
       "      <td>0.198582</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_genetic_words</th>\n",
       "      <td>3</td>\n",
       "      <td>[0]</td>\n",
       "      <td>0.179669</td>\n",
       "      <td>0.179669</td>\n",
       "      <td>0.134752</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_structural_words</th>\n",
       "      <td>4</td>\n",
       "      <td>[0]</td>\n",
       "      <td>0.179669</td>\n",
       "      <td>0.177305</td>\n",
       "      <td>0.134752</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_general_chemical_words</th>\n",
       "      <td>5</td>\n",
       "      <td>[0]</td>\n",
       "      <td>0.186761</td>\n",
       "      <td>0.184397</td>\n",
       "      <td>0.125296</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_functional_group</th>\n",
       "      <td>6</td>\n",
       "      <td>[0]</td>\n",
       "      <td>0.307329</td>\n",
       "      <td>0.307329</td>\n",
       "      <td>0.229314</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_paper_artifacts</th>\n",
       "      <td>7</td>\n",
       "      <td>[0]</td>\n",
       "      <td>0.042553</td>\n",
       "      <td>0.042553</td>\n",
       "      <td>0.030733</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_no_terms</th>\n",
       "      <td>8</td>\n",
       "      <td>[0]</td>\n",
       "      <td>0.295508</td>\n",
       "      <td>0.286052</td>\n",
       "      <td>0.193853</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>structure_adjacent_mentions</th>\n",
       "      <td>9</td>\n",
       "      <td>[0]</td>\n",
       "      <td>0.120567</td>\n",
       "      <td>0.120567</td>\n",
       "      <td>0.066194</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>structure_sep_or</th>\n",
       "      <td>10</td>\n",
       "      <td>[0]</td>\n",
       "      <td>0.028369</td>\n",
       "      <td>0.028369</td>\n",
       "      <td>0.021277</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>structure_sep_comma</th>\n",
       "      <td>11</td>\n",
       "      <td>[0]</td>\n",
       "      <td>0.085106</td>\n",
       "      <td>0.085106</td>\n",
       "      <td>0.035461</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>structure_sep_via</th>\n",
       "      <td>12</td>\n",
       "      <td>[0]</td>\n",
       "      <td>0.014184</td>\n",
       "      <td>0.014184</td>\n",
       "      <td>0.009456</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>structure_sep_sym</th>\n",
       "      <td>13</td>\n",
       "      <td>[0]</td>\n",
       "      <td>0.127660</td>\n",
       "      <td>0.127660</td>\n",
       "      <td>0.070922</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>structure_sep_adverb</th>\n",
       "      <td>14</td>\n",
       "      <td>[]</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_oxidation_words</th>\n",
       "      <td>15</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.141844</td>\n",
       "      <td>0.141844</td>\n",
       "      <td>0.132388</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>structure_sep_verb</th>\n",
       "      <td>16</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.028369</td>\n",
       "      <td>0.028369</td>\n",
       "      <td>0.028369</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>structure_sep_conversion_words</th>\n",
       "      <td>17</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.044917</td>\n",
       "      <td>0.044917</td>\n",
       "      <td>0.040189</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_combustion_words</th>\n",
       "      <td>18</td>\n",
       "      <td>[]</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_neutralization_words</th>\n",
       "      <td>19</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.004728</td>\n",
       "      <td>0.004728</td>\n",
       "      <td>0.004728</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_combination_words</th>\n",
       "      <td>20</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.016548</td>\n",
       "      <td>0.016548</td>\n",
       "      <td>0.016548</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_decomposition_words</th>\n",
       "      <td>21</td>\n",
       "      <td>[]</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_replacement_words</th>\n",
       "      <td>22</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.007092</td>\n",
       "      <td>0.007092</td>\n",
       "      <td>0.007092</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_reaction_words</th>\n",
       "      <td>23</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.432624</td>\n",
       "      <td>0.432624</td>\n",
       "      <td>0.411348</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_reaction_component_words</th>\n",
       "      <td>24</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.113475</td>\n",
       "      <td>0.113475</td>\n",
       "      <td>0.108747</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_comparison_words</th>\n",
       "      <td>25</td>\n",
       "      <td>[0]</td>\n",
       "      <td>0.252955</td>\n",
       "      <td>0.248227</td>\n",
       "      <td>0.177305</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_one_chem</th>\n",
       "      <td>26</td>\n",
       "      <td>[]</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_react</th>\n",
       "      <td>27</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.033097</td>\n",
       "      <td>0.033097</td>\n",
       "      <td>0.030733</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_measure_words</th>\n",
       "      <td>28</td>\n",
       "      <td>[0]</td>\n",
       "      <td>0.408983</td>\n",
       "      <td>0.406619</td>\n",
       "      <td>0.286052</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_experiment_words</th>\n",
       "      <td>29</td>\n",
       "      <td>[0]</td>\n",
       "      <td>0.446809</td>\n",
       "      <td>0.427896</td>\n",
       "      <td>0.276596</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_concentration</th>\n",
       "      <td>30</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.044917</td>\n",
       "      <td>0.044917</td>\n",
       "      <td>0.044917</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>structure_sep_and</th>\n",
       "      <td>31</td>\n",
       "      <td>[0]</td>\n",
       "      <td>0.137116</td>\n",
       "      <td>0.127660</td>\n",
       "      <td>0.096927</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>structure_next_to_conversion_words</th>\n",
       "      <td>32</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.236407</td>\n",
       "      <td>0.236407</td>\n",
       "      <td>0.226950</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>structure_conversion_by</th>\n",
       "      <td>33</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.011820</td>\n",
       "      <td>0.011820</td>\n",
       "      <td>0.011820</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>structure_conversion_of</th>\n",
       "      <td>34</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.130024</td>\n",
       "      <td>0.130024</td>\n",
       "      <td>0.125296</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_react_sym</th>\n",
       "      <td>35</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.004728</td>\n",
       "      <td>0.004728</td>\n",
       "      <td>0.004728</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_interconvert_sym</th>\n",
       "      <td>36</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.002364</td>\n",
       "      <td>0.002364</td>\n",
       "      <td>0.002364</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>structure_jtsui_pattern_1</th>\n",
       "      <td>37</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.042553</td>\n",
       "      <td>0.042553</td>\n",
       "      <td>0.037825</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>structure_jtsui_pattern_2</th>\n",
       "      <td>38</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.009456</td>\n",
       "      <td>0.009456</td>\n",
       "      <td>0.007092</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>structure_jtsui_pattern_3</th>\n",
       "      <td>39</td>\n",
       "      <td>[]</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>structure_jtsui_pattern_4</th>\n",
       "      <td>40</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.047281</td>\n",
       "      <td>0.047281</td>\n",
       "      <td>0.042553</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>structure_jtsui_pattern_5</th>\n",
       "      <td>41</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.007092</td>\n",
       "      <td>0.007092</td>\n",
       "      <td>0.007092</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>structure_jtsui_pattern_1_copy</th>\n",
       "      <td>42</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.042553</td>\n",
       "      <td>0.042553</td>\n",
       "      <td>0.037825</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>structure_jtsui_pattern_2_copy</th>\n",
       "      <td>43</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.009456</td>\n",
       "      <td>0.009456</td>\n",
       "      <td>0.007092</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>structure_jtsui_pattern_3_copy</th>\n",
       "      <td>44</td>\n",
       "      <td>[]</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>structure_jtsui_pattern_4_copy</th>\n",
       "      <td>45</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.047281</td>\n",
       "      <td>0.047281</td>\n",
       "      <td>0.042553</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>structure_jtsui_pattern_5_copy</th>\n",
       "      <td>46</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.007092</td>\n",
       "      <td>0.007092</td>\n",
       "      <td>0.007092</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_impact_words</th>\n",
       "      <td>47</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.099291</td>\n",
       "      <td>0.099291</td>\n",
       "      <td>0.096927</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_creation_words</th>\n",
       "      <td>48</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.068558</td>\n",
       "      <td>0.068558</td>\n",
       "      <td>0.063830</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_catalyze_words</th>\n",
       "      <td>49</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.056738</td>\n",
       "      <td>0.056738</td>\n",
       "      <td>0.052009</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_emzyze_words</th>\n",
       "      <td>50</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.108747</td>\n",
       "      <td>0.108747</td>\n",
       "      <td>0.101655</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>includes_reaction_words2</th>\n",
       "      <td>51</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.432624</td>\n",
       "      <td>0.432624</td>\n",
       "      <td>0.411348</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                     j Polarity  Coverage  Overlaps  Conflicts\n",
       "includes_solution_words              0      [0]  0.018913  0.018913   0.018913\n",
       "includes_mixture_words               1       []  0.000000  0.000000   0.000000\n",
       "includes_physical_words              2      [0]  0.283688  0.283688   0.198582\n",
       "includes_genetic_words               3      [0]  0.179669  0.179669   0.134752\n",
       "includes_structural_words            4      [0]  0.179669  0.177305   0.134752\n",
       "includes_general_chemical_words      5      [0]  0.186761  0.184397   0.125296\n",
       "includes_functional_group            6      [0]  0.307329  0.307329   0.229314\n",
       "includes_paper_artifacts             7      [0]  0.042553  0.042553   0.030733\n",
       "includes_no_terms                    8      [0]  0.295508  0.286052   0.193853\n",
       "structure_adjacent_mentions          9      [0]  0.120567  0.120567   0.066194\n",
       "structure_sep_or                    10      [0]  0.028369  0.028369   0.021277\n",
       "structure_sep_comma                 11      [0]  0.085106  0.085106   0.035461\n",
       "structure_sep_via                   12      [0]  0.014184  0.014184   0.009456\n",
       "structure_sep_sym                   13      [0]  0.127660  0.127660   0.070922\n",
       "structure_sep_adverb                14       []  0.000000  0.000000   0.000000\n",
       "includes_oxidation_words            15      [1]  0.141844  0.141844   0.132388\n",
       "structure_sep_verb                  16      [1]  0.028369  0.028369   0.028369\n",
       "structure_sep_conversion_words      17      [1]  0.044917  0.044917   0.040189\n",
       "includes_combustion_words           18       []  0.000000  0.000000   0.000000\n",
       "includes_neutralization_words       19      [1]  0.004728  0.004728   0.004728\n",
       "includes_combination_words          20      [1]  0.016548  0.016548   0.016548\n",
       "includes_decomposition_words        21       []  0.000000  0.000000   0.000000\n",
       "includes_replacement_words          22      [1]  0.007092  0.007092   0.007092\n",
       "includes_reaction_words             23      [1]  0.432624  0.432624   0.411348\n",
       "includes_reaction_component_words   24      [1]  0.113475  0.113475   0.108747\n",
       "includes_comparison_words           25      [0]  0.252955  0.248227   0.177305\n",
       "includes_one_chem                   26       []  0.000000  0.000000   0.000000\n",
       "includes_react                      27      [1]  0.033097  0.033097   0.030733\n",
       "includes_measure_words              28      [0]  0.408983  0.406619   0.286052\n",
       "includes_experiment_words           29      [0]  0.446809  0.427896   0.276596\n",
       "includes_concentration              30      [1]  0.044917  0.044917   0.044917\n",
       "structure_sep_and                   31      [0]  0.137116  0.127660   0.096927\n",
       "structure_next_to_conversion_words  32      [1]  0.236407  0.236407   0.226950\n",
       "structure_conversion_by             33      [1]  0.011820  0.011820   0.011820\n",
       "structure_conversion_of             34      [1]  0.130024  0.130024   0.125296\n",
       "includes_react_sym                  35      [1]  0.004728  0.004728   0.004728\n",
       "includes_interconvert_sym           36      [1]  0.002364  0.002364   0.002364\n",
       "structure_jtsui_pattern_1           37      [1]  0.042553  0.042553   0.037825\n",
       "structure_jtsui_pattern_2           38      [1]  0.009456  0.009456   0.007092\n",
       "structure_jtsui_pattern_3           39       []  0.000000  0.000000   0.000000\n",
       "structure_jtsui_pattern_4           40      [1]  0.047281  0.047281   0.042553\n",
       "structure_jtsui_pattern_5           41      [1]  0.007092  0.007092   0.007092\n",
       "structure_jtsui_pattern_1_copy      42      [1]  0.042553  0.042553   0.037825\n",
       "structure_jtsui_pattern_2_copy      43      [1]  0.009456  0.009456   0.007092\n",
       "structure_jtsui_pattern_3_copy      44       []  0.000000  0.000000   0.000000\n",
       "structure_jtsui_pattern_4_copy      45      [1]  0.047281  0.047281   0.042553\n",
       "structure_jtsui_pattern_5_copy      46      [1]  0.007092  0.007092   0.007092\n",
       "includes_impact_words               47      [1]  0.099291  0.099291   0.096927\n",
       "includes_creation_words             48      [1]  0.068558  0.068558   0.063830\n",
       "includes_catalyze_words             49      [1]  0.056738  0.056738   0.052009\n",
       "includes_emzyze_words               50      [1]  0.108747  0.108747   0.101655\n",
       "includes_reaction_words2            51      [1]  0.432624  0.432624   0.411348"
      ]
     },
     "execution_count": 99,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from snorkel.labeling import LFAnalysis\n",
    "\n",
    "LFAnalysis(L=L_test, lfs=lfs).lf_summary()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.9952718676122931"
      ]
     },
     "execution_count": 100,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "LFAnalysis(L_test).label_coverage()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [],
   "source": [
    "majority_model = MajorityLabelVoter()\n",
    "df_test[\"label_voter\"] = majority_model.predict(L=L_test)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([ 1, -1, -1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1, -1, -1,\n",
       "        1,  1,  1,  1,  1,  1,  1,  1,  1,  0,  1,  0,  1,  1,  1,  1,  1,\n",
       "        1,  0,  1,  1,  1,  1,  1,  1,  1,  1,  0,  1,  1,  1,  1,  1,  0,\n",
       "        1,  0,  1,  0,  0,  0,  0,  0,  0, -1,  1, -1,  0,  0, -1, -1,  0,\n",
       "       -1,  0,  1,  0, -1,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  1,\n",
       "        0,  0,  0,  1,  0,  0,  0,  0,  0,  0,  0,  0,  0, -1,  0, -1,  0,\n",
       "        0, -1,  0,  0,  0,  0,  0, -1,  0,  0,  1,  1, -1,  0,  0,  0,  0,\n",
       "        0,  1,  0,  1,  0,  1,  1,  0,  0,  0,  0,  0,  0,  0,  0,  0,  1,\n",
       "        0,  0,  0,  0, -1,  0,  0, -1,  1,  0,  1, -1,  0, -1,  0,  1,  0,\n",
       "        1,  1,  0,  0,  0,  0,  0, -1,  1,  1,  0, -1,  0,  0,  0,  0,  0,\n",
       "       -1,  0,  0,  0,  0,  0,  0,  0,  0,  0, -1,  1, -1,  1,  1,  0,  0,\n",
       "        0, -1,  0, -1,  0,  0,  0,  1,  1,  0,  0, -1,  0,  0,  0,  0,  0,\n",
       "        0,  0,  0,  0,  1,  0,  1,  0,  0,  1,  0,  0,  0,  0,  1,  0,  1,\n",
       "        0,  0,  0,  0,  0, -1,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0, -1,\n",
       "        0,  0,  1,  0,  0, -1,  0,  0,  0,  0,  0,  0,  0,  0, -1,  1, -1,\n",
       "        0,  1,  0,  0,  0,  1,  0,  0,  1,  0,  0,  0,  0, -1,  0,  0,  0,\n",
       "        0,  0,  0,  0,  0,  0, -1,  0,  0,  0,  1,  0,  0,  0,  0,  0, -1,\n",
       "        0,  0,  0,  0,  0,  0,  0,  0, -1,  0, -1,  0,  1,  0,  0,  0,  0,\n",
       "        0,  0,  0,  0,  1, -1, -1,  0,  0,  0,  0,  0,  0,  0,  0,  0,  1,\n",
       "        0,  1,  1,  0,  0,  0,  0,  0, -1,  0,  0,  0, -1, -1,  0,  0,  0,\n",
       "        0,  0,  0,  0,  0,  0,  0,  1,  0,  0,  0,  0,  0,  0,  1,  0, -1,\n",
       "        0,  1, -1, -1,  0,  0,  1,  1,  0,  0,  1,  1,  0,  0, -1,  0, -1,\n",
       "        0,  0,  0,  1, -1,  0,  0,  0,  0, -1, -1,  1,  1, -1,  0,  0,  0,\n",
       "        0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  1,  0,  0,  0,  0,  0,\n",
       "        0,  1,  1,  0,  1,  0,  0,  1,  1,  1,  1,  1,  1,  1,  1])"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "RangeIndex(start=0, stop=423, step=1)"
      ]
     },
     "execution_count": 102,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "display(majority_model.predict(L=L_test))\n",
    "df_test.index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sentence</th>\n",
       "      <th>chemicals</th>\n",
       "      <th>truth</th>\n",
       "      <th>substrates</th>\n",
       "      <th>products</th>\n",
       "      <th>text</th>\n",
       "      <th>label_voter</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>the enzyme cyclo-oxygenase catalyses the oxyge...</td>\n",
       "      <td>[prostaglandins, arachidonic acid]</td>\n",
       "      <td>1</td>\n",
       "      <td>arachidonic, acid</td>\n",
       "      <td>prostaglandins</td>\n",
       "      <td>[the enzyme cyclo-oxygenase catalyses the oxyg...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>although the exact site of decarboxylation of ...</td>\n",
       "      <td>[levodopa, dopamine]</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[although the exact site of decarboxylation of...</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>the enzyme dimethylarginine dimethylaminohydro...</td>\n",
       "      <td>[citrulline, arginine, methylamines, dimethyla...</td>\n",
       "      <td>1</td>\n",
       "      <td>arginine</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[the enzyme dimethylarginine dimethylaminohydr...</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>background and aims glutamic acid decarboxylas...</td>\n",
       "      <td>[gaba, glutamate]</td>\n",
       "      <td>1</td>\n",
       "      <td>glutamate</td>\n",
       "      <td>gamma-aminobutyric, acid, GABA</td>\n",
       "      <td>[background and aims glutamic acid decarboxyla...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>mazindane 26 was found to be a pro-drug, oxidi...</td>\n",
       "      <td>[5-h, mazindol, mazindane, 5-oh]</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[mazindane 26 was found to be a pro-drug, oxid...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>418</th>\n",
       "      <td>only two exceptions had been reported-the ente...</td>\n",
       "      <td>[n-alpha-acetylornithine, ornithine]</td>\n",
       "      <td>1</td>\n",
       "      <td>ornithine</td>\n",
       "      <td>N-alpha-acetylornithine</td>\n",
       "      <td>[only two exceptions had been reported-the ent...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>419</th>\n",
       "      <td>the energetics is here reported for the action...</td>\n",
       "      <td>[l-beta-lysine, l-lysine]</td>\n",
       "      <td>1</td>\n",
       "      <td>l-lysine, l-beta-lysine</td>\n",
       "      <td>l-lysine, l-beta-lysine</td>\n",
       "      <td>[the energetics is here reported for the actio...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420</th>\n",
       "      <td>estradiol 17 beta-hydroxysteroid dehydrogenase...</td>\n",
       "      <td>[estrone, estradiol]</td>\n",
       "      <td>1</td>\n",
       "      <td>estrone, estradiol</td>\n",
       "      <td>estrone, estradiol</td>\n",
       "      <td>[estradiol 17 beta-hydroxysteroid dehydrogenas...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>421</th>\n",
       "      <td>5-hydroxymethyltryptophan and 5-hydroxy-4-meth...</td>\n",
       "      <td>[5-hydroxy-4-methyltryptophan, 5-methyltryptop...</td>\n",
       "      <td>1</td>\n",
       "      <td>5-methyltryptophan</td>\n",
       "      <td>5-Hydroxymethyltryptophan, 5-hydroxy-4-methylt...</td>\n",
       "      <td>[5-hydroxymethyltryptophan and 5-hydroxy-4-met...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>422</th>\n",
       "      <td>we show that tres from mycobacterium smegmatis...</td>\n",
       "      <td>[trehalose, maltose]</td>\n",
       "      <td>1</td>\n",
       "      <td>maltose, trehalose</td>\n",
       "      <td>maltose, trehalose</td>\n",
       "      <td>[we show that tres from mycobacterium smegmati...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>423 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                              sentence  \\\n",
       "0    the enzyme cyclo-oxygenase catalyses the oxyge...   \n",
       "1    although the exact site of decarboxylation of ...   \n",
       "2    the enzyme dimethylarginine dimethylaminohydro...   \n",
       "3    background and aims glutamic acid decarboxylas...   \n",
       "4    mazindane 26 was found to be a pro-drug, oxidi...   \n",
       "..                                                 ...   \n",
       "418  only two exceptions had been reported-the ente...   \n",
       "419  the energetics is here reported for the action...   \n",
       "420  estradiol 17 beta-hydroxysteroid dehydrogenase...   \n",
       "421  5-hydroxymethyltryptophan and 5-hydroxy-4-meth...   \n",
       "422  we show that tres from mycobacterium smegmatis...   \n",
       "\n",
       "                                             chemicals  truth  \\\n",
       "0                   [prostaglandins, arachidonic acid]      1   \n",
       "1                                 [levodopa, dopamine]      1   \n",
       "2    [citrulline, arginine, methylamines, dimethyla...      1   \n",
       "3                                    [gaba, glutamate]      1   \n",
       "4                     [5-h, mazindol, mazindane, 5-oh]      1   \n",
       "..                                                 ...    ...   \n",
       "418               [n-alpha-acetylornithine, ornithine]      1   \n",
       "419                          [l-beta-lysine, l-lysine]      1   \n",
       "420                               [estrone, estradiol]      1   \n",
       "421  [5-hydroxy-4-methyltryptophan, 5-methyltryptop...      1   \n",
       "422                               [trehalose, maltose]      1   \n",
       "\n",
       "                  substrates  \\\n",
       "0          arachidonic, acid   \n",
       "1                        NaN   \n",
       "2                   arginine   \n",
       "3                  glutamate   \n",
       "4                        NaN   \n",
       "..                       ...   \n",
       "418                ornithine   \n",
       "419  l-lysine, l-beta-lysine   \n",
       "420       estrone, estradiol   \n",
       "421       5-methyltryptophan   \n",
       "422       maltose, trehalose   \n",
       "\n",
       "                                              products  \\\n",
       "0                                       prostaglandins   \n",
       "1                                                  NaN   \n",
       "2                                                  NaN   \n",
       "3                       gamma-aminobutyric, acid, GABA   \n",
       "4                                                  NaN   \n",
       "..                                                 ...   \n",
       "418                            N-alpha-acetylornithine   \n",
       "419                            l-lysine, l-beta-lysine   \n",
       "420                                 estrone, estradiol   \n",
       "421  5-Hydroxymethyltryptophan, 5-hydroxy-4-methylt...   \n",
       "422                                 maltose, trehalose   \n",
       "\n",
       "                                                  text  label_voter  \n",
       "0    [the enzyme cyclo-oxygenase catalyses the oxyg...            1  \n",
       "1    [although the exact site of decarboxylation of...           -1  \n",
       "2    [the enzyme dimethylarginine dimethylaminohydr...           -1  \n",
       "3    [background and aims glutamic acid decarboxyla...            1  \n",
       "4    [mazindane 26 was found to be a pro-drug, oxid...            1  \n",
       "..                                                 ...          ...  \n",
       "418  [only two exceptions had been reported-the ent...            1  \n",
       "419  [the energetics is here reported for the actio...            1  \n",
       "420  [estradiol 17 beta-hydroxysteroid dehydrogenas...            1  \n",
       "421  [5-hydroxymethyltryptophan and 5-hydroxy-4-met...            1  \n",
       "422  [we show that tres from mycobacterium smegmati...            1  \n",
       "\n",
       "[423 rows x 7 columns]"
      ]
     },
     "execution_count": 103,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "60"
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#oof\n",
    "\n",
    "len(df_test[df_test[\"truth\"] == 1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sentence</th>\n",
       "      <th>chemicals</th>\n",
       "      <th>truth</th>\n",
       "      <th>substrates</th>\n",
       "      <th>products</th>\n",
       "      <th>text</th>\n",
       "      <th>label_voter</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>the enzyme cyclo-oxygenase catalyses the oxyge...</td>\n",
       "      <td>[prostaglandins, arachidonic acid]</td>\n",
       "      <td>1</td>\n",
       "      <td>arachidonic, acid</td>\n",
       "      <td>prostaglandins</td>\n",
       "      <td>[the enzyme cyclo-oxygenase catalyses the oxyg...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>background and aims glutamic acid decarboxylas...</td>\n",
       "      <td>[gaba, glutamate]</td>\n",
       "      <td>1</td>\n",
       "      <td>glutamate</td>\n",
       "      <td>gamma-aminobutyric, acid, GABA</td>\n",
       "      <td>[background and aims glutamic acid decarboxyla...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>mazindane 26 was found to be a pro-drug, oxidi...</td>\n",
       "      <td>[5-h, mazindol, mazindane, 5-oh]</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[mazindane 26 was found to be a pro-drug, oxid...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>interestingly, the allele of pro1 was shown to...</td>\n",
       "      <td>[gamma-glutamyl, l-proline, l-glutamate]</td>\n",
       "      <td>1</td>\n",
       "      <td>L-glutamate</td>\n",
       "      <td>L-proline</td>\n",
       "      <td>[interestingly, the allele of pro1 was shown t...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>in addition to the racemization it also cataly...</td>\n",
       "      <td>[l-ser, pyruvate]</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[in addition to the racemization it also catal...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>418</th>\n",
       "      <td>only two exceptions had been reported-the ente...</td>\n",
       "      <td>[n-alpha-acetylornithine, ornithine]</td>\n",
       "      <td>1</td>\n",
       "      <td>ornithine</td>\n",
       "      <td>N-alpha-acetylornithine</td>\n",
       "      <td>[only two exceptions had been reported-the ent...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>419</th>\n",
       "      <td>the energetics is here reported for the action...</td>\n",
       "      <td>[l-beta-lysine, l-lysine]</td>\n",
       "      <td>1</td>\n",
       "      <td>l-lysine, l-beta-lysine</td>\n",
       "      <td>l-lysine, l-beta-lysine</td>\n",
       "      <td>[the energetics is here reported for the actio...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>420</th>\n",
       "      <td>estradiol 17 beta-hydroxysteroid dehydrogenase...</td>\n",
       "      <td>[estrone, estradiol]</td>\n",
       "      <td>1</td>\n",
       "      <td>estrone, estradiol</td>\n",
       "      <td>estrone, estradiol</td>\n",
       "      <td>[estradiol 17 beta-hydroxysteroid dehydrogenas...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>421</th>\n",
       "      <td>5-hydroxymethyltryptophan and 5-hydroxy-4-meth...</td>\n",
       "      <td>[5-hydroxy-4-methyltryptophan, 5-methyltryptop...</td>\n",
       "      <td>1</td>\n",
       "      <td>5-methyltryptophan</td>\n",
       "      <td>5-Hydroxymethyltryptophan, 5-hydroxy-4-methylt...</td>\n",
       "      <td>[5-hydroxymethyltryptophan and 5-hydroxy-4-met...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>422</th>\n",
       "      <td>we show that tres from mycobacterium smegmatis...</td>\n",
       "      <td>[trehalose, maltose]</td>\n",
       "      <td>1</td>\n",
       "      <td>maltose, trehalose</td>\n",
       "      <td>maltose, trehalose</td>\n",
       "      <td>[we show that tres from mycobacterium smegmati...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>105 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                              sentence  \\\n",
       "0    the enzyme cyclo-oxygenase catalyses the oxyge...   \n",
       "3    background and aims glutamic acid decarboxylas...   \n",
       "4    mazindane 26 was found to be a pro-drug, oxidi...   \n",
       "5    interestingly, the allele of pro1 was shown to...   \n",
       "6    in addition to the racemization it also cataly...   \n",
       "..                                                 ...   \n",
       "418  only two exceptions had been reported-the ente...   \n",
       "419  the energetics is here reported for the action...   \n",
       "420  estradiol 17 beta-hydroxysteroid dehydrogenase...   \n",
       "421  5-hydroxymethyltryptophan and 5-hydroxy-4-meth...   \n",
       "422  we show that tres from mycobacterium smegmatis...   \n",
       "\n",
       "                                             chemicals  truth  \\\n",
       "0                   [prostaglandins, arachidonic acid]      1   \n",
       "3                                    [gaba, glutamate]      1   \n",
       "4                     [5-h, mazindol, mazindane, 5-oh]      1   \n",
       "5             [gamma-glutamyl, l-proline, l-glutamate]      1   \n",
       "6                                    [l-ser, pyruvate]      1   \n",
       "..                                                 ...    ...   \n",
       "418               [n-alpha-acetylornithine, ornithine]      1   \n",
       "419                          [l-beta-lysine, l-lysine]      1   \n",
       "420                               [estrone, estradiol]      1   \n",
       "421  [5-hydroxy-4-methyltryptophan, 5-methyltryptop...      1   \n",
       "422                               [trehalose, maltose]      1   \n",
       "\n",
       "                  substrates  \\\n",
       "0          arachidonic, acid   \n",
       "3                  glutamate   \n",
       "4                        NaN   \n",
       "5                L-glutamate   \n",
       "6                        NaN   \n",
       "..                       ...   \n",
       "418                ornithine   \n",
       "419  l-lysine, l-beta-lysine   \n",
       "420       estrone, estradiol   \n",
       "421       5-methyltryptophan   \n",
       "422       maltose, trehalose   \n",
       "\n",
       "                                              products  \\\n",
       "0                                       prostaglandins   \n",
       "3                       gamma-aminobutyric, acid, GABA   \n",
       "4                                                  NaN   \n",
       "5                                            L-proline   \n",
       "6                                                  NaN   \n",
       "..                                                 ...   \n",
       "418                            N-alpha-acetylornithine   \n",
       "419                            l-lysine, l-beta-lysine   \n",
       "420                                 estrone, estradiol   \n",
       "421  5-Hydroxymethyltryptophan, 5-hydroxy-4-methylt...   \n",
       "422                                 maltose, trehalose   \n",
       "\n",
       "                                                  text  label_voter  \n",
       "0    [the enzyme cyclo-oxygenase catalyses the oxyg...            1  \n",
       "3    [background and aims glutamic acid decarboxyla...            1  \n",
       "4    [mazindane 26 was found to be a pro-drug, oxid...            1  \n",
       "5    [interestingly, the allele of pro1 was shown t...            1  \n",
       "6    [in addition to the racemization it also catal...            1  \n",
       "..                                                 ...          ...  \n",
       "418  [only two exceptions had been reported-the ent...            1  \n",
       "419  [the energetics is here reported for the actio...            1  \n",
       "420  [estradiol 17 beta-hydroxysteroid dehydrogenas...            1  \n",
       "421  [5-hydroxymethyltryptophan and 5-hydroxy-4-met...            1  \n",
       "422  [we show that tres from mycobacterium smegmati...            1  \n",
       "\n",
       "[105 rows x 7 columns]"
      ]
     },
     "execution_count": 105,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_test[df_test[\"label_voter\"] == 1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sentence</th>\n",
       "      <th>chemicals</th>\n",
       "      <th>truth</th>\n",
       "      <th>substrates</th>\n",
       "      <th>products</th>\n",
       "      <th>text</th>\n",
       "      <th>label_voter</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>omega class gsts have dehydroascorbate reducta...</td>\n",
       "      <td>[monomethylarsonate, arsenic]</td>\n",
       "      <td>0</td>\n",
       "      <td>monomethylarsonate</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[omega class gsts have dehydroascorbate reduct...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>n-acetylglucosamine-1-phosphodiester alpha-n-a...</td>\n",
       "      <td>[n-acetylglucosamine-1-phosphodiester, phospho...</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>mannose, 6-phosphate</td>\n",
       "      <td>[n-acetylglucosamine-1-phosphodiester alpha-n-...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>cgmp inhibited hydrolysis of camp, and camp in...</td>\n",
       "      <td>[cgmp, camp]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[cgmp inhibited hydrolysis of camp, and camp i...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84</th>\n",
       "      <td>on phosphorylation of ser40 by protein kinase ...</td>\n",
       "      <td>[h4biopterin, ser40]</td>\n",
       "      <td>0</td>\n",
       "      <td>Ser40</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[on phosphorylation of ser40 by protein kinase...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>thus, our studies provide new insight into the...</td>\n",
       "      <td>[catecholamine, h4biopterin]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[thus, our studies provide new insight into th...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>112</th>\n",
       "      <td>only in the last couple of years a consensus h...</td>\n",
       "      <td>[bh4, no]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[only in the last couple of years a consensus ...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113</th>\n",
       "      <td>the nos-catalyzed oxidation of nadph in the ab...</td>\n",
       "      <td>[nadph, no]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[the nos-catalyzed oxidation of nadph in the a...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>120</th>\n",
       "      <td>conclusions in the maximum registered dosage, ...</td>\n",
       "      <td>[nabumetone, meloxicam, thromboxane]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[conclusions in the maximum registered dosage,...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122</th>\n",
       "      <td>methods studies were performed on enzymatic ac...</td>\n",
       "      <td>[3h-gaba, 3h-glutamate, plp]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[methods studies were performed on enzymatic a...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>124</th>\n",
       "      <td>the results reinforce previous assumptions tha...</td>\n",
       "      <td>[eicosanoid, dopamine]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[the results reinforce previous assumptions th...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>125</th>\n",
       "      <td>among the [fe-s] cluster biosynthetic proteins...</td>\n",
       "      <td>[l-cysteine, fe-s, sulphur]</td>\n",
       "      <td>0</td>\n",
       "      <td>l-cysteine</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[among the [fe-s] cluster biosynthetic protein...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135</th>\n",
       "      <td>these results suggest that the altered gamma-g...</td>\n",
       "      <td>[gamma-glutamyl, l-proline]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>L-proline</td>\n",
       "      <td>[these results suggest that the altered gamma-...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>144</th>\n",
       "      <td>in shoots, therefore, the pathway from formate...</td>\n",
       "      <td>[ser, formate]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[in shoots, therefore, the pathway from format...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>146</th>\n",
       "      <td>compared with wt, glyd plants accumulated 5-fo...</td>\n",
       "      <td>[glyd, ser, gly]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[compared with wt, glyd plants accumulated 5-f...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151</th>\n",
       "      <td>the protein exhibited modest h2o2-dependent pe...</td>\n",
       "      <td>[guaiacol, abts]</td>\n",
       "      <td>0</td>\n",
       "      <td>guaiacol, potassium, iodide, 2,2, acid, ABTS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[the protein exhibited modest h2o2-dependent p...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>153</th>\n",
       "      <td>purified paoh1/smo oxidizes both spermine km=1...</td>\n",
       "      <td>[spermidine, spermine]</td>\n",
       "      <td>0</td>\n",
       "      <td>spermine, N</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[purified paoh1/smo oxidizes both spermine km=...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>154</th>\n",
       "      <td>the purified human enzyme also does not oxidiz...</td>\n",
       "      <td>[spermine, polyamine]</td>\n",
       "      <td>0</td>\n",
       "      <td>spermine</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[the purified human enzyme also does not oxidi...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>161</th>\n",
       "      <td>ten iu/ml urokinase was also incubated with po...</td>\n",
       "      <td>[ct, 1o2]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[ten iu/ml urokinase was also incubated with p...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162</th>\n",
       "      <td>after 0 to 80 minutes 37 degrees c, 50-microl ...</td>\n",
       "      <td>[ct, citrate, arginine]</td>\n",
       "      <td>0</td>\n",
       "      <td>arginine</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[after 0 to 80 minutes 37 degrees c, 50-microl...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>181</th>\n",
       "      <td>qr2 catalyzes the two-electron reduction of me...</td>\n",
       "      <td>[n-alkylated, menadione]</td>\n",
       "      <td>0</td>\n",
       "      <td>menadione, N-alkylated, N-ribosylated, nicotin...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[qr2 catalyzes the two-electron reduction of m...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>183</th>\n",
       "      <td>in contrast, the inhibitors quinacrine and chl...</td>\n",
       "      <td>[quinacrine, chloroquine]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[in contrast, the inhibitors quinacrine and ch...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>184</th>\n",
       "      <td>the quinone substrate menadione, on the other ...</td>\n",
       "      <td>[quinone, menadione]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[the quinone substrate menadione, on the other...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>194</th>\n",
       "      <td>this alternate conformation now causes a steri...</td>\n",
       "      <td>[udp-gal, gal]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[this alternate conformation now causes a ster...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>195</th>\n",
       "      <td>we previously reported that aminoglycoside ant...</td>\n",
       "      <td>[aminoglycoside, glucose]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[we previously reported that aminoglycoside an...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208</th>\n",
       "      <td>thus, it appears that the low conversion of py...</td>\n",
       "      <td>[plp, pyridoxal, pyridoxine]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[thus, it appears that the low conversion of p...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210</th>\n",
       "      <td>glufosinate inhibits glutamine synthetase and ...</td>\n",
       "      <td>[glutamine, glufosinate]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>glutamine</td>\n",
       "      <td>[glufosinate inhibits glutamine synthetase and...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>213</th>\n",
       "      <td>reduced synthesis of nitric oxide no contribut...</td>\n",
       "      <td>[nitric-oxide, l-arginine, no]</td>\n",
       "      <td>0</td>\n",
       "      <td>L-arginine</td>\n",
       "      <td>nitric, oxide, NO</td>\n",
       "      <td>[reduced synthesis of nitric oxide no contribu...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>218</th>\n",
       "      <td>however, despite activation of l-arginine upta...</td>\n",
       "      <td>[nor-noha, l-arginine]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[however, despite activation of l-arginine upt...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>220</th>\n",
       "      <td>these data suggest that endothelial no synthes...</td>\n",
       "      <td>[l-arginine, no]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NO</td>\n",
       "      <td>[these data suggest that endothelial no synthe...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>240</th>\n",
       "      <td>on the basis of our findings with structurally...</td>\n",
       "      <td>[nadh, arylhydroxylamine]</td>\n",
       "      <td>0</td>\n",
       "      <td>arylhydroxylamine</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[on the basis of our findings with structurall...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>253</th>\n",
       "      <td>exogenous all-trans-retinol, all-trans-13,14-d...</td>\n",
       "      <td>[all-trans-13,14-dihydroretinol, dihydroretino...</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[exogenous all-trans-retinol, all-trans-13,14-...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>256</th>\n",
       "      <td>conclusions the fads2 deletion may prevent the...</td>\n",
       "      <td>[pufas, ala]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[conclusions the fads2 deletion may prevent th...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>260</th>\n",
       "      <td>replacement of nacl by licl or choline chlorid...</td>\n",
       "      <td>[nacl, licl]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[replacement of nacl by licl or choline chlori...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>263</th>\n",
       "      <td>inhibition of endogenous production of h2s by ...</td>\n",
       "      <td>[pag, sp]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[inhibition of endogenous production of h2s by...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>282</th>\n",
       "      <td>affinity of v782a for cgmp, vardenafil, silden...</td>\n",
       "      <td>[tadalafil, ibmx, sildenafil, vardenafil, cgmp]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[affinity of v782a for cgmp, vardenafil, silde...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>301</th>\n",
       "      <td>second, probucol, an antiatherogenic compound ...</td>\n",
       "      <td>[alpha-tocopherol, probucol]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[second, probucol, an antiatherogenic compound...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>310</th>\n",
       "      <td>thymidine kinase-1 tk-1 and thymidylate syntha...</td>\n",
       "      <td>[thymidine, thymidylate, pyrimidine]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>pyrimidine</td>\n",
       "      <td>[thymidine kinase-1 tk-1 and thymidylate synth...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>322</th>\n",
       "      <td>we have previously reported that tzds reduce e...</td>\n",
       "      <td>[tzds, estrogen]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>estrogen</td>\n",
       "      <td>[we have previously reported that tzds reduce ...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>324</th>\n",
       "      <td>tzds inhibited estradiol production by up to 8...</td>\n",
       "      <td>[tzds, estradiol]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[tzds inhibited estradiol production by up to ...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>325</th>\n",
       "      <td>the results indicate that the glutathione conj...</td>\n",
       "      <td>[bpde, glutathione]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[the results indicate that the glutathione con...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>347</th>\n",
       "      <td>tension studies demonstrated a rapid, concentr...</td>\n",
       "      <td>[tes, 17î²-estradiol]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[tension studies demonstrated a rapid, concent...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>354</th>\n",
       "      <td>production of reactive oxygen species ros in t...</td>\n",
       "      <td>[dihydroethidium, oxygen]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[production of reactive oxygen species ros in ...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>358</th>\n",
       "      <td>our results suggest that arsenite can potentia...</td>\n",
       "      <td>[nadph, arsenite, superoxide, no]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>superoxide</td>\n",
       "      <td>[our results suggest that arsenite can potenti...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>363</th>\n",
       "      <td>dimethylarsinic acid dmasv, the major urinary ...</td>\n",
       "      <td>[dmas, arsenic]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[dimethylarsinic acid dmasv, the major urinary...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>364</th>\n",
       "      <td>although glutathione s-transferase omega 1 gst...</td>\n",
       "      <td>[dmas, arsenic, glutathione]</td>\n",
       "      <td>0</td>\n",
       "      <td>DMAs</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[although glutathione s-transferase omega 1 gs...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>367</th>\n",
       "      <td>assessment of dmasv-reducing activity of rat l...</td>\n",
       "      <td>[dmas, dehydroascorbate, gsh, thiol, gssg]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[assessment of dmasv-reducing activity of rat ...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>368</th>\n",
       "      <td>although thioredoxin reductase trr inhibitors ...</td>\n",
       "      <td>[nadph, dmas, aurothioglucose]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[although thioredoxin reductase trr inhibitors...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>377</th>\n",
       "      <td>moreover, a new mechanism for controlling endo...</td>\n",
       "      <td>[nadph, no]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[moreover, a new mechanism for controlling end...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>385</th>\n",
       "      <td>taken together, these data demonstrate that ba...</td>\n",
       "      <td>[dxm, baicalin]</td>\n",
       "      <td>0</td>\n",
       "      <td>DXM</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[taken together, these data demonstrate that b...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>386</th>\n",
       "      <td>the mechanisms by which oxysterols may influen...</td>\n",
       "      <td>[cholesterol, oxysterols]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[the mechanisms by which oxysterols may influe...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>402</th>\n",
       "      <td>thiotepa cyp2b6 inhibitor, 25 mum and the mono...</td>\n",
       "      <td>[thiotepa, troleandomycin, ketamine]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[thiotepa cyp2b6 inhibitor, 25 mum and the mon...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>409</th>\n",
       "      <td>the biological activities of 6-dhsg were atten...</td>\n",
       "      <td>[nac, gsh, 6-dhsg, glutathione, sulfhydryl, asc]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[the biological activities of 6-dhsg were atte...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>410</th>\n",
       "      <td>gs-6-dhsg showed reduced biological activities...</td>\n",
       "      <td>[6-dhsg, gs-6-dhsg]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[gs-6-dhsg showed reduced biological activitie...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>412</th>\n",
       "      <td>platelet-induced cox-2-dependent pge2 synthesi...</td>\n",
       "      <td>[rofecoxib, pge2]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PGE2</td>\n",
       "      <td>[platelet-induced cox-2-dependent pge2 synthes...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                              sentence  \\\n",
       "53   omega class gsts have dehydroascorbate reducta...   \n",
       "61   n-acetylglucosamine-1-phosphodiester alpha-n-a...   \n",
       "70   cgmp inhibited hydrolysis of camp, and camp in...   \n",
       "84   on phosphorylation of ser40 by protein kinase ...   \n",
       "88   thus, our studies provide new insight into the...   \n",
       "112  only in the last couple of years a consensus h...   \n",
       "113  the nos-catalyzed oxidation of nadph in the ab...   \n",
       "120  conclusions in the maximum registered dosage, ...   \n",
       "122  methods studies were performed on enzymatic ac...   \n",
       "124  the results reinforce previous assumptions tha...   \n",
       "125  among the [fe-s] cluster biosynthetic proteins...   \n",
       "135  these results suggest that the altered gamma-g...   \n",
       "144  in shoots, therefore, the pathway from formate...   \n",
       "146  compared with wt, glyd plants accumulated 5-fo...   \n",
       "151  the protein exhibited modest h2o2-dependent pe...   \n",
       "153  purified paoh1/smo oxidizes both spermine km=1...   \n",
       "154  the purified human enzyme also does not oxidiz...   \n",
       "161  ten iu/ml urokinase was also incubated with po...   \n",
       "162  after 0 to 80 minutes 37 degrees c, 50-microl ...   \n",
       "181  qr2 catalyzes the two-electron reduction of me...   \n",
       "183  in contrast, the inhibitors quinacrine and chl...   \n",
       "184  the quinone substrate menadione, on the other ...   \n",
       "194  this alternate conformation now causes a steri...   \n",
       "195  we previously reported that aminoglycoside ant...   \n",
       "208  thus, it appears that the low conversion of py...   \n",
       "210  glufosinate inhibits glutamine synthetase and ...   \n",
       "213  reduced synthesis of nitric oxide no contribut...   \n",
       "218  however, despite activation of l-arginine upta...   \n",
       "220  these data suggest that endothelial no synthes...   \n",
       "240  on the basis of our findings with structurally...   \n",
       "253  exogenous all-trans-retinol, all-trans-13,14-d...   \n",
       "256  conclusions the fads2 deletion may prevent the...   \n",
       "260  replacement of nacl by licl or choline chlorid...   \n",
       "263  inhibition of endogenous production of h2s by ...   \n",
       "282  affinity of v782a for cgmp, vardenafil, silden...   \n",
       "301  second, probucol, an antiatherogenic compound ...   \n",
       "310  thymidine kinase-1 tk-1 and thymidylate syntha...   \n",
       "322  we have previously reported that tzds reduce e...   \n",
       "324  tzds inhibited estradiol production by up to 8...   \n",
       "325  the results indicate that the glutathione conj...   \n",
       "347  tension studies demonstrated a rapid, concentr...   \n",
       "354  production of reactive oxygen species ros in t...   \n",
       "358  our results suggest that arsenite can potentia...   \n",
       "363  dimethylarsinic acid dmasv, the major urinary ...   \n",
       "364  although glutathione s-transferase omega 1 gst...   \n",
       "367  assessment of dmasv-reducing activity of rat l...   \n",
       "368  although thioredoxin reductase trr inhibitors ...   \n",
       "377  moreover, a new mechanism for controlling endo...   \n",
       "385  taken together, these data demonstrate that ba...   \n",
       "386  the mechanisms by which oxysterols may influen...   \n",
       "402  thiotepa cyp2b6 inhibitor, 25 mum and the mono...   \n",
       "409  the biological activities of 6-dhsg were atten...   \n",
       "410  gs-6-dhsg showed reduced biological activities...   \n",
       "412  platelet-induced cox-2-dependent pge2 synthesi...   \n",
       "\n",
       "                                             chemicals  truth  \\\n",
       "53                       [monomethylarsonate, arsenic]      0   \n",
       "61   [n-acetylglucosamine-1-phosphodiester, phospho...      0   \n",
       "70                                        [cgmp, camp]      0   \n",
       "84                                [h4biopterin, ser40]      0   \n",
       "88                        [catecholamine, h4biopterin]      0   \n",
       "112                                          [bh4, no]      0   \n",
       "113                                        [nadph, no]      0   \n",
       "120               [nabumetone, meloxicam, thromboxane]      0   \n",
       "122                       [3h-gaba, 3h-glutamate, plp]      0   \n",
       "124                             [eicosanoid, dopamine]      0   \n",
       "125                        [l-cysteine, fe-s, sulphur]      0   \n",
       "135                        [gamma-glutamyl, l-proline]      0   \n",
       "144                                     [ser, formate]      0   \n",
       "146                                   [glyd, ser, gly]      0   \n",
       "151                                   [guaiacol, abts]      0   \n",
       "153                             [spermidine, spermine]      0   \n",
       "154                              [spermine, polyamine]      0   \n",
       "161                                          [ct, 1o2]      0   \n",
       "162                            [ct, citrate, arginine]      0   \n",
       "181                           [n-alkylated, menadione]      0   \n",
       "183                          [quinacrine, chloroquine]      0   \n",
       "184                               [quinone, menadione]      0   \n",
       "194                                     [udp-gal, gal]      0   \n",
       "195                          [aminoglycoside, glucose]      0   \n",
       "208                       [plp, pyridoxal, pyridoxine]      0   \n",
       "210                           [glutamine, glufosinate]      0   \n",
       "213                     [nitric-oxide, l-arginine, no]      0   \n",
       "218                             [nor-noha, l-arginine]      0   \n",
       "220                                   [l-arginine, no]      0   \n",
       "240                          [nadh, arylhydroxylamine]      0   \n",
       "253  [all-trans-13,14-dihydroretinol, dihydroretino...      0   \n",
       "256                                       [pufas, ala]      0   \n",
       "260                                       [nacl, licl]      0   \n",
       "263                                          [pag, sp]      0   \n",
       "282    [tadalafil, ibmx, sildenafil, vardenafil, cgmp]      0   \n",
       "301                       [alpha-tocopherol, probucol]      0   \n",
       "310               [thymidine, thymidylate, pyrimidine]      0   \n",
       "322                                   [tzds, estrogen]      0   \n",
       "324                                  [tzds, estradiol]      0   \n",
       "325                                [bpde, glutathione]      0   \n",
       "347                              [tes, 17î²-estradiol]      0   \n",
       "354                          [dihydroethidium, oxygen]      0   \n",
       "358                  [nadph, arsenite, superoxide, no]      0   \n",
       "363                                    [dmas, arsenic]      0   \n",
       "364                       [dmas, arsenic, glutathione]      0   \n",
       "367         [dmas, dehydroascorbate, gsh, thiol, gssg]      0   \n",
       "368                     [nadph, dmas, aurothioglucose]      0   \n",
       "377                                        [nadph, no]      0   \n",
       "385                                    [dxm, baicalin]      0   \n",
       "386                          [cholesterol, oxysterols]      0   \n",
       "402               [thiotepa, troleandomycin, ketamine]      0   \n",
       "409   [nac, gsh, 6-dhsg, glutathione, sulfhydryl, asc]      0   \n",
       "410                                [6-dhsg, gs-6-dhsg]      0   \n",
       "412                                  [rofecoxib, pge2]      0   \n",
       "\n",
       "                                            substrates              products  \\\n",
       "53                                  monomethylarsonate                   NaN   \n",
       "61                                                 NaN  mannose, 6-phosphate   \n",
       "70                                                 NaN                   NaN   \n",
       "84                                               Ser40                   NaN   \n",
       "88                                                 NaN                   NaN   \n",
       "112                                                NaN                   NaN   \n",
       "113                                                NaN                   NaN   \n",
       "120                                                NaN                   NaN   \n",
       "122                                                NaN                   NaN   \n",
       "124                                                NaN                   NaN   \n",
       "125                                         l-cysteine                   NaN   \n",
       "135                                                NaN             L-proline   \n",
       "144                                                NaN                   NaN   \n",
       "146                                                NaN                   NaN   \n",
       "151       guaiacol, potassium, iodide, 2,2, acid, ABTS                   NaN   \n",
       "153                                        spermine, N                   NaN   \n",
       "154                                           spermine                   NaN   \n",
       "161                                                NaN                   NaN   \n",
       "162                                           arginine                   NaN   \n",
       "181  menadione, N-alkylated, N-ribosylated, nicotin...                   NaN   \n",
       "183                                                NaN                   NaN   \n",
       "184                                                NaN                   NaN   \n",
       "194                                                NaN                   NaN   \n",
       "195                                                NaN                   NaN   \n",
       "208                                                NaN                   NaN   \n",
       "210                                                NaN             glutamine   \n",
       "213                                         L-arginine     nitric, oxide, NO   \n",
       "218                                                NaN                   NaN   \n",
       "220                                                NaN                    NO   \n",
       "240                                  arylhydroxylamine                   NaN   \n",
       "253                                                NaN                   NaN   \n",
       "256                                                NaN                   NaN   \n",
       "260                                                NaN                   NaN   \n",
       "263                                                NaN                   NaN   \n",
       "282                                                NaN                   NaN   \n",
       "301                                                NaN                   NaN   \n",
       "310                                                NaN            pyrimidine   \n",
       "322                                                NaN              estrogen   \n",
       "324                                                NaN                   NaN   \n",
       "325                                                NaN                   NaN   \n",
       "347                                                NaN                   NaN   \n",
       "354                                                NaN                   NaN   \n",
       "358                                                NaN            superoxide   \n",
       "363                                                NaN                   NaN   \n",
       "364                                               DMAs                   NaN   \n",
       "367                                                NaN                   NaN   \n",
       "368                                                NaN                   NaN   \n",
       "377                                                NaN                   NaN   \n",
       "385                                                DXM                   NaN   \n",
       "386                                                NaN                   NaN   \n",
       "402                                                NaN                   NaN   \n",
       "409                                                NaN                   NaN   \n",
       "410                                                NaN                   NaN   \n",
       "412                                                NaN                  PGE2   \n",
       "\n",
       "                                                  text  label_voter  \n",
       "53   [omega class gsts have dehydroascorbate reduct...            1  \n",
       "61   [n-acetylglucosamine-1-phosphodiester alpha-n-...            1  \n",
       "70   [cgmp inhibited hydrolysis of camp, and camp i...            1  \n",
       "84   [on phosphorylation of ser40 by protein kinase...            1  \n",
       "88   [thus, our studies provide new insight into th...            1  \n",
       "112  [only in the last couple of years a consensus ...            1  \n",
       "113  [the nos-catalyzed oxidation of nadph in the a...            1  \n",
       "120  [conclusions in the maximum registered dosage,...            1  \n",
       "122  [methods studies were performed on enzymatic a...            1  \n",
       "124  [the results reinforce previous assumptions th...            1  \n",
       "125  [among the [fe-s] cluster biosynthetic protein...            1  \n",
       "135  [these results suggest that the altered gamma-...            1  \n",
       "144  [in shoots, therefore, the pathway from format...            1  \n",
       "146  [compared with wt, glyd plants accumulated 5-f...            1  \n",
       "151  [the protein exhibited modest h2o2-dependent p...            1  \n",
       "153  [purified paoh1/smo oxidizes both spermine km=...            1  \n",
       "154  [the purified human enzyme also does not oxidi...            1  \n",
       "161  [ten iu/ml urokinase was also incubated with p...            1  \n",
       "162  [after 0 to 80 minutes 37 degrees c, 50-microl...            1  \n",
       "181  [qr2 catalyzes the two-electron reduction of m...            1  \n",
       "183  [in contrast, the inhibitors quinacrine and ch...            1  \n",
       "184  [the quinone substrate menadione, on the other...            1  \n",
       "194  [this alternate conformation now causes a ster...            1  \n",
       "195  [we previously reported that aminoglycoside an...            1  \n",
       "208  [thus, it appears that the low conversion of p...            1  \n",
       "210  [glufosinate inhibits glutamine synthetase and...            1  \n",
       "213  [reduced synthesis of nitric oxide no contribu...            1  \n",
       "218  [however, despite activation of l-arginine upt...            1  \n",
       "220  [these data suggest that endothelial no synthe...            1  \n",
       "240  [on the basis of our findings with structurall...            1  \n",
       "253  [exogenous all-trans-retinol, all-trans-13,14-...            1  \n",
       "256  [conclusions the fads2 deletion may prevent th...            1  \n",
       "260  [replacement of nacl by licl or choline chlori...            1  \n",
       "263  [inhibition of endogenous production of h2s by...            1  \n",
       "282  [affinity of v782a for cgmp, vardenafil, silde...            1  \n",
       "301  [second, probucol, an antiatherogenic compound...            1  \n",
       "310  [thymidine kinase-1 tk-1 and thymidylate synth...            1  \n",
       "322  [we have previously reported that tzds reduce ...            1  \n",
       "324  [tzds inhibited estradiol production by up to ...            1  \n",
       "325  [the results indicate that the glutathione con...            1  \n",
       "347  [tension studies demonstrated a rapid, concent...            1  \n",
       "354  [production of reactive oxygen species ros in ...            1  \n",
       "358  [our results suggest that arsenite can potenti...            1  \n",
       "363  [dimethylarsinic acid dmasv, the major urinary...            1  \n",
       "364  [although glutathione s-transferase omega 1 gs...            1  \n",
       "367  [assessment of dmasv-reducing activity of rat ...            1  \n",
       "368  [although thioredoxin reductase trr inhibitors...            1  \n",
       "377  [moreover, a new mechanism for controlling end...            1  \n",
       "385  [taken together, these data demonstrate that b...            1  \n",
       "386  [the mechanisms by which oxysterols may influe...            1  \n",
       "402  [thiotepa cyp2b6 inhibitor, 25 mum and the mon...            1  \n",
       "409  [the biological activities of 6-dhsg were atte...            1  \n",
       "410  [gs-6-dhsg showed reduced biological activitie...            1  \n",
       "412  [platelet-induced cox-2-dependent pge2 synthes...            1  "
      ]
     },
     "execution_count": 106,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_false = df_test[df_test[\"truth\"] == 0]\n",
    "df_fp = df_false[df_false[\"label_voter\"] == 1]\n",
    "\n",
    "df_fp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([-1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1,\n",
       "       -1, -1, -1, -1, -1, -1,  1, -1, -1, -1, -1, -1,  0, -1, -1,  1, -1,\n",
       "       -1, -1, -1,  1, -1, -1, -1, -1,  1, -1, -1, -1, -1, -1, -1, -1, -1,\n",
       "        1])"
      ]
     },
     "execution_count": 107,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "L_test[51]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sentence</th>\n",
       "      <th>chemicals</th>\n",
       "      <th>truth</th>\n",
       "      <th>substrates</th>\n",
       "      <th>products</th>\n",
       "      <th>text</th>\n",
       "      <th>label_voter</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>metabolism of vitamin a, all-trans-retinol, le...</td>\n",
       "      <td>[11-cis-retinaldehyde, all-trans-retinol]</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[metabolism of vitamin a, all-trans-retinol, l...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>unlike mouse retsat mretsat, zretsat a had an ...</td>\n",
       "      <td>[all-trans-13,14-dihydroretinol, all-trans-7,8...</td>\n",
       "      <td>1</td>\n",
       "      <td>all-trans-retinol</td>\n",
       "      <td>all-trans-13,14-dihydroretinol, all-trans-7,8-...</td>\n",
       "      <td>[unlike mouse retsat mretsat, zretsat a had an...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>tas-102 is a novel drug containing trifluoroth...</td>\n",
       "      <td>[tas-102, trifluorothymidine]</td>\n",
       "      <td>1</td>\n",
       "      <td>TAS-102, trifluorothymidine</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[tas-102 is a novel drug containing trifluorot...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>these findings, derived from a variety of anal...</td>\n",
       "      <td>[no, peroxynitrite, superoxide, androgen, xant...</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>peroxynitrite, superoxide, NO</td>\n",
       "      <td>[these findings, derived from a variety of ana...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>pip5k1b encodes phosphatidylinositol 4-phospha...</td>\n",
       "      <td>[phosphatidylinositol 4-phosphate, phosphatidy...</td>\n",
       "      <td>1</td>\n",
       "      <td>phosphatidylinositol, 4-phosphate, PI</td>\n",
       "      <td>phosphatidylinositol-4,5-bisphosphate, PI</td>\n",
       "      <td>[pip5k1b encodes phosphatidylinositol 4-phosph...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             sentence  \\\n",
       "26  metabolism of vitamin a, all-trans-retinol, le...   \n",
       "28  unlike mouse retsat mretsat, zretsat a had an ...   \n",
       "35  tas-102 is a novel drug containing trifluoroth...   \n",
       "44  these findings, derived from a variety of anal...   \n",
       "50  pip5k1b encodes phosphatidylinositol 4-phospha...   \n",
       "\n",
       "                                            chemicals  truth  \\\n",
       "26          [11-cis-retinaldehyde, all-trans-retinol]      1   \n",
       "28  [all-trans-13,14-dihydroretinol, all-trans-7,8...      1   \n",
       "35                      [tas-102, trifluorothymidine]      1   \n",
       "44  [no, peroxynitrite, superoxide, androgen, xant...      1   \n",
       "50  [phosphatidylinositol 4-phosphate, phosphatidy...      1   \n",
       "\n",
       "                               substrates  \\\n",
       "26                                    NaN   \n",
       "28                      all-trans-retinol   \n",
       "35            TAS-102, trifluorothymidine   \n",
       "44                                    NaN   \n",
       "50  phosphatidylinositol, 4-phosphate, PI   \n",
       "\n",
       "                                             products  \\\n",
       "26                                                NaN   \n",
       "28  all-trans-13,14-dihydroretinol, all-trans-7,8-...   \n",
       "35                                                NaN   \n",
       "44                      peroxynitrite, superoxide, NO   \n",
       "50          phosphatidylinositol-4,5-bisphosphate, PI   \n",
       "\n",
       "                                                 text  label_voter  \n",
       "26  [metabolism of vitamin a, all-trans-retinol, l...            0  \n",
       "28  [unlike mouse retsat mretsat, zretsat a had an...            0  \n",
       "35  [tas-102 is a novel drug containing trifluorot...            0  \n",
       "44  [these findings, derived from a variety of ana...            0  \n",
       "50  [pip5k1b encodes phosphatidylinositol 4-phosph...            0  "
      ]
     },
     "execution_count": 108,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_true = df_test[df_test[\"truth\"] == 1]\n",
    "df_fn = df_true[df_true[\"label_voter\"] == 0]\n",
    "\n",
    "df_fn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_tp = df_true[df_true['label_voter'] == 1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SENTENCE:\n",
      "the enzyme cyclo-oxygenase catalyses the oxygenation of arachidonic acid, leading to the formation of prostaglandins\n",
      "\n",
      "CHEMICALS:\n",
      "['prostaglandins', 'arachidonic acid']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1  1  1  1]\n",
      "\n",
      "NEGS: 1 POS: 7\n",
      "\n",
      "['includes_general_chemical_words']\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'structure_conversion_of', 'includes_impact_words', 'includes_catalyze_words', 'includes_emzyze_words', 'includes_reaction_words2']\n",
      "\n",
      "SENTENCE:\n",
      "background and aims glutamic acid decarboxylase gad, ec 41115 catalyses the conversion of glutamate to gamma-aminobutyric acid gaba\n",
      "\n",
      "CHEMICALS:\n",
      "['gaba', 'glutamate']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1  1 -1  1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1  1 -1 -1\n",
      " -1  1 -1  1]\n",
      "\n",
      "NEGS: 1 POS: 7\n",
      "\n",
      "['includes_physical_words']\n",
      "\n",
      "['includes_reaction_words', 'structure_next_to_conversion_words', 'structure_conversion_of', 'structure_jtsui_pattern_4', 'structure_jtsui_pattern_4_copy', 'includes_catalyze_words', 'includes_reaction_words2']\n",
      "\n",
      "SENTENCE:\n",
      "mazindane 26 was found to be a pro-drug, oxidizing 5-h --> 5-oh to mazindol on rat striatal membranes and hek-hdat cells\n",
      "\n",
      "CHEMICALS:\n",
      "['5-h', 'mazindol', 'mazindane', '5-oh']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1  1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1  0 -1 -1  1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 1 POS: 6\n",
      "\n",
      "['includes_experiment_words']\n",
      "\n",
      "['includes_oxidation_words', 'structure_sep_conversion_words', 'includes_reaction_words', 'structure_next_to_conversion_words', 'includes_react_sym', 'includes_reaction_words2']\n",
      "\n",
      "SENTENCE:\n",
      "interestingly, the allele of pro1 was shown to enhance the activities of gamma-glutamyl kinase and gamma-glutamyl phosphate reductase, both of which catalyze the first two steps of l-proline synthesis from l-glutamate and which together may form a complex in vivo\n",
      "\n",
      "CHEMICALS:\n",
      "['gamma-glutamyl', 'l-proline', 'l-glutamate']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1  0  0 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      "  1 -1 -1 -1  0 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      "  1  1 -1  1]\n",
      "\n",
      "NEGS: 4 POS: 7\n",
      "\n",
      "['includes_physical_words', 'includes_genetic_words', 'includes_functional_group', 'includes_measure_words']\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'includes_reaction_component_words', 'structure_next_to_conversion_words', 'includes_creation_words', 'includes_catalyze_words', 'includes_reaction_words2']\n",
      "\n",
      "SENTENCE:\n",
      "in addition to the racemization it also catalyzes specific elimination of l-ser to pyruvate\n",
      "\n",
      "CHEMICALS:\n",
      "['l-ser', 'pyruvate']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1  1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1  1 -1 -1]\n",
      "\n",
      "NEGS: 0 POS: 4\n",
      "\n",
      "[]\n",
      "\n",
      "['structure_sep_conversion_words', 'structure_next_to_conversion_words', 'structure_conversion_of', 'includes_catalyze_words']\n",
      "\n",
      "SENTENCE:\n",
      "we concluded that fdh has no direct role in the regulation of the above two pathways of ser synthesis the breakdown of formate to co2 by the fdh reaction is the primary and preferred fate of the organic acid in arabidopsis\n",
      "\n",
      "CHEMICALS:\n",
      "['co(2)', 'ser', 'formate']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1  1 -1 -1 -1 -1  1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      "  1 -1 -1  1]\n",
      "\n",
      "NEGS: 1 POS: 6\n",
      "\n",
      "['includes_paper_artifacts']\n",
      "\n",
      "['includes_reaction_words', 'includes_react', 'structure_next_to_conversion_words', 'structure_conversion_of', 'includes_creation_words', 'includes_reaction_words2']\n",
      "\n",
      "SENTENCE:\n",
      "the ratio between the gdc/shmt and c1-thf synthase/shmt pathways of ser synthesis from [alpha-13c]gly and [13c]formate, respectively, in arabidopsis shoots was 21  1 in roots, 9  1\n",
      "\n",
      "CHEMICALS:\n",
      "['ser', 'formate', 'gly']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      "  1 -1 -1  1]\n",
      "\n",
      "NEGS: 1 POS: 3\n",
      "\n",
      "['includes_measure_words']\n",
      "\n",
      "['includes_reaction_words', 'includes_creation_words', 'includes_reaction_words2']\n",
      "\n",
      "SENTENCE:\n",
      "one of the enzymes responsible for the production of ka, kynurenine aminotransferase i kati, also catalyses the reversible transamination of glutamine to oxoglutaramic acid gtk, ec 26115\n",
      "\n",
      "CHEMICALS:\n",
      "['glutamine', 'kynurenine', 'oxoglutaramic acid', 'ka']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1  0 -1 -1 -1 -1  0 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1  1\n",
      "  1 -1 -1 -1 -1 -1 -1 -1  1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1  1  1  1]\n",
      "\n",
      "NEGS: 2 POS: 8\n",
      "\n",
      "['includes_functional_group', 'structure_sep_comma']\n",
      "\n",
      "['structure_sep_conversion_words', 'includes_reaction_words', 'includes_reaction_component_words', 'structure_next_to_conversion_words', 'structure_conversion_of', 'includes_catalyze_words', 'includes_emzyze_words', 'includes_reaction_words2']\n",
      "\n",
      "SENTENCE:\n",
      "cpt i carnitine palmitoyltransferase i catalyses the conversion of palmitoyl-coa into palmitoylcarnitine in the presence of l-carnitine, facilitating the entry of fatty acids into mitochondria\n",
      "\n",
      "CHEMICALS:\n",
      "['palmitoyl-coa', 'carnitine', 'palmitoylcarnitine', 'l-carnitine']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1  1 -1  1 -1 -1  1 -1 -1  1 -1  1 -1 -1  1 -1 -1\n",
      " -1  1 -1  1]\n",
      "\n",
      "NEGS: 0 POS: 10\n",
      "\n",
      "[]\n",
      "\n",
      "['structure_sep_conversion_words', 'includes_reaction_words', 'structure_next_to_conversion_words', 'structure_conversion_of', 'structure_jtsui_pattern_1', 'structure_jtsui_pattern_4', 'structure_jtsui_pattern_1_copy', 'structure_jtsui_pattern_4_copy', 'includes_catalyze_words', 'includes_reaction_words2']\n",
      "\n",
      "SENTENCE:\n",
      "purpose the fluoropyrimidine carbamate capecitabine is converted to 5-fluorouracil 5-fu by thymidine phosphorylase tp inside target tissues\n",
      "\n",
      "CHEMICALS:\n",
      "['thymidine', '5-fluorouracil', 'capecitabine', '5-fu']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1  0 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1  0  0 -1 -1  1 -1 -1 -1 -1 -1  1 -1  1 -1 -1  1 -1  1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 4 POS: 7\n",
      "\n",
      "['structure_adjacent_mentions', 'structure_sep_sym', 'includes_measure_words', 'includes_experiment_words']\n",
      "\n",
      "['includes_reaction_words', 'structure_next_to_conversion_words', 'structure_jtsui_pattern_2', 'structure_jtsui_pattern_4', 'structure_jtsui_pattern_2_copy', 'structure_jtsui_pattern_4_copy', 'includes_reaction_words2']\n",
      "\n",
      "SENTENCE:\n",
      "spermine was rapidly taken up and metabolized to spermidine and an unknown, possibly acetamidopropanal, by spermidine/spermine n1-acetyltransferase ssat and polyamine oxidase pao\n",
      "\n",
      "CHEMICALS:\n",
      "['spermidine', 'polyamine', 'spermine']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 2 POS: 3\n",
      "\n",
      "['includes_functional_group', 'structure_sep_sym']\n",
      "\n",
      "['includes_reaction_words', 'structure_next_to_conversion_words', 'includes_reaction_words2']\n",
      "\n",
      "SENTENCE:\n",
      "carnitine acetyltransferases crat catalyze the reversible conversion of acetyl-coa and carnitine to acetylcarnitine and free coa\n",
      "\n",
      "CHEMICALS:\n",
      "['carnitine', 'acetylcarnitine', 'acetyl-coa', 'coa']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1  0  1 -1  1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1  1 -1 -1\n",
      " -1  1 -1  1]\n",
      "\n",
      "NEGS: 1 POS: 8\n",
      "\n",
      "['structure_sep_and']\n",
      "\n",
      "['structure_sep_conversion_words', 'includes_reaction_words', 'structure_next_to_conversion_words', 'structure_conversion_of', 'structure_jtsui_pattern_4', 'structure_jtsui_pattern_4_copy', 'includes_catalyze_words', 'includes_reaction_words2']\n",
      "\n",
      "SENTENCE:\n",
      "l-serine dehydratase sdh, a member of the beta-family of pyridoxal phosphate-dependent plp enzymes, catalyzes the deamination of l-serine and l-threonine to yield pyruvate or 2-oxobutyrate\n",
      "\n",
      "CHEMICALS:\n",
      "['2-oxobutyrate', 'l-serine', 'pyruvate', 'l-threonine']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1  0  0 -1  0  1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1  1  1  1]\n",
      "\n",
      "NEGS: 4 POS: 6\n",
      "\n",
      "['structure_sep_or', 'includes_measure_words', 'includes_experiment_words', 'structure_sep_and']\n",
      "\n",
      "['includes_reaction_words', 'structure_next_to_conversion_words', 'structure_conversion_of', 'includes_catalyze_words', 'includes_emzyze_words', 'includes_reaction_words2']\n",
      "\n",
      "SENTENCE:\n",
      "astrocytes may play a role in these manifestations because astrocytes are essential in the regulation of released glutamate and its conversion to glutamine through the enzyme glutamine synthetase gs\n",
      "\n",
      "CHEMICALS:\n",
      "['glutamine', 'glutamate']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1  0 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1  1 -1  1\n",
      " -1 -1  1  1]\n",
      "\n",
      "NEGS: 2 POS: 7\n",
      "\n",
      "['includes_physical_words', 'includes_functional_group']\n",
      "\n",
      "['includes_reaction_words', 'structure_next_to_conversion_words', 'structure_jtsui_pattern_4', 'structure_jtsui_pattern_4_copy', 'includes_impact_words', 'includes_emzyze_words', 'includes_reaction_words2']\n",
      "\n",
      "SENTENCE:\n",
      "conclusions total vitamin b6 is abnormally high in autism, consistent with previous reports of an impaired pyridoxal kinase for the conversion of pyridoxine and pyridoxal to plp\n",
      "\n",
      "CHEMICALS:\n",
      "['plp', 'pyridoxal', 'pyridoxine']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1  0 -1 -1  0  1 -1  1 -1 -1  1 -1 -1  1 -1  1 -1 -1  1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 3 POS: 9\n",
      "\n",
      "['includes_structural_words', 'includes_measure_words', 'structure_sep_and']\n",
      "\n",
      "['structure_sep_conversion_words', 'includes_reaction_words', 'structure_next_to_conversion_words', 'structure_conversion_of', 'structure_jtsui_pattern_1', 'structure_jtsui_pattern_4', 'structure_jtsui_pattern_1_copy', 'structure_jtsui_pattern_4_copy', 'includes_reaction_words2']\n",
      "\n",
      "SENTENCE:\n",
      "this enzyme catalyzes the s-adenosyl-l-methionine sam-dependent methylation of glycine to form sarcosine\n",
      "\n",
      "CHEMICALS:\n",
      "['glycine', 'sarcosine', 's-adenosyl-l-methionine', 'sam']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1  0 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1  0 -1 -1  1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1  1  1  1]\n",
      "\n",
      "NEGS: 3 POS: 6\n",
      "\n",
      "['structure_adjacent_mentions', 'structure_sep_sym', 'includes_experiment_words']\n",
      "\n",
      "['includes_reaction_words', 'structure_next_to_conversion_words', 'structure_conversion_of', 'includes_catalyze_words', 'includes_emzyze_words', 'includes_reaction_words2']\n",
      "\n",
      "SENTENCE:\n",
      "estrogens are biosynthesised from androgens by the cyp450 enzyme complex called aromatase\n",
      "\n",
      "CHEMICALS:\n",
      "['androgens', 'estrogens']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1  1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1  1  1]\n",
      "\n",
      "NEGS: 1 POS: 5\n",
      "\n",
      "['includes_genetic_words']\n",
      "\n",
      "['includes_reaction_words', 'structure_next_to_conversion_words', 'structure_conversion_by', 'includes_emzyze_words', 'includes_reaction_words2']\n",
      "\n",
      "SENTENCE:\n",
      "background histamine synthesized by histidine decarboxylase hdc from l-histidine is a major chemical mediator in the development of nasal allergy which is characterized by nasal hypersensitivity\n",
      "\n",
      "CHEMICALS:\n",
      "['l-histidine', 'histamine', 'histidine']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 1 POS: 3\n",
      "\n",
      "['includes_functional_group']\n",
      "\n",
      "['includes_reaction_words', 'structure_next_to_conversion_words', 'includes_reaction_words2']\n",
      "\n",
      "SENTENCE:\n",
      "methionine synthase reductase mtrr is an enzyme involved in the conversion of hcy to methionine\n",
      "\n",
      "CHEMICALS:\n",
      "['methionine', 'hcy']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1  1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1  1 -1  1 -1 -1  1 -1 -1  1 -1  1 -1 -1  1 -1 -1\n",
      " -1 -1  1  1]\n",
      "\n",
      "NEGS: 1 POS: 11\n",
      "\n",
      "['includes_genetic_words']\n",
      "\n",
      "['includes_oxidation_words', 'structure_sep_conversion_words', 'includes_reaction_words', 'structure_next_to_conversion_words', 'structure_conversion_of', 'structure_jtsui_pattern_1', 'structure_jtsui_pattern_4', 'structure_jtsui_pattern_1_copy', 'structure_jtsui_pattern_4_copy', 'includes_emzyze_words', 'includes_reaction_words2']\n",
      "\n",
      "SENTENCE:\n",
      "gpxs reduce hydroperoxides to the corresponding alcohols by means of glutathione gsh\n",
      "\n",
      "CHEMICALS:\n",
      "['alcohols', 'hydroperoxides', 'gsh', 'glutathione']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1  0 -1 -1  0 -1 -1 -1  0 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1  1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 3 POS: 6\n",
      "\n",
      "['includes_functional_group', 'structure_adjacent_mentions', 'structure_sep_sym']\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'structure_next_to_conversion_words', 'structure_jtsui_pattern_1', 'structure_jtsui_pattern_1_copy', 'includes_reaction_words2']\n",
      "\n",
      "SENTENCE:\n",
      "adenine phosphoribosyltransferase plays a role in purine salvage by catalyzing the direct conversion of adenine to adenosine monophosphate\n",
      "\n",
      "CHEMICALS:\n",
      "['adenine', 'adenosine monophosphate', 'purine']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1  0 -1 -1 -1  1 -1  1 -1 -1  1 -1 -1  1 -1  1 -1 -1  1 -1 -1\n",
      " -1  1 -1  1]\n",
      "\n",
      "NEGS: 2 POS: 10\n",
      "\n",
      "['includes_general_chemical_words', 'includes_measure_words']\n",
      "\n",
      "['structure_sep_conversion_words', 'includes_reaction_words', 'structure_next_to_conversion_words', 'structure_conversion_of', 'structure_jtsui_pattern_1', 'structure_jtsui_pattern_4', 'structure_jtsui_pattern_1_copy', 'structure_jtsui_pattern_4_copy', 'includes_catalyze_words', 'includes_reaction_words2']\n",
      "\n",
      "SENTENCE:\n",
      "they all expressed ass, but not ornithine transcarbamylase otc, the enzyme that converts ornithine, the product of degradation of arginine with rharg, to citrulline, which is converted back to arginine via ass\n",
      "\n",
      "CHEMICALS:\n",
      "['rharg', 'arginine', 'citrulline', 'ornithine']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1  1\n",
      "  1 -1 -1 -1 -1 -1 -1 -1  1 -1  1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1  1 -1 -1\n",
      " -1 -1  1  1]\n",
      "\n",
      "NEGS: 1 POS: 9\n",
      "\n",
      "['includes_physical_words']\n",
      "\n",
      "['structure_sep_conversion_words', 'includes_reaction_words', 'includes_reaction_component_words', 'structure_next_to_conversion_words', 'structure_conversion_of', 'structure_jtsui_pattern_4', 'structure_jtsui_pattern_4_copy', 'includes_emzyze_words', 'includes_reaction_words2']\n",
      "\n",
      "SENTENCE:\n",
      "specifically, all-trans-13,14-dihydroretinol is transiently oxidized to all-trans-13,14-dihydroretinoic acid before being oxidized further by cyp26 enzymes\n",
      "\n",
      "CHEMICALS:\n",
      "['all-trans-13,14-dihydroretinol', 'all-trans-13,14-dihydroretinoic acid']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1  1  1]\n",
      "\n",
      "NEGS: 0 POS: 4\n",
      "\n",
      "[]\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'includes_emzyze_words', 'includes_reaction_words2']\n",
      "\n",
      "SENTENCE:\n",
      "ldh is responsible for pyruvate conversion to lactate through glycolysis\n",
      "\n",
      "CHEMICALS:\n",
      "['lactate', 'pyruvate']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1  1 -1  1 -1 -1  1 -1  1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 0 POS: 7\n",
      "\n",
      "[]\n",
      "\n",
      "['includes_reaction_words', 'structure_next_to_conversion_words', 'structure_jtsui_pattern_2', 'structure_jtsui_pattern_4', 'structure_jtsui_pattern_2_copy', 'structure_jtsui_pattern_4_copy', 'includes_reaction_words2']\n",
      "\n",
      "SENTENCE:\n",
      "in kidneys, stimulation of adenylyl cyclase causes egress of camp, conversion of camp to amp by ecto-phosphodiesterase, and metabolism of amp to adenosine by ecto-5'-nucleotidase\n",
      "\n",
      "CHEMICALS:\n",
      "['adenosine', 'amp', 'camp']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1  0  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1  0 -1 -1 -1  1  1  1 -1 -1  1 -1 -1  1 -1  1 -1 -1  1 -1  1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 3 POS: 11\n",
      "\n",
      "['includes_general_chemical_words', 'includes_functional_group', 'includes_measure_words']\n",
      "\n",
      "['structure_sep_conversion_words', 'includes_reaction_words', 'structure_next_to_conversion_words', 'structure_conversion_by', 'structure_conversion_of', 'structure_jtsui_pattern_1', 'structure_jtsui_pattern_4', 'structure_jtsui_pattern_1_copy', 'structure_jtsui_pattern_4_copy', 'includes_impact_words', 'includes_reaction_words2']\n",
      "\n",
      "SENTENCE:\n",
      "hydrogen sulphide h2s is synthesized from l-cysteine via the action of cystathionine-gamma-lyase cse and cystathionine-beta-synthase cbs\n",
      "\n",
      "CHEMICALS:\n",
      "['l-cysteine', 'hydrogen sulphide']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1  0 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 2 POS: 3\n",
      "\n",
      "['includes_physical_words', 'includes_measure_words']\n",
      "\n",
      "['includes_reaction_words', 'structure_next_to_conversion_words', 'includes_reaction_words2']\n",
      "\n",
      "SENTENCE:\n",
      "adenylosuccinate synthetase adss catalyzes the mg2+ dependent condensation of a molecule of imp with aspartate to form adenylosuccinate, in a reaction driven by the hydrolysis of gtp to gdp\n",
      "\n",
      "CHEMICALS:\n",
      "['mg2+', 'gtp', 'adenylosuccinate', 'imp', 'gdp', 'aspartate']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1  0 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1  1 -1  0 -1 -1  1 -1  1 -1 -1  1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1\n",
      " -1  1 -1  1]\n",
      "\n",
      "NEGS: 3 POS: 9\n",
      "\n",
      "['includes_physical_words', 'includes_general_chemical_words', 'includes_experiment_words']\n",
      "\n",
      "['structure_sep_conversion_words', 'includes_reaction_words', 'includes_react', 'structure_next_to_conversion_words', 'structure_conversion_of', 'structure_jtsui_pattern_1', 'structure_jtsui_pattern_1_copy', 'includes_catalyze_words', 'includes_reaction_words2']\n",
      "\n",
      "SENTENCE:\n",
      "risperidone is metabolized to its active metabolite, 9-hydroxyrisperidone, mainly by the cytochrome p450 enzymes cyp2d6 and 3a4\n",
      "\n",
      "CHEMICALS:\n",
      "['risperidone', '9-hydroxyrisperidone']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1  1  1]\n",
      "\n",
      "NEGS: 0 POS: 3\n",
      "\n",
      "[]\n",
      "\n",
      "['includes_reaction_words', 'includes_emzyze_words', 'includes_reaction_words2']\n",
      "\n",
      "SENTENCE:\n",
      "pge2 is synthesized from arachidonic acid by cyclooxygenases cox and prostaglandin e synthases pges and mediates its biological activity through binding to the four prostanoid receptors ep1 through ep4\n",
      "\n",
      "CHEMICALS:\n",
      "['pge(2)', 'arachidonic acid']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1  0 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1  1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 2 POS: 5\n",
      "\n",
      "['includes_physical_words', 'includes_general_chemical_words']\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'structure_next_to_conversion_words', 'structure_conversion_by', 'includes_reaction_words2']\n",
      "\n",
      "SENTENCE:\n",
      "aldo-keto reductases akrs metabolize a wide range of substrates, including polycyclic aromatic hydrocarbons pahs, generating metabolites o-quinones and reactive oxygen species ros, which are capable of initiating and promoting carcinogenesis\n",
      "\n",
      "CHEMICALS:\n",
      "['pahs', 'aldo-keto', 'oxygen', 'ros', 'o-quinones']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1  0  0  0 -1 -1 -1 -1 -1 -1 -1 -1 -1  1  1 -1 -1 -1 -1 -1 -1  1\n",
      "  1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 3 POS: 6\n",
      "\n",
      "['includes_genetic_words', 'includes_structural_words', 'includes_general_chemical_words']\n",
      "\n",
      "['includes_oxidation_words', 'structure_sep_verb', 'includes_reaction_words', 'includes_reaction_component_words', 'includes_react', 'includes_reaction_words2']\n",
      "\n",
      "SENTENCE:\n",
      "the ultimate carcinogenic phase i bp metabolite anti-bp-7,8-dihydrodiol-9,10-epoxide bpde can be detoxified by glutathione conjugate formation catalyzed by glutathione s-transferases\n",
      "\n",
      "CHEMICALS:\n",
      "['bpde', 'glutathione']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1  0 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1  0 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1  1 -1  1]\n",
      "\n",
      "NEGS: 3 POS: 4\n",
      "\n",
      "['includes_genetic_words', 'includes_functional_group', 'includes_measure_words']\n",
      "\n",
      "['includes_reaction_words', 'structure_next_to_conversion_words', 'includes_catalyze_words', 'includes_reaction_words2']\n",
      "\n",
      "SENTENCE:\n",
      "the mitochondrial respiratory chain complex iv cytochrome c oxidase is a multi-subunit enzyme that transfers electrons from cytochrome c to molecular oxygen, yielding water\n",
      "\n",
      "CHEMICALS:\n",
      "['oxygen', 'water', 'cytochrome c']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1  0  0  0  0 -1 -1 -1 -1 -1 -1 -1 -1 -1  1  1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1  1  1]\n",
      "\n",
      "NEGS: 4 POS: 6\n",
      "\n",
      "['includes_physical_words', 'includes_genetic_words', 'includes_structural_words', 'includes_general_chemical_words']\n",
      "\n",
      "['includes_oxidation_words', 'structure_sep_verb', 'includes_reaction_words', 'structure_next_to_conversion_words', 'includes_emzyze_words', 'includes_reaction_words2']\n",
      "\n",
      "SENTENCE:\n",
      "aldose reductase ar catalyzes the reduction of toxic lipid aldehydes to their alcohol products and mediates inflammatory signals triggered by lipopolysaccharide lps\n",
      "\n",
      "CHEMICALS:\n",
      "['alcohol', 'aldehydes', 'aldose']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1  0  0 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      "  1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1  1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1\n",
      "  1  1 -1  1]\n",
      "\n",
      "NEGS: 3 POS: 9\n",
      "\n",
      "['includes_physical_words', 'includes_genetic_words', 'includes_functional_group']\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'includes_reaction_component_words', 'structure_next_to_conversion_words', 'structure_jtsui_pattern_1', 'structure_jtsui_pattern_1_copy', 'includes_creation_words', 'includes_catalyze_words', 'includes_reaction_words2']\n",
      "\n",
      "SENTENCE:\n",
      "leukotriene a4 hydrolase lta4h is a cystolic enzyme that stereospecifically catalyzes the transformation of lta4 to ltb4\n",
      "\n",
      "CHEMICALS:\n",
      "['lta(4)', 'ltb(4)']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1  1  1  1]\n",
      "\n",
      "NEGS: 1 POS: 4\n",
      "\n",
      "['includes_genetic_words']\n",
      "\n",
      "['includes_reaction_words', 'includes_catalyze_words', 'includes_emzyze_words', 'includes_reaction_words2']\n",
      "\n",
      "SENTENCE:\n",
      "each recombinant protein metabolized aa and la to epoxides and hydroxy derivatives\n",
      "\n",
      "CHEMICALS:\n",
      "['epoxides', 'aa', 'la', 'hydroxy']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1  1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1  0  1 -1 -1 -1 -1  1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 2 POS: 7\n",
      "\n",
      "['includes_functional_group', 'structure_sep_and']\n",
      "\n",
      "['structure_sep_conversion_words', 'includes_combination_words', 'includes_reaction_words', 'structure_next_to_conversion_words', 'structure_jtsui_pattern_1', 'structure_jtsui_pattern_1_copy', 'includes_reaction_words2']\n",
      "\n",
      "SENTENCE:\n",
      "we conclude that the four new cyp2j isoforms might be involved in the metabolism of aa and la to bioactive lipids in mouse hepatic and extrahepatic tissues\n",
      "\n",
      "CHEMICALS:\n",
      "['aa', 'la', 'bioactive lipids']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1  0 -1  0  1 -1  1 -1 -1  1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 2 POS: 7\n",
      "\n",
      "['includes_experiment_words', 'structure_sep_and']\n",
      "\n",
      "['structure_sep_conversion_words', 'includes_reaction_words', 'structure_next_to_conversion_words', 'structure_conversion_of', 'structure_jtsui_pattern_1', 'structure_jtsui_pattern_1_copy', 'includes_reaction_words2']\n",
      "\n",
      "SENTENCE:\n",
      "as predicted, tes enhanced the production of both peroxynitrite precursors ie, superoxide and nitic oxide, and xanthine oxidase was identified as the likely source of tes-stimulated superoxide production\n",
      "\n",
      "CHEMICALS:\n",
      "['tes', 'peroxynitrite', 'superoxide', 'xanthine']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      "  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      "  1 -1 -1  1]\n",
      "\n",
      "NEGS: 0 POS: 4\n",
      "\n",
      "[]\n",
      "\n",
      "['includes_reaction_words', 'includes_reaction_component_words', 'includes_creation_words', 'includes_reaction_words2']\n",
      "\n",
      "SENTENCE:\n",
      "type 2 11beta-hydroxysteroid dehydrogenase encoded by the hsd11b2 gene converts cortisol to inactive cortisone, and alteration in this enzymatic activity might affect glucose homeostasis by affecting circulating levels or tissue availability of glucocorticoids\n",
      "\n",
      "CHEMICALS:\n",
      "['cortisone', 'cortisol', '11î²-hydroxysteroid', 'glucose']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1  0  0 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1  0  0 -1 -1  1 -1 -1 -1 -1  1 -1 -1  1 -1  1 -1 -1  1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 5 POS: 7\n",
      "\n",
      "['includes_genetic_words', 'includes_structural_words', 'includes_no_terms', 'includes_measure_words', 'includes_experiment_words']\n",
      "\n",
      "['includes_reaction_words', 'structure_next_to_conversion_words', 'structure_jtsui_pattern_1', 'structure_jtsui_pattern_4', 'structure_jtsui_pattern_1_copy', 'structure_jtsui_pattern_4_copy', 'includes_reaction_words2']\n",
      "\n",
      "SENTENCE:\n",
      "carboxylesterases hydrolyze esters, amides, and thioesters to produce carboxylic acids and resulting alcohols, amines, and thiols, respectively\n",
      "\n",
      "CHEMICALS:\n",
      "['amides', 'alcohols', 'amines', 'carboxylic acids', 'thioesters', 'thiols', 'esters']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1  0  0 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 3 POS: 4\n",
      "\n",
      "['includes_general_chemical_words', 'includes_functional_group', 'structure_sep_comma']\n",
      "\n",
      "['includes_reaction_words', 'structure_next_to_conversion_words', 'includes_impact_words', 'includes_reaction_words2']\n",
      "\n",
      "SENTENCE:\n",
      "uridine 5'-diphosphate- glucuronosyltransferases are colocalized with carboxylesterases and have the potential to further metabolize carboxylic acids to acyl glucuronides, but it is currently unknown if acyl glucuronides, being esters, also interact with carboxylesterases\n",
      "\n",
      "CHEMICALS:\n",
      "['carboxylic acids', 'esters', 'acyl glucuronides']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1  1  1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1  0 -1 -1 -1  1 -1 -1 -1 -1  1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 2 POS: 7\n",
      "\n",
      "['includes_functional_group', 'includes_measure_words']\n",
      "\n",
      "['structure_sep_verb', 'structure_sep_conversion_words', 'includes_reaction_words', 'structure_next_to_conversion_words', 'structure_jtsui_pattern_1', 'structure_jtsui_pattern_1_copy', 'includes_reaction_words2']\n",
      "\n",
      "SENTENCE:\n",
      "cytochrome p450 epoxygenase 2j2 cyp2j2 metabolizes arachidonic acids to form epoxyeicosatrienoic acids eets, which possess various beneficial effects on the cardiovascular system\n",
      "\n",
      "CHEMICALS:\n",
      "['eets', 'arachidonic acids']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1  1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 0 POS: 6\n",
      "\n",
      "[]\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'structure_next_to_conversion_words', 'structure_jtsui_pattern_1', 'structure_jtsui_pattern_1_copy', 'includes_reaction_words2']\n",
      "\n",
      "SENTENCE:\n",
      "ketamine is primarily metabolized to norketamine by hepatic cytochrome p450 cyp 2b6 and cyp3a4-mediated n-demethylation\n",
      "\n",
      "CHEMICALS:\n",
      "['norketamine', 'ketamine']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1  1  1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1  1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 1 POS: 6\n",
      "\n",
      "['includes_functional_group']\n",
      "\n",
      "['includes_reaction_words', 'structure_next_to_conversion_words', 'structure_conversion_by', 'structure_jtsui_pattern_5', 'structure_jtsui_pattern_5_copy', 'includes_reaction_words2']\n",
      "\n",
      "SENTENCE:\n",
      "6-dhsg was metabolised by gsh to form a gsh conjugate gs-6-dhsg in raw 2647 cells, via a potential mechanism involving the catalytic activity of glutathione-s-transferase gst\n",
      "\n",
      "CHEMICALS:\n",
      "['6-dhsg', 'gs-6-dhsg', 'gsh']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1  0 -1 -1  1 -1 -1 -1 -1  1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 1 POS: 5\n",
      "\n",
      "['includes_experiment_words']\n",
      "\n",
      "['includes_reaction_words', 'structure_next_to_conversion_words', 'structure_jtsui_pattern_1', 'structure_jtsui_pattern_1_copy', 'includes_reaction_words2']\n",
      "\n",
      "SENTENCE:\n",
      "as for flux through serine hydroxymethyltransferase and gcs, the conversion of serine to glycine occurred fairly rapidly, followed by gcs-mediated slow decarboxylation of the accumulated glycine\n",
      "\n",
      "CHEMICALS:\n",
      "['glycine', 'serine']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1  0 -1 -1 -1  1 -1  1 -1 -1  1 -1 -1  1 -1  1 -1 -1  1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 1 POS: 9\n",
      "\n",
      "['includes_measure_words']\n",
      "\n",
      "['structure_sep_conversion_words', 'includes_reaction_words', 'structure_next_to_conversion_words', 'structure_conversion_of', 'structure_jtsui_pattern_1', 'structure_jtsui_pattern_4', 'structure_jtsui_pattern_1_copy', 'structure_jtsui_pattern_4_copy', 'includes_reaction_words2']\n",
      "\n",
      "SENTENCE:\n",
      "phosphoserine aminotransferase psat ec 26152, a member of subgroup iv of the aminotransferases, catalyses the conversion of 3-phosphohydroxypyruvate to l-phosphoserine\n",
      "\n",
      "CHEMICALS:\n",
      "['3-phosphohydroxypyruvate', 'l-phosphoserine']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1  0 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1  0 -1 -1 -1  1 -1  1 -1 -1  1 -1 -1  1 -1  1 -1 -1  1 -1 -1\n",
      " -1  1 -1  1]\n",
      "\n",
      "NEGS: 3 POS: 10\n",
      "\n",
      "['includes_structural_words', 'includes_functional_group', 'includes_measure_words']\n",
      "\n",
      "['structure_sep_conversion_words', 'includes_reaction_words', 'structure_next_to_conversion_words', 'structure_conversion_of', 'structure_jtsui_pattern_1', 'structure_jtsui_pattern_4', 'structure_jtsui_pattern_1_copy', 'structure_jtsui_pattern_4_copy', 'includes_catalyze_words', 'includes_reaction_words2']\n",
      "\n",
      "SENTENCE:\n",
      "we therefore propose that in p putida methionine is converted to methanethiol and then oxidized to methanesulfonate\n",
      "\n",
      "CHEMICALS:\n",
      "['methionine', 'methanethiol', 'methanesulfonate']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1  0  0 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1  0 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1  1 -1  1 -1 -1  1 -1  1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 3 POS: 8\n",
      "\n",
      "['includes_functional_group', 'includes_paper_artifacts', 'includes_comparison_words']\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'structure_next_to_conversion_words', 'structure_jtsui_pattern_2', 'structure_jtsui_pattern_4', 'structure_jtsui_pattern_2_copy', 'structure_jtsui_pattern_4_copy', 'includes_reaction_words2']\n",
      "\n",
      "SENTENCE:\n",
      "only two exceptions had been reported-the enterobacteriaceae and myxococcus xanthus members of the gamma and delta groups of the class proteobacteria, respectively-in which ornithine is produced from n-alpha-acetylornithine by a deacylase, acetylornithinase aoase arge encoded\n",
      "\n",
      "CHEMICALS:\n",
      "['n-alpha-acetylornithine', 'ornithine']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1  0 -1  0 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1 -1  0 -1 -1 -1  1  1 -1 -1 -1 -1  1 -1 -1  1 -1  1 -1 -1  1 -1\n",
      " -1 -1 -1 -1]\n",
      "\n",
      "NEGS: 4 POS: 6\n",
      "\n",
      "['includes_physical_words', 'includes_structural_words', 'includes_functional_group', 'includes_measure_words']\n",
      "\n",
      "['structure_next_to_conversion_words', 'structure_conversion_by', 'structure_jtsui_pattern_2', 'structure_jtsui_pattern_5', 'structure_jtsui_pattern_2_copy', 'structure_jtsui_pattern_5_copy']\n",
      "\n",
      "SENTENCE:\n",
      "the energetics is here reported for the action of lysine 2,3-aminomutase lam, which catalyzes the interconversion of l-lysine and l-beta-lysine\n",
      "\n",
      "CHEMICALS:\n",
      "['l-beta-lysine', 'l-lysine']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1  0 -1 -1  1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1  1 -1 -1\n",
      " -1  1 -1  1]\n",
      "\n",
      "NEGS: 1 POS: 6\n",
      "\n",
      "['structure_sep_and']\n",
      "\n",
      "['includes_reaction_words', 'structure_conversion_of', 'structure_jtsui_pattern_4', 'structure_jtsui_pattern_4_copy', 'includes_catalyze_words', 'includes_reaction_words2']\n",
      "\n",
      "SENTENCE:\n",
      "estradiol 17 beta-hydroxysteroid dehydrogenase 17 beta hsd mediates the interconversion of estrone and estradiol in endocrine-responsive tissues such as the breast\n",
      "\n",
      "CHEMICALS:\n",
      "['estrone', 'estradiol']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1  0 -1  0 -1 -1  1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1  1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 2 POS: 5\n",
      "\n",
      "['includes_experiment_words', 'structure_sep_and']\n",
      "\n",
      "['includes_reaction_words', 'structure_conversion_of', 'structure_jtsui_pattern_4', 'structure_jtsui_pattern_4_copy', 'includes_reaction_words2']\n",
      "\n",
      "SENTENCE:\n",
      "5-hydroxymethyltryptophan and 5-hydroxy-4-methyltryptophan are the products from 5-methyltryptophan\n",
      "\n",
      "CHEMICALS:\n",
      "['5-hydroxy-4-methyltryptophan', '5-methyltryptophan', '5-hydroxymethyltryptophan']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      "  1 -1 -1 -1  0 -1 -1  0  1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1  1 -1\n",
      " -1 -1 -1 -1]\n",
      "\n",
      "NEGS: 2 POS: 4\n",
      "\n",
      "['includes_measure_words', 'structure_sep_and']\n",
      "\n",
      "['includes_reaction_component_words', 'structure_next_to_conversion_words', 'structure_jtsui_pattern_5', 'structure_jtsui_pattern_5_copy']\n",
      "\n",
      "SENTENCE:\n",
      "we show that tres from mycobacterium smegmatis, as well as recombinant tres produced in escherichia coli, has amylase activity in addition to the maltose <--> trehalose interconverting activity referred to as mtase\n",
      "\n",
      "CHEMICALS:\n",
      "['trehalose', 'maltose']\n",
      "\n",
      "LABELS:\n",
      "[-1 -1 -1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "NEGS: 1 POS: 5\n",
      "\n",
      "['includes_paper_artifacts']\n",
      "\n",
      "['includes_combination_words', 'includes_reaction_words', 'includes_react_sym', 'includes_interconvert_sym', 'includes_reaction_words2']\n",
      "\n"
     ]
    }
   ],
   "source": [
    "oof = 0\n",
    "for i in df_tp.index:\n",
    "    negs = []\n",
    "    neg_count = 0\n",
    "    pos = []\n",
    "    pos_count = 0\n",
    "    \n",
    "    for j in range(len(L_test[i])):\n",
    "        if L_test[i][j] == 0:\n",
    "            negs.append(str(lfs[j].name))\n",
    "            neg_count = neg_count + 1\n",
    "        if L_test[i][j] == 1:\n",
    "            pos.append(str(lfs[j].name))\n",
    "            pos_count = pos_count + 1\n",
    "    print(\"SENTENCE:\")\n",
    "    print(df_tp.iloc[oof]['sentence'])\n",
    "    print('')\n",
    "    print(\"CHEMICALS:\")\n",
    "    print(df_tp.iloc[oof]['chemicals'])\n",
    "    print(\"\")\n",
    "    oof = oof + 1\n",
    "    print(\"LABELS:\")\n",
    "    print(L_test[i])\n",
    "    print('')\n",
    "    print(\"NEGS: \" + str(neg_count) + \" POS: \" + str(pos_count))\n",
    "    print('')\n",
    "    print(negs)\n",
    "    print('')\n",
    "    print(pos)\n",
    "    print('')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "the enzyme cyclo-oxygenase catalyses the oxygenation of arachidonic acid, leading to the formation of prostaglandins\n",
      "\n",
      "although the exact site of decarboxylation of exogenous levodopa to dopamine in the brain is unknown, most striatal aadc is located in nigrostriatal dopaminergic nerve terminals\n",
      "\n",
      "the enzyme dimethylarginine dimethylaminohydrolase ddah specifically hydrolyzes these asymmetrically methylated arginine residues to citrulline and methylamines\n",
      "\n",
      "background and aims glutamic acid decarboxylase gad, ec 41115 catalyses the conversion of glutamate to gamma-aminobutyric acid gaba\n",
      "\n",
      "mazindane 26 was found to be a pro-drug, oxidizing 5-h --> 5-oh to mazindol on rat striatal membranes and hek-hdat cells\n",
      "\n",
      "interestingly, the allele of pro1 was shown to enhance the activities of gamma-glutamyl kinase and gamma-glutamyl phosphate reductase, both of which catalyze the first two steps of l-proline synthesis from l-glutamate and which together may form a complex in vivo\n",
      "\n",
      "in addition to the racemization it also catalyzes specific elimination of l-ser to pyruvate\n",
      "\n",
      "we concluded that fdh has no direct role in the regulation of the above two pathways of ser synthesis the breakdown of formate to co2 by the fdh reaction is the primary and preferred fate of the organic acid in arabidopsis\n",
      "\n",
      "the ratio between the gdc/shmt and c1-thf synthase/shmt pathways of ser synthesis from [alpha-13c]gly and [13c]formate, respectively, in arabidopsis shoots was 21  1 in roots, 9  1\n",
      "\n",
      "one of the enzymes responsible for the production of ka, kynurenine aminotransferase i kati, also catalyses the reversible transamination of glutamine to oxoglutaramic acid gtk, ec 26115\n",
      "\n",
      "cpt i carnitine palmitoyltransferase i catalyses the conversion of palmitoyl-coa into palmitoylcarnitine in the presence of l-carnitine, facilitating the entry of fatty acids into mitochondria\n",
      "\n",
      "purpose the fluoropyrimidine carbamate capecitabine is converted to 5-fluorouracil 5-fu by thymidine phosphorylase tp inside target tissues\n",
      "\n",
      "spermine was rapidly taken up and metabolized to spermidine and an unknown, possibly acetamidopropanal, by spermidine/spermine n1-acetyltransferase ssat and polyamine oxidase pao\n",
      "\n",
      "carnitine acetyltransferases crat catalyze the reversible conversion of acetyl-coa and carnitine to acetylcarnitine and free coa\n",
      "\n",
      "l-serine dehydratase sdh, a member of the beta-family of pyridoxal phosphate-dependent plp enzymes, catalyzes the deamination of l-serine and l-threonine to yield pyruvate or 2-oxobutyrate\n",
      "\n",
      "beta-1,4-galactosyltransferase i beta4gal-t1 normally transfers gal from udp-gal to glcnac in the presence of mn2+ ion gal-t activity and also transfers glc from udp-glc to glcnac glc-t activity, albeit at only 03% efficiency\n",
      "\n",
      "a cysteine residue in place of the catalytic tyrosine or serine residues found in other gsts was shown to form a mixed disulfide with glutathione\n",
      "\n",
      "astrocytes may play a role in these manifestations because astrocytes are essential in the regulation of released glutamate and its conversion to glutamine through the enzyme glutamine synthetase gs\n",
      "\n",
      "conclusions total vitamin b6 is abnormally high in autism, consistent with previous reports of an impaired pyridoxal kinase for the conversion of pyridoxine and pyridoxal to plp\n",
      "\n",
      "this enzyme catalyzes the s-adenosyl-l-methionine sam-dependent methylation of glycine to form sarcosine\n",
      "\n",
      "estrogens are biosynthesised from androgens by the cyp450 enzyme complex called aromatase\n",
      "\n",
      "background histamine synthesized by histidine decarboxylase hdc from l-histidine is a major chemical mediator in the development of nasal allergy which is characterized by nasal hypersensitivity\n",
      "\n",
      "methionine synthase reductase mtrr is an enzyme involved in the conversion of hcy to methionine\n",
      "\n",
      "gpxs reduce hydroperoxides to the corresponding alcohols by means of glutathione gsh\n",
      "\n",
      "adenine phosphoribosyltransferase plays a role in purine salvage by catalyzing the direct conversion of adenine to adenosine monophosphate\n",
      "\n",
      "they all expressed ass, but not ornithine transcarbamylase otc, the enzyme that converts ornithine, the product of degradation of arginine with rharg, to citrulline, which is converted back to arginine via ass\n",
      "\n",
      "metabolism of vitamin a, all-trans-retinol, leads to the formation of 11-cis-retinaldehyde, the visual chromophore, and all-trans-retinoic acid, which is involved in the regulation of gene expression through the retinoic acid receptor\n",
      "\n",
      "specifically, all-trans-13,14-dihydroretinol is transiently oxidized to all-trans-13,14-dihydroretinoic acid before being oxidized further by cyp26 enzymes\n",
      "\n",
      "unlike mouse retsat mretsat, zretsat a had an altered bond specificity saturating either the 13-14 or 7-8 double bonds of all-trans-retinol to produce either all-trans-13,14-dihydroretinol or all-trans-7,8-dihydroretinol, respectively\n",
      "\n",
      "ldh is responsible for pyruvate conversion to lactate through glycolysis\n",
      "\n",
      "in kidneys, stimulation of adenylyl cyclase causes egress of camp, conversion of camp to amp by ecto-phosphodiesterase, and metabolism of amp to adenosine by ecto-5'-nucleotidase\n",
      "\n",
      "hydrogen sulphide h2s is synthesized from l-cysteine via the action of cystathionine-gamma-lyase cse and cystathionine-beta-synthase cbs\n",
      "\n",
      "adenylosuccinate synthetase adss catalyzes the mg2+ dependent condensation of a molecule of imp with aspartate to form adenylosuccinate, in a reaction driven by the hydrolysis of gtp to gdp\n",
      "\n",
      "risperidone is metabolized to its active metabolite, 9-hydroxyrisperidone, mainly by the cytochrome p450 enzymes cyp2d6 and 3a4\n",
      "\n",
      "pge2 is synthesized from arachidonic acid by cyclooxygenases cox and prostaglandin e synthases pges and mediates its biological activity through binding to the four prostanoid receptors ep1 through ep4\n",
      "\n",
      "tas-102 is a novel drug containing trifluorothymidine, which is phosphorylated by tk-1 to its active monophosphated form, that in turn can inhibit ts\n",
      "\n",
      "aldo-keto reductases akrs metabolize a wide range of substrates, including polycyclic aromatic hydrocarbons pahs, generating metabolites o-quinones and reactive oxygen species ros, which are capable of initiating and promoting carcinogenesis\n",
      "\n",
      "the ultimate carcinogenic phase i bp metabolite anti-bp-7,8-dihydrodiol-9,10-epoxide bpde can be detoxified by glutathione conjugate formation catalyzed by glutathione s-transferases\n",
      "\n",
      "the mitochondrial respiratory chain complex iv cytochrome c oxidase is a multi-subunit enzyme that transfers electrons from cytochrome c to molecular oxygen, yielding water\n",
      "\n",
      "aldose reductase ar catalyzes the reduction of toxic lipid aldehydes to their alcohol products and mediates inflammatory signals triggered by lipopolysaccharide lps\n",
      "\n",
      "leukotriene a4 hydrolase lta4h is a cystolic enzyme that stereospecifically catalyzes the transformation of lta4 to ltb4\n",
      "\n",
      "each recombinant protein metabolized aa and la to epoxides and hydroxy derivatives\n",
      "\n",
      "we conclude that the four new cyp2j isoforms might be involved in the metabolism of aa and la to bioactive lipids in mouse hepatic and extrahepatic tissues\n",
      "\n",
      "as predicted, tes enhanced the production of both peroxynitrite precursors ie, superoxide and nitic oxide, and xanthine oxidase was identified as the likely source of tes-stimulated superoxide production\n",
      "\n",
      "these findings, derived from a variety of analytical and functional approaches, provide evidence for a novel nongenomic signaling mechanism for androgen action in the microvasculature tes-stimulated vasodilation mediated primarily by peroxynitrite formed from xanthine oxidase-generated superoxide and no\n",
      "\n",
      "type 2 11beta-hydroxysteroid dehydrogenase encoded by the hsd11b2 gene converts cortisol to inactive cortisone, and alteration in this enzymatic activity might affect glucose homeostasis by affecting circulating levels or tissue availability of glucocorticoids\n",
      "\n",
      "carboxylesterases hydrolyze esters, amides, and thioesters to produce carboxylic acids and resulting alcohols, amines, and thiols, respectively\n",
      "\n",
      "uridine 5'-diphosphate- glucuronosyltransferases are colocalized with carboxylesterases and have the potential to further metabolize carboxylic acids to acyl glucuronides, but it is currently unknown if acyl glucuronides, being esters, also interact with carboxylesterases\n",
      "\n",
      "cytochrome p450 epoxygenase 2j2 cyp2j2 metabolizes arachidonic acids to form epoxyeicosatrienoic acids eets, which possess various beneficial effects on the cardiovascular system\n",
      "\n",
      "ketamine is primarily metabolized to norketamine by hepatic cytochrome p450 cyp 2b6 and cyp3a4-mediated n-demethylation\n",
      "\n",
      "pip5k1b encodes phosphatidylinositol 4-phosphate 5-kinase beta type i pip5k1beta, an enzyme functionally linked to actin cytoskeleton dynamics that phosphorylates phosphatidylinositol 4-phosphate [pi4p] to generate phosphatidylinositol-4,5-bisphosphate [pi4,5p2]\n",
      "\n",
      "6-dhsg was metabolised by gsh to form a gsh conjugate gs-6-dhsg in raw 2647 cells, via a potential mechanism involving the catalytic activity of glutathione-s-transferase gst\n",
      "\n",
      "as for flux through serine hydroxymethyltransferase and gcs, the conversion of serine to glycine occurred fairly rapidly, followed by gcs-mediated slow decarboxylation of the accumulated glycine\n",
      "\n",
      "phosphoserine aminotransferase psat ec 26152, a member of subgroup iv of the aminotransferases, catalyses the conversion of 3-phosphohydroxypyruvate to l-phosphoserine\n",
      "\n",
      "we therefore propose that in p putida methionine is converted to methanethiol and then oxidized to methanesulfonate\n",
      "\n",
      "only two exceptions had been reported-the enterobacteriaceae and myxococcus xanthus members of the gamma and delta groups of the class proteobacteria, respectively-in which ornithine is produced from n-alpha-acetylornithine by a deacylase, acetylornithinase aoase arge encoded\n",
      "\n",
      "the energetics is here reported for the action of lysine 2,3-aminomutase lam, which catalyzes the interconversion of l-lysine and l-beta-lysine\n",
      "\n",
      "estradiol 17 beta-hydroxysteroid dehydrogenase 17 beta hsd mediates the interconversion of estrone and estradiol in endocrine-responsive tissues such as the breast\n",
      "\n",
      "5-hydroxymethyltryptophan and 5-hydroxy-4-methyltryptophan are the products from 5-methyltryptophan\n",
      "\n",
      "we show that tres from mycobacterium smegmatis, as well as recombinant tres produced in escherichia coli, has amylase activity in addition to the maltose <--> trehalose interconverting activity referred to as mtase\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for sentence in df_true[\"sentence\"]:\n",
    "    print(sentence)\n",
    "    print(\"\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "metabolism of vitamin a, all-trans-retinol, leads to the formation of 11-cis-retinaldehyde, the visual chromophore, and all-trans-retinoic acid, which is involved in the regulation of gene expression through the retinoic acid receptor\n",
      "\n",
      "unlike mouse retsat mretsat, zretsat a had an altered bond specificity saturating either the 13-14 or 7-8 double bonds of all-trans-retinol to produce either all-trans-13,14-dihydroretinol or all-trans-7,8-dihydroretinol, respectively\n",
      "\n",
      "tas-102 is a novel drug containing trifluorothymidine, which is phosphorylated by tk-1 to its active monophosphated form, that in turn can inhibit ts\n",
      "\n",
      "these findings, derived from a variety of analytical and functional approaches, provide evidence for a novel nongenomic signaling mechanism for androgen action in the microvasculature tes-stimulated vasodilation mediated primarily by peroxynitrite formed from xanthine oxidase-generated superoxide and no\n",
      "\n",
      "pip5k1b encodes phosphatidylinositol 4-phosphate 5-kinase beta type i pip5k1beta, an enzyme functionally linked to actin cytoskeleton dynamics that phosphorylates phosphatidylinositol 4-phosphate [pi4p] to generate phosphatidylinositol-4,5-bisphosphate [pi4,5p2]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for sentence in df_fn[\"sentence\"]:\n",
    "    print(sentence)\n",
    "    print(\"\")\n",
    "    \n",
    "#to_csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "metabolism of vitamin a, all-trans-retinol, leads to the formation of 11-cis-retinaldehyde, the visual chromophore, and all-trans-retinoic acid, which is involved in the regulation of gene expression through the retinoic acid receptor\n",
      "\n",
      "['11-cis-retinaldehyde', 'all-trans-retinol']\n",
      "\n",
      "[-1 -1  0  0  0 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1  0 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "['includes_physical_words', 'includes_genetic_words', 'includes_structural_words', 'includes_functional_group', 'includes_measure_words']\n",
      "\n",
      "['includes_reaction_words', 'structure_conversion_of', 'includes_impact_words', 'includes_reaction_words2']\n",
      "\n",
      "unlike mouse retsat mretsat, zretsat a had an altered bond specificity saturating either the 13-14 or 7-8 double bonds of all-trans-retinol to produce either all-trans-13,14-dihydroretinol or all-trans-7,8-dihydroretinol, respectively\n",
      "\n",
      "['all-trans-13,14-dihydroretinol', 'all-trans-7,8-dihydroretinol', 'all-trans-retinol']\n",
      "\n",
      "[ 0 -1 -1 -1  0 -1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1 -1 -1  0 -1 -1  1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1 -1]\n",
      "\n",
      "['includes_solution_words', 'includes_structural_words', 'structure_sep_or', 'includes_experiment_words']\n",
      "\n",
      "['structure_next_to_conversion_words', 'structure_conversion_of']\n",
      "\n",
      "tas-102 is a novel drug containing trifluorothymidine, which is phosphorylated by tk-1 to its active monophosphated form, that in turn can inhibit ts\n",
      "\n",
      "['tas-102', 'trifluorothymidine']\n",
      "\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1  0  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "['includes_no_terms', 'includes_measure_words', 'includes_experiment_words']\n",
      "\n",
      "['includes_reaction_words', 'includes_reaction_words2']\n",
      "\n",
      "these findings, derived from a variety of analytical and functional approaches, provide evidence for a novel nongenomic signaling mechanism for androgen action in the microvasculature tes-stimulated vasodilation mediated primarily by peroxynitrite formed from xanthine oxidase-generated superoxide and no\n",
      "\n",
      "['no', 'peroxynitrite', 'superoxide', 'androgen', 'xanthine']\n",
      "\n",
      "[-1 -1  0  0  0  0  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1  0  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      "  1 -1 -1  1]\n",
      "\n",
      "['includes_physical_words', 'includes_genetic_words', 'includes_structural_words', 'includes_general_chemical_words', 'includes_functional_group', 'structure_sep_and']\n",
      "\n",
      "['includes_reaction_words', 'structure_next_to_conversion_words', 'includes_creation_words', 'includes_reaction_words2']\n",
      "\n",
      "pip5k1b encodes phosphatidylinositol 4-phosphate 5-kinase beta type i pip5k1beta, an enzyme functionally linked to actin cytoskeleton dynamics that phosphorylates phosphatidylinositol 4-phosphate [pi4p] to generate phosphatidylinositol-4,5-bisphosphate [pi4,5p2]\n",
      "\n",
      "['phosphatidylinositol 4-phosphate', 'phosphatidylinositol-4,5-bisphosphate']\n",
      "\n",
      "[-1 -1 -1  0  0  0  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1  1  1]\n",
      "\n",
      "['includes_genetic_words', 'includes_structural_words', 'includes_general_chemical_words', 'includes_functional_group', 'includes_measure_words']\n",
      "\n",
      "['includes_reaction_words', 'includes_emzyze_words', 'includes_reaction_words2']\n",
      "\n"
     ]
    }
   ],
   "source": [
    "oof = 0\n",
    "for i in df_fn.index:\n",
    "    negs = []\n",
    "    pos = []\n",
    "    \n",
    "    for j in range(len(L_test[i])):\n",
    "        if L_test[i][j] == 0:\n",
    "            negs.append(str(lfs[j].name))\n",
    "        if L_test[i][j] == 1:\n",
    "            pos.append(str(lfs[j].name))\n",
    "    \n",
    "    print(df_fn.iloc[oof]['sentence'])\n",
    "    print('')\n",
    "    print(df_fn.iloc[oof]['chemicals'])\n",
    "    print(\"\")\n",
    "    oof = oof + 1\n",
    "    print(L_test[i])\n",
    "    print('')\n",
    "    print(negs)\n",
    "    print('')\n",
    "    print(pos)\n",
    "    print('')\n",
    "\n",
    "#for each sentence, find indices for all 0 labels and 1 labels, find corresponding LFs,\n",
    "#make df? "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "omega class gsts have dehydroascorbate reductase and thioltransferase activities and also catalyze the reduction of monomethylarsonate, an intermediate in the pathway of arsenic biotransformation\n",
      "\n",
      "['monomethylarsonate', 'arsenic']\n",
      "\n",
      "[-1 -1 -1  0 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1  1 -1  1]\n",
      "\n",
      "['includes_genetic_words', 'includes_functional_group']\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'includes_catalyze_words', 'includes_reaction_words2']\n",
      "\n",
      "n-acetylglucosamine-1-phosphodiester alpha-n-acetylglucosaminidase ec 31445 phosphodiester alpha-glcnacase catalyzes the second step in the synthesis of the mannose 6-phosphate determinant required for efficient intracellular targeting of newly synthesized lysosomal hydrolases to the lysosome\n",
      "\n",
      "['n-acetylglucosamine-1-phosphodiester', 'phosphodiester']\n",
      "\n",
      "[-1 -1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      "  1 -1 -1 -1  0  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1  1 -1  1]\n",
      "\n",
      "['includes_functional_group', 'includes_measure_words', 'includes_experiment_words']\n",
      "\n",
      "['includes_reaction_words', 'includes_reaction_component_words', 'includes_catalyze_words', 'includes_reaction_words2']\n",
      "\n",
      "cgmp inhibited hydrolysis of camp, and camp inhibited cgmp hydrolysis with ic50 values of 14 and 039 microm, respectively\n",
      "\n",
      "['cgmp', 'camp']\n",
      "\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "['includes_no_terms']\n",
      "\n",
      "['includes_reaction_words', 'includes_reaction_words2']\n",
      "\n",
      "on phosphorylation of ser40 by protein kinase a, the affinity for h4biopterin increased [s]05 = 11 +/- 2 microm and the negative cooperativity was amplified h = 027 +/- 003\n",
      "\n",
      "['h4biopterin', 'ser40']\n",
      "\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1  0 -1 -1  0 -1 -1 -1  1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "['includes_comparison_words', 'includes_measure_words']\n",
      "\n",
      "['includes_reaction_words', 'structure_next_to_conversion_words', 'structure_conversion_of', 'includes_reaction_words2']\n",
      "\n",
      "thus, our studies provide new insight into the regulation of th by the concentration of h4biopterin which may have significant implications for the physiological regulation of catecholamine biosynthesis in neuroendocrine cells\n",
      "\n",
      "['catecholamine', 'h4biopterin']\n",
      "\n",
      "[-1 -1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1  0  0  1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "['includes_functional_group', 'includes_measure_words', 'includes_experiment_words']\n",
      "\n",
      "['includes_reaction_words', 'includes_concentration', 'structure_conversion_of', 'includes_reaction_words2']\n",
      "\n",
      "only in the last couple of years a consensus has been reached on what constitutes the main role of bh4 in no synthesis\n",
      "\n",
      "['bh4', 'no']\n",
      "\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      "  1 -1 -1  1]\n",
      "\n",
      "['structure_sep_via']\n",
      "\n",
      "['includes_reaction_words', 'includes_creation_words', 'includes_reaction_words2']\n",
      "\n",
      "the nos-catalyzed oxidation of nadph in the absence of substrate or pterin that does not result in no production\n",
      "\n",
      "['nadph', 'no']\n",
      "\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      "  1 -1 -1 -1  0 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      "  1  1 -1  1]\n",
      "\n",
      "['includes_measure_words']\n",
      "\n",
      "['includes_reaction_words', 'includes_reaction_component_words', 'structure_conversion_of', 'includes_impact_words', 'includes_creation_words', 'includes_catalyze_words', 'includes_reaction_words2']\n",
      "\n",
      "conclusions in the maximum registered dosage, nabumetone inhibits thromboxane production much more than meloxicam, signifying less cox-2 selectivity of the former\n",
      "\n",
      "['nabumetone', 'meloxicam', 'thromboxane']\n",
      "\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1  1\n",
      "  1  0 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      "  1 -1 -1  1]\n",
      "\n",
      "['includes_no_terms', 'includes_comparison_words', 'includes_measure_words']\n",
      "\n",
      "['structure_sep_verb', 'includes_reaction_words', 'includes_reaction_component_words', 'includes_creation_words', 'includes_reaction_words2']\n",
      "\n",
      "methods studies were performed on enzymatic activity of rgad67/65 by production of 3h-gaba from 3h-glutamate, enzyme kinetics, binding to the enzyme cofactor pyridoxal phosphate plp, stability according to differences in ph, temperature and duration of storage, and antigenic reactivity with various gad-specific antisera\n",
      "\n",
      "['3h-gaba', '3h-glutamate', 'plp']\n",
      "\n",
      "[-1 -1  0 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      "  1  0 -1  1  0  0 -1 -1  1 -1 -1 -1 -1  1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1\n",
      " -1 -1  1  1]\n",
      "\n",
      "['includes_physical_words', 'includes_general_chemical_words', 'includes_comparison_words', 'includes_measure_words', 'includes_experiment_words']\n",
      "\n",
      "['includes_reaction_words', 'includes_reaction_component_words', 'includes_react', 'structure_next_to_conversion_words', 'structure_jtsui_pattern_1', 'structure_jtsui_pattern_1_copy', 'includes_emzyze_words', 'includes_reaction_words2']\n",
      "\n",
      "the results reinforce previous assumptions that dopamine may interact with eicosanoid metabolism by means of d2 receptor activation, and implicate an involvement of cpla2 and cox-2 in this effect\n",
      "\n",
      "['eicosanoid', 'dopamine']\n",
      "\n",
      "[-1 -1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "['includes_functional_group']\n",
      "\n",
      "['includes_reaction_words', 'includes_impact_words', 'includes_reaction_words2']\n",
      "\n",
      "among the [fe-s] cluster biosynthetic proteins are included a pyridoxal phosphate-dependent enzyme nifs that is involved in the activation of sulphur from l-cysteine, and a molecular scaffold protein nifu upon which [fe-s] cluster precursors are formed\n",
      "\n",
      "['l-cysteine', 'fe-s', 'sulphur']\n",
      "\n",
      "[-1 -1  0 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1  0  0 -1 -1  1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1  1  1]\n",
      "\n",
      "['includes_physical_words', 'includes_structural_words', 'includes_measure_words', 'includes_experiment_words']\n",
      "\n",
      "['includes_reaction_words', 'structure_next_to_conversion_words', 'structure_conversion_of', 'includes_emzyze_words', 'includes_reaction_words2']\n",
      "\n",
      "these results suggest that the altered gamma-glutamyl kinase results in stabilization of the complex or has an indirect effect on gamma-glutamyl phosphate reductase activity, which leads to an increase in l-proline production in saccharomyces cerevisiae\n",
      "\n",
      "['gamma-glutamyl', 'l-proline']\n",
      "\n",
      "[-1 -1  0  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      "  1  0 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      "  1 -1 -1  1]\n",
      "\n",
      "['includes_physical_words', 'includes_genetic_words', 'includes_comparison_words', 'includes_measure_words']\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'includes_reaction_component_words', 'includes_impact_words', 'includes_creation_words', 'includes_reaction_words2']\n",
      "\n",
      "in shoots, therefore, the pathway from formate plays only a small role in ser synthesis in the case of roots, results indicated that the 9  1 ratio was as a result of greater fluxes of 13c through both pathways together with a relatively higher contribution from the c1-thf synthase/shmt route than in shoots\n",
      "\n",
      "['ser', 'formate']\n",
      "\n",
      "[-1 -1 -1 -1 -1 -1  0  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1  0 -1 -1  0 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      "  1 -1 -1  1]\n",
      "\n",
      "['includes_functional_group', 'includes_paper_artifacts', 'includes_comparison_words', 'includes_measure_words']\n",
      "\n",
      "['includes_reaction_words', 'structure_next_to_conversion_words', 'includes_impact_words', 'includes_creation_words', 'includes_reaction_words2']\n",
      "\n",
      "compared with wt, glyd plants accumulated 5-fold more gly than wt after supplying [alpha-13c]gly for 24 h the accumulation of ser from [alpha-13c]gly was reduced by 25% in the same time period\n",
      "\n",
      "['glyd', 'ser', 'gly']\n",
      "\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1  0 -1 -1  0 -1 -1 -1  1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "['includes_comparison_words', 'includes_measure_words']\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'structure_next_to_conversion_words', 'structure_conversion_of', 'includes_reaction_words2']\n",
      "\n",
      "the protein exhibited modest h2o2-dependent peroxidase activities with guaiacol, potassium iodide, and 2,2'-azino-bis3-ethylbenzothiazoline-6-sulfonic acid abts\n",
      "\n",
      "['guaiacol', 'abts']\n",
      "\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "['includes_experiment_words']\n",
      "\n",
      "['includes_reaction_words', 'includes_reaction_words2']\n",
      "\n",
      "purified paoh1/smo oxidizes both spermine km=16 microm and n1-acetylspermine km=51 microm, but does not oxidize spermidine\n",
      "\n",
      "['spermidine', 'spermine']\n",
      "\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "[]\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'includes_reaction_words2']\n",
      "\n",
      "the purified human enzyme also does not oxidize eight representative antitumor polyamine analogues however, specific oligamine analogues were found to be potent inhibitors of the oxidation of spermine by paoh1/smo\n",
      "\n",
      "['spermine', 'polyamine']\n",
      "\n",
      "[-1 -1 -1 -1 -1 -1  0 -1  0 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1  0 -1 -1  1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1  1  1]\n",
      "\n",
      "['includes_functional_group', 'includes_no_terms', 'includes_experiment_words']\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'structure_next_to_conversion_words', 'structure_conversion_of', 'includes_emzyze_words', 'includes_reaction_words2']\n",
      "\n",
      "ten iu/ml urokinase was also incubated with pooled plasma of stroke patients, that was previously oxidized with the singlet oxygen 1o2 donor chloramine t ct, to destroy plasmatic pai-1 and alpha2-antiplasmin\n",
      "\n",
      "['ct', '1o2']\n",
      "\n",
      "[-1 -1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1  0 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "['includes_functional_group', 'includes_measure_words']\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'structure_next_to_conversion_words', 'includes_reaction_words2']\n",
      "\n",
      "after 0 to 80 minutes 37 degrees c, 50-microl samples were withdrawn and added to 100 microl 15 m arginine, ph 87, and oxidized with 50 microl of 20 mm ct for determination of plasmin activity, 10 microl thereof was incubated with 150 microl 15 m arginine, ph 87, and 100 microl 20 mm ct preoxidized 15 minutes 37 degrees c pooled normal citrate buffered edta-plasma for 30 minutes 37 degrees c\n",
      "\n",
      "['ct', 'citrate', 'arginine']\n",
      "\n",
      "[ 0 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1  1 -1 -1 -1  1\n",
      " -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "['includes_solution_words', 'includes_physical_words', 'includes_measure_words']\n",
      "\n",
      "['includes_oxidation_words', 'includes_neutralization_words', 'includes_reaction_words', 'includes_reaction_words2']\n",
      "\n",
      "qr2 catalyzes the two-electron reduction of menadione via the oxidation of n-alkylated or n-ribosylated nicotinamides\n",
      "\n",
      "['n-alkylated', 'menadione']\n",
      "\n",
      "[-1 -1 -1 -1 -1  0  0 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1  1 -1  1]\n",
      "\n",
      "['includes_general_chemical_words', 'includes_functional_group']\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'structure_conversion_of', 'includes_catalyze_words', 'includes_reaction_words2']\n",
      "\n",
      "in contrast, the inhibitors quinacrine and chloroquine bind exclusively to the reduced enzyme\n",
      "\n",
      "['quinacrine', 'chloroquine']\n",
      "\n",
      "[-1 -1  0 -1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1  1  1]\n",
      "\n",
      "['includes_physical_words', 'includes_no_terms', 'structure_sep_and']\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'includes_emzyze_words', 'includes_reaction_words2']\n",
      "\n",
      "the quinone substrate menadione, on the other hand, binds nonspecifically to both forms of the enzyme\n",
      "\n",
      "['quinone', 'menadione']\n",
      "\n",
      "[-1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      "  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1  1  1]\n",
      "\n",
      "['includes_physical_words']\n",
      "\n",
      "['includes_reaction_words', 'includes_reaction_component_words', 'includes_emzyze_words', 'includes_reaction_words2']\n",
      "\n",
      "this alternate conformation now causes a steric hindrance to the o4 hydroxyl group of the gal moiety of udp-gal, probably causing the dissociation of udp-gal and the reduced kcat of the gal-t reaction\n",
      "\n",
      "['udp-gal', 'gal']\n",
      "\n",
      "[-1 -1 -1  0  0 -1  0 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1  1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "['includes_genetic_words', 'includes_structural_words', 'includes_functional_group']\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'includes_react', 'structure_conversion_of', 'includes_impact_words', 'includes_reaction_words2']\n",
      "\n",
      "we previously reported that aminoglycoside antibiotics reduce sglt1-dependent glucose transport in pig proximal tubular epithelial llc-pk1 cells in parallel with the order of their nephrotoxicity\n",
      "\n",
      "['aminoglycoside', 'glucose']\n",
      "\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1  0  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "['includes_measure_words', 'includes_experiment_words']\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'includes_reaction_words2']\n",
      "\n",
      "thus, it appears that the low conversion of pyridoxal and pyridoxine to plp results in low levels of plp, which is the active cofactor for 113 known enzymatic reactions, including the formation of many key neurotransmitters\n",
      "\n",
      "['plp', 'pyridoxal', 'pyridoxine']\n",
      "\n",
      "[-1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1  1  0 -1 -1  0  1 -1  1 -1 -1  1 -1 -1  1 -1  1 -1 -1  1 -1  1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "['includes_general_chemical_words', 'includes_measure_words', 'structure_sep_and']\n",
      "\n",
      "['structure_sep_conversion_words', 'includes_reaction_words', 'includes_react', 'structure_next_to_conversion_words', 'structure_conversion_of', 'structure_jtsui_pattern_1', 'structure_jtsui_pattern_4', 'structure_jtsui_pattern_1_copy', 'structure_jtsui_pattern_4_copy', 'includes_impact_words', 'includes_reaction_words2']\n",
      "\n",
      "glufosinate inhibits glutamine synthetase and blocks biosynthesis of glutamine\n",
      "\n",
      "['glutamine', 'glufosinate']\n",
      "\n",
      "[-1 -1 -1 -1 -1 -1  0 -1  0 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "['includes_functional_group', 'includes_no_terms']\n",
      "\n",
      "['structure_sep_verb', 'includes_reaction_words', 'includes_reaction_words2']\n",
      "\n",
      "reduced synthesis of nitric oxide no contributes to the endothelial dysfunction and may be related to limited availability of l-arginine, the common substrate of constitutive nitric-oxide synthase nos and cytosolic arginase i and mitochondrial arginase ii\n",
      "\n",
      "['nitric-oxide', 'l-arginine', 'no']\n",
      "\n",
      "[-1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      "  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "['includes_physical_words']\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'includes_reaction_component_words', 'includes_reaction_words2']\n",
      "\n",
      "however, despite activation of l-arginine uptake, the inhibition of arginase activity by nor-noha was still significant\n",
      "\n",
      "['nor-noha', 'l-arginine']\n",
      "\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1  1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1 -1]\n",
      "\n",
      "['includes_no_terms']\n",
      "\n",
      "['structure_next_to_conversion_words', 'structure_conversion_of']\n",
      "\n",
      "these data suggest that endothelial no synthesis depends on the activity of arginase ii in mitochondria and l-arginine carriers in cell membrane\n",
      "\n",
      "['l-arginine', 'no']\n",
      "\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      "  1 -1 -1  1]\n",
      "\n",
      "['includes_experiment_words']\n",
      "\n",
      "['includes_reaction_words', 'includes_creation_words', 'includes_reaction_words2']\n",
      "\n",
      "on the basis of our findings with structurally similar arylhydroxylamine metabolites of therapeutic drugs, we hypothesized that the reductive detoxification of arylhydroxylamine carcinogens was catalyzed by nadh cytochrome b5 reductase b5r and cytochrome b5 cyt b5\n",
      "\n",
      "['nadh', 'arylhydroxylamine']\n",
      "\n",
      "[-1 -1 -1  0  0 -1  0 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1  0 -1 -1 -1  0 -1 -1  1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1  1 -1  1]\n",
      "\n",
      "['includes_genetic_words', 'includes_structural_words', 'includes_functional_group', 'includes_comparison_words', 'includes_experiment_words']\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'structure_next_to_conversion_words', 'structure_conversion_of', 'includes_catalyze_words', 'includes_reaction_words2']\n",
      "\n",
      "exogenous all-trans-retinol, all-trans-13,14-dihydroretinol, or all-trans-7,8-dihydroretinol led to the strong induction of the expression of the retinoic acid-metabolizing enzyme, cyp26a1, arguing for an active signaling function of dihydroretinoid metabolites in zebrafish\n",
      "\n",
      "['all-trans-13,14-dihydroretinol', 'dihydroretinoid', 'all-trans-7,8-dihydroretinol', 'all-trans-retinol']\n",
      "\n",
      "[-1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1  1  1]\n",
      "\n",
      "['includes_physical_words', 'structure_sep_comma']\n",
      "\n",
      "['includes_reaction_words', 'includes_emzyze_words', 'includes_reaction_words2']\n",
      "\n",
      "conclusions the fads2 deletion may prevent the conversion of ala into very-long-chain pufas\n",
      "\n",
      "['pufas', 'ala']\n",
      "\n",
      "[-1 -1 -1 -1  0 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1  1 -1  1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1  1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "['includes_structural_words', 'includes_no_terms']\n",
      "\n",
      "['includes_reaction_words', 'structure_next_to_conversion_words', 'structure_conversion_of', 'structure_jtsui_pattern_4', 'structure_jtsui_pattern_4_copy', 'includes_reaction_words2']\n",
      "\n",
      "replacement of nacl by licl or choline chloride abolished transport activity in shr and wky pte cells\n",
      "\n",
      "['nacl', 'licl']\n",
      "\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1\n",
      " -1 -1 -1 -1 -1  0 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1 -1]\n",
      "\n",
      "['includes_experiment_words']\n",
      "\n",
      "['includes_replacement_words', 'structure_next_to_conversion_words']\n",
      "\n",
      "inhibition of endogenous production of h2s by pag significantly suppressed sp concentration, ppt-a expression and nk1-r expression in the acini\n",
      "\n",
      "['pag', 'sp']\n",
      "\n",
      "[-1 -1  0 -1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      "  1 -1 -1 -1 -1 -1  1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1 -1]\n",
      "\n",
      "['includes_physical_words', 'includes_no_terms']\n",
      "\n",
      "['includes_reaction_component_words', 'includes_concentration', 'structure_next_to_conversion_words']\n",
      "\n",
      "affinity of v782a for cgmp, vardenafil, sildenafil, tadalafil, or ibmx was reduced 55-, 23-, 10-, 3-, and 12-fold, respectively\n",
      "\n",
      "['tadalafil', 'ibmx', 'sildenafil', 'vardenafil', 'cgmp']\n",
      "\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  0 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "['structure_sep_comma']\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'includes_reaction_words2']\n",
      "\n",
      "second, probucol, an antiatherogenic compound reported to be an inactivator of abca1 reduced hepatic alpha-tocopherol secretion\n",
      "\n",
      "['alpha-tocopherol', 'probucol']\n",
      "\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "['includes_measure_words']\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'includes_reaction_words2']\n",
      "\n",
      "thymidine kinase-1 tk-1 and thymidylate synthase ts are key enzymes for salvage and de novo pyrimidine synthesis, respectively\n",
      "\n",
      "['thymidine', 'thymidylate', 'pyrimidine']\n",
      "\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      "  1 -1  1  1]\n",
      "\n",
      "[]\n",
      "\n",
      "['includes_reaction_words', 'includes_creation_words', 'includes_emzyze_words', 'includes_reaction_words2']\n",
      "\n",
      "we have previously reported that tzds reduce estrogen synthesis by inhibiting aromatase activity in human granulosa cells hgc\n",
      "\n",
      "['tzds', 'estrogen']\n",
      "\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1  1  1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      "  1 -1 -1  1]\n",
      "\n",
      "['includes_no_terms', 'includes_experiment_words']\n",
      "\n",
      "['includes_oxidation_words', 'structure_sep_verb', 'includes_reaction_words', 'includes_creation_words', 'includes_reaction_words2']\n",
      "\n",
      "tzds inhibited estradiol production by up to 84% in hgc/mouse osteoblast co-cultures\n",
      "\n",
      "['tzds', 'estradiol']\n",
      "\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1\n",
      "  1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      "  1 -1 -1 -1]\n",
      "\n",
      "['includes_no_terms', 'includes_experiment_words']\n",
      "\n",
      "['structure_sep_verb', 'includes_reaction_component_words', 'includes_creation_words']\n",
      "\n",
      "the results indicate that the glutathione conjugates of bpde are formed and excreted mainly to the apical and to a minor extent to the basolateral side of polarized caco-2 monolayers\n",
      "\n",
      "['bpde', 'glutathione']\n",
      "\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "[]\n",
      "\n",
      "['includes_reaction_words', 'includes_impact_words', 'includes_reaction_words2']\n",
      "\n",
      "tension studies demonstrated a rapid, concentration-dependent, vasodilatory response to tes that did not involve protein synthesis or aromatization to 17beta-estradiol\n",
      "\n",
      "['tes', '17î²-estradiol']\n",
      "\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1  0  1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "['includes_experiment_words']\n",
      "\n",
      "['includes_reaction_words', 'includes_concentration', 'structure_next_to_conversion_words', 'includes_reaction_words2']\n",
      "\n",
      "production of reactive oxygen species ros in the endothelium of rabbit aortic valve leaflets and endothelium-denuded ria and aortic rings was assessed by imaging of dihydroethidium\n",
      "\n",
      "['dihydroethidium', 'oxygen']\n",
      "\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      "  1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1 -1]\n",
      "\n",
      "[]\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_component_words', 'includes_react']\n",
      "\n",
      "our results suggest that arsenite can potentiate edhf-type relaxations via a mechanism that is dependent on hydrogen peroxide, thus demonstrating that dismutation of the superoxide anion generated by nadph oxidase can potentially offset loss of no bioavailability under conditions of reduced enos activity\n",
      "\n",
      "['nadph', 'arsenite', 'superoxide', 'no']\n",
      "\n",
      "[-1 -1 -1  0  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1  0  0 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "['includes_genetic_words', 'includes_structural_words', 'includes_measure_words', 'includes_experiment_words']\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'structure_next_to_conversion_words', 'includes_impact_words', 'includes_reaction_words2']\n",
      "\n",
      "dimethylarsinic acid dmasv, the major urinary metabolite of inorganic arsenic, is weakly cytotoxic, whereas its reduced form, dimethylarsinous acid dmasiii, is highly toxic\n",
      "\n",
      "['dmas', 'arsenic']\n",
      "\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "['includes_measure_words']\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'includes_reaction_words2']\n",
      "\n",
      "although glutathione s-transferase omega 1 gsto1 and arsenic methyltransferase have been shown or thought to catalyze dmasv reduction, their role in dmasv reduction in vivo, or in cell extracts is uncertain\n",
      "\n",
      "['dmas', 'arsenic', 'glutathione']\n",
      "\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1  1 -1  1]\n",
      "\n",
      "['includes_experiment_words']\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'includes_catalyze_words', 'includes_reaction_words2']\n",
      "\n",
      "assessment of dmasv-reducing activity of rat liver cytosol revealed that reduction of dmasv required cytosolic protein and gsh and was inhibited by thiol reagents, gssg and dehydroascorbate\n",
      "\n",
      "['dmas', 'dehydroascorbate', 'gsh', 'thiol', 'gssg']\n",
      "\n",
      "[-1 -1 -1 -1 -1 -1  0 -1  0 -1 -1 -1 -1 -1 -1  1  1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1  0  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "['includes_functional_group', 'includes_no_terms', 'structure_sep_and']\n",
      "\n",
      "['includes_oxidation_words', 'structure_sep_verb', 'includes_reaction_words', 'structure_next_to_conversion_words', 'includes_reaction_words2']\n",
      "\n",
      "although thioredoxin reductase trr inhibitors aurothioglucose and sbiii inhibited cytosolic dmasv reduction, recombinant rat trr plus nadph, alone or when added to the cytosol, failed to support dmasv reduction\n",
      "\n",
      "['nadph', 'dmas', 'aurothioglucose']\n",
      "\n",
      "[-1 -1 -1  0 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1  1 -1 -1  1\n",
      " -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "['includes_genetic_words', 'includes_no_terms', 'includes_measure_words']\n",
      "\n",
      "['includes_oxidation_words', 'includes_combination_words', 'includes_reaction_words', 'includes_reaction_words2']\n",
      "\n",
      "moreover, a new mechanism for controlling endothelial function after sirt1 activation involves a decreased pgc-1alpha acetylation and the subsequent pparalpha activation, resulting in both decreased nadph oxidase-driven ros production and no inactivation\n",
      "\n",
      "['nadph', 'no']\n",
      "\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      "  1  0 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "['includes_comparison_words', 'includes_measure_words']\n",
      "\n",
      "['includes_reaction_words', 'includes_reaction_component_words', 'includes_impact_words', 'includes_reaction_words2']\n",
      "\n",
      "taken together, these data demonstrate that baicalin inhibits the metabolism of dxm in a concentration-dependent manner in rats, possibly through inhibiting hepatic cyp2d and cyp3a activities\n",
      "\n",
      "['dxm', 'baicalin']\n",
      "\n",
      "[-1 -1 -1 -1 -1 -1  0 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1  0  1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "['includes_functional_group', 'includes_no_terms', 'includes_experiment_words']\n",
      "\n",
      "['includes_reaction_words', 'includes_concentration', 'structure_conversion_of', 'includes_reaction_words2']\n",
      "\n",
      "the mechanisms by which oxysterols may influence proliferation are manifold they control the transcription and the turnover of the key enzyme in cholesterol synthesis, 3-hydroxy-3-methylglutaryl coa reductase, by binding to insig-1, insig-2 and liver x receptors\n",
      "\n",
      "['cholesterol', 'oxysterols']\n",
      "\n",
      "[-1 -1  0  0 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      "  1 -1  1  1]\n",
      "\n",
      "['includes_physical_words', 'includes_genetic_words', 'includes_functional_group']\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'includes_creation_words', 'includes_emzyze_words', 'includes_reaction_words2']\n",
      "\n",
      "thiotepa cyp2b6 inhibitor, 25 mum and the monoclonal antibody against cyp2b6 but not troleandomycin cyp3a4 inhibitor, 25 mum or the monoclonal antibody against cyp3a4 inhibited ketamine n-demethylation at clinically relevant concentrations\n",
      "\n",
      "['thiotepa', 'troleandomycin', 'ketamine']\n",
      "\n",
      "[-1 -1 -1 -1 -1 -1  0 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "['includes_functional_group', 'includes_no_terms']\n",
      "\n",
      "['includes_reaction_words', 'includes_concentration', 'includes_reaction_words2']\n",
      "\n",
      "the biological activities of 6-dhsg were attenuated by sulfhydryl antioxidants such as glutathione gsh or n-acetyl cysteine nac, but not ascorbic acid asc\n",
      "\n",
      "['nac', 'gsh', '6-dhsg', 'glutathione', 'sulfhydryl', 'asc']\n",
      "\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1 -1  0 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "['structure_adjacent_mentions', 'structure_sep_sym']\n",
      "\n",
      "['includes_reaction_words', 'structure_next_to_conversion_words', 'includes_reaction_words2']\n",
      "\n",
      "gs-6-dhsg showed reduced biological activities compared with 6-dhsg in multiple biological assays\n",
      "\n",
      "['6-dhsg', 'gs-6-dhsg']\n",
      "\n",
      "[-1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      " -1 -1 -1  1]\n",
      "\n",
      "['includes_physical_words', 'includes_comparison_words']\n",
      "\n",
      "['includes_oxidation_words', 'includes_reaction_words', 'includes_reaction_words2']\n",
      "\n",
      "platelet-induced cox-2-dependent pge2 synthesis in ht29 cells was involved in downregulation of p21waf1/cip1 and upregulation of cyclinb1, since these effects were prevented by rofecoxiba selective cox-2 inhibitor and rescued by exogenous pge2\n",
      "\n",
      "['rofecoxib', 'pge2']\n",
      "\n",
      "[-1 -1 -1 -1 -1 -1 -1 -1  0 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1  1\n",
      " -1 -1 -1 -1 -1  0 -1 -1  1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1\n",
      "  1 -1 -1  1]\n",
      "\n",
      "['includes_no_terms', 'includes_experiment_words']\n",
      "\n",
      "['includes_reaction_words', 'structure_next_to_conversion_words', 'includes_creation_words', 'includes_reaction_words2']\n",
      "\n"
     ]
    }
   ],
   "source": [
    "oof = 0\n",
    "for i in df_fp.index:\n",
    "    negs = []\n",
    "    pos = []\n",
    "    \n",
    "    for j in range(len(L_test[i])):\n",
    "        if L_test[i][j] == 0:\n",
    "            negs.append(str(lfs[j].name))\n",
    "        if L_test[i][j] == 1:\n",
    "            pos.append(str(lfs[j].name))\n",
    "    \n",
    "    print(df_fp.iloc[oof]['sentence'])\n",
    "    print('')\n",
    "    print(df_fp.iloc[oof]['chemicals'])\n",
    "    print('')\n",
    "    oof = oof + 1\n",
    "    print(L_test[i])\n",
    "    print('')\n",
    "    print(negs)\n",
    "    print('')\n",
    "    print(pos)\n",
    "    print('')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 207,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-207-a881e3605e50>:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_test['label_voter'] = df_true['label_voter'].replace(-1, 1)\n"
     ]
    }
   ],
   "source": [
    "\n",
    "df_true[df_true[\"label_voter\"] == -1]\n",
    "\n",
    "df_test['label_voter'] = df_test['label_voter'].replace(-1, 1)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 213,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sentence</th>\n",
       "      <th>chemicals</th>\n",
       "      <th>truth</th>\n",
       "      <th>substrates</th>\n",
       "      <th>products</th>\n",
       "      <th>text</th>\n",
       "      <th>label_voter</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>the enzyme cyclo-oxygenase catalyses the oxyge...</td>\n",
       "      <td>[arachidonic acid, prostaglandins]</td>\n",
       "      <td>1</td>\n",
       "      <td>arachidonic, acid</td>\n",
       "      <td>prostaglandins</td>\n",
       "      <td>[the enzyme cyclo-oxygenase catalyses the oxyg...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>although the exact site of decarboxylation of ...</td>\n",
       "      <td>[levodopa, dopamine]</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[although the exact site of decarboxylation of...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>the enzyme dimethylarginine dimethylaminohydro...</td>\n",
       "      <td>[citrulline, arginine, methylamines, dimethyla...</td>\n",
       "      <td>1</td>\n",
       "      <td>arginine</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[the enzyme dimethylarginine dimethylaminohydr...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>background and aims glutamic acid decarboxylas...</td>\n",
       "      <td>[gaba, glutamate]</td>\n",
       "      <td>1</td>\n",
       "      <td>glutamate</td>\n",
       "      <td>gamma-aminobutyric, acid, GABA</td>\n",
       "      <td>[background and aims glutamic acid decarboxyla...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>mazindane 26 was found to be a pro-drug, oxidi...</td>\n",
       "      <td>[mazindol, mazindane, 5-oh, 5-h]</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[mazindane 26 was found to be a pro-drug, oxid...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>interestingly, the allele of pro1 was shown to...</td>\n",
       "      <td>[l-glutamate, l-proline, gamma-glutamyl]</td>\n",
       "      <td>1</td>\n",
       "      <td>L-glutamate</td>\n",
       "      <td>L-proline</td>\n",
       "      <td>[interestingly, the allele of pro1 was shown t...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>in addition to the racemization it also cataly...</td>\n",
       "      <td>[pyruvate, l-ser]</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[in addition to the racemization it also catal...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>we concluded that fdh has no direct role in th...</td>\n",
       "      <td>[ser, formate, co(2)]</td>\n",
       "      <td>1</td>\n",
       "      <td>formate</td>\n",
       "      <td>CO</td>\n",
       "      <td>[we concluded that fdh has no direct role in t...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>the ratio between the gdc/shmt and c1-thf synt...</td>\n",
       "      <td>[ser, formate, gly]</td>\n",
       "      <td>1</td>\n",
       "      <td>[, [</td>\n",
       "      <td>Ser</td>\n",
       "      <td>[the ratio between the gdc/shmt and c1-thf syn...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>one of the enzymes responsible for the product...</td>\n",
       "      <td>[ka, kynurenine, oxoglutaramic acid, glutamine]</td>\n",
       "      <td>1</td>\n",
       "      <td>glutamine</td>\n",
       "      <td>KA, oxoglutaramic, acid</td>\n",
       "      <td>[one of the enzymes responsible for the produc...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>cpt i carnitine palmitoyltransferase i catalys...</td>\n",
       "      <td>[palmitoyl-coa, palmitoylcarnitine, l-carnitin...</td>\n",
       "      <td>1</td>\n",
       "      <td>palmitoyl-CoA</td>\n",
       "      <td>palmitoylcarnitine</td>\n",
       "      <td>[cpt i carnitine palmitoyltransferase i cataly...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>purpose the fluoropyrimidine carbamate capecit...</td>\n",
       "      <td>[5-fu, capecitabine, thymidine, 5-fluorouracil]</td>\n",
       "      <td>1</td>\n",
       "      <td>fluoropyrimidine, carbamate, capecitabine</td>\n",
       "      <td>5-fluorouracil, 5-FU</td>\n",
       "      <td>[purpose the fluoropyrimidine carbamate capeci...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>spermine was rapidly taken up and metabolized ...</td>\n",
       "      <td>[spermidine, spermine, polyamine]</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[spermine was rapidly taken up and metabolized...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>carnitine acetyltransferases crat catalyze the...</td>\n",
       "      <td>[acetylcarnitine, acetyl-coa, coa, carnitine]</td>\n",
       "      <td>1</td>\n",
       "      <td>acetyl-CoA, carnitine</td>\n",
       "      <td>acetylcarnitine</td>\n",
       "      <td>[carnitine acetyltransferases crat catalyze th...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>l-serine dehydratase sdh, a member of the beta...</td>\n",
       "      <td>[l-threonine, pyruvate, l-serine, 2-oxobutyrate]</td>\n",
       "      <td>1</td>\n",
       "      <td>L-serine, L-threonine</td>\n",
       "      <td>pyruvate, 2-oxobutyrate</td>\n",
       "      <td>[l-serine dehydratase sdh, a member of the bet...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>beta-1,4-galactosyltransferase i beta4gal-t1 n...</td>\n",
       "      <td>[udp-gal, glcnac, glc, gal, udp-glc]</td>\n",
       "      <td>1</td>\n",
       "      <td>UDP-Gal, UDP-Glc</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[beta-1,4-galactosyltransferase i beta4gal-t1 ...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>a cysteine residue in place of the catalytic t...</td>\n",
       "      <td>[cysteine, glutathione, tyrosine, serine, disu...</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[a cysteine residue in place of the catalytic ...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>astrocytes may play a role in these manifestat...</td>\n",
       "      <td>[glutamine, glutamate]</td>\n",
       "      <td>1</td>\n",
       "      <td>glutamate</td>\n",
       "      <td>glutamine</td>\n",
       "      <td>[astrocytes may play a role in these manifesta...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>conclusions total vitamin b6 is abnormally hig...</td>\n",
       "      <td>[plp, pyridoxal, pyridoxine]</td>\n",
       "      <td>1</td>\n",
       "      <td>pyridoxine, pyridoxal</td>\n",
       "      <td>PLP</td>\n",
       "      <td>[conclusions total vitamin b6 is abnormally hi...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>this enzyme catalyzes the s-adenosyl-l-methion...</td>\n",
       "      <td>[glycine, sarcosine, s-adenosyl-l-methionine, ...</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[this enzyme catalyzes the s-adenosyl-l-methio...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>estrogens are biosynthesised from androgens by...</td>\n",
       "      <td>[androgens, estrogens]</td>\n",
       "      <td>1</td>\n",
       "      <td>androgens</td>\n",
       "      <td>Estrogens</td>\n",
       "      <td>[estrogens are biosynthesised from androgens b...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>background histamine synthesized by histidine ...</td>\n",
       "      <td>[l-histidine, histidine, histamine]</td>\n",
       "      <td>1</td>\n",
       "      <td>L-histidine</td>\n",
       "      <td>Histamine</td>\n",
       "      <td>[background histamine synthesized by histidine...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>methionine synthase reductase mtrr is an enzym...</td>\n",
       "      <td>[methionine, hcy]</td>\n",
       "      <td>1</td>\n",
       "      <td>Hcy</td>\n",
       "      <td>methionine</td>\n",
       "      <td>[methionine synthase reductase mtrr is an enzy...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>gpxs reduce hydroperoxides to the correspondin...</td>\n",
       "      <td>[gsh, hydroperoxides, glutathione, alcohols]</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>alcohols</td>\n",
       "      <td>[gpxs reduce hydroperoxides to the correspondi...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>adenine phosphoribosyltransferase plays a role...</td>\n",
       "      <td>[purine, adenosine monophosphate, adenine]</td>\n",
       "      <td>1</td>\n",
       "      <td>adenine</td>\n",
       "      <td>adenosine, monophosphate</td>\n",
       "      <td>[adenine phosphoribosyltransferase plays a rol...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>they all expressed ass, but not ornithine tran...</td>\n",
       "      <td>[citrulline, arginine, rharg, ornithine]</td>\n",
       "      <td>1</td>\n",
       "      <td>ornithine, arginine, arginine</td>\n",
       "      <td>citrulline</td>\n",
       "      <td>[they all expressed ass, but not ornithine tra...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>metabolism of vitamin a, all-trans-retinol, le...</td>\n",
       "      <td>[all-trans-retinol, 11-cis-retinaldehyde]</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[metabolism of vitamin a, all-trans-retinol, l...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>specifically, all-trans-13,14-dihydroretinol i...</td>\n",
       "      <td>[all-trans-13,14-dihydroretinoic acid, all-tra...</td>\n",
       "      <td>1</td>\n",
       "      <td>all-trans-13,14-dihydroretinol</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[specifically, all-trans-13,14-dihydroretinol ...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>unlike mouse retsat mretsat, zretsat a had an ...</td>\n",
       "      <td>[all-trans-7,8-dihydroretinol, all-trans-retin...</td>\n",
       "      <td>1</td>\n",
       "      <td>all-trans-retinol</td>\n",
       "      <td>all-trans-13,14-dihydroretinol, all-trans-7,8-...</td>\n",
       "      <td>[unlike mouse retsat mretsat, zretsat a had an...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>ldh is responsible for pyruvate conversion to ...</td>\n",
       "      <td>[lactate, pyruvate]</td>\n",
       "      <td>1</td>\n",
       "      <td>pyruvate</td>\n",
       "      <td>lactate</td>\n",
       "      <td>[ldh is responsible for pyruvate conversion to...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>in kidneys, stimulation of adenylyl cyclase ca...</td>\n",
       "      <td>[camp, amp, adenosine]</td>\n",
       "      <td>1</td>\n",
       "      <td>cAMP, AMP</td>\n",
       "      <td>AMP, adenosine</td>\n",
       "      <td>[in kidneys, stimulation of adenylyl cyclase c...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>hydrogen sulphide h2s is synthesized from l-cy...</td>\n",
       "      <td>[l-cysteine, hydrogen sulphide]</td>\n",
       "      <td>1</td>\n",
       "      <td>L-cysteine</td>\n",
       "      <td>Hydrogen, sulphide, H</td>\n",
       "      <td>[hydrogen sulphide h2s is synthesized from l-c...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>adenylosuccinate synthetase adss catalyzes the...</td>\n",
       "      <td>[gdp, mg2+, imp, aspartate, adenylosuccinate, ...</td>\n",
       "      <td>1</td>\n",
       "      <td>Mg2+</td>\n",
       "      <td>adenylosuccinate</td>\n",
       "      <td>[adenylosuccinate synthetase adss catalyzes th...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>risperidone is metabolized to its active metab...</td>\n",
       "      <td>[9-hydroxyrisperidone, risperidone]</td>\n",
       "      <td>1</td>\n",
       "      <td>Risperidone</td>\n",
       "      <td>9-hydroxyrisperidone</td>\n",
       "      <td>[risperidone is metabolized to its active meta...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>pge2 is synthesized from arachidonic acid by c...</td>\n",
       "      <td>[arachidonic acid, pge(2)]</td>\n",
       "      <td>1</td>\n",
       "      <td>arachidonic, acid</td>\n",
       "      <td>PGE</td>\n",
       "      <td>[pge2 is synthesized from arachidonic acid by ...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>tas-102 is a novel drug containing trifluoroth...</td>\n",
       "      <td>[tas-102, trifluorothymidine]</td>\n",
       "      <td>1</td>\n",
       "      <td>TAS-102, trifluorothymidine</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[tas-102 is a novel drug containing trifluorot...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>aldo-keto reductases akrs metabolize a wide ra...</td>\n",
       "      <td>[o-quinones, aldo-keto, oxygen, ros, pahs]</td>\n",
       "      <td>1</td>\n",
       "      <td>polycyclic, aromatic, PAHs, o-quinones</td>\n",
       "      <td>oxygen</td>\n",
       "      <td>[aldo-keto reductases akrs metabolize a wide r...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>the ultimate carcinogenic phase i bp metabolit...</td>\n",
       "      <td>[glutathione, bpde]</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>glutathione</td>\n",
       "      <td>[the ultimate carcinogenic phase i bp metaboli...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>the mitochondrial respiratory chain complex iv...</td>\n",
       "      <td>[cytochrome c, oxygen, water]</td>\n",
       "      <td>1</td>\n",
       "      <td>oxygen</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[the mitochondrial respiratory chain complex i...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>here, we demonstrate that peptidyl-prolyl cis/...</td>\n",
       "      <td>[hne]</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[here, we demonstrate that peptidyl-prolyl cis...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>aldose reductase ar catalyzes the reduction of...</td>\n",
       "      <td>[alcohol, aldose, aldehydes]</td>\n",
       "      <td>1</td>\n",
       "      <td>aldehydes</td>\n",
       "      <td>alcohol</td>\n",
       "      <td>[aldose reductase ar catalyzes the reduction o...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>leukotriene a4 hydrolase lta4h is a cystolic e...</td>\n",
       "      <td>[lta(4), ltb(4)]</td>\n",
       "      <td>1</td>\n",
       "      <td>LTA</td>\n",
       "      <td>LTB</td>\n",
       "      <td>[leukotriene a4 hydrolase lta4h is a cystolic ...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>each recombinant protein metabolized aa and la...</td>\n",
       "      <td>[aa, hydroxy, epoxides, la]</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[each recombinant protein metabolized aa and l...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>we conclude that the four new cyp2j isoforms m...</td>\n",
       "      <td>[aa, bioactive lipids, la]</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[we conclude that the four new cyp2j isoforms ...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>as predicted, tes enhanced the production of b...</td>\n",
       "      <td>[peroxynitrite, xanthine, superoxide, tes]</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>superoxide</td>\n",
       "      <td>[as predicted, tes enhanced the production of ...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>these findings, derived from a variety of anal...</td>\n",
       "      <td>[androgen, superoxide, peroxynitrite, xanthine...</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>peroxynitrite, superoxide, NO</td>\n",
       "      <td>[these findings, derived from a variety of ana...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>type 2 11beta-hydroxysteroid dehydrogenase enc...</td>\n",
       "      <td>[cortisol, cortisone, 11î²-hydroxysteroid, glu...</td>\n",
       "      <td>1</td>\n",
       "      <td>cortisol</td>\n",
       "      <td>cortisone</td>\n",
       "      <td>[type 2 11beta-hydroxysteroid dehydrogenase en...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>carboxylesterases hydrolyze esters, amides, an...</td>\n",
       "      <td>[amides, thioesters, alcohols, amines, carboxy...</td>\n",
       "      <td>1</td>\n",
       "      <td>esters, amides, thioesters</td>\n",
       "      <td>carboxylic, acids, alcohols, amines, thiols</td>\n",
       "      <td>[carboxylesterases hydrolyze esters, amides, a...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>uridine 5'-diphosphate- glucuronosyltransferas...</td>\n",
       "      <td>[acyl glucuronides, carboxylic acids, esters]</td>\n",
       "      <td>1</td>\n",
       "      <td>carboxylic, acids</td>\n",
       "      <td>acyl, glucuronides, acyl, glucuronides</td>\n",
       "      <td>[uridine 5'-diphosphate- glucuronosyltransfera...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>cytochrome p450 epoxygenase 2j2 cyp2j2 metabol...</td>\n",
       "      <td>[arachidonic acids, eets]</td>\n",
       "      <td>1</td>\n",
       "      <td>arachidonic, acids</td>\n",
       "      <td>epoxyeicosatrienoic, acids, EETs</td>\n",
       "      <td>[cytochrome p450 epoxygenase 2j2 cyp2j2 metabo...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>ketamine is primarily metabolized to norketami...</td>\n",
       "      <td>[ketamine, norketamine]</td>\n",
       "      <td>1</td>\n",
       "      <td>Ketamine</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[ketamine is primarily metabolized to norketam...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>6-dhsg was metabolised by gsh to form a gsh co...</td>\n",
       "      <td>[gsh, 6-dhsg, gs-6-dhsg]</td>\n",
       "      <td>1</td>\n",
       "      <td>6-DHSG</td>\n",
       "      <td>GS-6-DHSG</td>\n",
       "      <td>[6-dhsg was metabolised by gsh to form a gsh c...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             sentence  \\\n",
       "0   the enzyme cyclo-oxygenase catalyses the oxyge...   \n",
       "1   although the exact site of decarboxylation of ...   \n",
       "2   the enzyme dimethylarginine dimethylaminohydro...   \n",
       "3   background and aims glutamic acid decarboxylas...   \n",
       "4   mazindane 26 was found to be a pro-drug, oxidi...   \n",
       "5   interestingly, the allele of pro1 was shown to...   \n",
       "6   in addition to the racemization it also cataly...   \n",
       "7   we concluded that fdh has no direct role in th...   \n",
       "8   the ratio between the gdc/shmt and c1-thf synt...   \n",
       "9   one of the enzymes responsible for the product...   \n",
       "10  cpt i carnitine palmitoyltransferase i catalys...   \n",
       "11  purpose the fluoropyrimidine carbamate capecit...   \n",
       "12  spermine was rapidly taken up and metabolized ...   \n",
       "13  carnitine acetyltransferases crat catalyze the...   \n",
       "15  l-serine dehydratase sdh, a member of the beta...   \n",
       "16  beta-1,4-galactosyltransferase i beta4gal-t1 n...   \n",
       "18  a cysteine residue in place of the catalytic t...   \n",
       "21  astrocytes may play a role in these manifestat...   \n",
       "22  conclusions total vitamin b6 is abnormally hig...   \n",
       "23  this enzyme catalyzes the s-adenosyl-l-methion...   \n",
       "24  estrogens are biosynthesised from androgens by...   \n",
       "25  background histamine synthesized by histidine ...   \n",
       "26  methionine synthase reductase mtrr is an enzym...   \n",
       "27  gpxs reduce hydroperoxides to the correspondin...   \n",
       "28  adenine phosphoribosyltransferase plays a role...   \n",
       "29  they all expressed ass, but not ornithine tran...   \n",
       "30  metabolism of vitamin a, all-trans-retinol, le...   \n",
       "31  specifically, all-trans-13,14-dihydroretinol i...   \n",
       "32  unlike mouse retsat mretsat, zretsat a had an ...   \n",
       "33  ldh is responsible for pyruvate conversion to ...   \n",
       "34  in kidneys, stimulation of adenylyl cyclase ca...   \n",
       "35  hydrogen sulphide h2s is synthesized from l-cy...   \n",
       "36  adenylosuccinate synthetase adss catalyzes the...   \n",
       "37  risperidone is metabolized to its active metab...   \n",
       "38  pge2 is synthesized from arachidonic acid by c...   \n",
       "39  tas-102 is a novel drug containing trifluoroth...   \n",
       "40  aldo-keto reductases akrs metabolize a wide ra...   \n",
       "41  the ultimate carcinogenic phase i bp metabolit...   \n",
       "43  the mitochondrial respiratory chain complex iv...   \n",
       "44  here, we demonstrate that peptidyl-prolyl cis/...   \n",
       "45  aldose reductase ar catalyzes the reduction of...   \n",
       "46  leukotriene a4 hydrolase lta4h is a cystolic e...   \n",
       "47  each recombinant protein metabolized aa and la...   \n",
       "48  we conclude that the four new cyp2j isoforms m...   \n",
       "49  as predicted, tes enhanced the production of b...   \n",
       "50  these findings, derived from a variety of anal...   \n",
       "51  type 2 11beta-hydroxysteroid dehydrogenase enc...   \n",
       "52  carboxylesterases hydrolyze esters, amides, an...   \n",
       "53  uridine 5'-diphosphate- glucuronosyltransferas...   \n",
       "54  cytochrome p450 epoxygenase 2j2 cyp2j2 metabol...   \n",
       "55  ketamine is primarily metabolized to norketami...   \n",
       "57  6-dhsg was metabolised by gsh to form a gsh co...   \n",
       "\n",
       "                                            chemicals  truth  \\\n",
       "0                  [arachidonic acid, prostaglandins]      1   \n",
       "1                                [levodopa, dopamine]      1   \n",
       "2   [citrulline, arginine, methylamines, dimethyla...      1   \n",
       "3                                   [gaba, glutamate]      1   \n",
       "4                    [mazindol, mazindane, 5-oh, 5-h]      1   \n",
       "5            [l-glutamate, l-proline, gamma-glutamyl]      1   \n",
       "6                                   [pyruvate, l-ser]      1   \n",
       "7                               [ser, formate, co(2)]      1   \n",
       "8                                 [ser, formate, gly]      1   \n",
       "9     [ka, kynurenine, oxoglutaramic acid, glutamine]      1   \n",
       "10  [palmitoyl-coa, palmitoylcarnitine, l-carnitin...      1   \n",
       "11    [5-fu, capecitabine, thymidine, 5-fluorouracil]      1   \n",
       "12                  [spermidine, spermine, polyamine]      1   \n",
       "13      [acetylcarnitine, acetyl-coa, coa, carnitine]      1   \n",
       "15   [l-threonine, pyruvate, l-serine, 2-oxobutyrate]      1   \n",
       "16               [udp-gal, glcnac, glc, gal, udp-glc]      1   \n",
       "18  [cysteine, glutathione, tyrosine, serine, disu...      1   \n",
       "21                             [glutamine, glutamate]      1   \n",
       "22                       [plp, pyridoxal, pyridoxine]      1   \n",
       "23  [glycine, sarcosine, s-adenosyl-l-methionine, ...      1   \n",
       "24                             [androgens, estrogens]      1   \n",
       "25                [l-histidine, histidine, histamine]      1   \n",
       "26                                  [methionine, hcy]      1   \n",
       "27       [gsh, hydroperoxides, glutathione, alcohols]      1   \n",
       "28         [purine, adenosine monophosphate, adenine]      1   \n",
       "29           [citrulline, arginine, rharg, ornithine]      1   \n",
       "30          [all-trans-retinol, 11-cis-retinaldehyde]      1   \n",
       "31  [all-trans-13,14-dihydroretinoic acid, all-tra...      1   \n",
       "32  [all-trans-7,8-dihydroretinol, all-trans-retin...      1   \n",
       "33                                [lactate, pyruvate]      1   \n",
       "34                             [camp, amp, adenosine]      1   \n",
       "35                    [l-cysteine, hydrogen sulphide]      1   \n",
       "36  [gdp, mg2+, imp, aspartate, adenylosuccinate, ...      1   \n",
       "37                [9-hydroxyrisperidone, risperidone]      1   \n",
       "38                         [arachidonic acid, pge(2)]      1   \n",
       "39                      [tas-102, trifluorothymidine]      1   \n",
       "40         [o-quinones, aldo-keto, oxygen, ros, pahs]      1   \n",
       "41                                [glutathione, bpde]      1   \n",
       "43                      [cytochrome c, oxygen, water]      1   \n",
       "44                                              [hne]      1   \n",
       "45                       [alcohol, aldose, aldehydes]      1   \n",
       "46                                   [lta(4), ltb(4)]      1   \n",
       "47                        [aa, hydroxy, epoxides, la]      1   \n",
       "48                         [aa, bioactive lipids, la]      1   \n",
       "49         [peroxynitrite, xanthine, superoxide, tes]      1   \n",
       "50  [androgen, superoxide, peroxynitrite, xanthine...      1   \n",
       "51  [cortisol, cortisone, 11î²-hydroxysteroid, glu...      1   \n",
       "52  [amides, thioesters, alcohols, amines, carboxy...      1   \n",
       "53      [acyl glucuronides, carboxylic acids, esters]      1   \n",
       "54                          [arachidonic acids, eets]      1   \n",
       "55                            [ketamine, norketamine]      1   \n",
       "57                           [gsh, 6-dhsg, gs-6-dhsg]      1   \n",
       "\n",
       "                                   substrates  \\\n",
       "0                           arachidonic, acid   \n",
       "1                                         NaN   \n",
       "2                                    arginine   \n",
       "3                                   glutamate   \n",
       "4                                         NaN   \n",
       "5                                 L-glutamate   \n",
       "6                                         NaN   \n",
       "7                                     formate   \n",
       "8                                        [, [   \n",
       "9                                   glutamine   \n",
       "10                              palmitoyl-CoA   \n",
       "11  fluoropyrimidine, carbamate, capecitabine   \n",
       "12                                        NaN   \n",
       "13                      acetyl-CoA, carnitine   \n",
       "15                      L-serine, L-threonine   \n",
       "16                           UDP-Gal, UDP-Glc   \n",
       "18                                        NaN   \n",
       "21                                  glutamate   \n",
       "22                      pyridoxine, pyridoxal   \n",
       "23                                        NaN   \n",
       "24                                  androgens   \n",
       "25                                L-histidine   \n",
       "26                                        Hcy   \n",
       "27                                        NaN   \n",
       "28                                    adenine   \n",
       "29              ornithine, arginine, arginine   \n",
       "30                                        NaN   \n",
       "31             all-trans-13,14-dihydroretinol   \n",
       "32                          all-trans-retinol   \n",
       "33                                   pyruvate   \n",
       "34                                  cAMP, AMP   \n",
       "35                                 L-cysteine   \n",
       "36                                       Mg2+   \n",
       "37                                Risperidone   \n",
       "38                          arachidonic, acid   \n",
       "39                TAS-102, trifluorothymidine   \n",
       "40     polycyclic, aromatic, PAHs, o-quinones   \n",
       "41                                        NaN   \n",
       "43                                     oxygen   \n",
       "44                                        NaN   \n",
       "45                                  aldehydes   \n",
       "46                                        LTA   \n",
       "47                                        NaN   \n",
       "48                                        NaN   \n",
       "49                                        NaN   \n",
       "50                                        NaN   \n",
       "51                                   cortisol   \n",
       "52                 esters, amides, thioesters   \n",
       "53                          carboxylic, acids   \n",
       "54                         arachidonic, acids   \n",
       "55                                   Ketamine   \n",
       "57                                     6-DHSG   \n",
       "\n",
       "                                             products  \\\n",
       "0                                      prostaglandins   \n",
       "1                                                 NaN   \n",
       "2                                                 NaN   \n",
       "3                      gamma-aminobutyric, acid, GABA   \n",
       "4                                                 NaN   \n",
       "5                                           L-proline   \n",
       "6                                                 NaN   \n",
       "7                                                  CO   \n",
       "8                                                 Ser   \n",
       "9                             KA, oxoglutaramic, acid   \n",
       "10                                 palmitoylcarnitine   \n",
       "11                               5-fluorouracil, 5-FU   \n",
       "12                                                NaN   \n",
       "13                                    acetylcarnitine   \n",
       "15                            pyruvate, 2-oxobutyrate   \n",
       "16                                                NaN   \n",
       "18                                                NaN   \n",
       "21                                          glutamine   \n",
       "22                                                PLP   \n",
       "23                                                NaN   \n",
       "24                                          Estrogens   \n",
       "25                                          Histamine   \n",
       "26                                         methionine   \n",
       "27                                           alcohols   \n",
       "28                           adenosine, monophosphate   \n",
       "29                                         citrulline   \n",
       "30                                                NaN   \n",
       "31                                                NaN   \n",
       "32  all-trans-13,14-dihydroretinol, all-trans-7,8-...   \n",
       "33                                            lactate   \n",
       "34                                     AMP, adenosine   \n",
       "35                              Hydrogen, sulphide, H   \n",
       "36                                   adenylosuccinate   \n",
       "37                               9-hydroxyrisperidone   \n",
       "38                                                PGE   \n",
       "39                                                NaN   \n",
       "40                                             oxygen   \n",
       "41                                        glutathione   \n",
       "43                                                NaN   \n",
       "44                                                NaN   \n",
       "45                                            alcohol   \n",
       "46                                                LTB   \n",
       "47                                                NaN   \n",
       "48                                                NaN   \n",
       "49                                         superoxide   \n",
       "50                      peroxynitrite, superoxide, NO   \n",
       "51                                          cortisone   \n",
       "52        carboxylic, acids, alcohols, amines, thiols   \n",
       "53             acyl, glucuronides, acyl, glucuronides   \n",
       "54                   epoxyeicosatrienoic, acids, EETs   \n",
       "55                                                NaN   \n",
       "57                                          GS-6-DHSG   \n",
       "\n",
       "                                                 text  label_voter  \n",
       "0   [the enzyme cyclo-oxygenase catalyses the oxyg...          1.0  \n",
       "1   [although the exact site of decarboxylation of...          1.0  \n",
       "2   [the enzyme dimethylarginine dimethylaminohydr...          1.0  \n",
       "3   [background and aims glutamic acid decarboxyla...          1.0  \n",
       "4   [mazindane 26 was found to be a pro-drug, oxid...          1.0  \n",
       "5   [interestingly, the allele of pro1 was shown t...          1.0  \n",
       "6   [in addition to the racemization it also catal...          1.0  \n",
       "7   [we concluded that fdh has no direct role in t...          1.0  \n",
       "8   [the ratio between the gdc/shmt and c1-thf syn...          1.0  \n",
       "9   [one of the enzymes responsible for the produc...          1.0  \n",
       "10  [cpt i carnitine palmitoyltransferase i cataly...          1.0  \n",
       "11  [purpose the fluoropyrimidine carbamate capeci...          1.0  \n",
       "12  [spermine was rapidly taken up and metabolized...          1.0  \n",
       "13  [carnitine acetyltransferases crat catalyze th...          1.0  \n",
       "15  [l-serine dehydratase sdh, a member of the bet...          1.0  \n",
       "16  [beta-1,4-galactosyltransferase i beta4gal-t1 ...          1.0  \n",
       "18  [a cysteine residue in place of the catalytic ...          1.0  \n",
       "21  [astrocytes may play a role in these manifesta...          1.0  \n",
       "22  [conclusions total vitamin b6 is abnormally hi...          1.0  \n",
       "23  [this enzyme catalyzes the s-adenosyl-l-methio...          1.0  \n",
       "24  [estrogens are biosynthesised from androgens b...          1.0  \n",
       "25  [background histamine synthesized by histidine...          1.0  \n",
       "26  [methionine synthase reductase mtrr is an enzy...          1.0  \n",
       "27  [gpxs reduce hydroperoxides to the correspondi...          1.0  \n",
       "28  [adenine phosphoribosyltransferase plays a rol...          1.0  \n",
       "29  [they all expressed ass, but not ornithine tra...          1.0  \n",
       "30  [metabolism of vitamin a, all-trans-retinol, l...          1.0  \n",
       "31  [specifically, all-trans-13,14-dihydroretinol ...          1.0  \n",
       "32  [unlike mouse retsat mretsat, zretsat a had an...          1.0  \n",
       "33  [ldh is responsible for pyruvate conversion to...          1.0  \n",
       "34  [in kidneys, stimulation of adenylyl cyclase c...          1.0  \n",
       "35  [hydrogen sulphide h2s is synthesized from l-c...          1.0  \n",
       "36  [adenylosuccinate synthetase adss catalyzes th...          1.0  \n",
       "37  [risperidone is metabolized to its active meta...          1.0  \n",
       "38  [pge2 is synthesized from arachidonic acid by ...          1.0  \n",
       "39  [tas-102 is a novel drug containing trifluorot...          1.0  \n",
       "40  [aldo-keto reductases akrs metabolize a wide r...          1.0  \n",
       "41  [the ultimate carcinogenic phase i bp metaboli...          1.0  \n",
       "43  [the mitochondrial respiratory chain complex i...          1.0  \n",
       "44  [here, we demonstrate that peptidyl-prolyl cis...          1.0  \n",
       "45  [aldose reductase ar catalyzes the reduction o...          1.0  \n",
       "46  [leukotriene a4 hydrolase lta4h is a cystolic ...          1.0  \n",
       "47  [each recombinant protein metabolized aa and l...          1.0  \n",
       "48  [we conclude that the four new cyp2j isoforms ...          1.0  \n",
       "49  [as predicted, tes enhanced the production of ...          1.0  \n",
       "50  [these findings, derived from a variety of ana...          1.0  \n",
       "51  [type 2 11beta-hydroxysteroid dehydrogenase en...          1.0  \n",
       "52  [carboxylesterases hydrolyze esters, amides, a...          1.0  \n",
       "53  [uridine 5'-diphosphate- glucuronosyltransfera...          1.0  \n",
       "54  [cytochrome p450 epoxygenase 2j2 cyp2j2 metabo...          1.0  \n",
       "55  [ketamine is primarily metabolized to norketam...          1.0  \n",
       "57  [6-dhsg was metabolised by gsh to form a gsh c...          1.0  "
      ]
     },
     "execution_count": 213,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_test[df_test[\"label_voter\"] == 1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 198,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "accuracy: 0.7581329561527581\n"
     ]
    },
    {
     "ename": "TypeError",
     "evalue": "'(0       False\n1       False\n2       False\n3       False\n4       False\n        ...  \n1409     True\n1410     True\n1411     True\n1412     True\n1413     True\nName: truth, Length: 1414, dtype: bool, 0        True\n1       False\n2       False\n3        True\n4        True\n        ...  \n1409    False\n1410    False\n1411     True\n1412    False\n1413    False\nName: label_voter, Length: 1414, dtype: bool)' is an invalid key",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-198-5c24ff26f79f>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 6\u001b[0;31m \u001b[0mdf_test\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mdf_test\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"truth\"\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdf_test\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"label_voter\"\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      7\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      8\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/anaconda3/lib/python3.8/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m__getitem__\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   2798\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mnlevels\u001b[0m \u001b[0;34m>\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2799\u001b[0m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_getitem_multilevel\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2800\u001b[0;31m             \u001b[0mindexer\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2801\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mis_integer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mindexer\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2802\u001b[0m                 \u001b[0mindexer\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0mindexer\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/anaconda3/lib/python3.8/site-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36mget_loc\u001b[0;34m(self, key, method, tolerance)\u001b[0m\n\u001b[1;32m   2644\u001b[0m                 )\n\u001b[1;32m   2645\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2646\u001b[0;31m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2647\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2648\u001b[0m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_maybe_cast_indexer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mTypeError\u001b[0m: '(0       False\n1       False\n2       False\n3       False\n4       False\n        ...  \n1409     True\n1410     True\n1411     True\n1412     True\n1413     True\nName: truth, Length: 1414, dtype: bool, 0        True\n1       False\n2       False\n3        True\n4        True\n        ...  \n1409    False\n1410    False\n1411     True\n1412    False\n1413    False\nName: label_voter, Length: 1414, dtype: bool)' is an invalid key"
     ]
    }
   ],
   "source": [
    "from sklearn.metrics import accuracy_score\n",
    "\n",
    "print(\"accuracy:\", accuracy_score(df_test[\"truth\"], df_test[\"label_voter\"]))\n",
    "\n",
    "\n",
    "df_test[df_test[\"truth\"] == 0, df_test[\"label_voter\"] == 1]\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'L_train' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-109-748e8989b292>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      2\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0mlabel_model\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mLabelModel\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcardinality\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m2\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mverbose\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 4\u001b[0;31m \u001b[0mlabel_model\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mL_train\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mL_train\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mn_epochs\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m500\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlog_freq\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m100\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mseed\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m123\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mclass_balance\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0.95\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;36m0.05\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      5\u001b[0m \u001b[0mdf_test\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"label_model\"\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlabel_model\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpredict\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mL\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mL_test\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      6\u001b[0m \u001b[0mdf_test\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"label_model_precent\"\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlabel_model\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpredict_proba\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mL\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mL_test\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'L_train' is not defined"
     ]
    }
   ],
   "source": [
    "from snorkel.labeling.model import LabelModel\n",
    "\n",
    "label_model = LabelModel(cardinality=2, verbose=True)\n",
    "label_model.fit(L_train=L_train, n_epochs=500, log_freq=100, seed=123, class_balance=[0.95, 0.05])\n",
    "df_test[\"label_model\"] = label_model.predict(L=L_test)\n",
    "df_test[\"label_model_precent\"] = label_model.predict_proba(L=L_test)[:,1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [],
   "source": [
    "def grab_second(probs):\n",
    "    return (probs[1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {},
   "outputs": [],
   "source": [
    "# df_train[\"label_probs\"] = np.apply_along_axis(grab_second, 1, majority_model.predict_proba(L=L_train))\n",
    "# df_train[\"label_probs\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_test.to_csv(r'../labeled.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 210,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0    1361\n",
       "1      53\n",
       "Name: truth, dtype: int64"
      ]
     },
     "execution_count": 210,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_test[\"truth\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 211,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sentence</th>\n",
       "      <th>chemicals</th>\n",
       "      <th>truth</th>\n",
       "      <th>substrates</th>\n",
       "      <th>products</th>\n",
       "      <th>text</th>\n",
       "      <th>label_voter</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>the enzyme cyclo-oxygenase catalyses the oxyge...</td>\n",
       "      <td>[arachidonic acid, prostaglandins]</td>\n",
       "      <td>1</td>\n",
       "      <td>arachidonic, acid</td>\n",
       "      <td>prostaglandins</td>\n",
       "      <td>[the enzyme cyclo-oxygenase catalyses the oxyg...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>although the exact site of decarboxylation of ...</td>\n",
       "      <td>[levodopa, dopamine]</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[although the exact site of decarboxylation of...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>the enzyme dimethylarginine dimethylaminohydro...</td>\n",
       "      <td>[citrulline, arginine, methylamines, dimethyla...</td>\n",
       "      <td>1</td>\n",
       "      <td>arginine</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[the enzyme dimethylarginine dimethylaminohydr...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>background and aims glutamic acid decarboxylas...</td>\n",
       "      <td>[gaba, glutamate]</td>\n",
       "      <td>1</td>\n",
       "      <td>glutamate</td>\n",
       "      <td>gamma-aminobutyric, acid, GABA</td>\n",
       "      <td>[background and aims glutamic acid decarboxyla...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>mazindane 26 was found to be a pro-drug, oxidi...</td>\n",
       "      <td>[mazindol, mazindane, 5-oh, 5-h]</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[mazindane 26 was found to be a pro-drug, oxid...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1409</th>\n",
       "      <td>water uptake of the polymer was only 28 and 02...</td>\n",
       "      <td>[pec]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[water uptake of the polymer was only 28 and 0...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1410</th>\n",
       "      <td>degradation of less hydrophilic pec41 with hig...</td>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[degradation of less hydrophilic pec41 with hi...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1411</th>\n",
       "      <td>by this mechanism, ce-responsive drug in vitro...</td>\n",
       "      <td>[pec]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[by this mechanism, ce-responsive drug in vitr...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1412</th>\n",
       "      <td>as expected, less bovine serum albumin bsa was...</td>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[as expected, less bovine serum albumin bsa wa...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1413</th>\n",
       "      <td>in conclusion, this work enabled the in vitro ...</td>\n",
       "      <td>[pec]</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[in conclusion, this work enabled the in vitro...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1414 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                               sentence  \\\n",
       "0     the enzyme cyclo-oxygenase catalyses the oxyge...   \n",
       "1     although the exact site of decarboxylation of ...   \n",
       "2     the enzyme dimethylarginine dimethylaminohydro...   \n",
       "3     background and aims glutamic acid decarboxylas...   \n",
       "4     mazindane 26 was found to be a pro-drug, oxidi...   \n",
       "...                                                 ...   \n",
       "1409  water uptake of the polymer was only 28 and 02...   \n",
       "1410  degradation of less hydrophilic pec41 with hig...   \n",
       "1411  by this mechanism, ce-responsive drug in vitro...   \n",
       "1412  as expected, less bovine serum albumin bsa was...   \n",
       "1413  in conclusion, this work enabled the in vitro ...   \n",
       "\n",
       "                                              chemicals  truth  \\\n",
       "0                    [arachidonic acid, prostaglandins]      1   \n",
       "1                                  [levodopa, dopamine]      1   \n",
       "2     [citrulline, arginine, methylamines, dimethyla...      1   \n",
       "3                                     [gaba, glutamate]      1   \n",
       "4                      [mazindol, mazindane, 5-oh, 5-h]      1   \n",
       "...                                                 ...    ...   \n",
       "1409                                              [pec]      0   \n",
       "1410                                                 []      0   \n",
       "1411                                              [pec]      0   \n",
       "1412                                                 []      0   \n",
       "1413                                              [pec]      0   \n",
       "\n",
       "             substrates                        products  \\\n",
       "0     arachidonic, acid                  prostaglandins   \n",
       "1                   NaN                             NaN   \n",
       "2              arginine                             NaN   \n",
       "3             glutamate  gamma-aminobutyric, acid, GABA   \n",
       "4                   NaN                             NaN   \n",
       "...                 ...                             ...   \n",
       "1409                NaN                             NaN   \n",
       "1410                NaN                             NaN   \n",
       "1411                NaN                             NaN   \n",
       "1412                NaN                             NaN   \n",
       "1413                NaN                             NaN   \n",
       "\n",
       "                                                   text  label_voter  \n",
       "0     [the enzyme cyclo-oxygenase catalyses the oxyg...          1.0  \n",
       "1     [although the exact site of decarboxylation of...          1.0  \n",
       "2     [the enzyme dimethylarginine dimethylaminohydr...          1.0  \n",
       "3     [background and aims glutamic acid decarboxyla...          1.0  \n",
       "4     [mazindane 26 was found to be a pro-drug, oxid...          1.0  \n",
       "...                                                 ...          ...  \n",
       "1409  [water uptake of the polymer was only 28 and 0...          NaN  \n",
       "1410  [degradation of less hydrophilic pec41 with hi...          NaN  \n",
       "1411  [by this mechanism, ce-responsive drug in vitr...          NaN  \n",
       "1412  [as expected, less bovine serum albumin bsa wa...          NaN  \n",
       "1413  [in conclusion, this work enabled the in vitro...          NaN  \n",
       "\n",
       "[1414 rows x 7 columns]"
      ]
     },
     "execution_count": 211,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "only_truth_df = df_test.dropna(subset=['truth'])\n",
    "only_truth_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 182,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sklearn as sk\n",
    "from sklearn.metrics import recall_score\n",
    "from sklearn.metrics import precision_score\n",
    "from sklearn.metrics import accuracy_score\n",
    "from sklearn.metrics import confusion_matrix\n",
    "from sklearn.metrics import balanced_accuracy_score\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 212,
   "metadata": {},
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "Input contains NaN, infinity or a value too large for dtype('float64').",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-212-3fc1e1de19c4>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0mno_abstain_df\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0monly_truth_df\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0monly_truth_df\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"label_voter\"\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m!=\u001b[0m \u001b[0;34m-\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"f1:\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msk\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmetrics\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mf1_score\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mno_abstain_df\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"truth\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mno_abstain_df\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"label_voter\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      3\u001b[0m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"recall:\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrecall_score\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mno_abstain_df\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"truth\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mno_abstain_df\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"label_voter\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"precision:\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mprecision_score\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mno_abstain_df\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"truth\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mno_abstain_df\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"label_voter\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"accuracy:\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maccuracy_score\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mno_abstain_df\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"truth\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mno_abstain_df\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"label_voter\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/anaconda3/lib/python3.8/site-packages/sklearn/metrics/_classification.py\u001b[0m in \u001b[0;36mf1_score\u001b[0;34m(y_true, y_pred, labels, pos_label, average, sample_weight, zero_division)\u001b[0m\n\u001b[1;32m   1111\u001b[0m     \u001b[0mmodified\u001b[0m \u001b[0;32mwith\u001b[0m\u001b[0;31m \u001b[0m\u001b[0;31m`\u001b[0m\u001b[0;31m`\u001b[0m\u001b[0mzero_division\u001b[0m\u001b[0;31m`\u001b[0m\u001b[0;31m`\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1112\u001b[0m     \"\"\"\n\u001b[0;32m-> 1113\u001b[0;31m     return fbeta_score(\n\u001b[0m\u001b[1;32m   1114\u001b[0m         \u001b[0my_true\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1115\u001b[0m         \u001b[0my_pred\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/anaconda3/lib/python3.8/site-packages/sklearn/metrics/_classification.py\u001b[0m in \u001b[0;36mfbeta_score\u001b[0;34m(y_true, y_pred, beta, labels, pos_label, average, sample_weight, zero_division)\u001b[0m\n\u001b[1;32m   1249\u001b[0m     \"\"\"\n\u001b[1;32m   1250\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1251\u001b[0;31m     _, _, f, _ = precision_recall_fscore_support(\n\u001b[0m\u001b[1;32m   1252\u001b[0m         \u001b[0my_true\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1253\u001b[0m         \u001b[0my_pred\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/anaconda3/lib/python3.8/site-packages/sklearn/metrics/_classification.py\u001b[0m in \u001b[0;36mprecision_recall_fscore_support\u001b[0;34m(y_true, y_pred, beta, labels, pos_label, average, warn_for, sample_weight, zero_division)\u001b[0m\n\u001b[1;32m   1532\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mbeta\u001b[0m \u001b[0;34m<\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1533\u001b[0m         \u001b[0;32mraise\u001b[0m \u001b[0mValueError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"beta should be >=0 in the F-beta score\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1534\u001b[0;31m     \u001b[0mlabels\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m_check_set_wise_labels\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0my_true\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0my_pred\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maverage\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlabels\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpos_label\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1535\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1536\u001b[0m     \u001b[0;31m# Calculate tp_sum, pred_sum, true_sum ###\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/anaconda3/lib/python3.8/site-packages/sklearn/metrics/_classification.py\u001b[0m in \u001b[0;36m_check_set_wise_labels\u001b[0;34m(y_true, y_pred, average, labels, pos_label)\u001b[0m\n\u001b[1;32m   1336\u001b[0m         \u001b[0;32mraise\u001b[0m \u001b[0mValueError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"average has to be one of \"\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0mstr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0maverage_options\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1337\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1338\u001b[0;31m     \u001b[0my_type\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0my_true\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0my_pred\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m_check_targets\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0my_true\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0my_pred\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1339\u001b[0m     \u001b[0;31m# Convert to Python primitive type to avoid NumPy type / Python str\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1340\u001b[0m     \u001b[0;31m# comparison. See https://github.com/numpy/numpy/issues/6784\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/anaconda3/lib/python3.8/site-packages/sklearn/metrics/_classification.py\u001b[0m in \u001b[0;36m_check_targets\u001b[0;34m(y_true, y_pred)\u001b[0m\n\u001b[1;32m     84\u001b[0m     \u001b[0mcheck_consistent_length\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0my_true\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0my_pred\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     85\u001b[0m     \u001b[0mtype_true\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mtype_of_target\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0my_true\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 86\u001b[0;31m     \u001b[0mtype_pred\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mtype_of_target\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0my_pred\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     87\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     88\u001b[0m     \u001b[0my_type\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m{\u001b[0m\u001b[0mtype_true\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtype_pred\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/anaconda3/lib/python3.8/site-packages/sklearn/utils/multiclass.py\u001b[0m in \u001b[0;36mtype_of_target\u001b[0;34m(y)\u001b[0m\n\u001b[1;32m    322\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0my\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mkind\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m\"f\"\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0many\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0my\u001b[0m \u001b[0;34m!=\u001b[0m \u001b[0my\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mastype\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mint\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    323\u001b[0m         \u001b[0;31m# [.1, .2, 3] or [[.1, .2, 3]] or [[1., .2]] and not [1., 2., 3.]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 324\u001b[0;31m         \u001b[0m_assert_all_finite\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0my\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    325\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0;34m\"continuous\"\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0msuffix\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    326\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/anaconda3/lib/python3.8/site-packages/sklearn/utils/validation.py\u001b[0m in \u001b[0;36m_assert_all_finite\u001b[0;34m(X, allow_nan, msg_dtype)\u001b[0m\n\u001b[1;32m    112\u001b[0m         ):\n\u001b[1;32m    113\u001b[0m             \u001b[0mtype_err\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m\"infinity\"\u001b[0m \u001b[0;32mif\u001b[0m \u001b[0mallow_nan\u001b[0m \u001b[0;32melse\u001b[0m \u001b[0;34m\"NaN, infinity\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 114\u001b[0;31m             raise ValueError(\n\u001b[0m\u001b[1;32m    115\u001b[0m                 msg_err.format(\n\u001b[1;32m    116\u001b[0m                     \u001b[0mtype_err\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmsg_dtype\u001b[0m \u001b[0;32mif\u001b[0m \u001b[0mmsg_dtype\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m \u001b[0;32melse\u001b[0m \u001b[0mX\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mValueError\u001b[0m: Input contains NaN, infinity or a value too large for dtype('float64')."
     ]
    }
   ],
   "source": [
    "no_abstain_df = only_truth_df[only_truth_df[\"label_voter\"] != -1]\n",
    "print(\"f1:\", sk.metrics.f1_score(no_abstain_df[\"truth\"], no_abstain_df[\"label_voter\"]))\n",
    "print(\"recall:\", recall_score(no_abstain_df[\"truth\"], no_abstain_df[\"label_voter\"]))\n",
    "print(\"precision:\", precision_score(no_abstain_df[\"truth\"], no_abstain_df[\"label_voter\"]))\n",
    "print(\"accuracy:\", accuracy_score(no_abstain_df[\"truth\"], no_abstain_df[\"label_voter\"]))\n",
    "print(\"balanced accuracy:\", balanced_accuracy_score(no_abstain_df[\"truth\"], no_abstain_df[\"label_voter\"]))\n",
    "tn, fp, fn, tp = confusion_matrix(no_abstain_df[\"truth\"], no_abstain_df[\"label_voter\"]).ravel()\n",
    "print(\"matrix:\", (tn, fp, fn, tp))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'\\nf1: 0.38317757009345793\\nrecall: 0.7454545454545455\\nprecision: 0.2578616352201258\\naccuracy: 0.8980694980694981\\nbalanced accuracy: 0.8251466275659824\\nmatrix: (1122, 118, 14, 41)\\nwith out js\\nf1: 0.4833333333333333\\nrecall: 0.5576923076923077\\nprecision: 0.4264705882352941\\naccuracy: 0.9533483822422875\\nbalanced accuracy: 0.763575989398229\\nmatrix: (1238, 39, 23, 29)\\n'"
      ]
     },
     "execution_count": 117,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'''\n",
    "f1: 0.38317757009345793\n",
    "recall: 0.7454545454545455\n",
    "precision: 0.2578616352201258\n",
    "accuracy: 0.8980694980694981\n",
    "balanced accuracy: 0.8251466275659824\n",
    "matrix: (1122, 118, 14, 41)\n",
    "*with out js lfs*\n",
    "f1: 0.4833333333333333\n",
    "recall: 0.5576923076923077\n",
    "precision: 0.4264705882352941\n",
    "accuracy: 0.9533483822422875\n",
    "balanced accuracy: 0.763575989398229\n",
    "matrix: (1238, 39, 23, 29)\n",
    "'''"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "f1: 0.1836734693877551\n",
      "recall: 0.15\n",
      "precision: 0.23684210526315788\n",
      "accuracy: 0.9434229137199435\n",
      "balanced accuracy: 0.5642909896602658\n",
      "matrix: (1325, 29, 51, 9)\n"
     ]
    }
   ],
   "source": [
    "no_abstain_df = only_truth_df[only_truth_df[\"label_model\"] != -1]\n",
    "print(\"f1:\", sk.metrics.f1_score(no_abstain_df[\"truth\"], no_abstain_df[\"label_model\"]))\n",
    "print(\"recall:\", recall_score(no_abstain_df[\"truth\"], no_abstain_df[\"label_model\"]))\n",
    "print(\"precision:\", precision_score(no_abstain_df[\"truth\"], no_abstain_df[\"label_model\"]))\n",
    "print(\"accuracy:\", accuracy_score(no_abstain_df[\"truth\"], no_abstain_df[\"label_model\"]))\n",
    "print(\"balanced accuracy:\", balanced_accuracy_score(no_abstain_df[\"truth\"], no_abstain_df[\"label_model\"]))\n",
    "tn, fp, fn, tp = confusion_matrix(no_abstain_df[\"truth\"], no_abstain_df[\"label_model\"]).ravel()\n",
    "print(\"matrix:\", (tn, fp, fn, tp))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'\\nf1: 0.14925373134328357\\nrecall: 0.08333333333333333\\nprecision: 0.7142857142857143\\naccuracy: 0.9596888260254597\\nbalanced accuracy: 0.5409281142294436\\nmatrix: (1352, 2, 55, 5)\\n'"
      ]
     },
     "execution_count": 141,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'''\n",
    "f1: 0.21951219512195122\n",
    "recall: 0.15\n",
    "precision: 0.4090909090909091\n",
    "accuracy: 0.9547383309759547\n",
    "balanced accuracy: 0.5701994091580502\n",
    "matrix: (1341, 13, 51, 9)\n",
    "*without js lfs*\n",
    "f1: 0.14925373134328357\n",
    "recall: 0.08333333333333333\n",
    "precision: 0.7142857142857143\n",
    "accuracy: 0.9596888260254597\n",
    "balanced accuracy: 0.5409281142294436\n",
    "matrix: (1352, 2, 55, 5)\n",
    "'''"
   ]
  }
 ],
 "metadata": {
  "interpreter": {
   "hash": "e1fffad673803e23a98281b7492a34c559235864658941242758d49259d94b3b"
  },
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
